---
title: "Data Analysis"
author: "Paul Christmann"
date: '2022-05-30'
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

## Load useful Libraries
```{r}
library(Rcpp)
library(tidyverse)
library(affy)
library(vsn)
library(AnnotationDbi)
library(hgu133plus2hsenstprobe)
library(hgu133plus2hsenstcdf) 
library(limma)
library(pheatmap)
library(hexbin)
library(biomaRt)
```


## Uses Annotated Dataset with only TRA Genes

Import the annotated dataset with only TRA genes. Information from the TRA data was merged 
```{r}
#setwd("C:/Users/david/Documents/Data Analysis Lokal/2022-topic-04-team-01/R Scripts")
#setwd("F:/Data Analysis Project/Github Repo/2022-topic-04-team-01/R Scripts") #Paul
setwd("C:/Users/merke/Documents/Vorlesungen Material/SS22/GitHub/2022-topic-04-team-01/R Scripts") #Verena
#setwd("C:/Daten/Data Analysis Projekt/Github Repo/2022-topic-04-team-01/R Scripts")
load("embryo_df_tra.RData")
```



Overview over the dataset

```{r}
colnames(embryo_df_tra)
dim(embryo_df_tra)
head(embryo_df_tra)
str(embryo_df_tra)
```


```{r}
#sum(apply(embryo_df_tra, 1, fun))
```





## Changing the Tissues into a vector
```{r}
library(tidyverse)

#Make Tissue names more accessible
 a = separate(embryo_df_tra, tissues, into = as.character(c(1:30)), sep="/")

create_tissue_vector = function(f) {
  f = unlist(f)
  f = f[!is.na(f)]
  return(f)
}

tissues = apply(a[,31:60],1, create_tissue_vector)
tissues = sapply(tissues, unlist)

a = a[,c(1:30, 61)] 
a %>% add_column(tissues = NA)
a$tissues = tissues

embryo_df_tissues = a
```

```{r}
save(embryo_df_tissues, file="embryo_df_tissues.RData")
```



## Modify expression data from Week9 to 4 comparison with accessible tissue vector
```{r}
setwd("C:/Users/merke/Documents/Vorlesungen Material/SS22/GitHub/2022-topic-04-team-01/R Scripts")
load("week9-4_diff.RData")

week9_4_diff_tra = embryo_df_tissues[embryo_df_tissues$ensembl.transcript %in% tra_diff$ensembl.transcript, ]

save(week9_4_diff_tra, file="week9_4_diff_tra.RData")

```



## Limma Analysis for all different Combinations

```{r}
library(limma)

#Differential gene expression: Limma Analysis 2, between weeks

limma_data = as.data.frame(embryo_df_tissues[1:18], rownames = embryo_df_tissues$ensembl.transcript)

weeks = c("week4", "week5", "week6", "week7", "week8", "week9")
f = factor(c("week4", "week4", "week4", "week5", "week5", "week5", "week6", "week6", "week6", "week7", "week7", "week7", "week8", "week8", "week8", "week9", "week9", "week9"), levels =weeks)
design = model.matrix(~0+f)
colnames(design) =  weeks
fit_2 = lmFit(limma_data, design)

contrast_all = makeContrasts(week9-week8, week9-week7, week9-week6, week9-week5, week9-week4, week8-week7,week8-week6, week8-week5, week8-week4, week7-week6, week7-week5, week7-week4, week6-week5, week6-week4, week5-week4, levels = design)
fit_all = contrasts.fit(fit_2, contrast_all)
fit_all = eBayes(fit_all)
results_all = decideTests(fit_all)
summary(results_all)

```

### Do not use this block, only a single comparison for test purposes
```{r}
# Ausgabe mit Gennamen für einen Vergleich

top.table = topTable(fit_all,1, n = Inf)
 n = sapply(rownames(top.table), function(x){
  embryo_df_tissues$ensembl.transcript[as.numeric(x)]
})
row.names(top.table) = n
top.table[which(top.table$adj.P.Val < 0.01), ]
```


```{r}
#Create list of all differentially expresssed genes
diff_genes_list_0.01 = c()

for (i in 1:15) {
 top.table = topTable(fit_all,i, n = Inf)
 n = sapply(rownames(top.table), function(x){
  embryo_df_tissues$ensembl.transcript[as.numeric(x)]
  })
  row.names(top.table) = n
  diff_genes_list_0.01 = c(diff_genes_list_0.01, rownames(top.table[which(top.table$adj.P.Val < 0.01), ]))
}

```

```{r}

length(diff_genes_list_0.01) #4538
diff_genes_list_0.01 = diff_genes_list_0.01[!duplicated(diff_genes_list_0.01)]
length(diff_genes_list_0.01) #1818

```

```{r}
#List with 5%

diff_genes_list_0.05 = c()

for (i in 1:15) {
 top.table = topTable(fit_all,i, n = Inf)
 n = sapply(rownames(top.table), function(x){
  embryo_df_tissues$ensembl.transcript[as.numeric(x)]
  })
  row.names(top.table) = n
  diff_genes_list_0.05 = c(diff_genes_list_0.05, rownames(top.table[which(top.table$adj.P.Val < 0.05), ]))
}
diff_genes_list_0.05 = diff_genes_list_0.05[!duplicated(diff_genes_list_0.05)]

```


### Final list of differentially expressed genes with an adjusted p-valuelower than 0.01:

```{r}
#Select differential genes from annotated dataframe
diff_genes_ann_0.01 = embryo_df_tissues[embryo_df_tissues$ensembl.transcript %in% diff_genes_list_0.01, ]
save(diff_genes_ann_0.01, file = "diff_genes_ann_0.01.RData")
```





## Test for single differential gene expression Graph: Frontal Cortex

```{r}
#Select genes only of target tissue
bool_contained = sapply(diff_genes_ann_0.01$tissues, function(x) {"Brain - Frontal Cortex" %in% x})
diff_genes_ann_0.01[bool_contained,]

```

```{r}
#Graph of mean gene expression over time
frontal_cortex = diff_genes_ann_0.01[bool_contained,]
frontal_cortex = frontal_cortex[, 1:19]
frontal_cortex_mean = data.frame(week4=rowMeans(frontal_cortex[,1:3]),week5=rowMeans(frontal_cortex[,4:6]), week6=rowMeans(frontal_cortex[,7:9]),week7=rowMeans(frontal_cortex[,10:12]), week8=rowMeans(frontal_cortex[,13:15]), week9=rowMeans(frontal_cortex[,16:18]))
frontal_cortex_graph = colMeans(frontal_cortex_mean)
frontal_cortex_graph = data.frame (Expression_Level = frontal_cortex_graph, Week = c(4:9))

frontal_cortex_plot = ggplot(frontal_cortex_graph, aes(x = Week, y= Expression_Level))  + geom_line(color = "#0000ff") + geom_point(color = "#0000ff") + labs(title = paste("Median expression level of all brain-related genes (GSE15744)"))  + theme_classic() 
  
```
```{r}
frontal_cortex_plot
frontal_cortex_mean
```

### Clustering of one tissue 

```{r}
#New Dataset with median of gene expression
data_temp = diff_genes_ann_0.01[bool_contained,]
data_abs = data.frame(Week4 = apply(data_temp[1:3], 1, median), Week5 =  apply(data_temp[4:6], 1, median), Week6 =  apply(data_temp[7:9], 1, median), Week7 =  apply(data_temp[10:12], 1, median), Week8 =  apply(data_temp[13:15], 1, median), Week9 =  apply(data_temp[16:18], 1, median))
row.names(data_abs) = data_temp$ensembl.transcript
```

```{r}
# New dataset with difference in gene expression

expression_difference = function(x,y) {
  return ((x-y)/((x+y)*0.5))
}

data = data.frame("5_4" = expression_difference(data_abs[,"Week5"], data_abs[,"Week4"]), 
                  "6_5" = expression_difference(data_abs[,"Week6"], data_abs[,"Week5"]),
                  "7_6" = expression_difference(data_abs[,"Week7"], data_abs[,"Week6"]),
                  "8_7" = expression_difference(data_abs[,"Week8"], data_abs[,"Week7"]),
                  "9_8" = expression_difference(data_abs[,"Week9"], data_abs[,"Week8"]))
row.names(data) = rownames(data_abs)
```


```{r}
# K-Means für die Werte
library(factoextra)
library(cluster)
library(stringr)
library(ggplot2)
library(ggsci)

# Daten Skalieren, Distance Matrix
data = scale(data)
dist_data = get_dist(data, method = "pearson")

# Find optimal cluster number by silhouette value
avg_sil = function(k) {
  km = kmeans(data, centers = k, nstart = 25)
  ss = silhouette(km$cluster, dist_data)
  mean(ss[, 3])
}
n_cluster = which.max(sapply(c(2:10), function(k) {avg_sil(k)})) +1

#Plot with clustered genes

km = kmeans(data, centers = n_cluster, nstart = 25)
data_plot_raw = cbind(data_abs, Cluster = km$cluster)

create_plot_data = function(dataset) {
  data_plot = data.frame(Week = c(sapply(c(4:9), function(x) {rep(x, n_cluster)})), Cluster = as.character(c(rep(c(1:n_cluster), 6))), Expression = NA)
  count = 0
  for(s in 4:9) {
    for(t in 1:n_cluster) {
      count = count + 1
      values = dataset[dataset$Cluster == t, str_glue("Week", s)]
      data_plot[count, "Expression"] = median(values)
      data_plot[count, "Cluster"] = str_glue(data_plot[count, "Cluster"], " [", km$size[t], " genes]")
   }
  }
  return(data_plot)
}
plot_data = create_plot_data(data_plot_raw)



ggplot(plot_data, aes(x = Week, y= Expression, group = Cluster))  + geom_line(aes(colour = Cluster)) + geom_point(aes(colour = Cluster, shape = Cluster)) + labs(title = paste("Expression of ", "Brain - Frontal Cortex", " genes", sep="")) + scale_color_npg() + theme_classic()


```




## Integrated function for 12 different plots

```{r}
library(factoextra)
library(cluster)
library(stringr)
library(ggplot2)
library(ggsci)
library(grid)
library(gridExtra)

#setwd("C:/Daten/Data Analysis Projekt/Github Repo/2022-topic-04-team-01/R Scripts")
load("diff_genes_ann_0.01.RData")

# List the tissues
tissues = c("Brain - Amygdala", "Brain - Anterior cingulate cortex", "Brain - Caudate", "Brain - Cerebellar Hemisphere", "Brain - Cerebellum", "Brain - Cortex", "Brain - Frontal Cortex", "Brain - Hippocampus", "Brain - Hypothalamus", "Brain - Nucleus accumbens", "Brain - Putamen", "Brain - Spinal cord", "Brain - Substantia nigra")


# Functions for later use
expression_difference = function(x,y) {
  return ((x-y)/((x+y)*0.5))
}

avg_sil = function(k, data) {
  km = kmeans(data, centers = k, nstart = 25)
  ss = silhouette(km$cluster, dist_data)
  mean(ss[, 3])
}

create_plot_data = function(dataset) {
  data_plot = data.frame(Week = c(sapply(c(4:9), function(x) {rep(x, n_cluster)})), Cluster = as.character(c(rep(c(1:n_cluster), 6))), Expression = NA)
  count = 0
  for(s in 4:9) {
    for(t in 1:n_cluster) {
      count = count + 1
      values = dataset[dataset$Cluster == t, str_glue("Week", s)]
      data_plot[count, "Expression"] = median(values)
      data_plot[count, "Cluster"] = str_glue(data_plot[count, "Cluster"], " [", km$size[t], " genes]")
   }
  }
  return(data_plot)
}


for (tissue in tissues) {
  # Select an modify the data used for kmeans
  bool_contained = sapply(diff_genes_ann_0.01$tissues, function(x) {tissue %in% x})
  data_temp = diff_genes_ann_0.01[bool_contained,]
  data_abs = data.frame(Week4 = apply(data_temp[1:3], 1, median), Week5 =  apply(data_temp[4:6], 1, median), Week6 =  apply(data_temp[7:9], 1, median),
                        Week7=apply(data_temp[10:12], 1, median), Week8 =  apply(data_temp[13:15], 1, median), Week9 =  apply(data_temp[16:18], 1, median))
  data = data.frame("5_4" = expression_difference(data_abs[,"Week5"], data_abs[,"Week4"]), 
                  "6_5" = expression_difference(data_abs[,"Week6"], data_abs[,"Week5"]),
                  "7_6" = expression_difference(data_abs[,"Week7"], data_abs[,"Week6"]),
                  "8_7" = expression_difference(data_abs[,"Week8"], data_abs[,"Week7"]),
                  "9_8" = expression_difference(data_abs[,"Week9"], data_abs[,"Week8"]))
  row.names(data) = rownames(data_abs)
  
  # Select optimum cluster number
  data = scale(data)
  dist_data = get_dist(data, method = "pearson")
  n_cluster = which.max(sapply(c(2:10), function(k) {avg_sil(k, data)})) +1
  
  # Create the data for our plot
  km = kmeans(data, centers = n_cluster, nstart = 25)
  data_plot_raw = cbind(data_abs, Cluster = km$cluster)
  plot_data = create_plot_data(data_plot_raw)
  
  # Create the Plot
  assign(str_glue(tissue, "_plot"), ggplot(plot_data, aes(x = Week, y= Expression, group = Cluster))  + geom_line(aes(colour = Cluster)) +
           geom_point(aes(colour = Cluster, shape = Cluster)) + labs(title = paste("Expression of ", tissue, " genes", sep="")) +
           scale_color_npg() + theme_classic() )
   
}
ff = sapply(tissues, function(x){str_glue(x, "_plot")})
brain_table = grid.arrange(get(ff[1]), get(ff[2]), get(ff[3]), get(ff[4]), get(ff[5]), get(ff[6]), get(ff[7]), get(ff[8]), get(ff[9]), get(ff[10]), get(ff[11]), get(ff[12]))
```


```{r}
# Get a tissue count for our final gene list

tissues = c("Adipose", "Artery", "Brain",  "Colon", "Esophagus", "Heart", "Kidney", "Liver", "Lung", "Muscle", "Skin", "Stomach", "Testis")
count_data = data.frame(tissue = tissues, count = NA)
for (tissue in tissues){
  
  count_data[count_data$tissue == tissue, "count"] = sum(bool_contained = sapply(diff_genes_ann_0.01$tissues, function(x) {grepl(tissue, toString(x))}))

}
count_data
```



## Ecactly the same with a cooler tissue selection
```{r}
library(factoextra)
library(cluster)
library(stringr)
library(ggplot2)
library(ggsci)
library(grid)
library(gridExtra)

#setwd("C:/Daten/Data Analysis Projekt/Github Repo/2022-topic-04-team-01/R Scripts")
load("diff_genes_ann_0.01.RData")

# List the tissues
tissues = c("Adipose", "Artery", "Brain",  "Colon", "Esophagus", "Heart", "Kidney", "Liver", "Lung", "Muscle", "Skin", "Stomach", "Testis")


# Functions for later use
expression_difference = function(x,y) {
  return ((x-y)/((x+y)*0.5))
}

avg_sil = function(k, data) {
  km = kmeans(data, centers = k, nstart = 25)
  ss = silhouette(km$cluster, dist_data)
  mean(ss[, 3])
}

create_plot_data = function(dataset) {
  data_plot = data.frame(Week = c(sapply(c(4:9), function(x) {rep(x, n_cluster)})), Cluster = as.character(c(rep(c(1:n_cluster), 6))), Expression = NA)
  count = 0
  for(s in 4:9) {
    for(t in 1:n_cluster) {
      count = count + 1
      values = dataset[dataset$Cluster == t, str_glue("Week", s)]
      data_plot[count, "Expression"] = median(values)
      data_plot[count, "Cluster"] = str_glue(data_plot[count, "Cluster"], " [", km$size[t], " genes]")
   }
  }
  return(data_plot)
}


for (tissue in tissues) {
  # Select an modify the data used for kmeans
  bool_contained = sapply(diff_genes_ann_0.01$tissues, function(x) {grepl(tissue, toString(x))})
  data_temp = diff_genes_ann_0.01[bool_contained,]
  data_abs = data.frame(Week4 = apply(data_temp[1:3], 1, median), Week5 =  apply(data_temp[4:6], 1, median), Week6 =  apply(data_temp[7:9], 1, median),
                        Week7=apply(data_temp[10:12], 1, median), Week8 =  apply(data_temp[13:15], 1, median), Week9 =  apply(data_temp[16:18], 1, median))
  data = data.frame("5_4" = expression_difference(data_abs[,"Week5"], data_abs[,"Week4"]), 
                  "6_5" = expression_difference(data_abs[,"Week6"], data_abs[,"Week5"]),
                  "7_6" = expression_difference(data_abs[,"Week7"], data_abs[,"Week6"]),
                  "8_7" = expression_difference(data_abs[,"Week8"], data_abs[,"Week7"]),
                  "9_8" = expression_difference(data_abs[,"Week9"], data_abs[,"Week8"]))
  row.names(data) = rownames(data_abs)
  
  # Select optimum cluster number
  data = scale(data)
  dist_data = get_dist(data, method = "pearson")
  n_cluster = which.max(sapply(c(2:10), function(k) {avg_sil(k, data)})) +1
  
  # Create the data for our plot
  km = kmeans(data, centers = n_cluster, nstart = 25)
  data_plot_raw = cbind(data_abs, Cluster = km$cluster)
  plot_data = create_plot_data(data_plot_raw)
  
  # Create the Plot
  assign(str_glue(tissue, "_plot"), ggplot(plot_data, aes(x = Week, y= Expression, group = Cluster))  + geom_line(aes(colour = Cluster)) +
           geom_point(aes(colour = Cluster, shape = Cluster)) + labs(title = paste("Expression of ", tissue, " genes", sep="")) +
           scale_color_npg() + theme_classic() )
   
}
ff = sapply(tissues, function(x){str_glue(x, "_plot")})
organs_table = grid.arrange(get(ff[1]), get(ff[2]), get(ff[3]), get(ff[4]), get(ff[5]), get(ff[6]), get(ff[7]), get(ff[8]), get(ff[9]), get(ff[10]), get(ff[11]), get(ff[12]))
```

















## Data cleaning
```{r}
glimpse(embryo_df_tra) #Chromosome.scaffold.name should be a factor data type
summary(embryo_df_tra)

embryo_df_tra <- embryo_df_tra %>%
  mutate(Chromosome.scaffold.name.fct = as.factor(Chromosome.scaffold.name))
summary(embryo_df_tra$Chromosome.scaffold.name.fct)


#lv = levels(embryo_df_tra$Chromosome.scaffold.name.fct) # weirde Chromosomennamen
#"CHR_HSCHR10_1_CTG2", "CHR_HSCHR11_1_CTG7", "CHR_HSCHR15_4_CTG8", "CHR_HSCHR22_1_CTG3", "CHR_HSCHR22_1_CTG7",  "CHR_HSCHR4_6_CTG12", "CHR_HSCHR7_2_CTG4_4", "CHR_HSCHR7_2_CTG6"
#sum(lv== "CHR_HSCHR10_1_CTG2" | lv== "CHR_HSCHR11_1_CTG7"| lv== "CHR_HSCHR15_4_CTG8"| lv== "CHR_HSCHR22_1_CTG3"| lv== "CHR_HSCHR22_1_CTG7"| lv==  "CHR_HSCHR4_6_CTG12"| lv== "CHR_HSCHR7_2_CTG4_4"| lv==  "CHR_HSCHR7_2_CTG6") # 8

embry_df_tra <- embryo_df_tra %>%
  filter(Chromosome.scaffold.name.fct != "CHR_HSCHR10_1_CTG2" & Chromosome.scaffold.name.fct != "CHR_HSCHR11_1_CTG7"& Chromosome.scaffold.name.fct!= "CHR_HSCHR15_4_CTG8"& Chromosome.scaffold.name.fct!= "CHR_HSCHR22_1_CTG3"& Chromosome.scaffold.name.fct!= "CHR_HSCHR22_1_CTG7"& Chromosome.scaffold.name.fct!=  "CHR_HSCHR4_6_CTG12"& Chromosome.scaffold.name.fct!= "CHR_HSCHR7_2_CTG4_4"& Chromosome.scaffold.name.fct!=  "CHR_HSCHR7_2_CTG6" )
embry_df_tra$Chromosome.scaffold.name.fct <- droplevels(embry_df_tra$Chromosome.scaffold.name.fct)
embry_df_tra$Chromosome.scaffold.name.fct <- factor(embry_df_tra$Chromosome.scaffold.name.fct, levels = c("1", "2","3","4","5","6","7","8","9","10","11","12","13","14","15","16","17","18","19", "20", "21","22", "X","Y", "empty"))
embry_df_tra$Chromosome.scaffold.name.fct <- droplevels(embry_df_tra$Chromosome.scaffold.name.fct)

summary(embry_df_tra$Chromosome.scaffold.name.fct)


```
Data cleaning tissue

```{r}
embryo_test <- embryo_df_tra %>%
  mutate(tiss.fct = as.factor(max.tissue))

summary(embryo_test$tiss.fct)
levels(embryo_test$tiss.fct)
head(embryo_df_tissues$tissues)
```




```{r}
glimpse(diff_genes_ann_0.01)

diff_genes_ann_0.01 %>%
  count(ensembl.transcript) %>%
  filter(n>1) # Es sind keine Dublikate mehr vorhanden

summary(as.factor(diff_genes_ann_0.01$max.tissue)) # von jedem tissue existiert noch min 1 differentiell exprimiertes Gen


```


```{r}
#pca ueber spalten (wochen)
pca_wochen <- prcomp(diff_genes_ann_0.01[,1:18],center=TRUE, scale.=TRUE)
summary(pca_wochen)
pca_wochen_df <- data.frame(pca_wochen$x[, 1:2])
```

```{r}
library(devtools)
install_github("vqv/ggbiplot")
library(ggbiplot)
library(psych)
```


```{r}
#plotting pca_wochen
ggbiplot(pca_wochen, groups = diff_genes_ann_0.01$max.tissue) #groese anpassen damit plot sichtbar
ggbiplot(pca_wochen, labels=colnames(diff_genes_ann_0.01)[1:18])
ggbiplot(pca_wochen, choices=c(2,3))
ggbiplot(pca_wochen, choices=c(3,4))
#ggbiplot(pca_wochen,labels=diff_genes_ann_0.01['max.tissue']) #nicht lesbar
#ggbiplot(pca_wochen,groups=c(diff_genes_ann_0.01['max.tissue']))

#ggplot
pca_wochen_plot <- ggplot(
  pca_wochen_df,
  aes(
    x = PC1,
    y = PC2,
    #color=c(diff_genes_ann_0.01['max.tissue'])
  )
) +
  geom_point() + # Plot individual points to make a scatterplot
  theme_classic() # Format as a classic-looking plot with no gridlines

# Print out the plot here
pca_wochen_plot
```


```{r}
pairs.panels(pca_wochen$x,
             gap=0,
             bg = c("red", "yellow", "blue"),
             pch=21)
pca_genes
```



```{r}
#pca ueber zeilen (genes)
pca_genes <- prcomp(
  t(diff_genes_ann_0.01[,1:18]), # transpose our data frame to obtain PC scores for genes, not weeks
  scale. = TRUE # we want the data scaled to have unit variance
  )
summary(pca_genes)
#pca_genes

pca_genes_df <- data.frame(pca_genes$x[, 1:2])
```

```{r}
#plotting pca_genes
ggbiplot(pca_genes, labels=colnames(diff_genes_ann_0.01)[1:18])
ggbiplot(pca_genes, choices=c(2,3))
ggbiplot(pca_genes, choices=c(3,4))


#ggplot
pca_genes_plot <- ggplot(
  pca_genes_df,
  aes(
    x = PC1,
    y = PC2,
    
  )
) +
  geom_point() + # Plot individual points to make a scatterplot
  theme_classic() # Format as a classic-looking plot with no gridlines

# Print out the plot here
pca_genes_plot
```
```{r}
pairs.panels(pca_genes$x,
             gap=0,
             bg = c("red", "yellow", "blue"),
             pch=21)
```


```{r}
head(diff_genes_ann_0.01)
dim(diff_genes_ann_0.01)
```






