---
title: "Data Analysis"
author: "Paul Christmann"
date: '2022-05-30'
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

## Uses Annotated Dataset with only TRA Genes


```{r}
#Import the annotated dataset with only TRA genes. Information from the TRA data was merged 
setwd("F:/Data Analysis Project/Github Repo/2022-topic-04-team-01/R Scripts")
load("embryo_df_tra.RData")
```


```{r}
#Overview over the dataset
colnames(embryo_df_tra)
dim(embryo_df_tra)

```






## Changing the Tissues into a vector
```{r}
#Make Tissue names more accessible
 a = separate(embryo_df_tra, tissues, into = as.character(c(1:30)), sep="/")

create_tissue_vector = function(f) {
  f = unlist(f)
  f = f[!is.na(f)]
  return(f)
}

tissues = apply(a[,31:60],1, create_tissue_vector)
tissues = sapply(tissues, unlist)

a = a[,c(1:30, 61)] 
a %>% add_column(tissues = NA)
a$tissues = tissues

embryo_df_tissues = a
```

```{r}
save(embryo_df_tissues, file="embryo_df_tissues.RData")
```



## Modify expression data from Week9 to 4 comparison with accessible tissue vector
```{r}
setwd("F:/Data Analysis Project/Github Repo/2022-topic-04-team-01/R Scripts")
load("week9-4_diff.RData")

week9_4_diff_tra = embryo_df_tissues[embryo_df_tissues$ensembl.transcript %in% tra_diff$ensembl.transcript, ]

save(week9_4_diff_tra, file="week9_4_diff_tra.RData")

```



## Limma Analysis for all different Combinations

```{r}
#Differential gene expression: Limma Analysis 2, between weeks

limma_data = as.data.frame(embryo_df_tissues[1:18], rownames = embryo_df_tissues$ensembl.transcript)

weeks = c("week4", "week5", "week6", "week7", "week8", "week9")
f = factor(c("week4", "week4", "week4", "week5", "week5", "week5", "week6", "week6", "week6", "week7", "week7", "week7", "week8", "week8", "week8", "week9", "week9", "week9"), levels =weeks)
design = model.matrix(~0+f)
colnames(design) =  weeks
fit_2 = lmFit(limma_data, design)

contrast_all = makeContrasts(week9-week8, week9-week7, week9-week6, week9-week5, week9-week4, week8-week7,week8-week6, week8-week5, week8-week4, week7-week6, week7-week5, week7-week4, week6-week5, week6-week4, week5-week4, levels = design)
fit_all = contrasts.fit(fit_2, contrast_all)
fit_all = eBayes(fit_all)
results_all = decideTests(fit_all)
summary(results_all)

```


```{r}
# Ausgabe mit Gennamen f√ºr einen Vergleich

top.table = topTable(fit_all,1, n = Inf)
 n = sapply(rownames(top.table), function(x){
  embryo_df_tissues$ensembl.transcript[as.numeric(x)]
})
row.names(top.table) = n
top.table[which(top.table$adj.P.Val < 0.01), ]
```


```{r}
diff_genes_list_0.01 = c()

for (i in 1:15) {
 top.table = topTable(fit_all,i, n = Inf)
 n = sapply(rownames(top.table), function(x){
  embryo_df_tissues$ensembl.transcript[as.numeric(x)]
  })
  row.names(top.table) = n
  diff_genes_list_0.01 = c(diff_genes_list_0.01, rownames(top.table[which(top.table$adj.P.Val < 0.01), ]))
}

```



```{r}

length(diff_genes_list_0.01)
diff_genes_list_0.01 = diff_genes_list_0.01[!duplicated(diff_genes_list_0.01)]
length(diff_genes_list_0.01)

```


```{r}
#List with 5%

diff_genes_list_0.05 = c()

for (i in 1:15) {
 top.table = topTable(fit_all,i, n = Inf)
 n = sapply(rownames(top.table), function(x){
  embryo_df_tissues$ensembl.transcript[as.numeric(x)]
  })
  row.names(top.table) = n
  diff_genes_list_0.05 = c(diff_genes_list_0.05, rownames(top.table[which(top.table$adj.P.Val < 0.05), ]))
}
diff_genes_list_0.05 = diff_genes_list_0.05[!duplicated(diff_genes_list_0.05)]

```


```{r}
length(diff_genes_list_0.05)
```




#Final list of differentially expressed genes with an adjusted p-valuelower than 0.01:

```{r}
diff_genes_ann_0.01 = embryo_df_tissues[embryo_df_tissues$ensembl.transcript %in% diff_genes_list_0.01, ]
save(diff_genes_ann_0.01, file = "diff_genes_ann_0.01.RData")
```

```{r}
diff_genes_ann_0.01
```






