---
title: "Skript4"
author: "Joshua Eigenmann"
date: '2022-06-07'
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

## Load useful Libraries

```{r}
library(Rcpp)
library(tidyverse)
library(affy)
library(vsn)
library(AnnotationDbi)
#library(hgu133plus2hsenstprobe)
#library(hgu133plus2hsenstcdf) 
library(limma)
library(pheatmap)
library(hexbin)
library(ggplot2)
#library(biomaRt)
```

## Uses Annotated Dataset with only TRA Genes

Import the annotated dataset with only TRA genes. Information from the TRA data was merged

```{r}
#setwd("C:/Users/david/Documents/Data Analysis Lokal/2022-topic-04-team-01/R Scripts")
#setwd("F:/Data Analysis Project/Github Repo/2022-topic-04-team-01/R Scripts") #Paul
setwd("C:/Data_Analysis")#Joshua (Desktop)
#setwd("C:/Daten/Data Analysis Projekt/Github Repo/2022-topic-04-team-01/R Scripts")
setwd("/Users/joshuaeigenmann/Dokumente/Github Repo/2022-topic-04-team-01/R Scripts")

load("/Users/joshuaeigenmann/Documents/GitHub/2022-topic-04-team-01/R Scripts/embryo_df_tissues.RData") #Joshua Mac
load("/Users/joshuaeigenmann/Documents/GitHub/2022-topic-04-team-01/R Scripts/week9_4_diff_tra.RData") #Joshua Mac

load("C:/Data_Analysis/embryo_df_tissues.RData")


#load("embryo_df_tra.RData") #Joshua Windows
#load("diff_genes_ann_0.01.RData") #Joshua Windows
```

Overview over the dataset

```{r}
colnames(embryo_df_tra)
dim(embryo_df_tra)
head(embryo_df_tra)
str(embryo_df_tra)
```

```{r}
#sum(apply(embryo_df_tra, 1, fun))
```

## Changing the Tissues into a vector

```{r}
library(tidyverse)

#Make Tissue names more accessible
 a = separate(embryo_df_tra, tissues, into = as.character(c(1:30)), sep="/")

create_tissue_vector = function(f) {
  f = unlist(f)
  f = f[!is.na(f)]
  return(f)
}

tissues = apply(a[,31:60],1, create_tissue_vector)
tissues = sapply(tissues, unlist)

a = a[,c(1:30, 61)] 
a %>% add_column(tissues = NA)
a$tissues = tissues

embryo_df_tissues = a
```

```{r}
save(embryo_df_tissues, file="embryo_df_tissues.RData")
```

## Modify expression data from Week9 to 4 comparison with accessible tissue vector

```{r}
setwd("C:/Daten/Data Analysis Projekt/Github Repo/2022-topic-04-team-01/R Scripts")
#setwd("C:/Data_Analysis")
load("week9_4_diff.RData")

week9_4_diff_tra = embryo_df_tissues[embryo_df_tissues$ensembl.transcript %in% tra_diff$ensembl.transcript, ]

save(week9_4_diff_tra, file="week9_4_diff_tra.RData")

```

## Limma Analysis for all different Combinations

```{r}
library(limma)

#Differential gene expression: Limma Analysis 2, between weeks

limma_data = as.data.frame(embryo_df_tissues[1:18], rownames = embryo_df_tissues$ensembl.transcript)

weeks = c("week4", "week5", "week6", "week7", "week8", "week9")
f = factor(c("week4", "week4", "week4", "week5", "week5", "week5", "week6", "week6", "week6", "week7", "week7", "week7", "week8", "week8", "week8", "week9", "week9", "week9"), levels =weeks)
design = model.matrix(~0+f)
colnames(design) =  weeks
fit_2 = lmFit(limma_data, design)

contrast_all = makeContrasts(week9-week8, week9-week7, week9-week6, week9-week5, week9-week4, week8-week7,week8-week6, week8-week5, week8-week4, week7-week6, week7-week5, week7-week4, week6-week5, week6-week4, week5-week4, levels = design)
fit_all = contrasts.fit(fit_2, contrast_all)
fit_all = eBayes(fit_all)
results_all = decideTests(fit_all)
summary(results_all)

```

### Do not use this block, only a single comparison for test purposes

```{r}
# Ausgabe mit Gennamen für einen Vergleich

top.table = topTable(fit_all,1, n = Inf)
head(top.table)
 n = sapply(rownames(top.table), function(x){
  embryo_df_tissues$ensembl.transcript[as.numeric(x)]
})
row.names(top.table) = n
top.table[which(top.table$adj.P.Val < 0.01), ]
```

```{r}
#Create list of all differentially expresssed genes
diff_genes_list_0.01 = c()

for (i in 1:15) {
 top.table = topTable(fit_all,i, n = Inf)
 n = sapply(rownames(top.table), function(x){
  embryo_df_tissues$ensembl.transcript[as.numeric(x)]
  })
  row.names(top.table) = n
  diff_genes_list_0.01 = c(diff_genes_list_0.01, rownames(top.table[which(top.table$adj.P.Val < 0.01), ]))
}

```

```{r}

length(diff_genes_list_0.01) #4538
diff_genes_list_0.01 = diff_genes_list_0.01[!duplicated(diff_genes_list_0.01)]
length(diff_genes_list_0.01) #1818

```

```{r}
#List with 5%

diff_genes_list_0.05 = c()

for (i in 1:15) {
 top.table = topTable(fit_all,i, n = Inf)
 n = sapply(rownames(top.table), function(x){
  embryo_df_tissues$ensembl.transcript[as.numeric(x)]
  })
  row.names(top.table) = n
  diff_genes_list_0.05 = c(diff_genes_list_0.05, rownames(top.table[which(top.table$adj.P.Val < 0.05), ]))
}
diff_genes_list_0.05 = diff_genes_list_0.05[!duplicated(diff_genes_list_0.05)]

```

### Final list of differentially expressed genes with an adjusted p-valuelower than 0.01:

```{r}
#Select differential genes from annotated dataframe
diff_genes_ann_0.01 = embryo_df_tissues[embryo_df_tissues$ensembl.transcript %in% diff_genes_list_0.01, ]
save(diff_genes_ann_0.01, file = "diff_genes_ann_0.01.RData")
```

## Test for single differential gene expression Graph: Frontal Cortex

```{r}
#Select genes only of target tissue
#chunk in console ausführen
bool_contained = sapply(diff_genes_ann_0.01$tissues, function(x) {"Brain - Frontal Cortex" %in% x})
diff_genes_ann_0.01[bool_contained,]

```

```{r}
#Graph of mean gene expression over time
frontal_cortex = diff_genes_ann_0.01[bool_contained,]
frontal_cortex = frontal_cortex[, 1:19]
frontal_cortex_mean = data.frame(week4=rowMeans(frontal_cortex[,1:3]),week5=rowMeans(frontal_cortex[,4:6]), week6=rowMeans(frontal_cortex[,7:9]),week7=rowMeans(frontal_cortex[,10:12]), week8=rowMeans(frontal_cortex[,13:15]), week9=rowMeans(frontal_cortex[,16:18]))
frontal_cortex_graph = colMeans(frontal_cortex_mean)
frontal_cortex_graph = data.frame (Expression_Level = frontal_cortex_graph, Week = c(4:9))

frontal_cortex_plot = ggplot(frontal_cortex_graph, aes(x = Week, y= Expression_Level))  + geom_line(color = "#0000ff") + geom_point(color = "#0000ff") + labs(title = paste("Median expression level of all brain-related genes (GSE15744)"))  + theme_classic() 
  
```

```{r}
frontal_cortex_plot
frontal_cortex_mean
```

### Clustering of one tissue

```{r}
#New Dataset with median of gene expression
data_temp = diff_genes_ann_0.01[bool_contained,]
data_abs = data.frame(Week4 = apply(data_temp[1:3], 1, median), Week5 =  apply(data_temp[4:6], 1, median), Week6 =  apply(data_temp[7:9], 1, median), Week7 =  apply(data_temp[10:12], 1, median), Week8 =  apply(data_temp[13:15], 1, median), Week9 =  apply(data_temp[16:18], 1, median))
row.names(data_abs) = data_temp$ensembl.transcript
```

```{r}
# New dataset with difference in gene expression

expression_difference = function(x,y) {
  return ((x-y)/((x+y)*0.5))
}

data = data.frame("5_4" = expression_difference(data_abs[,"Week5"], data_abs[,"Week4"]), 
                  "6_5" = expression_difference(data_abs[,"Week6"], data_abs[,"Week5"]),
                  "7_6" = expression_difference(data_abs[,"Week7"], data_abs[,"Week6"]),
                  "8_7" = expression_difference(data_abs[,"Week8"], data_abs[,"Week7"]),
                  "9_8" = expression_difference(data_abs[,"Week9"], data_abs[,"Week8"]))
row.names(data) = rownames(data_abs)
```

```{r}
# K-Means für die Werte
library(factoextra)
library(cluster)
library(stringr)
library(ggplot2)
library(ggsci)

# Daten Skalieren, Distance Matrix
data = scale(data)
dist_data = get_dist(data, method = "pearson")

# Find optimal cluster number by silhouette value
avg_sil = function(k) {
  km = kmeans(data, centers = k, nstart = 25)
  ss = silhouette(km$cluster, dist_data)
  mean(ss[, 3])
}
n_cluster = which.max(sapply(c(2:10), function(k) {avg_sil(k)})) +1

#Plot with clustered genes

km = kmeans(data, centers = n_cluster, nstart = 25)
data_plot_raw = cbind(data_abs, Cluster = km$cluster)

create_plot_data = function(dataset) {
  data_plot = data.frame(Week = c(sapply(c(4:9), function(x) {rep(x, n_cluster)})), Cluster = as.character(c(rep(c(1:n_cluster), 6))), Expression = NA)
  count = 0
  for(s in 4:9) {
    for(t in 1:n_cluster) {
      count = count + 1
      values = dataset[dataset$Cluster == t, str_glue("Week", s)]
      data_plot[count, "Expression"] = median(values)
      data_plot[count, "Cluster"] = str_glue(data_plot[count, "Cluster"], " [", km$size[t], " genes]")
   }
  }
  return(data_plot)
}
plot_data = create_plot_data(data_plot_raw)
ggplot(plot_data, aes(x = Week, y= Expression, group = Cluster))  + geom_line(aes(colour = Cluster)) + geom_point(aes(colour = Cluster, shape = Cluster)) + labs(title = paste("Expression of ", "Brain - Frontal Cortex", " genes", sep="")) + scale_color_npg() + theme_classic()
```

## Integrated function for 12 different plots

```{r}
library(factoextra)
library(cluster)
library(stringr)
library(ggplot2)
library(ggsci)
library(grid)
library(gridExtra)

#setwd("C:/Daten/Data Analysis Projekt/Github Repo/2022-topic-04-team-01/R Scripts")
load("diff_genes_ann_0.01.RData")

# List the tissues
tissues = c("Brain - Amygdala", "Brain - Anterior cingulate cortex", "Brain - Caudate", "Brain - Cerebellar Hemisphere", "Brain - Cerebellum", "Brain - Cortex", "Brain - Frontal Cortex", "Brain - Hippocampus", "Brain - Hypothalamus", "Brain - Nucleus accumbens", "Brain - Putamen", "Brain - Spinal cord", "Brain - Substantia nigra")

# Functions for later use
expression_difference = function(x,y) {
  return ((x-y)/((x+y)*0.5))
}

avg_sil = function(k, data) {
  km = kmeans(data, centers = k, nstart = 25)
  ss = silhouette(km$cluster, dist_data)
  mean(ss[, 3])
}

create_plot_data = function(dataset) {
  data_plot = data.frame(Week = c(sapply(c(4:9), function(x) {rep(x, n_cluster)})), Cluster = as.character(c(rep(c(1:n_cluster), 6))), Expression = NA)
  count = 0
  for(s in 4:9) {
    for(t in 1:n_cluster) {
      count = count + 1
      values = dataset[dataset$Cluster == t, str_glue("Week", s)]
      data_plot[count, "Expression"] = median(values)
      data_plot[count, "Cluster"] = str_glue(data_plot[count, "Cluster"], " [", km$size[t], " genes]")
   }
  }
  return(data_plot)
}


for (tissue in tissues) {
  # Select an modify the data used for kmeans
  bool_contained = sapply(diff_genes_ann_0.01$tissues, function(x) {tissue %in% x})
  data_temp = diff_genes_ann_0.01[bool_contained,]
  data_abs = data.frame(Week4 = apply(data_temp[1:3], 1, median), Week5 =  apply(data_temp[4:6], 1, median), Week6 =  apply(data_temp[7:9], 1, median),
                        Week7=apply(data_temp[10:12], 1, median), Week8 =  apply(data_temp[13:15], 1, median), Week9 =  apply(data_temp[16:18], 1, median))
  data = data.frame("5_4" = expression_difference(data_abs[,"Week5"], data_abs[,"Week4"]), 
                  "6_5" = expression_difference(data_abs[,"Week6"], data_abs[,"Week5"]),
                  "7_6" = expression_difference(data_abs[,"Week7"], data_abs[,"Week6"]),
                  "8_7" = expression_difference(data_abs[,"Week8"], data_abs[,"Week7"]),
                  "9_8" = expression_difference(data_abs[,"Week9"], data_abs[,"Week8"]))
  row.names(data) = rownames(data_abs)
  
  # Select optimum cluster number
  data = scale(data)
  dist_data = get_dist(data, method = "pearson")
  n_cluster = which.max(sapply(c(2:10), function(k) {avg_sil(k, data)})) +1
  
  # Create the data for our plot
  km = kmeans(data, centers = n_cluster, nstart = 25)
  data_plot_raw = cbind(data_abs, Cluster = km$cluster)
  plot_data = create_plot_data(data_plot_raw)
  
  # Create the Plot
  assign(str_glue(tissue, "_plot"), ggplot(plot_data, aes(x = Week, y= Expression, group = Cluster))  + geom_line(aes(colour = Cluster)) +
           geom_point(aes(colour = Cluster, shape = Cluster)) + labs(title = paste("Expression of ", tissue, " genes", sep="")) +
           scale_color_npg() + theme_classic() )
}
ff = sapply(tissues, function(x){str_glue(x, "_plot")})
brain_table = grid.arrange(get(ff[1]), get(ff[2]), get(ff[3]), get(ff[4]), get(ff[5]), get(ff[6]), get(ff[7]), get(ff[8]), get(ff[9]), get(ff[10]), get(ff[11]), get(ff[12]))
   
#Save the plot
dev.copy2pdf(file="C:/Data_Analysis/Clustering_Plots/Expression of tissues.pdf")
```

```{r}
# Get a tissue count for our final gene list

tissues = c("Adipose", "Artery", "Brain",  "Colon", "Esophagus", "Heart", "Kidney", "Liver", "Lung", "Muscle", "Skin", "Stomach", "Testis")
count_data = data.frame(tissue = tissues, count = NA)
for (tissue in tissues){
  
  count_data[count_data$tissue == tissue, "count"] = sum(bool_contained = sapply(diff_genes_ann_0.01$tissues, function(x) {grepl(tissue, toString(x))}))

}
count_data
```

## Ecactly the same with a cooler tissue selection

```{r}
library(factoextra)
library(cluster)
library(stringr)
library(ggplot2)
library(ggsci)
library(grid)
library(gridExtra)

#setwd("C:/Daten/Data Analysis Projekt/Github Repo/2022-topic-04-team-01/R Scripts")
load("diff_genes_ann_0.01.RData")

# List the tissues
tissues = c("Adipose", "Artery", "Brain",  "Colon", "Esophagus", "Heart", "Kidney", "Liver", "Lung", "Muscle", "Skin", "Stomach", "Testis")


# Functions for later use
expression_difference = function(x,y) {
  return ((x-y)/((x+y)*0.5))
}

avg_sil = function(k, data) {
  km = kmeans(data, centers = k, nstart = 25)
  ss = silhouette(km$cluster, dist_data)
  mean(ss[, 3])
}

create_plot_data = function(dataset) {
  data_plot = data.frame(Week = c(sapply(c(4:9), function(x) {rep(x, n_cluster)})), Cluster = as.character(c(rep(c(1:n_cluster), 6))), Expression = NA)
  count = 0
  for(s in 4:9) {
    for(t in 1:n_cluster) {
      count = count + 1
      values = dataset[dataset$Cluster == t, str_glue("Week", s)]
      data_plot[count, "Expression"] = median(values)
      data_plot[count, "Cluster"] = str_glue(data_plot[count, "Cluster"], " [", km$size[t], " genes]")
   }
  }
  return(data_plot)
}


for (tissue in tissues) {
  # Select an modify the data used for kmeans
  bool_contained = sapply(diff_genes_ann_0.01$tissues, function(x) {grepl(tissue, toString(x))})
  data_temp = diff_genes_ann_0.01[bool_contained,]
  data_abs = data.frame(Week4 = apply(data_temp[1:3], 1, median), Week5 =  apply(data_temp[4:6], 1, median), Week6 =  apply(data_temp[7:9], 1, median),
                        Week7=apply(data_temp[10:12], 1, median), Week8 =  apply(data_temp[13:15], 1, median), Week9 =  apply(data_temp[16:18], 1, median))
  data = data.frame("5_4" = expression_difference(data_abs[,"Week5"], data_abs[,"Week4"]), 
                  "6_5" = expression_difference(data_abs[,"Week6"], data_abs[,"Week5"]),
                  "7_6" = expression_difference(data_abs[,"Week7"], data_abs[,"Week6"]),
                  "8_7" = expression_difference(data_abs[,"Week8"], data_abs[,"Week7"]),
                  "9_8" = expression_difference(data_abs[,"Week9"], data_abs[,"Week8"]))
  row.names(data) = rownames(data_abs)
  
  # Select optimum cluster number
  data = scale(data)
  dist_data = get_dist(data, method = "pearson")
  n_cluster = which.max(sapply(c(2:10), function(k) {avg_sil(k, data)})) +1
  
  # Create the data for our plot
  km = kmeans(data, centers = n_cluster, nstart = 25)
  data_plot_raw = cbind(data_abs, Cluster = km$cluster)
  plot_data = create_plot_data(data_plot_raw)
  
  # Create the Plot
  assign(str_glue(tissue, "_plot"), ggplot(plot_data, aes(x = Week, y= Expression, group = Cluster))  + geom_line(aes(colour = Cluster)) +
           geom_point(aes(colour = Cluster, shape = Cluster)) + labs(title = paste("Expression of ", tissue, " genes", sep="")) +
           scale_color_npg() + theme_classic() )
   
}
ff = sapply(tissues, function(x){str_glue(x, "_plot")})
organs_table = grid.arrange(get(ff[1]), get(ff[2]), get(ff[3]), get(ff[4]), get(ff[5]), get(ff[6]), get(ff[7]), get(ff[8]), get(ff[9]), get(ff[10]), get(ff[11]), get(ff[12]))
```

## Data cleaning

```{r}
#einige Spalten sind Dublikate
glimpse(embryo_df_tissues) 
embryo_df_tissues <- embryo_df_tissues[,!names(embryo_df_tissues) %in% c("ensembl.gene", "ensembl.symbol", "ensembl.chrom", "max.tissue")] #removed duplicates
dim(embryo_df_tissues)

#Chromosome Spalte aufhübschen und unnötiges löschen
embryo_df_tissues$Chromosome.scaffold.name <- as.factor(embryo_df_tissues$Chromosome.scaffold.name) 

#Definition Embryo Test
embryo_test <- embryo_df_tra %>%
  mutate(tiss.fct = as.factor(max.tissue))

dim(embryo_test$Chromosome.scaffold.name)
embryo_df_tissues <- embryo_df_tissues %>%
  filter(Chromosome.scaffold.name != "CHR_HSCHR10_1_CTG2" & Chromosome.scaffold.name != "CHR_HSCHR11_1_CTG7"& Chromosome.scaffold.name!= "CHR_HSCHR15_4_CTG8"& Chromosome.scaffold.name!= "CHR_HSCHR22_1_CTG3"& Chromosome.scaffold.name!= "CHR_HSCHR22_1_CTG7"& Chromosome.scaffold.name!=  "CHR_HSCHR4_6_CTG12"& Chromosome.scaffold.name!= "CHR_HSCHR7_2_CTG4_4"& Chromosome.scaffold.name!=  "CHR_HSCHR7_2_CTG6" )
embryo_df_tissues$Chromosome.scaffold.name <- factor(embryo_df_tissues$Chromosome.scaffold.name, levels = c("1", "2","3","4","5","6","7","8","9","10","11","12","13","14","15","16","17","18","19", "20", "21","22", "X","Y"))

# Nach NA values suchen
sum(is.na(embryo_df_tissues$ensembl.entrez)) # diese Spalte enthält 501 NA - Values

embryo_test$tissues <- sapply(embryo_test$tissues, as.factor)

embryo_test %>%
  filter(tissues == str_extract(tissues, "Brain"))

summary(embryo_test$tiss.fct)
levels(embryo_test$tiss.fct)
head(embryo_df_tissues$tissues)

glimpse(diff_genes_ann_0.01)

diff_genes_ann_0.01 %>%
  count(ensembl.transcript) %>%
  filter(n>1) # Es sind keine Dublikate mehr vorhanden

summary(as.factor(diff_genes_ann_0.01$max.tissue)) # von jedem tissue existiert noch min 1 differentiell exprimiertes Gen

```

```{r}
#pca ueber spalten (wochen)
pca_wochen <- prcomp(diff_genes_ann_0.01[,1:18],center=TRUE, scale.=TRUE)
summary(pca_wochen)
pca_wochen_df <- data.frame(pca_wochen$x[, 1:2])
```

```{r}
install.packages("devtools")
library(devtools)
install_github("vqv/ggbiplot")
library(ggbiplot)
install.packages("psych")
library(psych)
```

```{r}
#plotting pca_wochen
ggbiplot(pca_wochen, groups = diff_genes_ann_0.01$max.tissue) #groese anpassen damit plot sichtbar
ggbiplot(pca_wochen, labels=colnames(diff_genes_ann_0.01)[1:18])
ggbiplot(pca_wochen, choices=c(2,3))
ggbiplot(pca_wochen, choices=c(3,4))
#ggbiplot(pca_wochen,labels=diff_genes_ann_0.01['max.tissue']) #nicht lesbar
#ggbiplot(pca_wochen,groups=c(diff_genes_ann_0.01['max.tissue']))

#ggplot
pca_wochen_plot <- ggplot(
  pca_wochen_df,
  aes(
    x = PC1,
    y = PC2,
    #color=c(diff_genes_ann_0.01['max.tissue'])
  )
) +
  geom_point() + # Plot individual points to make a scatterplot
  theme_classic() # Format as a classic-looking plot with no gridlines

# Print out the plot here
pca_wochen_plot
```

```{r}
pairs.panels(pca_wochen$x,
             gap=0,
             bg = c("red", "yellow", "blue"),
             pch=21)
```

```{r}
#pca ueber zeilen (genes)
pca_genes <- prcomp(
  t(diff_genes_ann_0.01[,1:18]), # transpose our data frame to obtain PC scores for samples, not genes
  scale. = TRUE # we want the data scaled to have unit variance for each gene
)
summary(pca_genes)
pca_genes_df <- data.frame(pca_genes$x[, 1:2])
```

```{r}
#plotting pca_genes
ggbiplot(pca_genes)
ggbiplot(pca_genes, choices=c(2,3))
ggbiplot(pca_genes, choices=c(3,4))


#ggplot
pca_genes_plot <- ggplot(
  pca_genes_df,
  aes(
    x = PC1,
    y = PC2,
    
  )
) +
  geom_point() + # Plot individual points to make a scatterplot
  theme_classic() # Format as a classic-looking plot with no gridlines

# Print out the plot here
pca_genes_plot
```

```{r}
head(diff_genes_ann_0.01)
dim(diff_genes_ann_0.01)
```

## Limma analysis

```{r}
# Zeile 598 bis 643 is Müll -> Ignorieren!!
load("/Users/joshuaeigenmann/Downloads/embryo_df_tra.RData") #Joshua(mac)
load("C:/Users/Eigenmann/Documents/GitHub/2022-topic-04-team-01/R Scripts/embryo_df_tra.RData") # Joshua(windows)

data.before.limma <- embryo_df_tra
data.before.limma <- data.frame(data.before.limma[,1:18]) 

```

# Limma: Radiation exposed vs Radiation not exposed

# No genes found for p\<0.01

```{r}
data.limma = lmFit(data.before.limma)
data.limma = eBayes(data.limma)
data.limma.subfit <- cbind(data.limma$coefficients,
                          data.limma$lods,
                          data.limma$p.value,
                          p.adjust(data.limma$p.value, "BH")
                          )
colnames(data.limma.subfit) <- c("coef", "Bstat", "P", "P.Adjust")

data.limma.pvalue.01 <- sum(data.limma.subfit[,4] < 0.01) 
data.limma.pvalue.05 <- sum(data.limma.subfit[,4] < 0.05) 
data.limma.pvalue.1 <- sum(data.limma.subfit[,4] < 0.1) 
data.limma.pvalue.5 <- sum(data.limma.subfit[,4] < 0.5) 
data.limma.pvalue.01
```

## Top differential genes: p\<0.1 & fc\>2

# Radiation exposed vs Radiation not exposed

# No genes found for

```{r}
topTable(data.limma)
data.limma.genes.fc2 <- topTable(data.limma, number = data.limma.pvalue.01, lfc = 1) 
# fc(x) = 2 ; log2(fc(x)) = log2(2) ; log2fc = 1
data.limma.p01andfc2 <- data.limma.genes.fc2[data.limma.genes.fc2$adj.P.Val < 0.01,]
data.limma.genes.all <- topTable(data.limma, number = length(data.limma$coefficients))
head(data.limma.p01andfc2)
```

# Limma Analyse von oben

```{r}
library(limma)

#Differential gene expression: Limma Analysis 2, between weeks

limma_data = as.data.frame(embryo_df_tissues[1:18], rownames = embryo_df_tissues$ensembl.transcript)

weeks = c("week4", "week5", "week6", "week7", "week8", "week9")
f = factor(c("week4", "week4", "week4", "week5", "week5", "week5", "week6", "week6", "week6", "week7", "week7", "week7", "week8", "week8", "week8", "week9", "week9", "week9"), levels =weeks)
design = model.matrix(~0+f)
colnames(design) =  weeks
fit_2 = lmFit(limma_data, design)

contrast_all = makeContrasts(week9-week8, week9-week7, week9-week6, week9-week5, week9-week4, week8-week7,week8-week6, week8-week5, week8-week4, week7-week6, week7-week5, week7-week4, week6-week5, week6-week4, week5-week4, levels = design)
fit_all = contrasts.fit(fit_2, contrast_all)
fit_all = eBayes(fit_all)
results_all = decideTests(fit_all)
summary(results_all)

# Ausgabe der Toptabel vom Vergleich Woche 4/9

top.table = topTable(fit_all,5, n = Inf)
head(top.table)
# n = sapply(rownames(top.table), function(x){
#  embryo_df_tissues$ensembl.transcript[as.numeric(x)]
#})
#row.names(top.table) = n
#top.table[which(top.table$adj.P.Val < 0.01), ]
```

## Volcano plot

\#<https://biocorecrg.github.io/CRG_RIntroduction/volcano-plots.html>

```{r}
# Volcano nach Internetanleitung

# The RDS format is used to save a single R object to a file, and to restore it.
# Extract that object in the current session:
tmp <- top.table

# remove rows that contain NA values
de <- tmp[complete.cases(tmp), ]

# The basic scatter plot: x is "log2FoldChange", y is "pvalue"
ggplot(data=de, aes(x=logFC, y=adj.P.Val)) + geom_point()

# Convert directly in the aes()
p <- ggplot(data=de, aes(x=llogFC, y=-log10(adj.P.Val))) + geom_point()
# Add more simple "theme"
p <- ggplot(data=de, aes(x=logFC, y=-log10(adj.P.Val))) + geom_point() + theme_minimal()
# Add vertical lines for log2FoldChange thresholds, and one horizontal line for the p-value threshold 
p2 <- p + geom_vline(xintercept=c(-0.5, 0.5), col="red") +
    geom_hline(yintercept=-log10(0.01), col="red")
# The significantly differentially expressed genes are the ones found in the upper-left and upper-right corners.
# Add a column to the data frame to specify if they are UP- or DOWN- regulated (log2FoldChange respectively positive or negative)

# add a column of NAs
de$diffexpressed <- "NO"
# if log2Foldchange > 0.6 and pvalue < 0.05, set as "UP" 
de$diffexpressed[de$logFC > 0.5 & de$adj.P.Val < 0.01] <- "UP"
# if log2Foldchange < -0.6 and pvalue < 0.05, set as "DOWN"
de$diffexpressed[de$logFC < -0.5 & de$adj.P.Val < 0.01] <- "DOWN"

# Re-plot but this time color the points with "diffexpressed"
p <- ggplot(data=de, aes(x=logFC, y=-log10(adj.P.Val), col=diffexpressed)) + geom_point() + theme_minimal()

# Add lines as before...
p2 <- p + geom_vline(xintercept=c(-0.5, 0.5), col="red") +
        geom_hline(yintercept=-log10(0.01), col="red")
## Change point color 

# 1. by default, it is assigned to the categories in an alphabetical order):
p3 <- p2 + scale_color_manual(values=c("blue", "black", "red"))

# 2. to automate a bit: ceate a named vector: the values are the colors to be used, the names are the categories they will be assigned to:
mycolors <- c("blue", "red", "black")
names(mycolors) <- c("DOWN", "UP", "NO")
p3 <- p2 + scale_colour_manual(values = mycolors)
# Now write down the name of genes beside the points...
# Create a new column "delabel" to de, that will contain the name of genes differentially expressed (NA in case they are not)
de$delabel <- "no"
#de$delabel[de$diffexpressed != "NO"] <- de$gene_symbol[de$diffexpressed != "NO"]

volcano <- ggplot(data=de, aes(x=logFC, y=-log10(adj.P.Val), col=diffexpressed, label=delabel)) + 
    geom_point() + 
    theme_minimal() +
    geom_text()

print(volcano)
pdf("C:/Users/Eigenmann/Documents/GitHub/2022-topic-04-team-01/R Scripts/volcano.pdf", height = 10, width = 10) #Joshua(windows)
#pdf("/Users/joshuaeigenmann/Documents/GitHub/2022-topic-04-team-01/R Scripts/volcano.pdf", height = 10, width = 10) #Joshua(mac)

# Finally, we can organize the labels nicely using the "ggrepel" package and the geom_text_repel() function
# load library
library(ggrepel)
# plot adding up all layers we have seen so far
volcano2 <- ggplot(data=top.table, aes(x=logFC, y=-log10(adj.P.Val), col=de$diffexpressed, label=de$delabel)) +
        geom_point() + 
        theme_minimal() +
        geom_text_repel() +
        scale_color_manual(values=c("blue", "black", "red")) +
        geom_vline(xintercept=c(-0.5, 0.5), col="red") +
        geom_hline(yintercept=-log10(0.01), col="red")

print(volcano2)
pdf("C:/Users/Eigenmann/Documents/GitHub/2022-topic-04-team-01/Plots/Volcano/volcano2.pdf", height = 10, width = 10) #Joshua(windows)
#pdf("/Users/joshuaeigenmann/Documents/GitHub/2022-topic-04-team-01/Plots/Volcano/volcano2.pdf", height = 10, width = 10) #Joshua(mac)


```

### 12. Volcano plot

# Help: <https://biocorecrg.github.io/CRG_RIntroduction/volcano-plots.html>

# Volcanizator function

```{r}
volcanizator <- function(data, logFC_thresh_upper, logFC_thresh_lower, max_pValue){
  
  # Point colours
  mycolours <- c("blue", "red", "black")
  names(mycolours) <- c("DOWN", "UP", "NO")
  
  # Add a column to the data frame to specify if they are UP- or DOWN- regulated (log2FoldChange respectively positive or negative)
  data$diffexpressed <- "NO"
  data$diffexpressed[data$logFC > logFC_thresh_upper & data$adj.P.Val < max_pValue] <- "UP"
  data$diffexpressed[data$logFC < logFC_thresh_lower & data$adj.P.Val < max_pValue] <- "DOWN"
  
  # Create a new column with names of differential expressed genes
  data$delabel <- NA
  #data$delabel[data$diffexpressed != "NO"] <- data$HGNC.symbol[data$diffexpressed != "NO"]
  
  # Plot
  ggplot(data, aes(x=logFC, y=-log10(adj.P.Val))) +
    geom_point(aes(colour = diffexpressed), size=0.9) +
    theme_minimal() +
    geom_text_repel(aes(label = delabel)) + #, force = 5, point.padding	= unit(5, "cm")
    geom_vline(xintercept=c(logFC_thresh_lower, logFC_thresh_upper), col="red") + # log2FoldChange thresholds
    geom_hline(yintercept=-log10(max_pValue), col="red") + # p-value threshold
    scale_colour_manual(values = mycolours) +
    geom_text(aes(label = delabel)) +
    labs(title = paste("Differential gene expression: week 4 to 9"),
         subtitle = "GEO dataset: GSE15744",
         caption = "p Value adjusted: Benjamini & Hochberg (1995)"
         ) +
    xlab("log2FC") +
    ylab("-log10(adj.P.Val)")
}
volcano3 <- volcanizator(top.table, 1, -1, 0.01)
volcano3
pdf("C:/Users/Eigenmann/Documents/GitHub/2022-topic-04-team-01/Plots/Differential_gene_expression(4-9).pdf", height = 10, width = 10)
pdf("C:/Users/Eigenmann/Documents/GitHub/2022-topic-04-team-01/Plots/Differential_gene_expression(4-9).pdf") #Joshua(windows)
#pdf("/Users/joshuaeigenmann/Documents/GitHub/2022-topic-04-team-01/R Scripts/volcano3.pdf", height = 10, width = 10) #Joshua(mac)
```

# Save the plots

```{r}
pdf("/Users/joshuaeigenmann/Documents/GitHub/2022-topic-04-team-01/R Scripts/volcano.pdf", height = 10, width = 10)
pdf("/Users/joshuaeigenmann/Documents/GitHub/2022-topic-04-team-01/R Scripts/volcano2.pdf", height = 10, width = 10)
pdf("/Users/joshuaeigenmann/Documents/GitHub/2022-topic-04-team-01/R Scripts/volcano3.pdf", height = 10, width = 10)


print(volcano)
print(volcano2)
print(volcano3)
```

```{r}
#load("/Users/joshuaeigenmann/Documents/GitHub/2022-topic-04-team-01/R Scripts/week9-4_diff.RData")
#week9_4_diff_tra bereits heruntergeladen
#week9_4_diff_tra = embryo_df_tissues[embryo_df_tissues$ensembl.transcript %in% tra_diff$ensembl.transcript, ]
#same
# load("/Users/joshuaeigenmann/Documents/GitHub/2022-topic-04-team-01/R Scripts/week9_4_diff_tra.RData")
load("C:/Users/Eigenmann/Documents/GitHub/2022-topic-04-team-01/R Scripts/week9_4_diff_tra.RData")
data.diff =week9_4_diff_tra
```

## Heatmap

```{r}
# Heatmap für Correlation zwischen den Wochen

#Installation der nötigen Pakete
install.packages("pheatmap")
library(pheatmap)

# Heatmap embryo_df_tra
correlation.mat = cor(embryo_df_tra[,1:18], embryo_df_tra[,1:18], method = "pearson")
correlation.mat
pheatmap(correlation.mat, main = "Correlations between diffrent week differences")

# Heatmap week9_4_diff_tra
correlation.mat.2 = cor(week9_4_diff_tra[,1:18], week9_4_diff_tra[,1:18])
correlation.mat.2
pheatmap(correlation.mat.2)

# Heatmap für Correlation zwischen dem Median der Wochen

#Aufstellen einer Matrix mit dem Median der Wochen: embryo_df_tra_modifi
embryo_df_tra_new = embryo_df_tra[,1:18]
embryo_df_tra_new_4 =embryo_df_tra_new[,1:3]
embryo_df_tra_new_4 = apply(embryo_df_tra_new_4,1,median)
embryo_df_tra_new_5 =embryo_df_tra_new[,4:6]
embryo_df_tra_new_5 = apply(embryo_df_tra_new_5,1,median)
embryo_df_tra_new_6 =embryo_df_tra_new[,7:9]
embryo_df_tra_new_6 = apply(embryo_df_tra_new_6,1,median)
embryo_df_tra_new_7 =embryo_df_tra_new[,10:12]
embryo_df_tra_new_7 = apply(embryo_df_tra_new_7,1,median)
embryo_df_tra_new_8 =embryo_df_tra_new[,13:15]
embryo_df_tra_new_8 = apply(embryo_df_tra_new_8,1,median)
embryo_df_tra_new_9 =embryo_df_tra_new[,16:18]
embryo_df_tra_new_9 = apply(embryo_df_tra_new_9,1,median)
embryo_df_tra_modifi =cbind(embryo_df_tra_new_4,embryo_df_tra_new_5,embryo_df_tra_new_6,embryo_df_tra_new_7,embryo_df_tra_new_8,embryo_df_tra_new_9)
colnames(embryo_df_tra_modifi) <- c('week_4','week_5','week_6','week_7','week_8','week_9')
embryo_df_tra_modifi

# Heatmap embryo_df_tra_modifi
correlation.mat_median = cor(embryo_df_tra_modifi, embryo_df_tra_modifi, method = "pearson")
correlation.mat_median
pheatmap(correlation.mat_median, main = "Correlations between diffrent week differences - whole data set")
pdf("/Users/joshuaeigenmann/Documents/GitHub/2022-topic-04-team-01/R Scripts/Correlations between diffrent week differences - whole data set.pdf", height = 10, width = 10)

#Aufstellen einer Matrix mit dem Median der Wochen: week9_4_diff_tra_modifi
week9_4_diff_tra_new = week9_4_diff_tra[,1:18]
week9_4_diff_tra_new_4 =week9_4_diff_tra_new[,1:3]
week9_4_diff_tra_new_4 = apply(week9_4_diff_tra_new_4,1,median)
week9_4_diff_tra_new_5 =week9_4_diff_tra_new[,4:6]
week9_4_diff_tra_new_5 = apply(week9_4_diff_tra_new_5,1,median)
week9_4_diff_tra_new_6 =week9_4_diff_tra_new[,7:9]
week9_4_diff_tra_new_6 = apply(week9_4_diff_tra_new_6,1,median)
week9_4_diff_tra_new_7 =week9_4_diff_tra_new[,10:12]
week9_4_diff_tra_new_7 = apply(week9_4_diff_tra_new_7,1,median)
week9_4_diff_tra_new_8 =week9_4_diff_tra_new[,13:15]
week9_4_diff_tra_new_8 = apply(week9_4_diff_tra_new_8,1,median)
week9_4_diff_tra_new_9 =week9_4_diff_tra_new[,16:18]
week9_4_diff_tra_new_9 = apply(week9_4_diff_tra_new_9,1,median)
week9_4_diff_tra_modifi =cbind(week9_4_diff_tra_new_4,week9_4_diff_tra_new_5,week9_4_diff_tra_new_6,week9_4_diff_tra_new_7,week9_4_diff_tra_new_8,week9_4_diff_tra_new_9)
colnames(week9_4_diff_tra_modifi) <- c('week_4','week_5','week_6','week_7','week_8','week_9')
week9_4_diff_tra_modifi

# Heatmap week9_4_diff_tra_modifi
correlation.mat_median.2 = cor(week9_4_diff_tra_modifi, week9_4_diff_tra_modifi, method = "pearson")
correlation.mat_median.2
pheatmap(correlation.mat_median.2, main = "Correlations between diffrent week differences")
pdf("/Users/joshuaeigenmann/Documents/GitHub/2022-topic-04-team-01/R Scripts/Correlations between diffrent week differences.pdf", height = 10, width = 10)
```

# Heatmap for tissues

```{r}
library(mgcv)
data.liver = as.data.frame(embryo_df_tissues %>%
                             filter(tissues == str_extract(tissues, "Liver")))
data.liver = t(data.liver[1:18])
dim(data.liver)
data.kidney = as.data.frame(embryo_df_tissues %>%filter(tissues == str_extract(tissues, "Kidney - Cortex")))
data.kidney =t(data.kidney[1:18])
dim(data.kidney)
correlation.matrix = cor(data.liver, data.kidney, method = "pearson")
pheatmap(correlation.matrix, main = "Correlations between liver and kidney - cortex")
pdf("/Users/joshuaeigenmann/Documents/GitHub/2022-topic-04-team-01/R Scripts/Correlations between liver and kidney.pdf", height = 10, width = 10)
```

```{r}
#Clustering embryo_df_tra
#Laden nötiger Librarys
install.packages("factoextra")
library("factoextra")
install.packages("pheatmap")
library(pheatmap)
library(cluster)
#Distanz Matrix
distanz = as.dist(correlation.mat)
pheatmap(distanz)
#Dendrogram
hclust <- hclust(distanz,method = "complete")
as.dendrogram(hclust)
plot(hclust)
#Darstellen verschiedener Cluster
lapply(2:10,function(k) {
  km = kmeans(correlation.mat, k, nstart = 25)
fviz_cluster(km, data = correlation.mat,
             ggtheme = theme_minimal(),
             geom='point',
             main = "Partitioning Clustering Plot"
             )
})
## Überprüfen der optimalen Clusteranzahl
#mit der Silhouette-Methode
lapply(2:10,function(k) {
  km = kmeans(correlation.mat, k, nstart = 25)
s = silhouette(km$cluster,distanz)
plot(s)
})
#Optimale Anzahl -> 2 Cluster
#Mit der WSS-Methode
wss = sapply(2:10,function(k) { 
  kmeans(x= correlation.mat, centers = k)$tot.withinss
})
plot(2:10,wss,type='b',pch=19,xlab="Number of clusters K",
     ylab="Total within-clusters sum of squares")
#Kein klarer Knick erkennbar
#-> mit fviz_nbclust
fviz_nbclust(correlation.mat, kmeans, method = "wss")
#Optimale Anzahl -> 2 Cluster [da hier Knick]
```

```{r}
#Clustering embryo_df_tra_modifi
#Laden nötiger Librarys
install.packages("factoextra")
library("factoextra")
install.packages("pheatmap")
library(pheatmap)
library(cluster)
#Distanz Matrix
distanz = as.dist(correlation.mat_median)
pheatmap(distanz)
#Dendrogram
hclust <- hclust(distanz,method = "complete")
as.dendrogram(hclust)
plot(hclust)
#Darstellen verschiedener Cluster
lapply(2:5,function(k) {
  km = kmeans(correlation.mat_median, k, nstart = 25)
fviz_cluster(km, data = correlation.mat_median,
             ggtheme = theme_minimal(),
             geom='point',
             main = "Partitioning Clustering Plot"
             )
})
## Überprüfen der optimalen Clusteranzahl
#mit der Silhouette-Methode
lapply(2:5,function(k) {
  km = kmeans(correlation.mat_median, k, nstart = 25)
s = silhouette(km$cluster,distanz)
plot(s)
})
#Optimale Anzahl -> 5 Cluster
#Mit der WSS-Methode
wss = sapply(2:5,function(k) { 
  kmeans(x= correlation.mat_median, centers = k)$tot.withinss
})
plot(2:5,wss,type='b',pch=19,xlab="Number of clusters K",
     ylab="Total within-clusters sum of squares")
# Knick erkennbar bei 2 Cluster
```

```{r}
#Clustering week9_4_diff_tra
#Laden nötiger Librarys
install.packages("factoextra")
library("factoextra")
install.packages("pheatmap")
library(pheatmap)
library(cluster)
#Distanz Matrix
distanz = as.dist(correlation.mat.2)
pheatmap(distanz)
#Dendrogram
hclust <- hclust(distanz,method = "complete")
as.dendrogram(hclust)
plot(hclust)
#Darstellen verschiedener Cluster
lapply(2:10,function(k) {
  km = kmeans(correlation.mat.2, k, nstart = 25)
fviz_cluster(km, data = correlation.mat.2,
             ggtheme = theme_minimal(),
             geom='point',
             main = "Partitioning Clustering Plot"
             )
})
## Überprüfen der optimalen Clusteranzahl
#mit der Silhouette-Methode
lapply(2:10,function(k) {
  km = kmeans(correlation.mat.2, k, nstart = 25)
s = silhouette(km$cluster,distanz)
plot(s)
})
#Optimale Anzahl -> 2 Cluster
#Mit der WSS-Methode
wss = sapply(2:10,function(k) { 
  kmeans(x= correlation.mat.2, centers = k)$tot.withinss
})
plot(2:10,wss,type='b',pch=19,xlab="Number of clusters K",
     ylab="Total within-clusters sum of squares")
#Kein klarer Knick erkennbar
#-> mit fviz_nbclust
fviz_nbclust(correlation.mat.2, kmeans, method = "wss")
#Optimale Anzahl -> 2 Cluster [da hier Knick]
```

```{r}
#Clustering week9_4_diff_tra_modifi
#Laden nötiger Librarys
install.packages("factoextra")
library("factoextra")
install.packages("pheatmap")
library(pheatmap)
library(cluster)
#Distanz Matrix
distanz = as.dist(correlation.mat_median.2)
pheatmap(distanz)
#Dendrogram
hclust <- hclust(distanz,method = "complete")
as.dendrogram(hclust)
plot(hclust)
#Darstellen verschiedener Cluster
lapply(2:5,function(k) {
  km = kmeans(correlation.mat_median.2, k, nstart = 25)
fviz_cluster(km, data = correlation.mat_median.2,
             ggtheme = theme_minimal(),
             geom='point',
             main = "Partitioning Clustering Plot"
             )
})

## Überprüfen der optimalen Clusteranzahl
#mit der Silhouette-Methode
lapply(2:5,function(k) {
  km = kmeans(correlation.mat_median.2, k)
s = silhouette(km$cluster,distanz)
plot(s)
})
#Optimale Anzahl -> 5 Cluster

#Mit der WSS-Methode
wss = sapply(2:5,function(k) { 
  kmeans(x= correlation.mat_median.2, centers = k)$tot.withinss
})
plot(2:5,wss,type='b',pch=19,xlab="Number of clusters K",
     ylab="Total within-clusters sum of squares")

#Knick bei 2 Clustern erkennbar erkennbar
```

```{r}
load(file = "embryo_df_tra.RData")
head(embryo_df_tra)
```

### Run the algorithm for all the tissues

```{r}
library(factoextra)
library(cluster)
library(stringr)
library(ggplot2)
library(ggsci)
library(grid)
library(gridExtra)

#setwd("C:/Daten/Data Analysis Projekt/Github Repo/2022-topic-04-team-01/R Scripts")
load("diff_genes_ann_0.01.RData")

# List the tissues
#tissues = c("Adipose - Subcutaneous", "Adipose - Visceral", "Adipose", "Cells - Transformed fibroblasts", "Skin - Not Sun Exposed", "Skin - Sun Exposed", "Skin", "Muscle - Skeletal", "Bladder", "Kidney - Cortex") #"Kidney - Medulla", "Kidney"
#tissues = c("Colon - Sigmoid", "Colon - Transverse", "Colon", "Esophagus - Gastroesophageal Junction", "Esophagus - Mucosa", "Esophagus - Muscularis", "Esophagus", "Liver", "Minor Salivary Gland", "Pancreas", "Small Intestine - Terminal Ileum", "Spleen", "Stomach")
#tissues = c("Breast - Mammary Tissue", "Cervix - Ectocervix", "Cervix - Endocervix", "Cervix", "Fallopian Tube", "Ovary", "Uterus", "Vagina", "Prostate", "Testis", "Adrenal Gland", "Pituitary", "Thyroid")
tissues = c("Artery - Aorta", "Artery - Coronary", "Artery - Tibial", "Artery", "Heart - Atrial Appendage", "Heart - Left Ventricle", "Heart", "Lung", "Whole Blood", "Cells - EBV-transformed lymphocytes")
#tissues = c("Brain - Amygdala", "Brain - Anterior cingulate cortex", "Brain - Caudate", "Brain - Cerebellar Hemisphere", "Brain - Cerebellum", "Brain - Cortex", "Brain - Frontal Cortex", "Brain - Hippocampus", "Brain - Hypothalamus", "Brain - Nucleus accumbens", "Brain - Putamen", "Brain - Spinal cord", "Brain - Substantia nigra", "Brain", "Nerve - Tibial")



# Functions for later use
expression_difference = function(x,y) {
  return ((x-y)/((x+y)*0.5))
}

avg_sil = function(k, data) {
  km = kmeans(data, centers = k, nstart = 25)
  ss = silhouette(km$cluster, dist_data)
  mean(ss[, 3])
}

create_plot_data = function(dataset) {
  data_plot = data.frame(Week = c(sapply(c(4:9), function(x) {rep(x, n_cluster)})), Cluster = as.character(c(rep(c(1:n_cluster), 6))), Expression = NA)
  count = 0
  for(s in 4:9) {
    for(t in 1:n_cluster) {
      count = count + 1
      values = dataset[dataset$Cluster == t, str_glue("Week", s)]
      data_plot[count, "Expression"] = median(values)
      data_plot[count, "Cluster"] = str_glue(data_plot[count, "Cluster"], " [", km$size[t], " genes]")
   }
  }
  return(data_plot)
}


for (tissue in tissues) {
  # Select an modify the data used for kmeans
  bool_contained = sapply(diff_genes_ann_0.01$tissues, function(x) {grepl(tissue, toString(x))})
  data_temp = diff_genes_ann_0.01[bool_contained,]
  data_abs = data.frame(Week4 = apply(data_temp[1:3], 1, median), Week5 =  apply(data_temp[4:6], 1, median), Week6 =  apply(data_temp[7:9], 1, median),
                        Week7=apply(data_temp[10:12], 1, median), Week8 =  apply(data_temp[13:15], 1, median), Week9 =  apply(data_temp[16:18], 1, median))
  data = data.frame("5_4" = expression_difference(data_abs[,"Week5"], data_abs[,"Week4"]), 
                  "6_5" = expression_difference(data_abs[,"Week6"], data_abs[,"Week5"]),
                  "7_6" = expression_difference(data_abs[,"Week7"], data_abs[,"Week6"]),
                  "8_7" = expression_difference(data_abs[,"Week8"], data_abs[,"Week7"]),
                  "9_8" = expression_difference(data_abs[,"Week9"], data_abs[,"Week8"]))
  row.names(data) = rownames(data_abs)
  
  # Select optimum cluster number
  data = scale(data)
  dist_data = get_dist(data, method = "pearson")
  n_cluster = which.max(sapply(c(2:10), function(k) {avg_sil(k, data)})) +1
  
  # Create the data for our plot
  km = kmeans(data, centers = n_cluster, nstart = 25)
  data_plot_raw = cbind(data_abs, Cluster = km$cluster)
  plot_data = create_plot_data(data_plot_raw)
  
  # Create the Plot
  assign(str_glue(tissue, "_plot"), ggplot(plot_data, aes(x = Week, y= Expression, group = Cluster))  + geom_line(aes(colour = Cluster)) +
           geom_point(aes(colour = Cluster, shape = Cluster)) + labs(title = tissue) +
           scale_color_npg() + theme_classic()  )  # + ylim(6.5,8.5)
   
}
ff = sapply(tissues, function(x){str_glue(x, "_plot")})
#grid.arrange(get(ff[1]), get(ff[2]), get(ff[3]), get(ff[4]), get(ff[5]), get(ff[6]), get(ff[7]), get(ff[8]), get(ff[9]), get(ff[10]), get(ff[11]), get(ff[12]), get(ff[13]), get(ff[14]), get(ff[15]))
```

### Search for exclusive TRAs

```{r}

length_tissues = c()
for (i in 1:1818) {
  length_tissues = c(length_tissues, length(diff_genes_ann_0.01$tissue[[i]]))
}
single_tissue = (length_tissues==1)

diff_single_df = diff_genes_ann_0.01[single_tissue,]

all_tissues = c("Adipose - Subcutaneous", "Adipose - Visceral", "Adipose", "Cells - Transformed fibroblasts", "Skin - Not Sun Exposed", "Skin - Sun Exposed", "Skin", "Muscle - Skeletal", "Bladder", "Kidney - Cortex", "Colon - Sigmoid", "Colon - Transverse", "Colon", "Esophagus - Gastroesophageal Junction", "Esophagus - Mucosa", "Esophagus - Muscularis", "Esophagus", "Liver", "Minor Salivary Gland", "Pancreas", "Small Intestine - Terminal Ileum", "Spleen", "Stomach", "Breast - Mammary Tissue", "Cervix - Ectocervix", "Cervix - Endocervix", "Cervix", "Fallopian Tube", "Ovary", "Uterus", "Vagina", "Prostate", "Testis", "Adrenal Gland", "Pituitary", "Thyroid", "Artery - Aorta", "Artery - Coronary", "Artery - Tibial", "Artery", "Heart - Atrial Appendage", "Heart - Left Ventricle", "Heart", "Lung", "Whole Blood", "Cells - EBV-transformed lymphocytes", "Brain - Amygdala", "Brain - Anterior cingulate cortex", "Brain - Caudate", "Brain - Cerebellar Hemisphere", "Brain - Cerebellum", "Brain - Cortex", "Brain - Frontal Cortex", "Brain - Hippocampus", "Brain - Hypothalamus", "Brain - Nucleus accumbens", "Brain - Putamen", "Brain - Spinal cord", "Brain - Substantia nigra", "Brain", "Nerve - Tibial")



count_data = data.frame(tissue = all_tissues, count = NA)
for (tissue in all_tissues){
  
  count_data[count_data$tissue == tissue, "count"] = sum(bool_contained = sapply(diff_single_df$tissues, function(x) {grepl(tissue, toString(x))}))

}
count_data
```

# Work with brain gene differentiators:

```{r}

brain_tissues = c("Brain - Amygdala", "Brain - Anterior cingulate cortex", "Brain - Caudate", "Brain - Cerebellar Hemisphere", "Brain - Cerebellum", "Brain - Cortex", "Brain - Frontal Cortex", "Brain - Hippocampus", "Brain - Hypothalamus", "Brain - Nucleus accumbens", "Brain - Putamen", "Brain - Spinal cord", "Brain - Substantia nigra")
for (brain_tiss in brain_tissues) {
 
  assign(str_glue(brain_tiss, "_single_db"), diff_genes_ann_0.01[(sapply(diff_genes_ann_0.01$tissues, function(x) {grepl(brain_tiss, toString(x))}) &
                                                                   !sapply(diff_genes_ann_0.01$tissues, function(x)
                                                                     {grepl(paste(brain_tissues[!brain_tissues%in%c(brain_tiss)], collapse = "|"),
                                                                            toString(x))})),])

}

```

```{r}
library(factoextra)
library(stringr)
library(ggplot2)
library(ggsci)
library(grid)
library(gridExtra)


brain_tissues = c("Brain - Amygdala", "Brain - Anterior cingulate cortex", "Brain - Caudate", "Brain - Cerebellar Hemisphere", "Brain - Cerebellum", "Brain - Cortex", "Brain - Frontal Cortex", "Brain - Hippocampus", "Brain - Hypothalamus", "Brain - Nucleus accumbens", "Brain - Putamen", "Brain - Spinal cord", "Brain - Substantia nigra")
count = 0

for (brain_tiss in brain_tissues) {
  count = count +1
  #Select the specific genes
  data_temp = diff_genes_ann_0.01[(sapply(diff_genes_ann_0.01$tissues, function(x) {grepl(brain_tiss, toString(x))}) &
                                                                   !sapply(diff_genes_ann_0.01$tissues, function(x)
                                                                     {grepl(paste(brain_tissues[!brain_tissues%in%c(brain_tiss)], collapse = "|"),
                                                                            toString(x))})),]
  if(dim(data_temp)[1] >0) {
      #Get the data for our plot
    data_raw = data.frame(Week4 = apply(data_temp[1:3], 1, median), Week5 =  apply(data_temp[4:6], 1, median), Week6 =  apply(data_temp[7:9], 1, median),
                          Week7=apply(data_temp[10:12], 1, median), Week8 =  apply(data_temp[13:15], 1, median), Week9 =  apply(data_temp[16:18], 1, median))
    rownames(data_raw) = data_temp$ensembl.transcript
    data_plot = data.frame(Week = c(sapply(c(4:9), function(x) {rep(x, dim(data_raw)[1])})), Transcript = factor(c(rep(rownames(data_raw), 6))), Expression = NA)
    for(s in 1:6) {
      for(t in 1:dim(data_raw)[1]) {
        data_plot$Expression[t + (s-1)*dim(data_raw)[1]] = data_raw[t, s]
     }
    }
  
    #Plot the data
    assign(str_glue(brain_tiss, "_plot1"), ggplot(data_plot, aes(x = Week, y= Expression, group = Transcript))   + geom_line(aes(colour = Transcript)) +
            geom_point(aes(colour = Transcript)) + labs(title = brain_tiss) + theme_classic() )
  }
}
ff = sapply(brain_tissues, function(x){str_glue(x, "_plot1")})
data_plot
#grid.arrange(get(ff[1]), get(ff[2]), get(ff[4]), get(ff[5]), get(ff[6]), get(ff[7]), get(ff[8]), get(ff[9]), get(ff[10]), get(ff[11]), get(ff[12]), get(ff[13]))
  # Brain - Caudata so ff[3] has no unique genes
```

```{r}
 
data_temp
data_raw
data_plot



```

#Look at the differentially expressed genes from the spleen in week4 Grouped by clusters

```{r}
#Get the dataset with the clusters
expression_difference = function(x,y) {
  return ((x-y)/((x+y)*0.5))
}

avg_sil = function(k, data) {
  km = kmeans(data, centers = k, nstart = 25)
  ss = silhouette(km$cluster, dist_data)
  mean(ss[, 3])
}
bool_contained = sapply(diff_genes_ann_0.01$tissues, function(x) {grepl("Spleen", toString(x))})
  data_temp = diff_genes_ann_0.01[bool_contained,]
  data_abs = data.frame(Week4 = apply(data_temp[1:3], 1, median), Week5 =  apply(data_temp[4:6], 1, median), Week6 =  apply(data_temp[7:9], 1, median),
                        Week7=apply(data_temp[10:12], 1, median), Week8 =  apply(data_temp[13:15], 1, median), Week9 =  apply(data_temp[16:18], 1, median))
  data = data.frame("5_4" = expression_difference(data_abs[,"Week5"], data_abs[,"Week4"]), 
                  "6_5" = expression_difference(data_abs[,"Week6"], data_abs[,"Week5"]),
                  "7_6" = expression_difference(data_abs[,"Week7"], data_abs[,"Week6"]),
                  "8_7" = expression_difference(data_abs[,"Week8"], data_abs[,"Week7"]),
                  "9_8" = expression_difference(data_abs[,"Week9"], data_abs[,"Week8"]))
  row.names(data_abs) = data_temp$ensembl.transcript
  row.names(data) = rownames(data_abs)
  
  data = scale(data)
  dist_data = get_dist(data, method = "pearson")
  n_cluster = which.max(sapply(c(2:10), function(k) {avg_sil(k, data)})) +1
  
  km = kmeans(data, centers = n_cluster, nstart = 25)
  data_plot_raw = cbind(data_abs, Cluster = km$cluster)

  spleen_genes_clustered = data_plot_raw

```

```{r}
#Compare the initial expression values of both clusters
spleen_genes_clustered_1 = spleen_genes_clustered[spleen_genes_clustered$Cluster==1, ]
spleen_genes_clustered_2 = spleen_genes_clustered[spleen_genes_clustered$Cluster==2, ]
boxplot(spleen_genes_clustered_1$Week4, spleen_genes_clustered_2$Week4)
#boxplot(spleen_genes_clustered_1$Week4, spleen_genes_clustered_1$Week9)
```

```{r}
head(spleen_genes_clustered_1)
head(spleen_genes_clustered_2)
```

### Not used, better method for annotation below

```{r}
# Annotate Transcripts with NCBI data
install.packages("rentrez")
library(rentrez)
setwd("C:/Daten/Data Analysis Projekt/Github Repo/2022-topic-04-team-01/R Scripts")
load(Homo_sapiens.gene_info.gz)

```

# Annoattion with NCBI data

```{r}
get_gene_info = function(transcript_ids){
  library(rentrez)
  library(XML)
  table = data.frame(Transcript = NA, Gene = NA, Protein = NA, Summary = NA)
  for(x in transcript_ids) {
    table = rbind(table, get_single_transcript(x))
  }
  table
}

get_single_transcript = function(single_transcript) {
   res = entrez_search(db = "gene", term = single_transcript)
   if(!length(res$ids)==0) {
     results = entrez_summary(db = "gene", id = res$ids)
      c(single_transcript, as.character(results[2]),  as.character(results[3]),  as.character(results[17]))
   }
   else{
        c(single_transcript, NA, NA, NA)
   }
}

spleen1_gene_description = get_gene_info(rownames(spleen_genes_clustered_1))
spleen2_gene_description = get_gene_info(rownames(spleen_genes_clustered_2))
```

```{r}
#Table verschönern
install.packages("gt")
library(gt)
install.packages("gtExtras")
library(gtExtras)
library(ggplot2)
library(tidyverse)
install.packages("svglite")
library(svglite)

Expression =  data.frame(c(1:length(spleen_genes_clustered_1[,1])))
Expression$Expression_over_time = sapply(c(1:length(spleen_genes_clustered_1[,1])), function(x) {list(as.numeric(c(as.numeric(spleen_genes_clustered_1[x,1]), as.numeric(spleen_genes_clustered_1[x,2]), as.numeric(spleen_genes_clustered_1[x,3]),
           as.numeric(spleen_genes_clustered_1[x,4]),as.numeric(spleen_genes_clustered_1[x,5]),as.numeric(spleen_genes_clustered_1[x,6]))))})

spleen1_gene_description_spl = cbind(spleen1_gene_description[-1,], Expression)
spleen1_gene_description_spl = spleen1_gene_description_spl[,-5]



table_spleen1 = gt(spleen1_gene_description_spl)|>  tab_header(title= "Spleen - Upregulated", subtitle = "List of differentially epxressed genes") |> fmt_markdown(columns = c(Transcript, Gene, Protein, Summary)) |> gt_theme_538() |> gt_plt_sparkline( column = Expression_over_time, type = "ref_median")
```

```{r}
# Jetzt noch für Spleen2
library(gt)
library(gtExtras)
library(ggplot2)
library(tidyverse)
library(svglite)
library(webshot)

Expression =  data.frame(c(1:length(spleen_genes_clustered_2[,1])))
Expression$Expression_over_time = sapply(c(1:length(spleen_genes_clustered_2[,1])), function(x) {list(as.numeric(c(as.numeric(spleen_genes_clustered_2[x,1]), as.numeric(spleen_genes_clustered_2[x,2]), as.numeric(spleen_genes_clustered_2[x,3]),
           as.numeric(spleen_genes_clustered_2[x,4]),as.numeric(spleen_genes_clustered_2[x,5]),as.numeric(spleen_genes_clustered_2[x,6]))))})

spleen2_gene_description_spl = cbind(spleen2_gene_description[-1,], Expression)
spleen2_gene_description_spl = spleen2_gene_description_spl[,-5]


table_spleen2 = gt(spleen2_gene_description_spl)|>  tab_header(title= "Spleen - Downregulated", subtitle = "List of differentially epxressed genes") |> fmt_markdown(columns = c(Transcript, Gene, Protein, Summary)) |> gt_theme_538() |> gt_plt_sparkline( column = Expression_over_time, type = "shaded")

```

```{r}
embryo_df_tra[embryo_df_tra$ensembl.transcript == "ENST00000233505",]
```
