---
title: "Test4"
author: "Paul Christmann"
date: '2022-07-12'
output: pdf_document
geometry: "left=1.5cm,right=1.5cm,top=1.5cm,bottom=1.5cm"
header-includes:
  - \usepackage[font={small,it}, labelfont={bf}]{caption}
---

```{r knitr_init, echo=FALSE, cache=FALSE}
library(knitr)
#library(rmdformats)
## Global options
#options(max.print="120")
opts_chunk$set(echo=FALSE,
	             cache=TRUE,
               prompt=FALSE,
               tidy=TRUE,
               comment=NA,
               message=FALSE,
               warning=FALSE)
#opts_knit$set(width=120)
#opts_knit$set(root.dir  = "~/")
```
```{r}
library(affy)
library(AnnotationDbi)
library(hgu133plus2hsenstprobe)
library(hgu133plus2hsenstcdf) 
library(Rcpp)
library(tidyverse)
library(vsn)
library(limma)
library(pheatmap)
library(hexbin)
library(biomaRt)
library(kableExtra)
library(Rfssa)
library(ggbiplot)
library(magrittr) 
library(dplyr)
library(stringr)
library(ggrepel)
library(readxl)
library(ggpubr)
library(grid)
library(gridExtra)
library(cowplot)
library(RCurl)
library(readxl)
library(ggforce)
library(VennDiagram)
library(png)
library(factoextra)
library(cluster)
library(ggplot2)
library(ggsci)
library(ggplotify)
library(clusterProfiler)
library(enrichplot)
library(org.Hs.eg.db)
library(ggupset)
library(gt)
library(gtExtras)
library(svglite)
library(webshot)
library(rentrez)
library(XML)
library(igraph)
library(ggnet)
library(viridis)
library(treemapify)
library(magick)
```

```{r}
library(tidyverse)
load("diff_genes_ann_0.01.RData")
load("embryo_df_tissues.RData")
```

```{r loading-data-from-github}
#make data accessible for everyone
githubURL <- "https://github.com/datascience-mobi-2022/2022-topic-04-team-01/blob/main/Report/data.GSE15744.RData"
load_github_data(githubURL)
```

```{r vsn-normalization, include=FALSE}
data.GSE15744.vsnrma <- vsnrma(data.GSE15744)
```

```{r silent-Data-Cleaning, include=FALSE}
#adjusting the microarray names by cutting the .cel
rownames(pData(data.GSE15744.vsnrma)) <- substr(rownames(pData(data.GSE15744)), 1, nchar(rownames(pData(data.GSE15744)))-4)
colnames(exprs(data.GSE15744.vsnrma)) <- substr(colnames(exprs(data.GSE15744)), 1, nchar(colnames(exprs(data.GSE15744)))-4)

temp_embryo_df = as.data.frame(exprs(data.GSE15744.vsnrma))
colnames(temp_embryo_df) = rownames(pData(data.GSE15744.vsnrma))
embryo_df = temp_embryo_df[63:95721,]
```

```{r Annotation, include=FALSE}
file_annotation = getURL("https://raw.githubusercontent.com/datascience-mobi-2022/2022-topic-04-team-01/main/R%20Scripts/ensembl_103_human.txt")
ensembl = as.data.frame(read.csv(text=file_annotation, sep = ","))   # Load the ensembl annotation data
TranscriptID_orig = rownames(embryo_df)
ensembl = ensembl %>% distinct(Transcript.stable.ID, .keep_all = TRUE) # Remove duplicate transcripts


embryo_df["Transcript.stable.ID"] = gsub("[//._].*$" ,"", rownames(embryo_df)) # Add column for mergign
embryo_df_ann = left_join(embryo_df, ensembl, by="Transcript.stable.ID")   # Combine the Datasets by their Transcript ID
rownames(embryo_df_ann) <- embryo_df_ann$Transcript.stable.ID
colnames(embryo_df_ann) = c("Week4_1", "Week4_2", "Week4_3", "Week5_1", "Week5_2", "Week5_3", "Week6_1", "Week6_2", "Week6_3", "Week7_1", "Week7_2", "Week7_3", "Week8_1", "Week8_2", "Week8_3", "Week9_1", "Week9_2", "Week9_3", "Transcript.stable.ID", "Gene.stable.ID", "Chromosome.scaffold.name", "HGNC.symbol", "AFFY.HG.U133.Plus.2.probe")


file_tra = getURL("https://raw.githubusercontent.com/datascience-mobi-2022/2022-topic-04-team-01/main/R%20Scripts/tra.2017.human.gtex.5x.table.tsv")
tra_gtex = as.data.frame(read.csv(text = file_tra, sep = "\t"))
embryo_df_tra = embryo_df_ann[embryo_df_ann$Transcript.stable.ID %in% tra_gtex$ensembl.transcript ,]
embryo_df_tra = embryo_df_tra %>%  dplyr::rename(ensembl.transcript = Transcript.stable.ID)
embryo_df_tra =  left_join(embryo_df_tra, tra_gtex, by="ensembl.transcript") 


#Make Tissue names more accessible by creating a vector for them
a = separate(embryo_df_tra, tissues, into = as.character(c(1:30)), sep="/")
create_tissue_vector = function(f) {
  f = unlist(f)
  f = f[!is.na(f)]
  return(f)
}
tissues = apply(a[,31:60],1, create_tissue_vector)
tissues = sapply(tissues, unlist)
a = a[,c(1:30, 61)] 
a %>% add_column(tissues = NA)
a$tissues = tissues
embryo_df_tissues = a
```

```{r further-silent-Data-Cleaning}
embryo_df_tissues <- embryo_df_tissues[,!names(embryo_df_tissues) %in% c("ensembl.gene", "ensembl.symbol", "ensembl.chrom")] 

#Chromosome Spalte aufhübschen und unnötiges löschen
embryo_df_tissues$Chromosome.scaffold.name <- as.factor(embryo_df_tissues$Chromosome.scaffold.name) 
embryo_df_tissues <- embryo_df_tissues %>%
  filter(Chromosome.scaffold.name != "CHR_HSCHR10_1_CTG2" & Chromosome.scaffold.name != "CHR_HSCHR11_1_CTG7"& Chromosome.scaffold.name!= "CHR_HSCHR15_4_CTG8"& Chromosome.scaffold.name!= "CHR_HSCHR22_1_CTG3"& Chromosome.scaffold.name!= "CHR_HSCHR22_1_CTG7"& Chromosome.scaffold.name!=  "CHR_HSCHR4_6_CTG12"& Chromosome.scaffold.name!= "CHR_HSCHR7_2_CTG4_4"& Chromosome.scaffold.name!=  "CHR_HSCHR7_2_CTG6" )
embryo_df_tissues$Chromosome.scaffold.name <- factor(embryo_df_tissues$Chromosome.scaffold.name, levels = c("1", "2","3","4","5","6","7","8","9","10","11","12","13","14","15","16","17","18","19", "20", "21","22", "X","Y"))

# Nach NA values suchen
#sum(is.na(embryo_df_tissues$ensembl.entrez)) # diese Spalte enthält 501 NA - Values
```

```{r, include = FALSE}

limma_data = as.data.frame(embryo_df_tissues[1:18], rownames = embryo_df_tissues$ensembl.transcript)

weeks = c("week4", "week5", "week6", "week7", "week8", "week9")
f = factor(c("week4", "week4", "week4", "week5", "week5", "week5", "week6", "week6", "week6", "week7", "week7", "week7", "week8", "week8", "week8", "week9", "week9", "week9"), levels =weeks)
design = model.matrix(~0+f)
colnames(design) =  weeks
fit_2 = lmFit(limma_data, design)
contrast_all = makeContrasts(week9-week8, week9-week7, week9-week6, week9-week5, week9-week4, week8-week7,week8-week6, week8-week5, week8-week4, week7-week6, week7-week5, week7-week4, week6-week5, week6-week4, week5-week4, levels = design)
fit_all = contrasts.fit(fit_2, contrast_all)
fit_all = eBayes(fit_all)
results_all = decideTests(fit_all)
#summary(results_all)


#Create list of all differentially expressed genes
diff_genes_list_0.01 = c()

for (i in 1:15) {
 top.table = topTable(fit_all,i, n = Inf)
 n = sapply(rownames(top.table), function(x){
  embryo_df_tissues$ensembl.transcript[as.numeric(x)]
  })
  row.names(top.table) = n
  diff_genes_list_0.01 = c(diff_genes_list_0.01, rownames(top.table[which(top.table$adj.P.Val < 0.01), ]))
}
length(diff_genes_list_0.01) #4522
diff_genes_list_0.01 = diff_genes_list_0.01[!duplicated(diff_genes_list_0.01)]
length(diff_genes_list_0.01) #1814

#Select differential genes from annotated dataframe
diff_genes_ann_0.01 = embryo_df_tissues[embryo_df_tissues$ensembl.transcript %in% diff_genes_list_0.01, ]

```

\newpage
## Using Principal component analysis to find biomarkes for developmental progression

Principal component analysis (PCA) was performed on a matrix containing all differentially expressed genes throughout the 6 weeks. This was performed to reduce dimension while keeping most of the data's variance. To get a better grasp of how much variance is explained by each principal component (PC), the percentage variance of each PC together as well as the cumulative variance were plotted. This visualization helps to determine how many PCs are needed to explain a significant amount of the data's variation. The chosen PCs can than be used for further analysis. The first three PCs already explain over 80% of the data's variance, with the first PC alone accounting for over 60% (Fig. \@ref(fig:PCA_analysis) A).

```{r, include=FALSE}
#pca on rows (genes)
pca_genes <- prcomp(
  t(diff_genes_ann_0.01[,1:18]), # transposed matrix
  scale. = TRUE # data scaled to have unit variance
)

#taking a first look
summary(pca_genes)
```

```{r pca-variance, fig.width = 5, fig.height=5}
#visualizing the variance of the PCAs
pc_eigenvalues <- pca_genes$sdev^2

pc_eigenvalues <- tibble(PC = factor(1:length(pc_eigenvalues)), 
                         variance = pc_eigenvalues) %>% 
  #column with percent variance
  mutate(pct = variance/sum(variance)*100) %>% 
  #column with cumulative variance explained
  mutate(pct_cum = cumsum(pct))


#actual plot
PCA_contribution <- pc_eigenvalues %>% 
  ggplot(aes(x = PC)) +
  geom_col(aes(y = pct)) +
  geom_line(aes(y = pct_cum, group = 1)) + 
  geom_point(aes(y = pct_cum)) +
  ggtitle('PCA variance') +
  theme(plot.title = element_text(hjust = 0.5,face = "bold")) +
  labs(x = "Principal component", y = "Fraction variance explained") +
  theme(aspect.ratio = 1)

```

```{r, include=FALSE}
 loadings = pca_genes$rotation[,"PC1"]
pca_df = data.frame(diff_genes_ann_0.01[,c("ensembl.transcript", "HGNC.symbol", "tissues")], Loadings = loadings)
pca_df[order(abs(pca_df$Loadings), decreasing = TRUE),]
```

```{r, include=FALSE}
test_genes = diff_genes_ann_0.01  #[bool_contained,]

#pca on rows (genes)
pca_genes <- prcomp(
  t(test_genes[,1:18]), # transposed matrix
  scale. = TRUE # data scaled to have unit variance
  )

#taking a first look
summary(pca_genes)

loadings = pca_genes$rotation[,"PC1"]
pca_df = data.frame(test_genes[,c("ensembl.transcript", "HGNC.symbol", "tissues")], Loadings = loadings)
pca_df[order(abs(pca_df$Loadings), decreasing = TRUE),]
test_genes
ranks_pca = data.frame(test_genes[,c("ensembl.transcript", "HGNC.symbol")], PC1 = NA, PC2 = NA, PC3 = NA, PC4 = NA, PC5 = NA)
PC_vector = c("PC1", "PC2", "PC3", "PC4", "PC5")
for (i in PC_vector) {
  loadings = pca_genes$rotation[,i]
  pca_df = data.frame(test_genes$ensembl.transcript, Loadings = loadings, Order = c(1:length(test_genes$ensembl.transcript)))
  pca_df = pca_df[order(abs(pca_df$Loadings), decreasing = TRUE),]
  pca_df$rank = c(1:length(test_genes$ensembl.transcript))
  pca_df = pca_df[order(pca_df$Order),]
  ranks_pca[,i] =  pca_df$rank
}

ranks_pca$sum = ranks_pca$PC1*4 - ranks_pca$PC2 - ranks_pca$PC3 - ranks_pca$PC4 - ranks_pca$PC5
ranks_pca[order(ranks_pca$sum),]

```

For further analysis, each transcript was awarded a rank depending on how much it contributes to each PC. Then, a score was calculated for maximum contribution to PC1 and minimal loadings for all other relevant PCs (PC2-5). We took a closer look at the five best socring transcripts for PC1, as this PC alone already explains the brunt of the variance of our data. These transcripts encode for three different proteins: Low-Density Lipoprotein Receptor Class A Domain-Containing Protein 4 (LDLRAD4) encoded by ENST00000399848, Calcium-activated potassium channel subunit beta-2 (KCNMB2) encoded by ENST00000432997 and ENST00000452583 and SLIT And NTRK-Like Protein 3 (SLITRK3) encoded by ENST00000241274 and ENST00000475390.

```{r, fig.width=5,fig.height=5, include = FALSE}

PCA1_first_five <- diff_genes_ann_0.01[c('9703','11797','13044','616','15025'),0:19]


#median of the transcripts for each week
PCA1_4 <- PCA1_first_five[,c(1,2,3)]
week4 <- t(data.frame(apply(PCA1_4,1,median)))

PCA1_5 <- PCA1_first_five[,c(4,5,6)]
week5 <- t(data.frame(apply(PCA1_5,1,median)))

PCA1_6 <- PCA1_first_five[,c(7,8,9)]
week6 <- t(data.frame(apply(PCA1_6,1,median)))

PCA1_7 <- PCA1_first_five[,c(10,11,12)]
week7 <- t(data.frame(apply(PCA1_7,1,median)))

PCA1_8 <- PCA1_first_five[,c(13,14,15)]
week8 <- t(data.frame(apply(PCA1_8,1,median)))

PCA1_9 <- PCA1_first_five[,c(16,17,18)]
week9 <- t(data.frame(apply(PCA1_9,1,median)))


#combining in a dataframe
first_five <- rbind(week4,week5,week6,week7,week8,week9)

#plotting the expression over weeks

png(filename = "PCA1_five.png", res = 100)
par(mar=c(4, 4, 4, 7), xpd=TRUE)
matplot(first_five, 
        type = 'b',
        pch=16,
        col = c('black','red','green','blue','orange'), 
        xlab = 'Weeks', 
        ylab = 'Expression Level', 
        main = 'Most contribution to PC1' , 
        xaxt='n'
)
axis(side = 1,
     at = 1:6, 
     labels = c(4,5,6,7,8,9)
)
legend('topright', 
       inset=c(-0.45,0), 
       legend = c('ENST00000399848 \n (LDLRAD4)\n ','ENST00000432997 \n (KCNMB2) \n','ENST00000452583 \n (KCNMB2) \n','ENST00000241274 \n (SLITRK3)\n','ENST00000475390 \n (SLITRK3)\n'),
       col=c('black','red','green','blue','orange'), 
       pch=16, 
       cex=0.6,
       title="Expressed genes"
)

dev.off()

first_five_plot <- ggdraw() + draw_image("PCA1_five.png") 

```

The transcripts of ENST00000432997 and ENST00000452583 encode for the same Protein (KCNMB2). Fitting to that, both transcripts consistently display the same expression level over all weeks. The same applies to the transcripts of ENST00000241274 and ENST00000475390, as they also encode for the same protein (SLITRK3) and therefore also display the same expression levels. The transcripts for LDLRAD4 show the highest overall expression levels. Their expression pattern also does not display many similarities to those of KCNMB2 and SLITRK3. However, KCNMB2 and SLITRK3 show a highl similar curve over all weeks (Fig. \@ref(fig:PCA_analysis) B).

```{r PCA_analysis, fig.width = 10, fig.height = 4}
lay = rbind(c(1,2))
PCA_plots = arrangeGrob(grobs = list(PCA_contribution, first_five_plot),layout_matrix = lay)

PCA_plots_output = as_ggplot(PCA_plots) +
  draw_plot_label(label = c("A", "B"), 
                  size = 15, 
                  x = c(0.01, 0.5), 
                  y = c(0.17, 0.17)
                  ) 

annotate_figure(PCA_plots_output, 
                top = text_grob("Principal component analysis", 
                                color = "black", 
                                face = "bold", 
                                size = 16
                                )
                )
```