---
title: "Test4"
author: "Paul Christmann"
date: '2022-07-12'
output: pdf_document
geometry: "left=1.5cm,right=1.5cm,top=1.5cm,bottom=1.5cm"
header-includes:
  - \usepackage[font={small,it}, labelfont={bf}]{caption}
---

```{r knitr_init, echo=FALSE, cache=FALSE}
library(knitr)
#library(rmdformats)
## Global options
#options(max.print="120")
opts_chunk$set(echo=FALSE,
	             cache=TRUE,
               prompt=FALSE,
               tidy=TRUE,
               comment=NA,
               message=FALSE,
               warning=FALSE)
#opts_knit$set(width=120)
#opts_knit$set(root.dir  = "~/")
```
```{r}
library(affy)
library(AnnotationDbi)
library(hgu133plus2hsenstprobe)
library(hgu133plus2hsenstcdf) 
library(Rcpp)
library(tidyverse)
library(vsn)
library(limma)
library(pheatmap)
library(hexbin)
library(biomaRt)
library(kableExtra)
library(Rfssa)
library(ggbiplot)
library(magrittr) 
library(dplyr)
library(stringr)
library(ggrepel)
library(readxl)
library(ggpubr)
library(grid)
library(gridExtra)
library(cowplot)
library(RCurl)
library(readxl)
library(ggforce)
library(VennDiagram)
library(png)
library(factoextra)
library(cluster)
library(ggplot2)
library(ggsci)
library(ggplotify)
library(clusterProfiler)
library(enrichplot)
library(org.Hs.eg.db)
library(ggupset)
library(gt)
library(gtExtras)
library(svglite)
library(webshot)
library(rentrez)
library(XML)
library(igraph)
library(ggnet)
library(viridis)
library(treemapify)
library(magick)
```



```{r loading-data-from-github}
#make data accessible for everyone
githubURL <- "https://github.com/datascience-mobi-2022/2022-topic-04-team-01/blob/main/Report/data.GSE15744.RData"
load_github_data(githubURL)
```

```{r vsn-normalization, include=FALSE}
data.GSE15744.vsnrma <- vsnrma(data.GSE15744)
```

```{r silent-Data-Cleaning, include=FALSE}
#adjusting the microarray names by cutting the .cel
rownames(pData(data.GSE15744.vsnrma)) <- substr(rownames(pData(data.GSE15744)), 1, nchar(rownames(pData(data.GSE15744)))-4)
colnames(exprs(data.GSE15744.vsnrma)) <- substr(colnames(exprs(data.GSE15744)), 1, nchar(colnames(exprs(data.GSE15744)))-4)

temp_embryo_df = as.data.frame(exprs(data.GSE15744.vsnrma))
colnames(temp_embryo_df) = rownames(pData(data.GSE15744.vsnrma))
embryo_df = temp_embryo_df[63:95721,]
```

```{r Annotation, include=FALSE}
file_annotation = getURL("https://raw.githubusercontent.com/datascience-mobi-2022/2022-topic-04-team-01/main/R%20Scripts/ensembl_103_human.txt")
ensembl = as.data.frame(read.csv(text=file_annotation, sep = ","))   # Load the ensembl annotation data
TranscriptID_orig = rownames(embryo_df)
ensembl = ensembl %>% distinct(Transcript.stable.ID, .keep_all = TRUE) # Remove duplicate transcripts


embryo_df["Transcript.stable.ID"] = gsub("[//._].*$" ,"", rownames(embryo_df)) # Add column for mergign
embryo_df_ann = left_join(embryo_df, ensembl, by="Transcript.stable.ID")   # Combine the Datasets by their Transcript ID
rownames(embryo_df_ann) <- embryo_df_ann$Transcript.stable.ID
colnames(embryo_df_ann) = c("Week4_1", "Week4_2", "Week4_3", "Week5_1", "Week5_2", "Week5_3", "Week6_1", "Week6_2", "Week6_3", "Week7_1", "Week7_2", "Week7_3", "Week8_1", "Week8_2", "Week8_3", "Week9_1", "Week9_2", "Week9_3", "Transcript.stable.ID", "Gene.stable.ID", "Chromosome.scaffold.name", "HGNC.symbol", "AFFY.HG.U133.Plus.2.probe")


file_tra = getURL("https://raw.githubusercontent.com/datascience-mobi-2022/2022-topic-04-team-01/main/R%20Scripts/tra.2017.human.gtex.5x.table.tsv")
tra_gtex = as.data.frame(read.csv(text = file_tra, sep = "\t"))
embryo_df_tra = embryo_df_ann[embryo_df_ann$Transcript.stable.ID %in% tra_gtex$ensembl.transcript ,]
embryo_df_tra = embryo_df_tra %>%  dplyr::rename(ensembl.transcript = Transcript.stable.ID)
embryo_df_tra =  left_join(embryo_df_tra, tra_gtex, by="ensembl.transcript") 


#Make Tissue names more accessible by creating a vector for them
a = separate(embryo_df_tra, tissues, into = as.character(c(1:30)), sep="/")
create_tissue_vector = function(f) {
  f = unlist(f)
  f = f[!is.na(f)]
  return(f)
}
tissues = apply(a[,31:60],1, create_tissue_vector)
tissues = sapply(tissues, unlist)
a = a[,c(1:30, 61)] 
a %>% add_column(tissues = NA)
a$tissues = tissues
embryo_df_tissues = a
```

```{r further-silent-Data-Cleaning}
embryo_df_tissues <- embryo_df_tissues[,!names(embryo_df_tissues) %in% c("ensembl.gene", "ensembl.symbol", "ensembl.chrom")] 

#Chromosome Spalte aufhübschen und unnötiges löschen
embryo_df_tissues$Chromosome.scaffold.name <- as.factor(embryo_df_tissues$Chromosome.scaffold.name) 
embryo_df_tissues <- embryo_df_tissues %>%
  filter(Chromosome.scaffold.name != "CHR_HSCHR10_1_CTG2" & Chromosome.scaffold.name != "CHR_HSCHR11_1_CTG7"& Chromosome.scaffold.name!= "CHR_HSCHR15_4_CTG8"& Chromosome.scaffold.name!= "CHR_HSCHR22_1_CTG3"& Chromosome.scaffold.name!= "CHR_HSCHR22_1_CTG7"& Chromosome.scaffold.name!=  "CHR_HSCHR4_6_CTG12"& Chromosome.scaffold.name!= "CHR_HSCHR7_2_CTG4_4"& Chromosome.scaffold.name!=  "CHR_HSCHR7_2_CTG6" )
embryo_df_tissues$Chromosome.scaffold.name <- factor(embryo_df_tissues$Chromosome.scaffold.name, levels = c("1", "2","3","4","5","6","7","8","9","10","11","12","13","14","15","16","17","18","19", "20", "21","22", "X","Y"))

# Nach NA values suchen
#sum(is.na(embryo_df_tissues$ensembl.entrez)) # diese Spalte enthält 501 NA - Values
```

```{r, include = FALSE}

limma_data = as.data.frame(embryo_df_tissues[1:18], rownames = embryo_df_tissues$ensembl.transcript)

weeks = c("week4", "week5", "week6", "week7", "week8", "week9")
f = factor(c("week4", "week4", "week4", "week5", "week5", "week5", "week6", "week6", "week6", "week7", "week7", "week7", "week8", "week8", "week8", "week9", "week9", "week9"), levels =weeks)
design = model.matrix(~0+f)
colnames(design) =  weeks
fit_2 = lmFit(limma_data, design)
contrast_all = makeContrasts(week9-week8, week9-week7, week9-week6, week9-week5, week9-week4, week8-week7,week8-week6, week8-week5, week8-week4, week7-week6, week7-week5, week7-week4, week6-week5, week6-week4, week5-week4, levels = design)
fit_all = contrasts.fit(fit_2, contrast_all)
fit_all = eBayes(fit_all)
results_all = decideTests(fit_all)
#summary(results_all)


#Create list of all differentially expressed genes
diff_genes_list_0.01 = c()

for (i in 1:15) {
 top.table = topTable(fit_all,i, n = Inf)
 n = sapply(rownames(top.table), function(x){
  embryo_df_tissues$ensembl.transcript[as.numeric(x)]
  })
  row.names(top.table) = n
  diff_genes_list_0.01 = c(diff_genes_list_0.01, rownames(top.table[which(top.table$adj.P.Val < 0.01), ]))
}
length(diff_genes_list_0.01) #4522
diff_genes_list_0.01 = diff_genes_list_0.01[!duplicated(diff_genes_list_0.01)]
length(diff_genes_list_0.01) #1814

#Select differential genes from annotated dataframe
diff_genes_ann_0.01 = embryo_df_tissues[embryo_df_tissues$ensembl.transcript %in% diff_genes_list_0.01, ]

```

\newpage
## TRAs can infer a basic timeline of organ development {#organ}

### Differentially expressed transcripts can be linked to all analyzed tissues {#organ-overview}

The TRA dataset covers 53 distinct tissues. For all of those, we found at least 40 differentially expressed transcripts within our dataset. The minimum were 46 stomach-linked transcripts, the highest reached 837 TRAs for the testes.

```{r, fig.width=8, fig.height=6, include=FALSE}
# Create a barplot for the TRA count of each tissue based on our dataset of differentially expressed genes

all_tissues = c("Adipose - Subcutaneous", "Adipose - Visceral", "Cells - Transformed fibroblasts", "Skin - Not Sun Exposed", "Skin - Sun Exposed",  "Muscle - Skeletal", "Bladder", "Kidney - Cortex", "Colon - Sigmoid", "Colon - Transverse", "Esophagus - Gastroesophageal Junction", "Esophagus - Mucosa", "Esophagus - Muscularis", "Liver", "Minor Salivary Gland", "Pancreas", "Small Intestine - Terminal Ileum", "Spleen", "Stomach", "Breast - Mammary Tissue", "Cervix - Ectocervix", "Cervix - Endocervix",  "Fallopian Tube", "Ovary", "Uterus", "Vagina", "Prostate", "Testis", "Adrenal Gland", "Pituitary", "Thyroid", "Artery - Aorta", "Artery - Coronary", "Artery - Tibial", "Heart - Atrial Appendage", "Heart - Left Ventricle", "Lung", "Whole Blood", "Cells - EBV-transformed lymphocytes", "Brain - Amygdala", "Brain - Anterior cingulate cortex", "Brain - Caudate", "Brain - Cerebellar Hemisphere", "Brain - Cerebellum", "Brain - Cortex", "Brain - Frontal Cortex", "Brain - Hippocampus", "Brain - Hypothalamus", "Brain - Nucleus accumbens", "Brain - Putamen", "Brain - Spinal cord", "Brain - Substantia nigra",  "Nerve - Tibial")
tissue_count = data.frame(tissue = all_tissues, count = sapply(all_tissues, function(y) {sum(sapply(diff_genes_ann_0.01$tissues, function(x) {grepl(y, toString(x))}))}))

tissue_count_barplot =  ggplot(tissue_count, aes(x = tissue, y = count, fill = tissue))  + geom_bar(stat = "identity") + labs(x = "Tissue", y = "Number of Transcripts")  + theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1)) + guides(fill = FALSE) 
```

```{r, fig.width=5, fig.height=5, include=FALSE}

select_tissues = c("Adipose", "Cells - Transformed fibroblasts", "Skin",  "Muscle - Skeletal", "Bladder", "Kidney - Cortex",  "Colon", "Esophagus", "Liver", "Minor Salivary Gland", "Pancreas", "Small Intestine - Terminal Ileum", "Spleen", "Stomach", "Breast - Mammary Tissue", "Cervix", "Fallopian Tube", "Ovary", "Uterus", "Vagina", "Prostate", "Testis", "Adrenal Gland", "Pituitary", "Thyroid",  "Artery", "Heart", "Lung", "Whole Blood", "Cells - EBV-transformed lymphocytes", "Brain - Spinal cord", "Brain", "Nerve - Tibial")



count_data = data.frame(tissue = select_tissues, count = NA)
for (tissue in select_tissues){
  
  count_data[count_data$tissue == tissue, "count"] = sum(bool_contained = sapply(diff_genes_ann_0.01$tissues, function(x) {grepl(tissue, toString(x))}))

}
cols = rainbow(33, s=.6, v=.9) #[sample(1:33,33)] #61
count_data$color = cols
treemap = ggplot(count_data, aes(area = count, label = tissue, fill = tissue)) + geom_treemap() + geom_treemap_text(colour = "white", place = "top", reflow = T) + scale_fill_manual(values=cols) + guides(fill = FALSE)
```

```{r, fig.width=12, fig.height=6, fig.cap= "A. The number of transcripts associated with each tissue, including subtissues, is displayed.  B. The plot shows the share of TRAs associated with each tissue. Subtissues are subsumed under their main tissue."}
lay = rbind(c(1,1,1,2), c(1,1,1, NA))
plot = arrangeGrob(grobs = list(tissue_count_barplot, treemap),layout_matrix = lay)
plot_output = as_ggplot(plot) + draw_plot_label(label = c("A", "B"), size = 15, x = c(0.05, 0.77), y = c(0.1,0.1)) 
annotate_figure(plot_output, top = text_grob("Differentially expressed transcripts of TRAs by tissue", color = "black", face = "bold", size = 16))
```

```{r, eval = FALSE}
# Mean number of tissues associated with each transcript
mean(sapply(diff_genes_ann_0.01$tissues, function(x) {length(x)}))
```

These numbers are sufficient for further analysis of the gene expression within individual tissues. Therefore, all results will be based on our dataset with differentially expressed genes from limma analysis. Nonetheless, it should be noted that there is a significant overlap between the TRAs associated with different tissues, especially as each transcript is on average linked to 7.4 different sub-tissues or tissues. This overlap is illustrated by Fig. \@ref(fig:tissue-links). Furthermore, a detailed heatmap for the shared TRAs between tissues is shown in the supplementary Fig. \@ref(fig:heatmap-tissues).

```{r tissue-links,fig.width= 14, fig.height=4, fig.cap= "Each tissue is displayed as a node with its size representing the number of transcripts associated with it. The edges show the shared TRAs between the linked tissues, with only links corresponding to more than 100 common TRAs visible."}

nodes = c("Adipose", "Cells - Transformed fibroblasts", "Skin",  "Muscle - Skeletal", "Bladder", "Kidney - Cortex",  "Colon", "Esophagus", "Liver", "Minor Salivary Gland", "Pancreas", "Small Intestine - Terminal Ileum", "Spleen", "Stomach", "Breast - Mammary Tissue", "Cervix", "Fallopian Tube", "Ovary", "Uterus", "Vagina", "Prostate", "Testis", "Adrenal Gland", "Pituitary", "Thyroid",  "Artery", "Heart", "Lung", "Whole Blood", "Cells - EBV-transformed lymphocytes", "Brain - Spinal cord", "Brain", "Nerve - Tibial")

edges = data.frame (rep(nodes, 32), NA)
for (i in 1:32) {
  a = c(edges[(i+1):33,1], edges[1:i,1])
  edges[(33*(i-1)+1):(33*i),2] = a
}
colnames(edges) = c("Tissue1", "Tissue2")
graph1 = graph_from_data_frame(edges, directed = FALSE)


node_weight = data.frame(tissue = nodes, weight = sapply(nodes, function(y) {sum(sapply(diff_genes_ann_0.01$tissues, function(x) {grepl(y, toString(x))}))}))

edge_combined = edges %>% unite(tissues, c("Tissue1", "Tissue2"), sep = "|")
edge_weight = data.frame(edges, weight = apply(edges,1, function(y) {sum(sapply(diff_genes_ann_0.01$tissues, function(x) {grepl(y[1], toString(x)) && grepl(y[2], toString(x))}))}))

edge_weight_adj = edge_weight
edge_weight_adj$weight = sapply(edge_weight_adj$weight, function(x) { if(x < 100) {x = 0} else {x = x/100}})
edge_weight_cleared = edge_weight_adj[!(edge_weight_adj$weight == 0),]


graph2 = ggnet2(edge_weight_cleared[,1:2], label = TRUE, edge.size = edge_weight_cleared$weight)
node_size = sapply(graph2$data$label, function(x) {node_weight[x,2]})

graph2 = ggnet2(edge_weight_cleared[,1:2], color = graph2$data$label, label = TRUE, color.palette = c() ,edge.size = edge_weight_cleared$weight, size = (ceiling(node_size/30))) + guides(size = FALSE, color = FALSE) + scale_color_viridis(option = "D", discrete = TRUE) + labs(title = "Overlap between the TRAs associated with different tissues") +  theme(plot.title = element_text(hjust = 0.5, face = "bold"), plot.margin = unit(c(0.2,6.5,0,6.5),"cm"))
graph2
```

### The expression of all TRAs associated with a tissue cannot be used to infer organ development {#organ-cluster}

In this research, we attempt to draw conclusions about the developmental state of a tissue based on the expression of genes associated with it alone. For all differencially expressed transcripts associated with one tissue, we analyzed the share of transcripts above a certain expression level over time, as shown in Fig. \@ref(fig:expression-plot)A. Furthermore, we observed trends within the median expression of these trascnrips (Fig. \@ref(fig:expression-plot)B). Since both metrics only depicted miniscule changes, we hypothesized that distinct, counteracting trends in expression exist within one tissue. Thus, k-means clustering was used to determine groups of TRAs with similar expression patterns. For each of these clusters, the median expression was plotted as shown in Fig. \@ref(fig:expression-plot)C. The clustering graphs for all tissues can be found on our github repository.

```{r expression-threshold, include=FALSE}
# Create dataframe with median expression levels for each transcript and week (combining the replicates)
embryo_df_tissues_median  = data.frame(Week4 = apply(embryo_df_tissues[1:3], 1, median), Week5 =  apply(embryo_df_tissues[4:6], 1, median), Week6 =  apply(embryo_df_tissues[7:9], 1, median), Week7=apply(embryo_df_tissues[10:12], 1, median), Week8 =  apply(embryo_df_tissues[13:15], 1, median), Week9 =  apply(embryo_df_tissues[16:18], 1, median))
embryo_df_tissues_median = cbind(embryo_df_tissues_median, embryo_df_tissues[,c(19,29)])


tissue = "Spleen"
threshold = c(6.6,6.8, 7.0, 7.2, 7.4, 7.6, 8.0)

# Select the data from a tissues
bool_contained = sapply(embryo_df_tissues_median$tissues, function(x) {grepl(tissue, toString(x))})
data_temp = embryo_df_tissues_median[bool_contained,]

#Determine the percentage of expressed genes
count = 0
a = c()
data_plot = data.frame(Week = c(sapply(c(4:9), function(x) {rep(x, length(threshold))})), Threshold = factor(c(rep(threshold, 6))), Percentage = NA)
  for(s in 4:9) {
    for(t in 1:length(threshold)) {
      count = count +1
      a  = c(a,(sum(data_temp[,str_glue("Week", s)] > threshold[t]))/(dim(data_temp)[1]))
    }
  }
data_plot$Percentage= a

  
#Plot the data
spleen_percentage_plot =  ggplot(data_plot, aes(x = Week, y= Percentage, group = Threshold))   + geom_line(aes(colour = Threshold)) + geom_point(aes(colour = Threshold, shape = Threshold)) + labs(title = "Share over threshold") + theme_classic() + ylim(0,1)

```

```{r expression-median, include=FALSE}
#Plot of median expression level of all spleen-related genes over time

#Select genes only of target tissue
bool_contained = sapply(diff_genes_ann_0.01$tissues, function(x) {"Spleen" %in% x})

#Graph of mean gene expression over time
spleen_genes = diff_genes_ann_0.01[bool_contained,]
spleen_genes = spleen_genes[, 1:19]
spleen_genes_mean = data.frame(week4=rowMeans(spleen_genes[,1:3]),week5=rowMeans(spleen_genes[,4:6]), week6=rowMeans(spleen_genes[,7:9]),week7=rowMeans(spleen_genes[,10:12]), week8=rowMeans(spleen_genes[,13:15]), week9=rowMeans(spleen_genes[,16:18]))
spleen_genes_graph = apply(spleen_genes_mean, 2, median)
spleen_genes_graph= data.frame (Expression = spleen_genes_graph, Week = c(4:9))

spleen_genes_plot = ggplot(spleen_genes_graph, aes(x = Week, y= Expression))  + geom_line(color = "#0000ff") + geom_point(color = "#0000ff") + labs(title = paste("Median expression"))  + theme_classic() +  ylim(6.5,8.5)

```

```{r expression-clustered, include=FALSE}
#Clustered plot of expression levels over time for the spleen

tissue = "Spleen"

# Functions for later use
expression_difference = function(x,y) {
  return ((x-y)/((x+y)*0.5))
}

avg_sil = function(k, data) {
  km = kmeans(data, centers = k, nstart = 25)
  ss = silhouette(km$cluster, dist_data)
  mean(ss[, 3])
}

create_plot_data = function(dataset) {
  data_plot = data.frame(Week = c(sapply(c(4:9), function(x) {rep(x, n_cluster)})), Cluster = as.character(c(rep(c(1:n_cluster), 6))), Expression = NA)
  count = 0
  for(s in 4:9) {
    for(t in 1:n_cluster) {
      count = count + 1
      values = dataset[dataset$Cluster == t, str_glue("Week", s)]
      data_plot[count, "Expression"] = median(values)
      data_plot[count, "Cluster"] = str_glue(data_plot[count, "Cluster"], " [", km$size[t], " genes]")
   }
  }
  return(data_plot)
}



# Select an modify the data used for kmeans
bool_contained = sapply(diff_genes_ann_0.01$tissues, function(x) {grepl(tissue, toString(x))})
data_temp = diff_genes_ann_0.01[bool_contained,]
data_abs = data.frame(Week4 = apply(data_temp[1:3], 1, median), Week5 =  apply(data_temp[4:6], 1, median), Week6 =  apply(data_temp[7:9], 1, median),
                        Week7=apply(data_temp[10:12], 1, median), Week8 =  apply(data_temp[13:15], 1, median), Week9 =  apply(data_temp[16:18], 1, median))
data = data.frame("5_4" = expression_difference(data_abs[,"Week5"], data_abs[,"Week4"]), 
                  "6_5" = expression_difference(data_abs[,"Week6"], data_abs[,"Week5"]),
                  "7_6" = expression_difference(data_abs[,"Week7"], data_abs[,"Week6"]),
                  "8_7" = expression_difference(data_abs[,"Week8"], data_abs[,"Week7"]),
                  "9_8" = expression_difference(data_abs[,"Week9"], data_abs[,"Week8"]))
row.names(data) = rownames(data_abs)
  
# Select optimum cluster number
data = scale(data)
dist_data = get_dist(data, method = "pearson")
n_cluster = which.max(sapply(c(2:10), function(k) {avg_sil(k, data)})) +1
  
# Create the data for our plot
km = kmeans(data, centers = n_cluster, nstart = 25)
data_plot_raw = cbind(data_abs, Cluster = km$cluster)
plot_data = create_plot_data(data_plot_raw)
  
# Create the Plot
spleen_clustered_plot =  ggplot(plot_data, aes(x = Week, y= Expression, group = Cluster))  + geom_line(aes(colour = Cluster)) +
          geom_point(aes(colour = Cluster, shape = Cluster)) + labs(title = "Clustered expression") +
          scale_color_npg() + theme_classic()  + ylim(6.5,8.5)
   
```

```{r expression-plot, fig.width= 10, fig.height= 3, fig.cap= "A. For different expression thresholds the share of differentially expressed transcripts with higher expressions than the threshold is depicted. B. The median expression for all spleen-associated TRAs is shown for each point in time. C. For k-means clustring, the silhouette score determined an optimum of two clusters. The median expression analog to B is plottet for each of these clusters. "}

plot = arrangeGrob(grobs = list(spleen_percentage_plot, spleen_genes_plot, spleen_clustered_plot), nrow=1, widths = c(3.3,2.7, 4), heights = c(0.7))
plot_output = as_ggplot(plot) + draw_plot_label(label = c("A", "B", "C"), size = 15, x = c(0.03, 0.35, 0.63), y = c(0.13, 0.13, 0.13)) 
annotate_figure(plot_output, top = text_grob("Expression over time of Spleen-related differentially expressed transcripts", color = "black", face = "bold", size = 16))
```

For many tissues, as shown here exemplary with the spleen, the clustering revealed two or more clusters that could be characterized as either an upregulation or a downregulation. In order to determine the indications for organ development, we analyzed the functions of the transcripts belonging to the two clusters.

### The clusters of up- and downregulated transcripts can be linked to distinct gene functions. {#organ-tables}

```{r spleen-clusters}
#Get the spleen dataset with the clusters
expression_difference = function(x,y) {
  return ((x-y)/((x+y)*0.5))
}

avg_sil = function(k, data) {
  km = kmeans(data, centers = k, nstart = 25)
  ss = silhouette(km$cluster, dist_data)
  mean(ss[, 3])
}
bool_contained = sapply(diff_genes_ann_0.01$tissues, function(x) {grepl("Spleen", toString(x))})
  data_temp = diff_genes_ann_0.01[bool_contained,]
  data_abs = data.frame(Week4 = apply(data_temp[1:3], 1, median), Week5 =  apply(data_temp[4:6], 1, median), Week6 =  apply(data_temp[7:9], 1, median),
                        Week7=apply(data_temp[10:12], 1, median), Week8 =  apply(data_temp[13:15], 1, median), Week9 =  apply(data_temp[16:18], 1, median))
  data = data.frame("5_4" = expression_difference(data_abs[,"Week5"], data_abs[,"Week4"]), 
                  "6_5" = expression_difference(data_abs[,"Week6"], data_abs[,"Week5"]),
                  "7_6" = expression_difference(data_abs[,"Week7"], data_abs[,"Week6"]),
                  "8_7" = expression_difference(data_abs[,"Week8"], data_abs[,"Week7"]),
                  "9_8" = expression_difference(data_abs[,"Week9"], data_abs[,"Week8"]))
  row.names(data_abs) = data_temp$ensembl.transcript
  row.names(data) = rownames(data_abs)
  
  data = scale(data)
  dist_data = get_dist(data, method = "pearson")
  n_cluster = which.max(sapply(c(2:10), function(k) {avg_sil(k, data)})) +1
  
  km = kmeans(data, centers = n_cluster, nstart = 25)
  data_plot_raw = cbind(data_abs, Cluster = km$cluster)

  spleen_genes_clustered = data_plot_raw

```

```{r spleen-boxplot, fig.width= 12, fig.height= 3, fig.cap = "A further look on the expression of transcripts in the up- and downregulated clusters shows that the upregulated transcripts are close to the minimum expression level between 6 and 7 in week 4 and showing expressions between 7 and 8.5 by week 9. In contrast, the downregulated genes have very high expression levels (8-9) by week 4 and decrease to a more moderate expression between 7 and 8.5 analogous to the upregulated transcripts."}
library(ggplot2)
#Compare the initial expression values of both clusters
spleen_genes_clustered_1 = spleen_genes_clustered[spleen_genes_clustered$Cluster==1, ]
spleen_genes_clustered_2 = spleen_genes_clustered[spleen_genes_clustered$Cluster==2, ]
spleen_boxplot_data = data.frame(Expression = NA, Week = NA, Cluster = NA)
i = 0
for (genelist in list(spleen_genes_clustered_1$Week4, spleen_genes_clustered_1$Week9, spleen_genes_clustered_2$Week4,  spleen_genes_clustered_2$Week9)) {
  i = i+1
  spleen_df = data.frame(Expression = genelist, Week = factor(rep((if(i%%2 == 0) {9} else {4}),length(genelist))), Cluster = rep((if(i<3) {"Up"} else
    {"Down"}),length(genelist)))
  spleen_boxplot_data = rbind(spleen_boxplot_data, spleen_df)
}
spleen_boxplot_data = spleen_boxplot_data[-1,]
spleen_boxplot = ggplot(spleen_boxplot_data, aes(x = Cluster, y = Expression, fill = Week)) + geom_boxplot(position=position_dodge(0.8)) + theme_classic() + labs(title = "Expression of spleen-associated transcripts") + theme(aspect.ratio = 1, plot.title = element_text(hjust = 0.5))
spleen_boxplot

```

As shown in Fig. \@ref(fig:spleen-boxplot), the spleen is a clear example of two dinstict clusters with one consisting of upregulated previously inactive genes and one with downregulated highly active genes. For all these differentially expressed transcripts, we used the NCBI gene database to get a functional annotation. Of the 98 upregulated genes, 48 had a functional annotation. 17 of those were clearly associated with immune system or blood functions and thus relevant for the functional thymus. We further found 157 downregulated genes. There, 70 were annotated and 45 of those displayed a relation to the cell cycle or cell division. The tables of the transcripts with a relevant function are visible in the suppplementary material (Fig. \@ref(fig:spleen-up-table) and Fig. \@ref(fig:spleen-down-table)).

### Overrepresentation Analysis can create plots that signify organ development {#organ-ora}

For this analysis, the eight tissues with the most meaningful results were chosen.In Fig. \@ref(fig:ORA-plot), the most important functions for these tissue were determined through overrepresentation analysis. In addition, the Expression of the associated transcripts was plotted.

```{r ORA-code, include = FALSE}
#Rearrange the selected plots so that we can combine the table with a plot 


# Spleen 
tissue = "Spleen"
selected = c(1,2,3,4,5)
file_tissue = "https://raw.githubusercontent.com/datascience-mobi-2022/2022-topic-04-team-01/main/Report/Spleen_table.png"
#Overrepresentation Analysis
bool_contained = sapply(embryo_df_tissues$tissues, function(x) {grepl(tissue, toString(x))})
embryo_df_tissue = embryo_df_tissues[bool_contained,]
deGenes = as.character(embryo_df_tissue$ensembl.entrez[!duplicated(embryo_df_tissue$Gene.stable.ID)])
geneUniverse = as.character(embryo_df_tissues$ensembl.entrez[!duplicated(embryo_df_tissues$Gene.stable.ID)])
enr_GO_tissue1 = enrichGO(gene = deGenes, ont = "BP",  OrgDb ="org.Hs.eg.db", universe = geneUniverse, readable=TRUE, pvalueCutoff = 0.05)
enr_GO_tissue = as.data.frame(enr_GO_tissue1)
#Create the table of most significant functions
if (dim(enr_GO_tissue)[1] <10) {npath = dim(enr_GO_tissue)[1]} else {npath = 10}
tissue_table = enr_GO_tissue[1:npath,c(2,6)]
colnames(tissue_table) = c("Function", "Adj_p_value")
#assign(str_glue(tissue,"_table"), gt(tissue_table)|>  tab_header(title= tissue) |> fmt_markdown(columns = c(Function)) |> gt_theme_538() |> fmt_scientific(columns = c(Adj_p_value), decimals = 0) |>  tab_style(style = list(cell_text(weight = "bold")), locations = cells_body(columns = Function, rows = selected)))
#gtsave(get(str_glue(tissue, "_table")),str_glue(tissue, "_table.png"))
download.file(file_tissue, str_glue(tissue, "_table.png"), mode = "wb")
assign(str_glue(tissue, "_t"), (ggdraw() + draw_image(str_glue(tissue, "_table.png"))))
#Plot the chosen functions
data_plot = data.frame(Week = c(sapply(c(4:9), function(x) {rep(x, length(selected))})), Function =
                             as.character(c(rep(enr_GO_tissue$Description[selected], 6)), Expression=NA))
for (i in selected) {
  xx = strsplit(enr_GO_tissue$geneID[i], "/")
  bool_contained1 = sapply(diff_genes_ann_0.01$HGNC.symbol, function(x) {grepl(x, toString(xx))})
  bool_contained1[is.na(bool_contained1)] = FALSE
  data_temp = diff_genes_ann_0.01[bool_contained1,]
  data_abs = data.frame(Week4 = apply(data_temp[1:3], 1, median), Week5 = apply(data_temp[4:6], 1, median), Week6 =  
                              apply(data_temp[7:9], 1, median), Week7=apply(data_temp[10:12], 1, median), Week8 = 
                              apply(data_temp[13:15], 1, median), Week9 =  apply(data_temp[16:18], 1, median))
  expr_path = colMeans(data_abs[1:6,])
  data_plot[data_plot$Function == enr_GO_tissue$Description[i],"Expression"] = expr_path
}
assign(str_glue(tissue, "_plot"), ggplot(data_plot, aes(x = Week, y= Expression, group = Function))  + geom_line(aes(colour =Function), position = position_dodge(0)) + geom_point(aes(colour = Function, shape = Function), position = position_dodge(0)) + labs(title = tissue) +  theme_classic() + scale_color_npg()  + theme(legend.title = element_text(size = 9), legend.text = element_text(size = 7), aspect.ratio = 1, legend.position = "bottom", legend.spacing.y = unit(0.1, "cm"),legend.key.size = unit(0.4, "cm"), plot.margin = unit(c(0.5,0.8,0.2,0),"cm")) + guides(colour = guide_legend(nrow = length(selected), title.position = "top",byrow = TRUE), shape = guide_legend(nrow = length(selected), title.position = "top",byrow = TRUE)))




# Brain 
tissue = "Brain"
selected = c(1,5,7)
file_tissue = "https://raw.githubusercontent.com/datascience-mobi-2022/2022-topic-04-team-01/main/Report/Brain_table.png"
#Overrepresentation Analysis
bool_contained = sapply(embryo_df_tissues$tissues, function(x) {grepl(tissue, toString(x))})
embryo_df_tissue = embryo_df_tissues[bool_contained,]
deGenes = as.character(embryo_df_tissue$ensembl.entrez[!duplicated(embryo_df_tissue$Gene.stable.ID)])
geneUniverse = as.character(embryo_df_tissues$ensembl.entrez[!duplicated(embryo_df_tissues$Gene.stable.ID)])
enr_GO_tissue1 = enrichGO(gene = deGenes, ont = "BP",  OrgDb ="org.Hs.eg.db", universe = geneUniverse, readable=TRUE, pvalueCutoff = 0.05)
enr_GO_tissue = as.data.frame(enr_GO_tissue1)
#Create the table of most significant functions
if (dim(enr_GO_tissue)[1] <10) {npath = dim(enr_GO_tissue)[1]} else {npath = 10}
tissue_table = enr_GO_tissue[1:npath,c(2,6)]
colnames(tissue_table) = c("Function", "Adj_p_value")
#assign(str_glue(tissue,"_table"), gt(tissue_table)|>  tab_header(title= tissue) |> fmt_markdown(columns = c(Function)) |> gt_theme_538() |> fmt_scientific(columns = c(Adj_p_value), decimals = 0) |>  tab_style(style = list(cell_text(weight = "bold")), locations = cells_body(columns = Function, rows = selected)))
#gtsave(get(str_glue(tissue, "_table")),str_glue(tissue, "_table.png"))
download.file(file_tissue, str_glue(tissue, "_table.png"), mode = "wb")
assign(str_glue(tissue, "_t"), (ggdraw() + draw_image(str_glue(tissue, "_table.png"))))
#Plot the chosen functions
data_plot = data.frame(Week = c(sapply(c(4:9), function(x) {rep(x, length(selected))})), Function =
                             as.character(c(rep(enr_GO_tissue$Description[selected], 6)), Expression=NA))
for (i in selected) {
  xx = strsplit(enr_GO_tissue$geneID[i], "/")
  bool_contained1 = sapply(diff_genes_ann_0.01$HGNC.symbol, function(x) {grepl(x, toString(xx))})
  bool_contained1[is.na(bool_contained1)] = FALSE
  data_temp = diff_genes_ann_0.01[bool_contained1,]
  data_abs = data.frame(Week4 = apply(data_temp[1:3], 1, median), Week5 = apply(data_temp[4:6], 1, median), Week6 =  
                              apply(data_temp[7:9], 1, median), Week7=apply(data_temp[10:12], 1, median), Week8 = 
                              apply(data_temp[13:15], 1, median), Week9 =  apply(data_temp[16:18], 1, median))
  expr_path = colMeans(data_abs[1:6,])
  data_plot[data_plot$Function == enr_GO_tissue$Description[i],"Expression"] = expr_path
}
assign(str_glue(tissue, "_plot"), ggplot(data_plot, aes(x = Week, y= Expression, group = Function))  + geom_line(aes(colour =Function), position = position_dodge(0)) + geom_point(aes(colour = Function, shape = Function), position = position_dodge(0)) + labs(title = tissue) +  theme_classic() + scale_color_npg()  + theme(legend.title = element_text(size = 9), legend.text = element_text(size = 7), aspect.ratio = 1, legend.position = "bottom", legend.spacing.y = unit(0.1, "cm"),legend.key.size = unit(0.4, "cm"), plot.margin = unit(c(0.5,0.8,0.2,0),"cm")) + guides(colour = guide_legend(nrow = length(selected), title.position = "top",byrow = TRUE), shape = guide_legend(nrow = length(selected), title.position = "top",byrow = TRUE)))


# Liver 
tissue = "Liver"
selected = c(1,3,5,7,9)
file_tissue = "https://raw.githubusercontent.com/datascience-mobi-2022/2022-topic-04-team-01/main/Report/Liver_table.png"
#Overrepresentation Analysis
bool_contained = sapply(embryo_df_tissues$tissues, function(x) {grepl(tissue, toString(x))})
embryo_df_tissue = embryo_df_tissues[bool_contained,]
deGenes = as.character(embryo_df_tissue$ensembl.entrez[!duplicated(embryo_df_tissue$Gene.stable.ID)])
geneUniverse = as.character(embryo_df_tissues$ensembl.entrez[!duplicated(embryo_df_tissues$Gene.stable.ID)])
enr_GO_tissue1 = enrichGO(gene = deGenes, ont = "BP",  OrgDb ="org.Hs.eg.db", universe = geneUniverse, readable=TRUE, pvalueCutoff = 0.05)
enr_GO_tissue = as.data.frame(enr_GO_tissue1)
#Create the table of most significant functions
if (dim(enr_GO_tissue)[1] <10) {npath = dim(enr_GO_tissue)[1]} else {npath = 10}
tissue_table = enr_GO_tissue[1:npath,c(2,6)]
colnames(tissue_table) = c("Function", "Adj_p_value")
#assign(str_glue(tissue,"_table"), gt(tissue_table)|>  tab_header(title= tissue) |> fmt_markdown(columns = c(Function)) |> gt_theme_538() |> fmt_scientific(columns = c(Adj_p_value), decimals = 0) |>  tab_style(style = list(cell_text(weight = "bold")), locations = cells_body(columns = Function, rows = selected)))
#gtsave(get(str_glue(tissue, "_table")),str_glue(tissue, "_table.png"))
download.file(file_tissue, str_glue(tissue, "_table.png"), mode = "wb")
assign(str_glue(tissue, "_t"), (ggdraw() + draw_image(str_glue(tissue, "_table.png"))))
#Plot the chosen functions
data_plot = data.frame(Week = c(sapply(c(4:9), function(x) {rep(x, length(selected))})), Function =
                             as.character(c(rep(enr_GO_tissue$Description[selected], 6)), Expression=NA))
for (i in selected) {
  xx = strsplit(enr_GO_tissue$geneID[i], "/")
  bool_contained1 = sapply(diff_genes_ann_0.01$HGNC.symbol, function(x) {grepl(x, toString(xx))})
  bool_contained1[is.na(bool_contained1)] = FALSE
  data_temp = diff_genes_ann_0.01[bool_contained1,]
  data_abs = data.frame(Week4 = apply(data_temp[1:3], 1, median), Week5 = apply(data_temp[4:6], 1, median), Week6 =  
                              apply(data_temp[7:9], 1, median), Week7=apply(data_temp[10:12], 1, median), Week8 = 
                              apply(data_temp[13:15], 1, median), Week9 =  apply(data_temp[16:18], 1, median))
  expr_path = colMeans(data_abs[1:6,])
  data_plot[data_plot$Function == enr_GO_tissue$Description[i],"Expression"] = expr_path
}
assign(str_glue(tissue, "_plot"), ggplot(data_plot, aes(x = Week, y= Expression, group = Function))  + geom_line(aes(colour =Function), position = position_dodge(0)) + geom_point(aes(colour = Function, shape = Function), position = position_dodge(0)) + labs(title = tissue) +  theme_classic() + scale_color_npg()  + theme(legend.title = element_text(size = 9), legend.text = element_text(size = 7), aspect.ratio = 1, legend.position = "bottom", legend.spacing.y = unit(0.1, "cm"),legend.key.size = unit(0.4, "cm"), plot.margin = unit(c(0.5,0.8,0.2,0),"cm")) + guides(colour = guide_legend(nrow = length(selected), title.position = "top",byrow = TRUE), shape = guide_legend(nrow = length(selected), title.position = "top",byrow = TRUE)))


# Heart 
tissue = "Heart"
selected = c(1,2,3,6,10)
file_tissue = "https://raw.githubusercontent.com/datascience-mobi-2022/2022-topic-04-team-01/main/Report/Heart_table.png"
#Overrepresentation Analysis
bool_contained = sapply(embryo_df_tissues$tissues, function(x) {grepl(tissue, toString(x))})
embryo_df_tissue = embryo_df_tissues[bool_contained,]
deGenes = as.character(embryo_df_tissue$ensembl.entrez[!duplicated(embryo_df_tissue$Gene.stable.ID)])
geneUniverse = as.character(embryo_df_tissues$ensembl.entrez[!duplicated(embryo_df_tissues$Gene.stable.ID)])
enr_GO_tissue1 = enrichGO(gene = deGenes, ont = "BP",  OrgDb ="org.Hs.eg.db", universe = geneUniverse, readable=TRUE, pvalueCutoff = 0.05)
enr_GO_tissue = as.data.frame(enr_GO_tissue1)
#Create the table of most significant functions
if (dim(enr_GO_tissue)[1] <10) {npath = dim(enr_GO_tissue)[1]} else {npath = 10}
tissue_table = enr_GO_tissue[1:npath,c(2,6)]
colnames(tissue_table) = c("Function", "Adj_p_value")
#assign(str_glue(tissue,"_table"), gt(tissue_table)|>  tab_header(title= tissue) |> fmt_markdown(columns = c(Function)) |> gt_theme_538() |> fmt_scientific(columns = c(Adj_p_value), decimals = 0) |>  tab_style(style = list(cell_text(weight = "bold")), locations = cells_body(columns = Function, rows = selected)))
#gtsave(get(str_glue(tissue, "_table")),str_glue(tissue, "_table.png"))
download.file(file_tissue, str_glue(tissue, "_table.png"), mode = "wb")
assign(str_glue(tissue, "_t"), (ggdraw() + draw_image(str_glue(tissue, "_table.png"))))
#Plot the chosen functions
data_plot = data.frame(Week = c(sapply(c(4:9), function(x) {rep(x, length(selected))})), Function =
                             as.character(c(rep(enr_GO_tissue$Description[selected], 6)), Expression=NA))
for (i in selected) {
  xx = strsplit(enr_GO_tissue$geneID[i], "/")
  bool_contained1 = sapply(diff_genes_ann_0.01$HGNC.symbol, function(x) {grepl(x, toString(xx))})
  bool_contained1[is.na(bool_contained1)] = FALSE
  data_temp = diff_genes_ann_0.01[bool_contained1,]
  data_abs = data.frame(Week4 = apply(data_temp[1:3], 1, median), Week5 = apply(data_temp[4:6], 1, median), Week6 =  
                              apply(data_temp[7:9], 1, median), Week7=apply(data_temp[10:12], 1, median), Week8 = 
                              apply(data_temp[13:15], 1, median), Week9 =  apply(data_temp[16:18], 1, median))
  expr_path = colMeans(data_abs[1:6,])
  data_plot[data_plot$Function == enr_GO_tissue$Description[i],"Expression"] = expr_path
}
assign(str_glue(tissue, "_plot"), ggplot(data_plot, aes(x = Week, y= Expression, group = Function))  + geom_line(aes(colour =Function), position = position_dodge(0)) + geom_point(aes(colour = Function, shape = Function), position = position_dodge(0)) + labs(title = tissue) +  theme_classic() + scale_color_npg()  + theme(legend.title = element_text(size = 9), legend.text = element_text(size = 7), aspect.ratio = 1, legend.position = "bottom", legend.spacing.y = unit(0.1, "cm"),legend.key.size = unit(0.4, "cm"), plot.margin = unit(c(0.5,0.8,0.2,0),"cm")) + guides(colour = guide_legend(nrow = length(selected), title.position = "top",byrow = TRUE), shape = guide_legend(nrow = length(selected), title.position = "top",byrow = TRUE)))


# Muscle - Skeletal
tissue = "Muscle - Skeletal"
selected = c(1,2,4,5,7)
file_tissue = "https://raw.githubusercontent.com/datascience-mobi-2022/2022-topic-04-team-01/main/Report/Muscle%20-%20Skeletal_table.png"
#Overrepresentation Analysis
bool_contained = sapply(embryo_df_tissues$tissues, function(x) {grepl(tissue, toString(x))})
embryo_df_tissue = embryo_df_tissues[bool_contained,]
deGenes = as.character(embryo_df_tissue$ensembl.entrez[!duplicated(embryo_df_tissue$Gene.stable.ID)])
geneUniverse = as.character(embryo_df_tissues$ensembl.entrez[!duplicated(embryo_df_tissues$Gene.stable.ID)])
enr_GO_tissue1 = enrichGO(gene = deGenes, ont = "BP",  OrgDb ="org.Hs.eg.db", universe = geneUniverse, readable=TRUE, pvalueCutoff = 0.05)
enr_GO_tissue = as.data.frame(enr_GO_tissue1)
#Create the table of most significant functions
if (dim(enr_GO_tissue)[1] <10) {npath = dim(enr_GO_tissue)[1]} else {npath = 10}
tissue_table = enr_GO_tissue[1:npath,c(2,6)]
colnames(tissue_table) = c("Function", "Adj_p_value")
#assign(str_glue(tissue,"_table"), gt(tissue_table)|>  tab_header(title= tissue) |> fmt_markdown(columns = c(Function)) |> gt_theme_538() |> fmt_scientific(columns = c(Adj_p_value), decimals = 0) |>  tab_style(style = list(cell_text(weight = "bold")), locations = cells_body(columns = Function, rows = selected)))
#gtsave(get(str_glue(tissue, "_table")),str_glue(tissue, "_table.png"))
download.file(file_tissue, str_glue(tissue, "_table.png"), mode = "wb")
assign(str_glue(tissue, "_t"), (ggdraw() + draw_image(str_glue(tissue, "_table.png"))))
#Plot the chosen functions
data_plot = data.frame(Week = c(sapply(c(4:9), function(x) {rep(x, length(selected))})), Function =
                             as.character(c(rep(enr_GO_tissue$Description[selected], 6)), Expression=NA))
for (i in selected) {
  xx = strsplit(enr_GO_tissue$geneID[i], "/")
  bool_contained1 = sapply(diff_genes_ann_0.01$HGNC.symbol, function(x) {grepl(x, toString(xx))})
  bool_contained1[is.na(bool_contained1)] = FALSE
  data_temp = diff_genes_ann_0.01[bool_contained1,]
  data_abs = data.frame(Week4 = apply(data_temp[1:3], 1, median), Week5 = apply(data_temp[4:6], 1, median), Week6 =  
                              apply(data_temp[7:9], 1, median), Week7=apply(data_temp[10:12], 1, median), Week8 = 
                              apply(data_temp[13:15], 1, median), Week9 =  apply(data_temp[16:18], 1, median))
  expr_path = colMeans(data_abs[1:6,])
  data_plot[data_plot$Function == enr_GO_tissue$Description[i],"Expression"] = expr_path
}
assign(str_glue(tissue, "_plot"), ggplot(data_plot, aes(x = Week, y= Expression, group = Function))  + geom_line(aes(colour =Function), position = position_dodge(0)) + geom_point(aes(colour = Function, shape = Function), position = position_dodge(0)) + labs(title = tissue) +  theme_classic() + scale_color_npg()  + theme(legend.title = element_text(size = 9), legend.text = element_text(size = 7), aspect.ratio = 1, legend.position = "bottom", legend.spacing.y = unit(0.1, "cm"),legend.key.size = unit(0.4, "cm"), plot.margin = unit(c(0.5,0.8,0.2,0),"cm")) + guides(colour = guide_legend(nrow = length(selected), title.position = "top",byrow = TRUE), shape = guide_legend(nrow = length(selected), title.position = "top",byrow = TRUE)))


# Testis
tissue = "Testis"
selected = c(1,4,7)
file_tissue = "https://raw.githubusercontent.com/datascience-mobi-2022/2022-topic-04-team-01/main/Report/Testis_table.png"
#Overrepresentation Analysis
bool_contained = sapply(embryo_df_tissues$tissues, function(x) {grepl(tissue, toString(x))})
embryo_df_tissue = embryo_df_tissues[bool_contained,]
deGenes = as.character(embryo_df_tissue$ensembl.entrez[!duplicated(embryo_df_tissue$Gene.stable.ID)])
geneUniverse = as.character(embryo_df_tissues$ensembl.entrez[!duplicated(embryo_df_tissues$Gene.stable.ID)])
enr_GO_tissue1 = enrichGO(gene = deGenes, ont = "BP",  OrgDb ="org.Hs.eg.db", universe = geneUniverse, readable=TRUE, pvalueCutoff = 0.05)
enr_GO_tissue = as.data.frame(enr_GO_tissue1)
#Create the table of most significant functions
if (dim(enr_GO_tissue)[1] <10) {npath = dim(enr_GO_tissue)[1]} else {npath = 10}
tissue_table = enr_GO_tissue[1:npath,c(2,6)]
colnames(tissue_table) = c("Function", "Adj_p_value")
#assign(str_glue(tissue,"_table"), gt(tissue_table)|>  tab_header(title= tissue) |> fmt_markdown(columns = c(Function)) |> gt_theme_538() |> fmt_scientific(columns = c(Adj_p_value), decimals = 0) |>  tab_style(style = list(cell_text(weight = "bold")), locations = cells_body(columns = Function, rows = selected)))
#gtsave(get(str_glue(tissue, "_table")),str_glue(tissue, "_table.png"))
download.file(file_tissue, str_glue(tissue, "_table.png"), mode = "wb")
assign(str_glue(tissue, "_t"), (ggdraw() + draw_image(str_glue(tissue, "_table.png"))))
#Plot the chosen functions
data_plot = data.frame(Week = c(sapply(c(4:9), function(x) {rep(x, length(selected))})), Function =
                             as.character(c(rep(enr_GO_tissue$Description[selected], 6)), Expression=NA))
for (i in selected) {
  xx = strsplit(enr_GO_tissue$geneID[i], "/")
  bool_contained1 = sapply(diff_genes_ann_0.01$HGNC.symbol, function(x) {grepl(x, toString(xx))})
  bool_contained1[is.na(bool_contained1)] = FALSE
  data_temp = diff_genes_ann_0.01[bool_contained1,]
  data_abs = data.frame(Week4 = apply(data_temp[1:3], 1, median), Week5 = apply(data_temp[4:6], 1, median), Week6 =  
                              apply(data_temp[7:9], 1, median), Week7=apply(data_temp[10:12], 1, median), Week8 = 
                              apply(data_temp[13:15], 1, median), Week9 =  apply(data_temp[16:18], 1, median))
  expr_path = colMeans(data_abs[1:6,])
  data_plot[data_plot$Function == enr_GO_tissue$Description[i],"Expression"] = expr_path
}
assign(str_glue(tissue, "_plot"), ggplot(data_plot, aes(x = Week, y= Expression, group = Function))  + geom_line(aes(colour =Function), position = position_dodge(0)) + geom_point(aes(colour = Function, shape = Function), position = position_dodge(0)) + labs(title = tissue) +  theme_classic() + scale_color_npg()  + theme(legend.title = element_text(size = 9), legend.text = element_text(size = 7), aspect.ratio = 1, legend.position = "bottom", legend.spacing.y = unit(0.1, "cm"),legend.key.size = unit(0.4, "cm"), plot.margin = unit(c(0.5,0.8,0.2,0),"cm")) + guides(colour = guide_legend(nrow = length(selected), title.position = "top",byrow = TRUE), shape = guide_legend(nrow = length(selected), title.position = "top",byrow = TRUE)))

#Stomach
tissue = "Stomach"
selected = c(1,2,4)
file_tissue = "https://raw.githubusercontent.com/datascience-mobi-2022/2022-topic-04-team-01/main/Report/Stomach_table.png"
#Overrepresentation Analysis
bool_contained = sapply(embryo_df_tissues$tissues, function(x) {grepl(tissue, toString(x))})
embryo_df_tissue = embryo_df_tissues[bool_contained,]
deGenes = as.character(embryo_df_tissue$ensembl.entrez[!duplicated(embryo_df_tissue$Gene.stable.ID)])
geneUniverse = as.character(embryo_df_tissues$ensembl.entrez[!duplicated(embryo_df_tissues$Gene.stable.ID)])
enr_GO_tissue1 = enrichGO(gene = deGenes, ont = "BP",  OrgDb ="org.Hs.eg.db", universe = geneUniverse, readable=TRUE, pvalueCutoff = 0.05)
enr_GO_tissue = as.data.frame(enr_GO_tissue1)
#Create the table of most significant functions
if (dim(enr_GO_tissue)[1] <10) {npath = dim(enr_GO_tissue)[1]} else {npath = 10}
tissue_table = enr_GO_tissue[1:npath,c(2,6)]
colnames(tissue_table) = c("Function", "Adj_p_value")
#assign(str_glue(tissue,"_table"), gt(tissue_table)|>  tab_header(title= tissue) |> fmt_markdown(columns = c(Function)) |> gt_theme_538() |> fmt_scientific(columns = c(Adj_p_value), decimals = 0) |>  tab_style(style = list(cell_text(weight = "bold")), locations = cells_body(columns = Function, rows = selected)))
#gtsave(get(str_glue(tissue, "_table")),str_glue(tissue, "_table.png"))
download.file(file_tissue, str_glue(tissue, "_table.png"), mode = "wb")
assign(str_glue(tissue, "_t"), (ggdraw() + draw_image(str_glue(tissue, "_table.png"))))
#Plot the chosen functions
data_plot = data.frame(Week = c(sapply(c(4:9), function(x) {rep(x, length(selected))})), Function =
                             as.character(c(rep(enr_GO_tissue$Description[selected], 6)), Expression=NA))
for (i in selected) {
  xx = strsplit(enr_GO_tissue$geneID[i], "/")
  bool_contained1 = sapply(diff_genes_ann_0.01$HGNC.symbol, function(x) {grepl(x, toString(xx))})
  bool_contained1[is.na(bool_contained1)] = FALSE
  data_temp = diff_genes_ann_0.01[bool_contained1,]
  data_abs = data.frame(Week4 = apply(data_temp[1:3], 1, median), Week5 = apply(data_temp[4:6], 1, median), Week6 =  
                              apply(data_temp[7:9], 1, median), Week7=apply(data_temp[10:12], 1, median), Week8 = 
                              apply(data_temp[13:15], 1, median), Week9 =  apply(data_temp[16:18], 1, median))
  expr_path = colMeans(data_abs[1:6,])
  data_plot[data_plot$Function == enr_GO_tissue$Description[i],"Expression"] = expr_path
}
assign(str_glue(tissue, "_plot"), ggplot(data_plot, aes(x = Week, y= Expression, group = Function))  + geom_line(aes(colour =Function), position = position_dodge(0)) + geom_point(aes(colour = Function, shape = Function), position = position_dodge(0)) + labs(title = tissue) +  theme_classic() + scale_color_npg()  + theme(legend.title = element_text(size = 9), legend.text = element_text(size = 7), aspect.ratio = 1, legend.position = "bottom", legend.spacing.y = unit(0.1, "cm"),legend.key.size = unit(0.4, "cm"), plot.margin = unit(c(0.5,0.8,0.8,0),"cm")) + guides(colour = guide_legend(nrow = length(selected), title.position = "top",byrow = TRUE), shape = guide_legend(nrow = length(selected), title.position = "top",byrow = TRUE)))


#Skin
tissue = "Skin"
selected = c(1,2,3,4)
file_tissue = "https://raw.githubusercontent.com/datascience-mobi-2022/2022-topic-04-team-01/main/Report/Skin_table.png"
#Overrepresentation Analysis
bool_contained = sapply(embryo_df_tissues$tissues, function(x) {grepl(tissue, toString(x))})
embryo_df_tissue = embryo_df_tissues[bool_contained,]
deGenes = as.character(embryo_df_tissue$ensembl.entrez[!duplicated(embryo_df_tissue$Gene.stable.ID)])
geneUniverse = as.character(embryo_df_tissues$ensembl.entrez[!duplicated(embryo_df_tissues$Gene.stable.ID)])
enr_GO_tissue1 = enrichGO(gene = deGenes, ont = "BP",  OrgDb ="org.Hs.eg.db", universe = geneUniverse, readable=TRUE, pvalueCutoff = 0.05)
enr_GO_tissue = as.data.frame(enr_GO_tissue1)
#Create the table of most significant functions
if (dim(enr_GO_tissue)[1] <10) {npath = dim(enr_GO_tissue)[1]} else {npath = 10}
tissue_table = enr_GO_tissue[1:npath,c(2,6)]
colnames(tissue_table) = c("Function", "Adj_p_value")
#assign(str_glue(tissue,"_table"), gt(tissue_table)|>  tab_header(title= tissue) |> fmt_markdown(columns = c(Function)) |> gt_theme_538() |> fmt_scientific(columns = c(Adj_p_value), decimals = 0) |>  tab_style(style = list(cell_text(weight = "bold")), locations = cells_body(columns = Function, rows = selected)))
#gtsave(get(str_glue(tissue, "_table")),str_glue(tissue, "_table.png"))
download.file(file_tissue, str_glue(tissue, "_table.png"), mode = "wb")
assign(str_glue(tissue, "_t"), (ggdraw() + draw_image(str_glue(tissue, "_table.png"))))
#Plot the chosen functions
data_plot = data.frame(Week = c(sapply(c(4:9), function(x) {rep(x, length(selected))})), Function =
                             as.character(c(rep(enr_GO_tissue$Description[selected], 6)), Expression=NA))
for (i in selected) {
  xx = strsplit(enr_GO_tissue$geneID[i], "/")
  bool_contained1 = sapply(diff_genes_ann_0.01$HGNC.symbol, function(x) {grepl(x, toString(xx))})
  bool_contained1[is.na(bool_contained1)] = FALSE
  data_temp = diff_genes_ann_0.01[bool_contained1,]
  data_abs = data.frame(Week4 = apply(data_temp[1:3], 1, median), Week5 = apply(data_temp[4:6], 1, median), Week6 =  
                              apply(data_temp[7:9], 1, median), Week7=apply(data_temp[10:12], 1, median), Week8 = 
                              apply(data_temp[13:15], 1, median), Week9 =  apply(data_temp[16:18], 1, median))
  expr_path = colMeans(data_abs[1:6,])
  data_plot[data_plot$Function == enr_GO_tissue$Description[i],"Expression"] = expr_path
}
assign(str_glue(tissue, "_plot"), ggplot(data_plot, aes(x = Week, y= Expression, group = Function))  + geom_line(aes(colour =Function), position = position_dodge(0)) + geom_point(aes(colour = Function, shape = Function), position = position_dodge(0)) + labs(title = tissue) +  theme_classic() + scale_color_npg()  + theme(legend.title = element_text(size = 9), legend.text = element_text(size = 7), aspect.ratio = 1, legend.position = "bottom", legend.spacing.y = unit(0.1, "cm"),legend.key.size = unit(0.4, "cm"), plot.margin = unit(c(0.5,0.8,0.8,0),"cm")) + guides(colour = guide_legend(nrow = length(selected), title.position = "top",byrow = TRUE), shape = guide_legend(nrow = length(selected), title.position = "top",byrow = TRUE)))

```

```{r ORA-plot, fig.width = 12, fig.height=15, fig.cap = "For eight tissues A to H, the tables show the 10 most significant results from overrepresentation analysis. There, the functional GO annotation of all transcripts in our dataset (not only the differentially expressed ones) was compared to the functions associated with all TRAs of one tissue. The significance was determined by an adjusted p-value. Of those 10 gene sets, up to 5 were chosen to represent different groups of functions and avoid the overlap of highly related processes. For each of those, the median expression for each week was calculated and plotted, as seen in the expression graphs."}
#Arrange the plots for all eight tissues
lay = rbind(c(1,2,2,3,4,4),c(5,6,6,7,8,8), c(9,10,10,11,12,12),c(13,14,14,15,16,16))
plot = arrangeGrob(grobs = list(Spleen_t,Spleen_plot, Brain_t, Brain_plot, Heart_t, Heart_plot, Liver_t, Liver_plot,get("Muscle - Skeletal_t"), get("Muscle - Skeletal_plot"), Testis_t, Testis_plot, Stomach_t, Stomach_plot, Skin_t, Skin_plot), norw =4)
plot_output = as_ggplot(plot) + draw_plot_label(label = c("A", "B", "C", "D", "E", "F", "G", "H"), size = 15, x = c(0.01, 0.51,0.01, 0.51,0.01, 0.51,0.01, 0.51), y = c(0.79,0.79,0.52,0.52,0.28,0.28,0.02,0.02)) 
annotate_figure(plot_output, top = text_grob("Main functions from overrepresentation analysis plotted by tissue", color = "black", face = "bold", size = 16))
```

For the spleen (Fig. \@ref(fig:ORA-plot)A), we determined a largely constant expression of immune-related genes throughout the time frame, with a slight increase in some functions from week 7-9.
The brain (Fig. \@ref(fig:ORA-plot)B) showed an increase in neuron projection morphogenesis from a previously inactive state (expression \< 6.8) in week 4 to a significant expression (\>7.5) by week 8. Synaptic signaling stayed at relatively constant expression levels.
Heart-associated functions (Fig. \@ref(fig:ORA-plot)C) can be grouped into two categories. Cardiac functions (cardiac muscle tissue development, heart contraction) are highly expressed in week 4 and fall continously until week 8. In contrast, general muscle gene sets are rising from originally lower expression levels during the observed time.
The liver (Fig. \@ref(fig:ORA-plot)D) shows no clear expression patterns, with some metabolic functions increasing through time (organic hydroxy compound metabolic process) while others stay mostly constant (cellular amino acid metabolic process) or fall (organic acid catabolic process). In contrast, skeletal muscle gene sets (Fig. \@ref(fig:ORA-plot)E) show a very clear trend. After a mostly slight increase between week 4 and 8, a sharp rise in expression levels is visible from week 8 to 9.
The testis-associated sets (Fig. \@ref(fig:ORA-plot)F) continuously decrease in expression from week 5 onward.
For the stomach (Fig. \@ref(fig:ORA-plot)G), we found a initially high expression in week 4 that then falls until week 6 and then increases again towards week 9.
Finally, the skin-associated functions (Fig. \@ref(fig:ORA-plot)H) all displayed a constant rise in expression levels from week 5 to 9.
