---
title: "Test3"
author: "Paul Christmann"
date: '2022-07-12'
output: pdf_document
geometry: "left=1.5cm,right=1.5cm,top=1.5cm,bottom=1.5cm"
header-includes:
  - \usepackage[font={small,it}, labelfont={bf}]{caption}
---

```{r knitr_init, echo=FALSE, cache=FALSE}
library(knitr)
#library(rmdformats)
## Global options
#options(max.print="120")
opts_chunk$set(echo=FALSE,
	             cache=TRUE,
               prompt=FALSE,
               tidy=TRUE,
               comment=NA,
               message=FALSE,
               warning=FALSE)
#opts_knit$set(width=120)
#opts_knit$set(root.dir  = "~/")
```

### The expression of all TRAs associated with a tissue cannot be used to infer organ development {#Organ-dev1}

In this research, we attempt to draw conclusions about the developmental state of a tissue based on the expression of genes associated with it alone. Therefore, we analyzed the share of differentially expressed transcripts above a certain expression level over time, as shown in Fig. ???A. Furthermore, we observed trends within the median expression of all differentially expressed transcripts associated with a tissue (Fig. ???B). Since both metrics only showed in miniscule changes, we hypothesized that distinct, counteracting trends in expression exsisted within one tissue. Thus, k-means clustering was used to determine groups of TRAs with similar expression patterns. For each of these clusters, the median expression was plotted as shown in Fig. ???C. 

```{r}
library(affy)
library(AnnotationDbi)
library(hgu133plus2hsenstprobe)
library(hgu133plus2hsenstcdf) 
library(Rcpp)
library(tidyverse)
library(vsn)
library(limma)
library(pheatmap)
library(hexbin)
library(biomaRt)
library(kableExtra)
library(Rfssa)
library(ggbiplot)
library(magrittr) 
library(dplyr)
library(stringr)
library(ggrepel)
library(readxl)
library(ggpubr)
library(grid)
library(gridExtra)
library(cowplot)
library(RCurl)
library(readxl)
library(ggforce)
library(VennDiagram)
library(png)
library(factoextra)
library(cluster)
library(ggplot2)
library(ggsci)
library(ggplotify)
library(clusterProfiler)
library(enrichplot)
library(org.Hs.eg.db)
library(ggupset)
library(gt)
library(gtExtras)
library(svglite)
library(webshot)
library(rentrez)
library(XML)
library(igraph)
library(ggnet)
library(viridis)
library(treemapify)
library(magick)
```


```{r}
library(tidyverse)
load("diff_genes_ann_0.01.RData")
load("embryo_df_tissues.RData")
```

## Normalising the data set

Intensity values of different chips are affected by sample preparation and array manufacturing and processing resulting in statistical variance and random fluctuation. To access the biological relevant variation the raw data needs to be transformed by normalisation. We chose the vsn rma normalization with its library *vsn* according to Huber *et al.* (2002).

```{r ORA-code, include = FALSE}
#Rearrange the selected plots so that we can combine the table with a plot 


# Spleen 
tissue = "Spleen"
selected = c(1,2,3,4,5)
file_tissue = "https://raw.githubusercontent.com/datascience-mobi-2022/2022-topic-04-team-01/main/Report/Spleen_table.png"
#Overrepresentation Analysis
bool_contained = sapply(embryo_df_tissues$tissues, function(x) {grepl(tissue, toString(x))})
embryo_df_tissue = embryo_df_tissues[bool_contained,]
deGenes = as.character(embryo_df_tissue$ensembl.entrez[!duplicated(embryo_df_tissue$Gene.stable.ID)])
geneUniverse = as.character(embryo_df_tissues$ensembl.entrez[!duplicated(embryo_df_tissues$Gene.stable.ID)])
enr_GO_tissue1 = enrichGO(gene = deGenes, ont = "BP",  OrgDb ="org.Hs.eg.db", universe = geneUniverse, readable=TRUE, pvalueCutoff = 0.05)
enr_GO_tissue = as.data.frame(enr_GO_tissue1)
#Create the table of most significant functions
if (dim(enr_GO_tissue)[1] <10) {npath = dim(enr_GO_tissue)[1]} else {npath = 10}
tissue_table = enr_GO_tissue[1:npath,c(2,6)]
colnames(tissue_table) = c("Function", "Adj_p_value")
#assign(str_glue(tissue,"_table"), gt(tissue_table)|>  tab_header(title= tissue) |> fmt_markdown(columns = c(Function)) |> gt_theme_538() |> fmt_scientific(columns = c(Adj_p_value), decimals = 0) |>  tab_style(style = list(cell_text(weight = "bold")), locations = cells_body(columns = Function, rows = selected)))
#gtsave(get(str_glue(tissue, "_table")),str_glue(tissue, "_table.png"))
download.file(file_tissue, str_glue(tissue, "_table.png"), mode = "wb")
assign(str_glue(tissue, "_t"), (ggdraw() + draw_image(str_glue(tissue, "_table.png"))))
#Plot the chosen functions
data_plot = data.frame(Week = c(sapply(c(4:9), function(x) {rep(x, length(selected))})), Function =
                             as.character(c(rep(enr_GO_tissue$Description[selected], 6)), Expression=NA))
for (i in selected) {
  xx = strsplit(enr_GO_tissue$geneID[i], "/")
  bool_contained1 = sapply(diff_genes_ann_0.01$HGNC.symbol, function(x) {grepl(x, toString(xx))})
  bool_contained1[is.na(bool_contained1)] = FALSE
  data_temp = diff_genes_ann_0.01[bool_contained1,]
  data_abs = data.frame(Week4 = apply(data_temp[1:3], 1, median), Week5 = apply(data_temp[4:6], 1, median), Week6 =  
                              apply(data_temp[7:9], 1, median), Week7=apply(data_temp[10:12], 1, median), Week8 = 
                              apply(data_temp[13:15], 1, median), Week9 =  apply(data_temp[16:18], 1, median))
  expr_path = colMeans(data_abs[1:6,])
  data_plot[data_plot$Function == enr_GO_tissue$Description[i],"Expression"] = expr_path
}
assign(str_glue(tissue, "_plot"), ggplot(data_plot, aes(x = Week, y= Expression, group = Function))  + geom_line(aes(colour =Function), position = position_dodge(0)) + geom_point(aes(colour = Function, shape = Function), position = position_dodge(0)) + labs(title = tissue) +  theme_classic() + scale_color_npg()  + theme(legend.title = element_text(size = 9), legend.text = element_text(size = 7), aspect.ratio = 1, legend.position = "bottom", legend.spacing.y = unit(0.1, "cm"),legend.key.size = unit(0.4, "cm"), plot.margin = unit(c(0.5,0.8,0.2,0),"cm")) + guides(colour = guide_legend(nrow = length(selected), title.position = "top",byrow = TRUE), shape = guide_legend(nrow = length(selected), title.position = "top",byrow = TRUE)))




# Brain 
tissue = "Brain"
selected = c(1,5,7)
file_tissue = "https://raw.githubusercontent.com/datascience-mobi-2022/2022-topic-04-team-01/main/Report/Brain_table.png"
#Overrepresentation Analysis
bool_contained = sapply(embryo_df_tissues$tissues, function(x) {grepl(tissue, toString(x))})
embryo_df_tissue = embryo_df_tissues[bool_contained,]
deGenes = as.character(embryo_df_tissue$ensembl.entrez[!duplicated(embryo_df_tissue$Gene.stable.ID)])
geneUniverse = as.character(embryo_df_tissues$ensembl.entrez[!duplicated(embryo_df_tissues$Gene.stable.ID)])
enr_GO_tissue1 = enrichGO(gene = deGenes, ont = "BP",  OrgDb ="org.Hs.eg.db", universe = geneUniverse, readable=TRUE, pvalueCutoff = 0.05)
enr_GO_tissue = as.data.frame(enr_GO_tissue1)
#Create the table of most significant functions
if (dim(enr_GO_tissue)[1] <10) {npath = dim(enr_GO_tissue)[1]} else {npath = 10}
tissue_table = enr_GO_tissue[1:npath,c(2,6)]
colnames(tissue_table) = c("Function", "Adj_p_value")
#assign(str_glue(tissue,"_table"), gt(tissue_table)|>  tab_header(title= tissue) |> fmt_markdown(columns = c(Function)) |> gt_theme_538() |> fmt_scientific(columns = c(Adj_p_value), decimals = 0) |>  tab_style(style = list(cell_text(weight = "bold")), locations = cells_body(columns = Function, rows = selected)))
#gtsave(get(str_glue(tissue, "_table")),str_glue(tissue, "_table.png"))
download.file(file_tissue, str_glue(tissue, "_table.png"), mode = "wb")
assign(str_glue(tissue, "_t"), (ggdraw() + draw_image(str_glue(tissue, "_table.png"))))
#Plot the chosen functions
data_plot = data.frame(Week = c(sapply(c(4:9), function(x) {rep(x, length(selected))})), Function =
                             as.character(c(rep(enr_GO_tissue$Description[selected], 6)), Expression=NA))
for (i in selected) {
  xx = strsplit(enr_GO_tissue$geneID[i], "/")
  bool_contained1 = sapply(diff_genes_ann_0.01$HGNC.symbol, function(x) {grepl(x, toString(xx))})
  bool_contained1[is.na(bool_contained1)] = FALSE
  data_temp = diff_genes_ann_0.01[bool_contained1,]
  data_abs = data.frame(Week4 = apply(data_temp[1:3], 1, median), Week5 = apply(data_temp[4:6], 1, median), Week6 =  
                              apply(data_temp[7:9], 1, median), Week7=apply(data_temp[10:12], 1, median), Week8 = 
                              apply(data_temp[13:15], 1, median), Week9 =  apply(data_temp[16:18], 1, median))
  expr_path = colMeans(data_abs[1:6,])
  data_plot[data_plot$Function == enr_GO_tissue$Description[i],"Expression"] = expr_path
}
assign(str_glue(tissue, "_plot"), ggplot(data_plot, aes(x = Week, y= Expression, group = Function))  + geom_line(aes(colour =Function), position = position_dodge(0)) + geom_point(aes(colour = Function, shape = Function), position = position_dodge(0)) + labs(title = tissue) +  theme_classic() + scale_color_npg()  + theme(legend.title = element_text(size = 9), legend.text = element_text(size = 7), aspect.ratio = 1, legend.position = "bottom", legend.spacing.y = unit(0.1, "cm"),legend.key.size = unit(0.4, "cm"), plot.margin = unit(c(0.5,0.8,0.2,0),"cm")) + guides(colour = guide_legend(nrow = length(selected), title.position = "top",byrow = TRUE), shape = guide_legend(nrow = length(selected), title.position = "top",byrow = TRUE)))


# Liver 
tissue = "Liver"
selected = c(1,3,5,7,9)
file_tissue = "https://raw.githubusercontent.com/datascience-mobi-2022/2022-topic-04-team-01/main/Report/Liver_table.png"
#Overrepresentation Analysis
bool_contained = sapply(embryo_df_tissues$tissues, function(x) {grepl(tissue, toString(x))})
embryo_df_tissue = embryo_df_tissues[bool_contained,]
deGenes = as.character(embryo_df_tissue$ensembl.entrez[!duplicated(embryo_df_tissue$Gene.stable.ID)])
geneUniverse = as.character(embryo_df_tissues$ensembl.entrez[!duplicated(embryo_df_tissues$Gene.stable.ID)])
enr_GO_tissue1 = enrichGO(gene = deGenes, ont = "BP",  OrgDb ="org.Hs.eg.db", universe = geneUniverse, readable=TRUE, pvalueCutoff = 0.05)
enr_GO_tissue = as.data.frame(enr_GO_tissue1)
#Create the table of most significant functions
if (dim(enr_GO_tissue)[1] <10) {npath = dim(enr_GO_tissue)[1]} else {npath = 10}
tissue_table = enr_GO_tissue[1:npath,c(2,6)]
colnames(tissue_table) = c("Function", "Adj_p_value")
#assign(str_glue(tissue,"_table"), gt(tissue_table)|>  tab_header(title= tissue) |> fmt_markdown(columns = c(Function)) |> gt_theme_538() |> fmt_scientific(columns = c(Adj_p_value), decimals = 0) |>  tab_style(style = list(cell_text(weight = "bold")), locations = cells_body(columns = Function, rows = selected)))
#gtsave(get(str_glue(tissue, "_table")),str_glue(tissue, "_table.png"))
download.file(file_tissue, str_glue(tissue, "_table.png"), mode = "wb")
assign(str_glue(tissue, "_t"), (ggdraw() + draw_image(str_glue(tissue, "_table.png"))))
#Plot the chosen functions
data_plot = data.frame(Week = c(sapply(c(4:9), function(x) {rep(x, length(selected))})), Function =
                             as.character(c(rep(enr_GO_tissue$Description[selected], 6)), Expression=NA))
for (i in selected) {
  xx = strsplit(enr_GO_tissue$geneID[i], "/")
  bool_contained1 = sapply(diff_genes_ann_0.01$HGNC.symbol, function(x) {grepl(x, toString(xx))})
  bool_contained1[is.na(bool_contained1)] = FALSE
  data_temp = diff_genes_ann_0.01[bool_contained1,]
  data_abs = data.frame(Week4 = apply(data_temp[1:3], 1, median), Week5 = apply(data_temp[4:6], 1, median), Week6 =  
                              apply(data_temp[7:9], 1, median), Week7=apply(data_temp[10:12], 1, median), Week8 = 
                              apply(data_temp[13:15], 1, median), Week9 =  apply(data_temp[16:18], 1, median))
  expr_path = colMeans(data_abs[1:6,])
  data_plot[data_plot$Function == enr_GO_tissue$Description[i],"Expression"] = expr_path
}
assign(str_glue(tissue, "_plot"), ggplot(data_plot, aes(x = Week, y= Expression, group = Function))  + geom_line(aes(colour =Function), position = position_dodge(0)) + geom_point(aes(colour = Function, shape = Function), position = position_dodge(0)) + labs(title = tissue) +  theme_classic() + scale_color_npg()  + theme(legend.title = element_text(size = 9), legend.text = element_text(size = 7), aspect.ratio = 1, legend.position = "bottom", legend.spacing.y = unit(0.1, "cm"),legend.key.size = unit(0.4, "cm"), plot.margin = unit(c(0.5,0.8,0.2,0),"cm")) + guides(colour = guide_legend(nrow = length(selected), title.position = "top",byrow = TRUE), shape = guide_legend(nrow = length(selected), title.position = "top",byrow = TRUE)))


# Heart 
tissue = "Heart"
selected = c(1,2,3,6,10)
file_tissue = "https://raw.githubusercontent.com/datascience-mobi-2022/2022-topic-04-team-01/main/Report/Heart_table.png"
#Overrepresentation Analysis
bool_contained = sapply(embryo_df_tissues$tissues, function(x) {grepl(tissue, toString(x))})
embryo_df_tissue = embryo_df_tissues[bool_contained,]
deGenes = as.character(embryo_df_tissue$ensembl.entrez[!duplicated(embryo_df_tissue$Gene.stable.ID)])
geneUniverse = as.character(embryo_df_tissues$ensembl.entrez[!duplicated(embryo_df_tissues$Gene.stable.ID)])
enr_GO_tissue1 = enrichGO(gene = deGenes, ont = "BP",  OrgDb ="org.Hs.eg.db", universe = geneUniverse, readable=TRUE, pvalueCutoff = 0.05)
enr_GO_tissue = as.data.frame(enr_GO_tissue1)
#Create the table of most significant functions
if (dim(enr_GO_tissue)[1] <10) {npath = dim(enr_GO_tissue)[1]} else {npath = 10}
tissue_table = enr_GO_tissue[1:npath,c(2,6)]
colnames(tissue_table) = c("Function", "Adj_p_value")
#assign(str_glue(tissue,"_table"), gt(tissue_table)|>  tab_header(title= tissue) |> fmt_markdown(columns = c(Function)) |> gt_theme_538() |> fmt_scientific(columns = c(Adj_p_value), decimals = 0) |>  tab_style(style = list(cell_text(weight = "bold")), locations = cells_body(columns = Function, rows = selected)))
#gtsave(get(str_glue(tissue, "_table")),str_glue(tissue, "_table.png"))
download.file(file_tissue, str_glue(tissue, "_table.png"), mode = "wb")
assign(str_glue(tissue, "_t"), (ggdraw() + draw_image(str_glue(tissue, "_table.png"))))
#Plot the chosen functions
data_plot = data.frame(Week = c(sapply(c(4:9), function(x) {rep(x, length(selected))})), Function =
                             as.character(c(rep(enr_GO_tissue$Description[selected], 6)), Expression=NA))
for (i in selected) {
  xx = strsplit(enr_GO_tissue$geneID[i], "/")
  bool_contained1 = sapply(diff_genes_ann_0.01$HGNC.symbol, function(x) {grepl(x, toString(xx))})
  bool_contained1[is.na(bool_contained1)] = FALSE
  data_temp = diff_genes_ann_0.01[bool_contained1,]
  data_abs = data.frame(Week4 = apply(data_temp[1:3], 1, median), Week5 = apply(data_temp[4:6], 1, median), Week6 =  
                              apply(data_temp[7:9], 1, median), Week7=apply(data_temp[10:12], 1, median), Week8 = 
                              apply(data_temp[13:15], 1, median), Week9 =  apply(data_temp[16:18], 1, median))
  expr_path = colMeans(data_abs[1:6,])
  data_plot[data_plot$Function == enr_GO_tissue$Description[i],"Expression"] = expr_path
}
assign(str_glue(tissue, "_plot"), ggplot(data_plot, aes(x = Week, y= Expression, group = Function))  + geom_line(aes(colour =Function), position = position_dodge(0)) + geom_point(aes(colour = Function, shape = Function), position = position_dodge(0)) + labs(title = tissue) +  theme_classic() + scale_color_npg()  + theme(legend.title = element_text(size = 9), legend.text = element_text(size = 7), aspect.ratio = 1, legend.position = "bottom", legend.spacing.y = unit(0.1, "cm"),legend.key.size = unit(0.4, "cm"), plot.margin = unit(c(0.5,0.8,0.2,0),"cm")) + guides(colour = guide_legend(nrow = length(selected), title.position = "top",byrow = TRUE), shape = guide_legend(nrow = length(selected), title.position = "top",byrow = TRUE)))


# Muscle - Skeletal
tissue = "Muscle - Skeletal"
selected = c(1,2,4,5,7)
file_tissue = "https://raw.githubusercontent.com/datascience-mobi-2022/2022-topic-04-team-01/main/Report/Muscle%20-%20Skeletal_table.png"
#Overrepresentation Analysis
bool_contained = sapply(embryo_df_tissues$tissues, function(x) {grepl(tissue, toString(x))})
embryo_df_tissue = embryo_df_tissues[bool_contained,]
deGenes = as.character(embryo_df_tissue$ensembl.entrez[!duplicated(embryo_df_tissue$Gene.stable.ID)])
geneUniverse = as.character(embryo_df_tissues$ensembl.entrez[!duplicated(embryo_df_tissues$Gene.stable.ID)])
enr_GO_tissue1 = enrichGO(gene = deGenes, ont = "BP",  OrgDb ="org.Hs.eg.db", universe = geneUniverse, readable=TRUE, pvalueCutoff = 0.05)
enr_GO_tissue = as.data.frame(enr_GO_tissue1)
#Create the table of most significant functions
if (dim(enr_GO_tissue)[1] <10) {npath = dim(enr_GO_tissue)[1]} else {npath = 10}
tissue_table = enr_GO_tissue[1:npath,c(2,6)]
colnames(tissue_table) = c("Function", "Adj_p_value")
#assign(str_glue(tissue,"_table"), gt(tissue_table)|>  tab_header(title= tissue) |> fmt_markdown(columns = c(Function)) |> gt_theme_538() |> fmt_scientific(columns = c(Adj_p_value), decimals = 0) |>  tab_style(style = list(cell_text(weight = "bold")), locations = cells_body(columns = Function, rows = selected)))
#gtsave(get(str_glue(tissue, "_table")),str_glue(tissue, "_table.png"))
download.file(file_tissue, str_glue(tissue, "_table.png"), mode = "wb")
assign(str_glue(tissue, "_t"), (ggdraw() + draw_image(str_glue(tissue, "_table.png"))))
#Plot the chosen functions
data_plot = data.frame(Week = c(sapply(c(4:9), function(x) {rep(x, length(selected))})), Function =
                             as.character(c(rep(enr_GO_tissue$Description[selected], 6)), Expression=NA))
for (i in selected) {
  xx = strsplit(enr_GO_tissue$geneID[i], "/")
  bool_contained1 = sapply(diff_genes_ann_0.01$HGNC.symbol, function(x) {grepl(x, toString(xx))})
  bool_contained1[is.na(bool_contained1)] = FALSE
  data_temp = diff_genes_ann_0.01[bool_contained1,]
  data_abs = data.frame(Week4 = apply(data_temp[1:3], 1, median), Week5 = apply(data_temp[4:6], 1, median), Week6 =  
                              apply(data_temp[7:9], 1, median), Week7=apply(data_temp[10:12], 1, median), Week8 = 
                              apply(data_temp[13:15], 1, median), Week9 =  apply(data_temp[16:18], 1, median))
  expr_path = colMeans(data_abs[1:6,])
  data_plot[data_plot$Function == enr_GO_tissue$Description[i],"Expression"] = expr_path
}
assign(str_glue(tissue, "_plot"), ggplot(data_plot, aes(x = Week, y= Expression, group = Function))  + geom_line(aes(colour =Function), position = position_dodge(0)) + geom_point(aes(colour = Function, shape = Function), position = position_dodge(0)) + labs(title = tissue) +  theme_classic() + scale_color_npg()  + theme(legend.title = element_text(size = 9), legend.text = element_text(size = 7), aspect.ratio = 1, legend.position = "bottom", legend.spacing.y = unit(0.1, "cm"),legend.key.size = unit(0.4, "cm"), plot.margin = unit(c(0.5,0.8,0.2,0),"cm")) + guides(colour = guide_legend(nrow = length(selected), title.position = "top",byrow = TRUE), shape = guide_legend(nrow = length(selected), title.position = "top",byrow = TRUE)))


# Testis
tissue = "Testis"
selected = c(1,4,7)
file_tissue = "https://raw.githubusercontent.com/datascience-mobi-2022/2022-topic-04-team-01/main/Report/Testis_table.png"
#Overrepresentation Analysis
bool_contained = sapply(embryo_df_tissues$tissues, function(x) {grepl(tissue, toString(x))})
embryo_df_tissue = embryo_df_tissues[bool_contained,]
deGenes = as.character(embryo_df_tissue$ensembl.entrez[!duplicated(embryo_df_tissue$Gene.stable.ID)])
geneUniverse = as.character(embryo_df_tissues$ensembl.entrez[!duplicated(embryo_df_tissues$Gene.stable.ID)])
enr_GO_tissue1 = enrichGO(gene = deGenes, ont = "BP",  OrgDb ="org.Hs.eg.db", universe = geneUniverse, readable=TRUE, pvalueCutoff = 0.05)
enr_GO_tissue = as.data.frame(enr_GO_tissue1)
#Create the table of most significant functions
if (dim(enr_GO_tissue)[1] <10) {npath = dim(enr_GO_tissue)[1]} else {npath = 10}
tissue_table = enr_GO_tissue[1:npath,c(2,6)]
colnames(tissue_table) = c("Function", "Adj_p_value")
#assign(str_glue(tissue,"_table"), gt(tissue_table)|>  tab_header(title= tissue) |> fmt_markdown(columns = c(Function)) |> gt_theme_538() |> fmt_scientific(columns = c(Adj_p_value), decimals = 0) |>  tab_style(style = list(cell_text(weight = "bold")), locations = cells_body(columns = Function, rows = selected)))
#gtsave(get(str_glue(tissue, "_table")),str_glue(tissue, "_table.png"))
download.file(file_tissue, str_glue(tissue, "_table.png"), mode = "wb")
assign(str_glue(tissue, "_t"), (ggdraw() + draw_image(str_glue(tissue, "_table.png"))))
#Plot the chosen functions
data_plot = data.frame(Week = c(sapply(c(4:9), function(x) {rep(x, length(selected))})), Function =
                             as.character(c(rep(enr_GO_tissue$Description[selected], 6)), Expression=NA))
for (i in selected) {
  xx = strsplit(enr_GO_tissue$geneID[i], "/")
  bool_contained1 = sapply(diff_genes_ann_0.01$HGNC.symbol, function(x) {grepl(x, toString(xx))})
  bool_contained1[is.na(bool_contained1)] = FALSE
  data_temp = diff_genes_ann_0.01[bool_contained1,]
  data_abs = data.frame(Week4 = apply(data_temp[1:3], 1, median), Week5 = apply(data_temp[4:6], 1, median), Week6 =  
                              apply(data_temp[7:9], 1, median), Week7=apply(data_temp[10:12], 1, median), Week8 = 
                              apply(data_temp[13:15], 1, median), Week9 =  apply(data_temp[16:18], 1, median))
  expr_path = colMeans(data_abs[1:6,])
  data_plot[data_plot$Function == enr_GO_tissue$Description[i],"Expression"] = expr_path
}
assign(str_glue(tissue, "_plot"), ggplot(data_plot, aes(x = Week, y= Expression, group = Function))  + geom_line(aes(colour =Function), position = position_dodge(0)) + geom_point(aes(colour = Function, shape = Function), position = position_dodge(0)) + labs(title = tissue) +  theme_classic() + scale_color_npg()  + theme(legend.title = element_text(size = 9), legend.text = element_text(size = 7), aspect.ratio = 1, legend.position = "bottom", legend.spacing.y = unit(0.1, "cm"),legend.key.size = unit(0.4, "cm"), plot.margin = unit(c(0.5,0.8,0.2,0),"cm")) + guides(colour = guide_legend(nrow = length(selected), title.position = "top",byrow = TRUE), shape = guide_legend(nrow = length(selected), title.position = "top",byrow = TRUE)))

#Stomach
tissue = "Stomach"
selected = c(1,2,4)
file_tissue = "https://raw.githubusercontent.com/datascience-mobi-2022/2022-topic-04-team-01/main/Report/Stomach_table.png"
#Overrepresentation Analysis
bool_contained = sapply(embryo_df_tissues$tissues, function(x) {grepl(tissue, toString(x))})
embryo_df_tissue = embryo_df_tissues[bool_contained,]
deGenes = as.character(embryo_df_tissue$ensembl.entrez[!duplicated(embryo_df_tissue$Gene.stable.ID)])
geneUniverse = as.character(embryo_df_tissues$ensembl.entrez[!duplicated(embryo_df_tissues$Gene.stable.ID)])
enr_GO_tissue1 = enrichGO(gene = deGenes, ont = "BP",  OrgDb ="org.Hs.eg.db", universe = geneUniverse, readable=TRUE, pvalueCutoff = 0.05)
enr_GO_tissue = as.data.frame(enr_GO_tissue1)
#Create the table of most significant functions
if (dim(enr_GO_tissue)[1] <10) {npath = dim(enr_GO_tissue)[1]} else {npath = 10}
tissue_table = enr_GO_tissue[1:npath,c(2,6)]
colnames(tissue_table) = c("Function", "Adj_p_value")
#assign(str_glue(tissue,"_table"), gt(tissue_table)|>  tab_header(title= tissue) |> fmt_markdown(columns = c(Function)) |> gt_theme_538() |> fmt_scientific(columns = c(Adj_p_value), decimals = 0) |>  tab_style(style = list(cell_text(weight = "bold")), locations = cells_body(columns = Function, rows = selected)))
#gtsave(get(str_glue(tissue, "_table")),str_glue(tissue, "_table.png"))
download.file(file_tissue, str_glue(tissue, "_table.png"), mode = "wb")
assign(str_glue(tissue, "_t"), (ggdraw() + draw_image(str_glue(tissue, "_table.png"))))
#Plot the chosen functions
data_plot = data.frame(Week = c(sapply(c(4:9), function(x) {rep(x, length(selected))})), Function =
                             as.character(c(rep(enr_GO_tissue$Description[selected], 6)), Expression=NA))
for (i in selected) {
  xx = strsplit(enr_GO_tissue$geneID[i], "/")
  bool_contained1 = sapply(diff_genes_ann_0.01$HGNC.symbol, function(x) {grepl(x, toString(xx))})
  bool_contained1[is.na(bool_contained1)] = FALSE
  data_temp = diff_genes_ann_0.01[bool_contained1,]
  data_abs = data.frame(Week4 = apply(data_temp[1:3], 1, median), Week5 = apply(data_temp[4:6], 1, median), Week6 =  
                              apply(data_temp[7:9], 1, median), Week7=apply(data_temp[10:12], 1, median), Week8 = 
                              apply(data_temp[13:15], 1, median), Week9 =  apply(data_temp[16:18], 1, median))
  expr_path = colMeans(data_abs[1:6,])
  data_plot[data_plot$Function == enr_GO_tissue$Description[i],"Expression"] = expr_path
}
assign(str_glue(tissue, "_plot"), ggplot(data_plot, aes(x = Week, y= Expression, group = Function))  + geom_line(aes(colour =Function), position = position_dodge(0)) + geom_point(aes(colour = Function, shape = Function), position = position_dodge(0)) + labs(title = tissue) +  theme_classic() + scale_color_npg()  + theme(legend.title = element_text(size = 9), legend.text = element_text(size = 7), aspect.ratio = 1, legend.position = "bottom", legend.spacing.y = unit(0.1, "cm"),legend.key.size = unit(0.4, "cm"), plot.margin = unit(c(0.5,0.8,0.8,0),"cm")) + guides(colour = guide_legend(nrow = length(selected), title.position = "top",byrow = TRUE), shape = guide_legend(nrow = length(selected), title.position = "top",byrow = TRUE)))


#Skin
tissue = "Skin"
selected = c(1,2,3,4)
file_tissue = "https://raw.githubusercontent.com/datascience-mobi-2022/2022-topic-04-team-01/main/Report/Skin_table.png"
#Overrepresentation Analysis
bool_contained = sapply(embryo_df_tissues$tissues, function(x) {grepl(tissue, toString(x))})
embryo_df_tissue = embryo_df_tissues[bool_contained,]
deGenes = as.character(embryo_df_tissue$ensembl.entrez[!duplicated(embryo_df_tissue$Gene.stable.ID)])
geneUniverse = as.character(embryo_df_tissues$ensembl.entrez[!duplicated(embryo_df_tissues$Gene.stable.ID)])
enr_GO_tissue1 = enrichGO(gene = deGenes, ont = "BP",  OrgDb ="org.Hs.eg.db", universe = geneUniverse, readable=TRUE, pvalueCutoff = 0.05)
enr_GO_tissue = as.data.frame(enr_GO_tissue1)
#Create the table of most significant functions
if (dim(enr_GO_tissue)[1] <10) {npath = dim(enr_GO_tissue)[1]} else {npath = 10}
tissue_table = enr_GO_tissue[1:npath,c(2,6)]
colnames(tissue_table) = c("Function", "Adj_p_value")
#assign(str_glue(tissue,"_table"), gt(tissue_table)|>  tab_header(title= tissue) |> fmt_markdown(columns = c(Function)) |> gt_theme_538() |> fmt_scientific(columns = c(Adj_p_value), decimals = 0) |>  tab_style(style = list(cell_text(weight = "bold")), locations = cells_body(columns = Function, rows = selected)))
#gtsave(get(str_glue(tissue, "_table")),str_glue(tissue, "_table.png"))
download.file(file_tissue, str_glue(tissue, "_table.png"), mode = "wb")
assign(str_glue(tissue, "_t"), (ggdraw() + draw_image(str_glue(tissue, "_table.png"))))
#Plot the chosen functions
data_plot = data.frame(Week = c(sapply(c(4:9), function(x) {rep(x, length(selected))})), Function =
                             as.character(c(rep(enr_GO_tissue$Description[selected], 6)), Expression=NA))
for (i in selected) {
  xx = strsplit(enr_GO_tissue$geneID[i], "/")
  bool_contained1 = sapply(diff_genes_ann_0.01$HGNC.symbol, function(x) {grepl(x, toString(xx))})
  bool_contained1[is.na(bool_contained1)] = FALSE
  data_temp = diff_genes_ann_0.01[bool_contained1,]
  data_abs = data.frame(Week4 = apply(data_temp[1:3], 1, median), Week5 = apply(data_temp[4:6], 1, median), Week6 =  
                              apply(data_temp[7:9], 1, median), Week7=apply(data_temp[10:12], 1, median), Week8 = 
                              apply(data_temp[13:15], 1, median), Week9 =  apply(data_temp[16:18], 1, median))
  expr_path = colMeans(data_abs[1:6,])
  data_plot[data_plot$Function == enr_GO_tissue$Description[i],"Expression"] = expr_path
}
assign(str_glue(tissue, "_plot"), ggplot(data_plot, aes(x = Week, y= Expression, group = Function))  + geom_line(aes(colour =Function), position = position_dodge(0)) + geom_point(aes(colour = Function, shape = Function), position = position_dodge(0)) + labs(title = tissue) +  theme_classic() + scale_color_npg()  + theme(legend.title = element_text(size = 9), legend.text = element_text(size = 7), aspect.ratio = 1, legend.position = "bottom", legend.spacing.y = unit(0.1, "cm"),legend.key.size = unit(0.4, "cm"), plot.margin = unit(c(0.5,0.8,0.8,0),"cm")) + guides(colour = guide_legend(nrow = length(selected), title.position = "top",byrow = TRUE), shape = guide_legend(nrow = length(selected), title.position = "top",byrow = TRUE)))

```


```{r ORA-plot, fig.width = 12, fig.height=15, fig.cap = "For eight tissues A to H, the tables show the 10 most significant results from overrepresentation analysis. There, the functional GO annotation of all transcripts in our dataset (not only the differentially expressed ones) was compared to the functions associated with all TRAs of one tissue. The significance was determined by an adjusted p-value. Of those 10 functions, up to 5 were chosen to represent different groups of functions and avoid the overlap of highly related processes. For each of those, the median expression for each week was calculated and plotted, as seen in the expression graphs."}
#Arrange the plots for all eight tissues
lay = rbind(c(1,2,2,3,4,4),c(5,6,6,7,8,8), c(9,10,10,11,12,12),c(13,14,14,15,16,16))
plot = arrangeGrob(grobs = list(Spleen_t,Spleen_plot, Brain_t, Brain_plot, Heart_t, Heart_plot, Liver_t, Liver_plot,get("Muscle - Skeletal_t"), get("Muscle - Skeletal_plot"), Testis_t, Testis_plot, Stomach_t, Stomach_plot, Skin_t, Skin_plot), norw =4)
plot_output = as_ggplot(plot) + draw_plot_label(label = c("A", "B", "C", "D", "E", "F", "G", "H"), size = 15, x = c(0.01, 0.51,0.01, 0.51,0.01, 0.51,0.01, 0.51), y = c(0.79,0.79,0.52,0.52,0.28,0.28,0.02,0.02)) 
annotate_figure(plot_output, top = text_grob("Main functions from overrepresentation analysis plotted by tissue", color = "black", face = "bold", size = 16))
```


For the spleen (Fig. \@ref(fig:ORA-plot)A), we determined a largely constant expression of immune-related genes throughout the time frame, with a slight increase in some functions from week 7-9. \
The brain (Fig. \@ref(fig:ORA-plot)B) showed an increase in neuron projection morphogenesis from a previously inactive state (expression < 6.8) in week 4 to a significant expression (>7.5) by week 8. Synaptic signaling stayed at relatively constant expression levels. \
Heart-associated functions (Fig. \@ref(fig:ORA-plot)C) can be grouped into two categories. Cardiac functions (cardiac muscle tissue development, heart contraction) are highly expressed in week 4 and fall continously until week 8. In contrast, general muscle gene sets are rising from originally lower expression levels during the observed time. \
The liver (Fig. \@ref(fig:ORA-plot)D) shows no clear expression patterns, with some metabolic functions increasing through time (organic hydroxy compound metabolic process) while others stay mostly constant (cellular amino acid metabolic process) or fall (organic acid catabolic process). 
In contrast, skeletal muscle gene sets (Fig. \@ref(fig:ORA-plot)E) show a very clear trend. After a mostly slight increase between week 4 and 8, a sharp rise in expression levels is visible from week 8 to 9. \
The testis-associated sets (Fig. \@ref(fig:ORA-plot)F) continuously decrease in expression from week 5 onward. \
For the stomach (Fig. \@ref(fig:ORA-plot)G), we found a initially high expression in week 4 that then falls until week 6 and then increases again towards week 9. \
Finally, the skin-associated functions (Fig. \@ref(fig:ORA-plot)H) all displayed a constant rise in expression levels from week 5 to 9. \




# Discussion


## Hypothesis: TRAs can infer a timeline of organ development similar to the results by Yi et al. 2010
In our analysis, we have shown that a number of TRAs are differentially expressed (section \@ref(organ-overview)) between week 4 and 9 of human embryonic development in each of the analyzed tissues. Nonetheless, the expression levels of TRAs associated with one tissue do not constitute a useful metric for the organ's development (section \@ref(organ-clustering)). This can be explained by the fact that within one tissue's TRAs, there are multiple groups of genes both distinct in expression patterns (clustering in section \@ref(organ-clustering)) and function (analysis of spleen gene functions in section \@ref(organ-tables)). Thus, we determined that the expression over time of functional gene sets linked to specific tissues through overrepresentation analysis is a more meaningful metric for organ development. 

This approach was used in section \@ref(organ-ora) for eight different tissues. 
For the spleen, the results of our analysis (Fig. \@ref(fig:ORA-plot)A) largely do not reflect the embryonic development (section \@ref(intro-tissues)). While some of the immune-related gene sets are already expressed in week 4, the spleen only develops by week 6 and contains immune cells by week 12. This shows that while the spleen plays a role in the immune system and such gene sets are therefore rightly linked to the spleen, the expression of these transcripts alone does not necessarily relate to the development of the organ. It is still noteworthy that functions related to the adaptive immune system increase in expression form week 7 onward, which correlates with the beginning of T-cell development in the thymus. \
The observed timeframe is an important part of brain development (section \@ref(intro-tissues)). This is also visible in the expression data (Fig. \@ref(fig:ORA-plot)B), with a already high but still continously increasing expression of synaptic gene sets. Furthermore, as the brain starts to form, the expression of neuron projection morphogenesis transcripts increases continuously from week 5 to 8. \
At week 4, the clearly heart-associated gene sets (Fig. \@ref(fig:ORA-plot)C) are at their highest expression level and decrease until week 8. The cardiac muscle tissue development transcripts still remain highly expressed (>7.5). This corresponds to the early development of the heart as noted in the introduction (section \@ref(intro-tissues)). It is noteworthy that the heart contraction gene set rises in expression again from week 8 to 9, but here an explanation is not possible without further analyzing the individual genes. \
The liver-associated TRAs showed no clear expression pattern (Fig. \@ref(fig:ORA-plot)D). Thus, even though the liver forms mostly during the analyzed timeframe (section \@ref(intro-tissues)), we cannot link the gene expression to the organ's development. The detected functions are mostly metabolic pathways whose activity could also be related to processes outside the liver. As a result, it is plausible that their expression is independent of liver development. \
The skeletal muscle functions are expressed only late within the observed time, as shown by the large increase in expression from  week 8 to 9 (Fig. \@ref(fig:ORA-plot)E). As muscle fibers begin to develop later than week 9 and the first related proteins appear from week 7 on (section \@ref(intro-tissues)), these expression data correspond well to the embryonic development. \
The testis gene sets decrease in expression from week 5 onward (Fig. \@ref(fig:ORA-plot)E). This is in contrast to the embryonic development, where the gonads start to form at around the same time (section \@ref(intro-tissues)). \
For the stomach, the expression pattern indicates a decrease until week 6 followed by rising expression levels until week 9 (Fig. \@ref(fig:ORA-plot)G). However, the literature indicates that these results are unrelated to the stomach development. Functions like digestion or peptide hormone secretion are impossible to occure at this time, since the specific cells needed for this only appear later in embryogenesis (section \@ref(intro-tissues)). Therfore, the cause of the changing expression would have to be determined through a more in-depth analysis of the involved genes. \
Finally, the skin shows an increased expression of related genes sets from week 5 through 9 (Fig. \@ref(fig:ORA-plot)H). This broadly reflects the embryonic development, with the epidermis starting to form in week 4 (section \@ref(intro-tissues)). We also found this expression pattern in the keratinization gene set that is suggested by literature as a good indicator for skin formation.\



## Dataset
We obtained the data set from Yi H *et al.* (2010). We chose this data set by the following criteria, it contains human embryonic data and it covers every week between the 4th and 9th week, which are interesting stages of embryogenesis and organ development. Three replica at each point in time were tested, hence data from 18 embryos were acquired. The timezone covers the Carnegie stages 10-23, finishing the process of embryogenesis and organogenesis. This peroid of embryogenesis is highly regulated with considerable differential gene expression. Overall, the data set suits the requirements for our purpose.

### Affymetrix U133 plus 2.0 human GeneChip array

The data was generated from embryos by using Affymetrix U133 plus 2.0 human GeneChip arrays. RNA microarrays are slides coted with oligonucleotides as matrices which screen for thousands of transcripts. [The HG-U133 Plus 2.0](https://www.affymetrix.com/support/technical/datasheets/human_datasheet.pdf) allows the detection of about 50,000 transcripts and uses quality control matrices. The Affymetrix chip include 62 control transcripts, whose intensities are imported together with the acquired data.

### Importing the data set
We downloaded the raw data to a local harddrive from the Gene Expression Omnibus with the Accession Number of [GSE15744](https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE15744). We imported it with the help of the library *affy* and is connected to the correct Annotation by the brainarray package. The *affy* package allows more manageable data analysis and manipulation of microarray intensity values.
