---
title: "Final Report"
author: "Gruppe 4"
date: '2022-07-03'
bibliography: references.bib
output: 
  html_document:
    number_sections: TRUE
    toc: TRUE
    theme: united
---

```{r knitr_init, echo=FALSE, cache=FALSE}
library(knitr)
#library(rmdformats)
## Global options
#options(max.print="120")
opts_chunk$set(echo=FALSE,
	             cache=TRUE,
               prompt=FALSE,
               tidy=TRUE,
               comment=NA,
               message=FALSE,
               warning=FALSE)
#opts_knit$set(width=120)
#opts_knit$set(root.dir  = "~/")
```

```{r}
library(affy)
library(AnnotationDbi)
library(hgu133plus2hsenstprobe)
library(hgu133plus2hsenstcdf) 
library(Rcpp)
library(tidyverse)
library(vsn)
library(limma)
library(pheatmap)
library(hexbin)
library(biomaRt)
library(kableExtra)
library(Rfssa)
library(ggbiplot)
library(magrittr) 
library(dplyr)
library(stringr)
library(ggrepel)
```

#Introduction
Our expression data was gained from embryos between week 4 to 9, hence it's essential to know what happens in organogenesis during this week in the most prominent organs: the liver, the gonades, the spleen, the heart the skeletal muscles and the brain.

Let us start with the liver. At week 4, the liver sprout further develops to hepathocytes and intrahepathic gallic pathways a network of venes develop. Between week 5 to 9 the production of gallic acid starts. Furthermore glykogengranula develop at week 8 and glykogensynthesis starts at week 9. Between week 8 and 9 the expression of cytochrome-P450-enzyme start (Deutsch, 2013).

*Quellen: https://www.ncbi.nlm.nih.gov/pmc/articles/PMC7498810/ *

The next organ are the gonades their progenitors develop next to the progenitor of kidney and are made up of three different cellforms, the coelomic epithelium form, the progenitor kidney form and the germline cell form. Those cells move to the gonades in week 5 to 6. They develop to ovaries and testis at week7 (Benninghoff; 1993). 

*Quellen:Benninghoff 1993 BENNINGHOFF, A.:Makroskopische Anatomie, Embryologie und Histologie des Menschen. 15. Auflage. München; Wien; Baltimore : Urban und Schwarzenberg, 1993*

The spleen

The shape of the heart resembels a tube at week 3. The different heart parts develope at week 4 due to curvation and growth and the ventricels are created. the fundamental layout of the heart is already present at week 4 to 5. Further remodelling takes place till week 7 ( Ulfig; 2009; Hikspoors; 2022)

*Quellen: https://viamedici.thieme.de/lernmodul/542754/532616/herz+entwicklung; https://www.nature.com/articles/s42003-022-03153-x *

The skeletal muscles start developing at week 5 ()

Major parts of the brain are developed by week 9. Characteristic neuronal development steps take place during this periode, e.g. neuronal proliferation, neuronal differentiation, neuronal immigration, synapse formation (O'Connell, 2009;Müller, 2018). Many signals influence the development of the brain e.g. chemokines.

*Quellen: https://nap.nationalacademies.org/catalog/12480/preventing-mental-emotional-and-behavioral-disorders-among-young-people-progress; https://link.springer.com/content/pdf/10.1007/978-3-662-55534-7.pdf *

Chemokines are a groupe of small proteins, interacting as chemoattractors on effector cells. They are classified in 4 groups (alpha to delta) depending on the position of their first cysteins (C). In the alpha groupe (or CXC) they are separeted by a single aminoacid. In the beta groupe (or CC) they are next to each other. In the gamma groupe (or C) there is only one cystein present. In the delta groupe (or CX3C) they are separeted by three aminoacids. (Yusuf,2005). Chemokines induce cell migration by binding to their respective receptors (a G-Protein coupled receptor), which are often shortend with an R. E.g. CXCR4, the receptor of the alpha class ligand CXCL12. The function of chemokines during embryonal development is a target of further research. Nevertheless, the CXCL12/CXCR4 signalling pathway plays an important role for the neuronal cell migration (Tiveron 2008).

*Quellen: https://link.springer.com/article/10.1007/s00429-005-0013-9; https://reader.elsevier.com/reader/sd/pii/S0959438808000561?token=EBA0F4DA59B38154C30F1D421B5B2862BBDCC72383E8C74729CDC6C06634C0F7065B6A96BBC72E3CC83E0D075D61D79C&originRegion=eu-west-1&originCreation=20220712150847 *


*Quellen: <https://link.springer.com/article/10.1007/s00429-005-0013-9>; <https://reader.elsevier.com/reader/sd/pii/S0959438808000561?token=EBA0F4DA59B38154C30F1D421B5B2862BBDCC72383E8C74729CDC6C06634C0F7065B6A96BBC72E3CC83E0D075D61D79C&originRegion=eu-west-1&originCreation=20220712150847>*

#Methods

## Programming language and Libraries

The freely available programming language R version 4.2.0 and its IDE RStudio were used to draw statistical conclusions and generate informative plots. The used code packages were installed from CRAN, an online network with submitted libraries for specific programming and statistical purpose. More precisely, some packages were downloaded from bioconductor, an open software library build by developers of the community specifically for biological assays. Packages for annotation purposes of microarrays were provided by brainarray.

Following libraries were used:
```{r table-with-all-libraries, eval = FALSE}
all_libraries <- tibble::tribble(
  ~Library...package,                        ~Version,                              ~Description,
              "affy",                              NA, "library used for unpacking the raw data",
               "vsn",                              NA,   "library for the vsn rma normalization",
     "AnnotationDbi",                              NA,                                        NA,
         "tidyverse", "library used for Datacleaning",                       "Library / package",
           "Version",                   "Description",                                    "affy"
  )

all_libraries %>%
  kable(all_libraries, digits = 3, row.names = FALSE, align = "c", caption = "Table 1: summary of all libraries used") %>%
  kable_styling(bootstrap_options = "striped", full_width = FALSE)

```


## Dataset

We obtained the data set from Yi H *et al.* (2010). We chose this data set by the following criteria, it contains human embryonic data and it covers every week between the 4th and 9th week, which are interesting stages of embryogenesis and organ development. Three replica at each point in time were tested, hence data from 18 embryos were acquired.
The timezone covers the Carnegie stages 10-23, finishing the process of embryogenesis and organogenesis. This peroid of embryogenesis is highly regulated with considerable differential gene expression.
Overall, the data set suits the requirements for our purpose.

### Affymetrix U133 plus 2.0 human GeneChip array

The data was generated from embryos by using Affymetrix U133 plus 2.0 human GeneChip arrays. RNA microarrays are slides coted with oligonucleotides as matrices which screen for thousands of transcripts. [The HG-U133 Plus 2.0](https://www.affymetrix.com/support/technical/datasheets/human_datasheet.pdf) allows the detection of about 50,000 transcripts and uses quality control matrices.

### Importing the data set

We downloaded the raw data to a local harddrive from the Gene Expression Omnibus with the Accession Number of [GSE15744](https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE15744). We imported it with the help of the library *affy* and is connected to the correct Annotation by the brainarray package. The *affy* package allows more manageable data analysis and manipulation of microarray intensity values.

```{r importing-the-data-from-harddrive, eval = FALSE}
setwd("C:/Users/merke/Documents/Vorlesungen Material/SS22/DataAnalysis/project/rawdata")
data.GSE15744 <- ReadAffy()
data.GSE15744@cdfName <- "HGU133Plus2_Hs_ENST"

# adjusting the microarray names by cutting the .cel
rownames(pData(data.GSE15744)) <- substr(rownames(pData(data.GSE15744)), 1, nchar(rownames(pData(data.GSE15744)))-4)
colnames(exprs(data.GSE15744)) <- substr(colnames(exprs(data.GSE15744)), 1, nchar(colnames(exprs(data.GSE15744)))-4)


setwd("C:/Users/merke/Documents/Vorlesungen Material/SS22/DataAnalysis/project/report")
save(data.GSE15744, file="data.GSE15744.RData")
```

To access the data remotely, we uploaded it to the cloud-based repository hosting service github. It can be imported with the Library *Rfssa*.

```{r loading-data-from-github, eval=FALSE}
#make data accessible for everyone
githubURL <- "https://github.com/datascience-mobi-2022/2022-topic-04-team-01/blob/main/Report/data.GSE15744.RData"
load_github_data(githubURL)
```

### Quality control of the suface images

To ensure that the microarrays are without surface damaged, we checked their images. We selected two images as an example 

As shown in Figure \@ref(fig:QC-surface-images) as an example, the surface of the chips are visible and show no spatial artefacts, fingerprints, irregular dye or stripes. Some differences in overall brightness are visible but marginal.
```{r QC-surface-images, fig.width= 4, fig.height= 4, fig.cap= "**Quality control: a selected surface image of the microarrays showing no damage or artefacts.** The microarray inspection shows no irregularities and every chip is accepted for further data analysis"}
surfaceimage.GSM394525 <- image(data.GSE15744[,13], col=rainbow(100, start = 0, end = 0.75)[100:1])
## To generate all images please use:
#for (i in 1:18) {
#  image(data.GSE15744[,i], col=rainbow(100, start = 0, end = 0.75)[100:1])
#} # GSM394501 looks a bit brighter, GSM394527 looks bright
```

### Quality control of RNA Degradation

We can further analyse the quality of the microarrays by checking for low RNA quality chips. Coted matrices degrades under unfavorable conditions, which negatively affects raw intensities (**doi: 10.1093/bioinformatics/bts629**). By plotting the RNA degradation for 3'--5' strand, we can compare the different chips (Figure \@ref(fig:RNA-Degradation-Plot)).

```{r RNA-Degradation-Plot, fig.width= 8, fig.height= 8, fig.cap= "**Quality control: RNA degradation plot shows slight irregularities and verifies the data.** Some crossing lines can be seen, especally the microarray GSM394519. We decided that the inconsistencies are minor though, and kept all microarrays to avoid the loss of potentionally relevant data."}
data.rnadegradation <- AffyRNAdeg(data.GSE15744)

plotAffyRNAdeg(data.rnadegradation, col=rainbow(18))
title(sub="RNA Degradation of 18 microarrays, GSE15744")

## Not scaled
#plotAffyRNAdeg(data.rnadegradation, col=rainbow(18), transform = "shift.only")
#title(sub="Human embryo (GSE15744, Raw data)")
```

## Normalising the data set

Intensity values of different chips are affected by sample preparation and array manufacturing and processing resulting in statistical variance and random fluctuation. To access the biological relevant variation the raw data needs to be transformed by normalisation. We chose the vsn rma normalization with its library *vsn* according to Huber *et al.* (2002).

```{r vsn-normalization}
data.GSE15744.vsnrma <- vsnrma(data.GSE15744)
```

The library vsn is designed to process microarray intensity values. It calibrates data and applies *generalized log*-transformation, which is an adjusted natural logharithm and preserves statistical significance.


### Quality control of the vsn normalization

To verify the transformed data intensity values, some test can be performed (Figure \@ref(fig:)).
After the normalization, the rank of the mean of the intensity values and their standard deviation should not correlate. Therefore we can plot the rank of the mean against the standard deviation to control the normalization method and should get a horizontal line indicated in red (Figure \@ref(fig:)A).
Another way to control the normalization is to visualize the intensity values. Here we have two options. We used boxplots to compare each of the 18 microarray separately by its mean, median and variance. This allows us to knock out unfitting arrays (Figure \@ref(fig:)B). The second option gives us the ability to zoom in even further. The intensity levels of three replica should be the same, since they were taken at the same time. We can use scatterplots to compare single intensity levels. With one of the replica applied on the x and y axis respectively, we should see a scatterplot following the linear function y = x since the same transcript should show the same intensity in both replica (Figure \@ref(fig:)C). 




```{r mean-sd-plot}
meansd.GSE15744.vsnrma <- meanSdPlot(data.GSE15744.vsnrma, xlab="Rank (mean)", ylab="Standard Deviation")
meansdplot.GSE15744.vsnrma <- meansd.GSE15744.vsnrma$gg + ggtitle("Mean-Sd-Plot of vsn normalized data GSE15744")
```
```{r boxplot-vsn-normalization}
boxplot.GSE15744.vsnrma <- boxplot(exprs(data.GSE15744.vsnrma), col = rainbow(15),
        cex.axis=0.5,
        las=2,
        ylab="Relative expression",
        main="Boxplots of microarrays with vsn normalized data")
```
```{r Scatter-plot}
exprs.GSE15744.vsnrma <- as.data.frame(exprs(data.GSE15744.vsnrma))
scatterplot.GSE15744.vsnrma <- ggplot(data = exprs.GSE15744.vsnrma,) +
  theme_bw() +
  geom_point(mapping = aes(exprs.GSE15744.vsnrma[,4], exprs.GSE15744.vsnrma[,5]), size = 0.5) +
  geom_abline(intersept = 0, slope = 1, col = "red", aes(size=1)) +
  labs(title = paste("Scatterplot of microarray", substr(colnames(data.GSE15744.vsnrma)[4], 1, nchar(colnames(data.GSE15744.vsnrma)[4])), "and", substr(colnames(data.GSE15744.vsnrma)[5], 1, nchar(colnames(data.GSE15744.vsnrma)[5])), "in week 5", sep=" ", collapse = NULL),
       x = colnames(data.GSE15744.vsnrma)[4],
       y = colnames(data.GSE15744.vsnrma)[5]) 
```

```{r QC-normalization-plots, fig.width=9, fig.height=6, fig.subcap= "The plots support the use of the vsn normalization on our data set. A: The red line is close to horizontal, although it shows some correlation at high intensity leves. B: The boxplots show nice allignment of the mean intensity values. Some outliers are given but can neglected given the 0.25 and 0.75 quantile. C: A selected Scatterplot is shown. Very slight banana shaped structure can be seen, but only marginal. Overall the quality control confirmes successful vsn normalization"}

OC_normalization_plot = arrangeGrob(grobs = list(meansd.GSE15744.vsnrma, boxplot.GSE15744.vsnrma, exprs.GSE15744.vsnrma), nrow=1, widths = c(3, 3, 3), heights = c(3))

OC_normalization_plot_output = as_ggplot(OC_normalization_plot) +
  draw_plot_label(label = c("A", "B", "C"), size = 15, x = c(0.03, 0.35, 0.63), y = c(0.13, 0.13, 0.13)) 

annotate_figure(OC_normalization_plot_output, top = text_grob("Quality control: verifying the normalization at different levels of detail", color = "black", face = "bold", size = 16))
```


## Data cleaning



Convert to dataframe and cut of Affymetrix Control genes

```{r, eval = FALSE}
temp_embryo_df = as.data.frame(exprs(data.GSE15744.vsnrma))
colnames(temp_embryo_df) = rownames(pData(data.GSE15744.vsnrma))
temp_embryo_df
embryo_df = temp_embryo_df[63:95721,]
```

## Annotation

```{r Annotation, eval=}
library(RCurl)
file_annotation = getURL("https://raw.githubusercontent.com/datascience-mobi-2022/2022-topic-04-team-01/main/R%20Scripts/ensembl_103_human.txt")
ensembl = as.data.frame(read.csv(text=file_annotation, sep = ","))   # Load the ensembl annotation data
TranscriptID_orig = rownames(embryo_df)
ensembl = ensembl %>% distinct(Transcript.stable.ID, .keep_all = TRUE) # Remove duplicate transcripts


embryo_df["Transcript.stable.ID"] = gsub("[//._].*$" ,"", rownames(embryo_df)) # Add column for mergign
embryo_df_ann = left_join(embryo_df, ensembl, by="Transcript.stable.ID")   # Combine the Datasets by their Transcript ID
rownames(embryo_df_ann) <- embryo_df_ann$Transcript.stable.ID
colnames(embryo_df_ann) = c("Week4_1", "Week4_2", "Week4_3", "Week5_1", "Week5_2", "Week5_3", "Week6_1", "Week6_2", "Week6_3", "Week7_1", "Week7_2", "Week7_3", "Week8_1", "Week8_2", "Week8_3", "Week9_1", "Week9_2", "Week9_3", "Transcript.stable.ID", "Gene.stable.ID", "Chromosome.scaffold.name", "HGNC.symbol", "AFFY.HG.U133.Plus.2.probe")


file_tra = getURL("https://raw.githubusercontent.com/datascience-mobi-2022/2022-topic-04-team-01/main/R%20Scripts/tra.2017.human.gtex.5x.table.tsv")
tra_gtex = as.data.frame(read.csv(text = file_tra, sep = "\t"))
embryo_df_tra = embryo_df_ann[embryo_df_ann$Transcript.stable.ID %in% tra_gtex$ensembl.transcript ,]
embryo_df_tra = embryo_df_tra %>%  dplyr::rename(ensembl.transcript = Transcript.stable.ID)
embryo_df_tra =  left_join(embryo_df_tra, tra_gtex, by="ensembl.transcript") 


#Make Tissue names more accessible by creating a vector for them
a = separate(embryo_df_tra, tissues, into = as.character(c(1:30)), sep="/")
create_tissue_vector = function(f) {
  f = unlist(f)
  f = f[!is.na(f)]
  return(f)
}
tissues = apply(a[,31:60],1, create_tissue_vector)
tissues = sapply(tissues, unlist)
a = a[,c(1:30, 61)] 
a %>% add_column(tissues = NA)
a$tissues = tissues
embryo_df_tissues = a
```

## Data Cleaning

```{r, eval = FALSE}
embryo_df_tissues <- embryo_df_tissues[,!names(embryo_df_tissues) %in% c("ensembl.gene", "ensembl.symbol", "ensembl.chrom")] #removed duplicates


#Chromosome Spalte aufhübschen und unnötiges löschen
embryo_df_tissues$Chromosome.scaffold.name <- as.factor(embryo_df_tissues$Chromosome.scaffold.name) 
embryo_df_tissues <- embryo_df_tissues %>%
  filter(Chromosome.scaffold.name != "CHR_HSCHR10_1_CTG2" & Chromosome.scaffold.name != "CHR_HSCHR11_1_CTG7"& Chromosome.scaffold.name!= "CHR_HSCHR15_4_CTG8"& Chromosome.scaffold.name!= "CHR_HSCHR22_1_CTG3"& Chromosome.scaffold.name!= "CHR_HSCHR22_1_CTG7"& Chromosome.scaffold.name!=  "CHR_HSCHR4_6_CTG12"& Chromosome.scaffold.name!= "CHR_HSCHR7_2_CTG4_4"& Chromosome.scaffold.name!=  "CHR_HSCHR7_2_CTG6" )
embryo_df_tissues$Chromosome.scaffold.name <- factor(embryo_df_tissues$Chromosome.scaffold.name, levels = c("1", "2","3","4","5","6","7","8","9","10","11","12","13","14","15","16","17","18","19", "20", "21","22", "X","Y"))

# Nach NA values suchen
sum(is.na(embryo_df_tissues$ensembl.entrez)) # diese Spalte enthält 501 NA - Values
```

Datacleaning complete. Now on to some Exploritory stuff.

# Results

## Limma analysis and Volcano plot

### 1. Limma analysis

```{r, eval = FALSE}
library(limma)

limma_data = as.data.frame(embryo_df_tissues[1:18], rownames = embryo_df_tissues$ensembl.transcript)

weeks = c("week4", "week5", "week6", "week7", "week8", "week9")
f = factor(c("week4", "week4", "week4", "week5", "week5", "week5", "week6", "week6", "week6", "week7", "week7", "week7", "week8", "week8", "week8", "week9", "week9", "week9"), levels =weeks)
design = model.matrix(~0+f)
colnames(design) =  weeks
fit_2 = lmFit(limma_data, design)
contrast_all = makeContrasts(week9-week8, week9-week7, week9-week6, week9-week5, week9-week4, week8-week7,week8-week6, week8-week5, week8-week4, week7-week6, week7-week5, week7-week4, week6-week5, week6-week4, week5-week4, levels = design)
fit_all = contrasts.fit(fit_2, contrast_all)
fit_all = eBayes(fit_all)
results_all = decideTests(fit_all)
#summary(results_all)


#Create list of all differentially expressed genes
diff_genes_list_0.01 = c()

for (i in 1:15) {
 top.table = topTable(fit_all,i, n = Inf)
 n = sapply(rownames(top.table), function(x){
  embryo_df_tissues$ensembl.transcript[as.numeric(x)]
  })
  row.names(top.table) = n
  diff_genes_list_0.01 = c(diff_genes_list_0.01, rownames(top.table[which(top.table$adj.P.Val < 0.01), ]))
}
length(diff_genes_list_0.01) #4522
diff_genes_list_0.01 = diff_genes_list_0.01[!duplicated(diff_genes_list_0.01)]
length(diff_genes_list_0.01) #1814

#Select differential genes from annotated dataframe
diff_genes_ann_0.01 = embryo_df_tissues[embryo_df_tissues$ensembl.transcript %in% diff_genes_list_0.01, ]

```

```{r, eval = FALSE}
#List with 5%

diff_genes_list_0.05 = c()

for (i in 1:15) {
 top.table = topTable(fit_all,i, n = Inf)
 n = sapply(rownames(top.table), function(x){
  embryo_df_tissues$ensembl.transcript[as.numeric(x)]
  })
  row.names(top.table) = n
  diff_genes_list_0.05 = c(diff_genes_list_0.05, rownames(top.table[which(top.table$adj.P.Val < 0.05), ]))
}
diff_genes_list_0.05 = diff_genes_list_0.05[!duplicated(diff_genes_list_0.05)]
```

### 2. Volcano plot

The gathered dataset from limma analysis of the differentially expression between week 4 to 9 was used to created a volcano plot. The negativ log10 of the adjusted P-value was ploted against the logFC value. The -log10(adjusted P-value) boundary was set at 2 which equals our targeted our targeted adjusted P-value of 0.01. The logFC boundary was -1 and 1, which reflects a doubling in the expression. The genes were categorized in two groups:

1.  up-regulated if the logFC value is larger than 1 and the adjusted P-value is smaller than 0.01
2.  down-regulated if the logFC value is smaller than 1 and the adjusted P-value is smaller than 0.01

Gene which didn't belong in those groups were declared as not differentially expressed. The differentially expressed genes between all weeks with an adjusted P-value smaller then 0.01 were further used.

```{r Volcano Plot}
# Create a dataset for the volcano plot from the limma analysis
# Ausgabe der Toptabel vom Vergleich Woche 4/9
top.table = topTable(fit_all,5, n = Inf)
head(top.table)

# Create a function for volcano plot
volcanizator <- function(data, logFC_thresh_upper, logFC_thresh_lower, max_pValue){
  
  # Point colours
  mycolours <- c("blue", "red", "black")
  names(mycolours) <- c("DOWN", "UP", "NO")
  
  # Add a column to the data frame to specify if they are UP- or DOWN- regulated (log2FoldChange respectively positive or negative)
  data$diffexpressed <- "NO"
  data$diffexpressed[data$logFC > logFC_thresh_upper & data$adj.P.Val < max_pValue] <- "UP"
  data$diffexpressed[data$logFC < logFC_thresh_lower & data$adj.P.Val < max_pValue] <- "DOWN"
  
  # Create a new column with names of differential expressed genes
  data$delabel <- NA
  #data$delabel[data$diffexpressed != "NO"] <- data$HGNC.symbol[data$diffexpressed != "NO"]
  
  # Plot
  ggplot(data, aes(x=logFC, y=-log10(adj.P.Val))) +
    geom_point(aes(colour = diffexpressed), size=0.9) +
    theme_minimal() +
    geom_text_repel(aes(label = delabel)) + #, force = 5, point.padding	= unit(5, "cm")
    geom_vline(xintercept=c(logFC_thresh_lower, logFC_thresh_upper), col="red") + # log2FoldChange thresholds
    geom_hline(yintercept=-log10(max_pValue), col="red") + # p-value threshold
    scale_colour_manual(values = mycolours) +
    geom_text(aes(label = delabel)) +
    labs(title = paste("Differential gene expression: week 4 to 9"),
         subtitle = "GEO dataset: GSE15744",
         caption = "p Value adjusted: Benjamini & Hochberg (1995)"
         ) +
    xlab("log2FC") +
    ylab("-log10(adj.P.Val)")
}
volcano <- volcanizator(top.table, 1, -1, 0.01)
volcano
```

## Hypothesis: Specific TRAs can be used as key biomarkers for the development of individual organs

### Principal component analysis

Principal component analysis (PCA) was performed on a matrix containing all differentially expressed genes throughout the 6 weeks. This was performed to reduce dimension while keeping most of the data's variance.

```{r}
#pca on rows (genes)
pca_genes <- prcomp(
  t(diff_genes_ann_0.01[,1:18]), # transposed matrix
  scale. = TRUE # data scaled to have unit variance
  )

#taking a first look
summary(pca_genes)
```

To get a better grasp of how much variance is explained by each PCA the percentage variance of each PCA (scree plot) together with the cumulative variance of the PCAs together was visualized in a plot.

```{r}
#visualizing the variance of the PCAs
pc_eigenvalues <- pca_genes$sdev^2

pc_eigenvalues <- tibble(PC = factor(1:length(pc_eigenvalues)), 
                         variance = pc_eigenvalues) %>% 
  #column with percent variance
  mutate(pct = variance/sum(variance)*100) %>% 
  #column with cumulative variance explained
  mutate(pct_cum = cumsum(pct))


#actual plot
pc_eigenvalues %>% 
  ggplot(aes(x = PC)) +
  geom_col(aes(y = pct)) +
  geom_line(aes(y = pct_cum, group = 1)) + 
  geom_point(aes(y = pct_cum)) +
  labs(x = "Principal component", y = "Fraction variance explained")
```

This visualization helps to determine how many PCAs are needed to explain a significant amount of the data's variation. The chosen PCAs can than be used for further analysis. Seeing as the first three PCAs already explain over 80% of the data's variance, only those three were chosen to be used for further analysis and visualization.

The first three PCAs were now plotted against each other. This helps to analyze the genes contributions to the PCAs. The more parallel a gene's vector is to the PCA axis, the more the gene contributes to that PCA. The lenght of the vectors shows how much variability of the gene is explained by the PCAs. The longer the vector, the better it is represented in this dimension. The angels between the vectors give insight about the correlation between the genes. Small angels demonstrate a high correlation whereas opposite angles demonstrate a high negative correlation.

```{r}
#PCA 1&2
ggbiplot(pca_genes)

#PCA 2&3
ggbiplot(pca_genes, choices=c(2,3))

#PCA 1&3
ggbiplot(pca_genes, choices=c(1,3))
```

## Hypothesis: Neural TRA expression patterns reflect morphological brain development

The annotated dataset after limma analysis was filtered to determine which genes are differentially expressed in a certain tissue. The discovered genes were examined with NCBI to determine their function. The genes of interest are categorized in three groups:

1.Genes of Ion channels 2.Genes for neuronal development 3.Genes for cytokines

### Ion channels

```{r,fig.height=2.5,fig.width=2.5}
# A vektor of ENST numbers and weeks was created
ENST = embryo_df_tissues$ensembl.transcript
week = c(4,5,6,7,8,9)

# ENST00000531293
data.ENST00000531293 = as.data.frame(embryo_df_tissues %>%
                             filter(ensembl.transcript == str_extract(ENST, "ENST00000531293")))
data.ENST00000531293 = data.ENST00000531293[,1:18]
week_4 =data.ENST00000531293[1:3]
week_4 = apply(week_4,1,median)
week_5 =data.ENST00000531293[4:6]
week_5 = apply(week_5,1,median)
week_6 =data.ENST00000531293[7:9]
week_6 = apply(week_6,1,median)
week_7 =data.ENST00000531293[10:12]
week_7 = apply(week_7,1,median)
week_8 =data.ENST00000531293[13:15]
week_8 = apply(week_8,1,median)
week_9 =data.ENST00000531293[16:18]
week_9 = apply(week_9,1,median)
data.ENST00000531293 = c(week_4,week_5,week_6,week_7,week_8,week_9)
data.ENST00000531293 =cbind(data.ENST00000531293,week)

# Summary
plot(data.ENST00000531293[,2],data.ENST00000531293[,1],type = "b",lwd=1,pch=19, main = "Geneexpression level during week 4 to 9", xlab = "weeks",ylab = "Expressionlevel",col = "red")
par(new=TRUE)
legend(3.8, 9.25, legend=c("ENST00000531293"),
       col=c("red"), lty=1, cex=0.3,title="Expressed genes")
```

Ion channels play an important role in the function of neurons. We discovered that ENST00000531293 is highly expressed in Nucleus accumbens. It shows a significant increase between weeks 7 to 9 and codes for SLN sarcolipin which is a Sarcoplasmic reticulum Ca(2+)-ATPases.

### Genes for neuronal development and function

```{r,fig.height=3.2,fig.width=3.2}
# ENST00000602349
data.ENST00000602349 = as.data.frame(embryo_df_tissues %>%
                             filter(ensembl.transcript == str_extract(ENST, "ENST00000602349")))
data.ENST00000602349 = data.ENST00000602349[,1:18]
week_4 =data.ENST00000602349[1:3]
week_4 = apply(week_4,1,median)
week_5 =data.ENST00000602349[4:6]
week_5 = apply(week_5,1,median)
week_6 =data.ENST00000602349[7:9]
week_6 = apply(week_6,1,median)
week_7 =data.ENST00000602349[10:12]
week_7 = apply(week_7,1,median)
week_8 =data.ENST00000602349[13:15]
week_8 = apply(week_8,1,median)
week_9 =data.ENST00000602349[16:18]
week_9 = apply(week_9,1,median)
data.ENST00000602349 = c(week_4,week_5,week_6,week_7,week_8,week_9) 
data.ENST00000602349 =cbind(data.ENST00000602349,week)

# ENST00000276646 & ENST00000529690
data.ENST00000276646 = as.data.frame(embryo_df_tissues %>%
                             filter(ensembl.transcript == str_extract(ENST, "ENST00000276646")))
data.ENST00000276646 = data.ENST00000276646[,1:18]
week_4 =data.ENST00000276646[1:3]
week_4 = apply(week_4,1,median)
week_5 =data.ENST00000276646[4:6]
week_5 = apply(week_5,1,median)
week_6 =data.ENST00000276646[7:9]
week_6 = apply(week_6,1,median)
week_7 =data.ENST00000276646[10:12]
week_7 = apply(week_7,1,median)
week_8 =data.ENST00000276646[13:15]
week_8 = apply(week_8,1,median)
week_9 =data.ENST00000276646[16:18]
week_9 = apply(week_9,1,median)
data.ENST00000276646 = c(week_4,week_5,week_6,week_7,week_8,week_9) 
data.ENST00000276646 =cbind(data.ENST00000276646,week)

data.ENST00000529690 = as.data.frame(embryo_df_tissues %>%
                             filter(ensembl.transcript == str_extract(ENST, "ENST00000529690")))
data.ENST00000529690 = data.ENST00000529690[,1:18]
week_4 =data.ENST00000529690[1:3]
week_4 = apply(week_4,1,median)
week_5 =data.ENST00000529690[4:6]
week_5 = apply(week_5,1,median)
week_6 =data.ENST00000529690[7:9]
week_6 = apply(week_6,1,median)
week_7 =data.ENST00000529690[10:12]
week_7 = apply(week_7,1,median)
week_8 =data.ENST00000529690[13:15]
week_8 = apply(week_8,1,median)
week_9 =data.ENST00000529690[16:18]
week_9 = apply(week_9,1,median)
data.ENST00000529690 = c(week_4,week_5,week_6,week_7,week_8,week_9) 
data.ENST00000529690 =cbind(data.ENST00000529690,week)

# ENST00000356660
data.ENST00000356660 = as.data.frame(embryo_df_tissues %>%
                             filter(ensembl.transcript == str_extract(ENST, "ENST00000356660")))
data.ENST00000356660 = data.ENST00000356660[,1:18]
week_4 =data.ENST00000356660[1:3]
week_4 = apply(week_4,1,median)
week_5 =data.ENST00000356660[4:6]
week_5 = apply(week_5,1,median)
week_6 =data.ENST00000356660[7:9]
week_6 = apply(week_6,1,median)
week_7 =data.ENST00000356660[10:12]
week_7 = apply(week_7,1,median)
week_8 =data.ENST00000356660[13:15]
week_8 = apply(week_8,1,median)
week_9 =data.ENST00000356660[16:18]
week_9 = apply(week_9,1,median)
data.ENST00000356660 = c(week_4,week_5,week_6,week_7,week_8,week_9) 
data.ENST00000356660 =cbind(data.ENST00000356660,week)

# ENST00000518312 & ENST00000521485
data.ENST00000518312 = as.data.frame(embryo_df_tissues %>%
                             filter(ensembl.transcript == str_extract(ENST, "ENST00000518312")))
data.ENST00000518312 = data.ENST00000518312[,1:18]
week_4 =data.ENST00000518312[1:3]
week_4 = apply(week_4,1,median)
week_5 =data.ENST00000518312[4:6]
week_5 = apply(week_5,1,median)
week_6 =data.ENST00000518312[7:9]
week_6 = apply(week_6,1,median)
week_7 =data.ENST00000518312[10:12]
week_7 = apply(week_7,1,median)
week_8 =data.ENST00000518312[13:15]
week_8 = apply(week_8,1,median)
week_9 =data.ENST00000518312[16:18]
week_9 = apply(week_9,1,median)
data.ENST00000518312 = c(week_4,week_5,week_6,week_7,week_8,week_9) 
data.ENST00000518312 =cbind(data.ENST00000518312,week)

data.ENST00000521485 = as.data.frame(embryo_df_tissues %>%
                             filter(ensembl.transcript == str_extract(ENST, "ENST00000521485")))
data.ENST00000521485 = data.ENST00000521485[,1:18]
week_4 =data.ENST00000521485[1:3]
week_4 = apply(week_4,1,median)
week_5 =data.ENST00000521485[4:6]
week_5 = apply(week_5,1,median)
week_6 =data.ENST00000521485[7:9]
week_6 = apply(week_6,1,median)
week_7 =data.ENST00000521485[10:12]
week_7 = apply(week_7,1,median)
week_8 =data.ENST00000521485[13:15]
week_8 = apply(week_8,1,median)
week_9 =data.ENST00000521485[16:18]
week_9 = apply(week_9,1,median)
data.ENST00000521485 = c(week_4,week_5,week_6,week_7,week_8,week_9) 
data.ENST00000521485 =cbind(data.ENST00000521485,week)

# ENST00000577440
data.ENST00000577440 = as.data.frame(embryo_df_tissues %>%
                             filter(ensembl.transcript == str_extract(ENST, "ENST00000577440")))
data.ENST00000577440 = data.ENST00000577440[,1:18]
week_4 =data.ENST00000577440[1:3]
week_4 = apply(week_4,1,median)
week_5 =data.ENST00000577440[4:6]
week_5 = apply(week_5,1,median)
week_6 =data.ENST00000577440[7:9]
week_6 = apply(week_6,1,median)
week_7 =data.ENST00000577440[10:12]
week_7 = apply(week_7,1,median)
week_8 =data.ENST00000577440[13:15]
week_8 = apply(week_8,1,median)
week_9 =data.ENST00000577440[16:18]
week_9 = apply(week_9,1,median)
data.ENST00000577440 = c(week_4,week_5,week_6,week_7,week_8,week_9) 
data.ENST00000577440 =cbind(data.ENST00000577440,week)

# ENST00000439476
data.ENST00000439476 = as.data.frame(embryo_df_tissues %>%
                             filter(ensembl.transcript == str_extract(ENST, "ENST00000439476")))
data.ENST00000439476 = data.ENST00000439476[,1:18]
week_4 =data.ENST00000439476[1:3]
week_4 = apply(week_4,1,median)
week_5 =data.ENST00000439476[4:6]
week_5 = apply(week_5,1,median)
week_6 =data.ENST00000439476[7:9]
week_6 = apply(week_6,1,median)
week_7 =data.ENST00000439476[10:12]
week_7 = apply(week_7,1,median)
week_8 =data.ENST00000439476[13:15]
week_8 = apply(week_8,1,median)
week_9 =data.ENST00000439476[16:18]
week_9 = apply(week_9,1,median)
data.ENST00000439476 = c(week_4,week_5,week_6,week_7,week_8,week_9) 
data.ENST00000439476 =cbind(data.ENST00000439476,week)


# ENST00000539563
data.ENST00000539563 = as.data.frame(embryo_df_tissues %>%
                             filter(ensembl.transcript == str_extract(ENST, "ENST00000539563")))
data.ENST00000539563 = data.ENST00000539563[,1:18]
week_4 =data.ENST00000539563[1:3]
week_4 = apply(week_4,1,median)
week_5 =data.ENST00000539563[4:6]
week_5 = apply(week_5,1,median)
week_6 =data.ENST00000539563[7:9]
week_6 = apply(week_6,1,median)
week_7 =data.ENST00000539563[10:12]
week_7 = apply(week_7,1,median)
week_8 =data.ENST00000539563[13:15]
week_8 = apply(week_8,1,median)
week_9 =data.ENST00000539563[16:18]
week_9 = apply(week_9,1,median)
data.ENST00000539563 = c(week_4,week_5,week_6,week_7,week_8,week_9) 
data.ENST00000539563 =cbind(data.ENST00000539563,week)

# Summary
plot(data.ENST00000602349[,2],data.ENST00000602349[,1],type = "b",lwd=1,pch=19, main = "Geneexpression level during week 4 to 9", xlab = "weeks",ylab = "Expressionlevel",col = "red", ylim = c(6.3,8.4))
par(new=TRUE)
plot(data.ENST00000276646[,2],data.ENST00000276646[,1],type = "b",lwd=1,pch=19, main = "Geneexpression level during week 4 to 9", xlab = "weeks",ylab = "Expressionlevel",col = "blue", ylim = c(6.3,8.4)) 
par(new=TRUE)
plot(data.ENST00000529690[,2],data.ENST00000529690[,1],type = "b",lwd=1,pch=19, main = "Geneexpression level during week 4 to 9", xlab = "weeks",ylab = "Expressionlevel",col = "orange", ylim = c(6.3,8.4))
par(new=TRUE)
plot(data.ENST00000356660[,2],data.ENST00000356660[,1],type = "b",lwd=1,pch=19, main = "Geneexpression level during week 4 to 9", xlab = "weeks",ylab = "Expressionlevel",col = "green", ylim =c(6.3,8.4))
par(new=TRUE)
plot(data.ENST00000518312[,2],data.ENST00000518312[,1],type = "b",lwd=1,pch=19, main = "Geneexpression level during week 4 to 9", xlab = "weeks",ylab = "Expressionlevel",col = "black", ylim = c(6.3,8.4))
par(new=TRUE)
plot(data.ENST00000521485[,2],data.ENST00000521485[,1],type = "b",lwd=1,pch=19, main = "Geneexpression level during week 4 to 9", xlab = "weeks",ylab = "Expressionlevel",col = "pink", ylim = c(6.3,8.4)) 
par(new=TRUE)
plot(data.ENST00000577440[,2],data.ENST00000577440[,1],type = "b",lwd=1,pch=19, main = "Geneexpression level during week 4 to 9", xlab = "weeks",ylab = "Expressionlevel",col = "purple", ylim = c(6.3,8.4))
par(new=TRUE)
plot(data.ENST00000439476[,2],data.ENST00000439476[,1],type = "b",lwd=1,pch=19, main = "Geneexpression level during week 4 to 9", xlab = "weeks",ylab = "Expressionlevel",col = "salmon", ylim = c(6.3,8.4))
par(new=TRUE)
plot(data.ENST00000539563[,2],data.ENST00000539563[,1],type = "b",lwd=1,pch=19, main = "Geneexpression level during week 4 to 9", xlab = "weeks",ylab = "Expressionlevel",col = "yellowgreen", ylim = c(6.3,8.4))
legend(3.8, 8.5, legend=c("ENST00000602349", "ENST00000276646", "ENST00000529690", "ENST00000356660", "ENST00000518312", "ENST00000521485", "ENST00000577440", "ENST00000439476", "ENST00000539563"),
       col=c("red", "blue", "orange", "green","black", "pink", "purple", "salmon", "yellowgreen"), lty=1, cex=0.3,title="Expressed genes")

```

The second groupe are genes with a specific function in the neuronal development and function.

#### Genes for neuronal function 

Therefore we discovered that ENST00000276646 and ENST00000529690 show an significant increase in gene expression over the weeks. Both genes were connected to the Cerebellar Hemisphere and were associated with SYBU (syntabulin). SYBU plays an important role as it contributes to activity-dependent presynaptic assembly in neuronal development.

#### Genes for axon guidance

Filtering mentioned four genes for axon guidance:

ENST00000602349 codes for NXPH1 (neurexophilin 1) which forms a thight complex with neurexins. Neurexins promote the adhesion between axons and dendrites. ENST00000602349 shows a strong increase, especially between week 7 to 8 and is connected to Anterior cingulate cortex.

ENST00000518312 and ENST00000521485 encode for SNAP91 (synaptosome associated protein 91) which interacts in regulation of clathrin-dependent endocytosis. Therefore SNAP91 is important for axonal functions of neurons like postsynaptic density. ENST00000518312 & ENST00000521485 were associated with Cerebellar Hemisphere and also show significant increase between weeks 7 to 8.

In addition ENST00000539563, encoding for LSAMP (limbic system associated membrane protein) interacts in axon guidance. The encoded preprotein is processed to neuronal surface glycoprotein which interacts as an adhesion molecule during axon guidance and neuronal growth in the developing limbic system. ENST00000539563 is associated to Putamen which is grouped in basal ganglia. Basal ganglia were associated with the lymbic system.

#### Genes for neuronal survival

ENST00000356660 and ENST00000439476 code for BDNF (brain derived neurotrophic factor). A binding of BDNF to its receptor promotes neuronal survival. Both genes show an identical decline in expression over the weeks. Nevertheles, ENST00000356660 is connected to Cerebellar Hemisphere and ENST00000439476 to Hippocampus.

#### Brain associated gene

Finally ENST00000577440 encoding for SEPTIN4 (septin 4) was identified. SEPTIN4 may regulate cytoskeletal organization. A defect in septin function disturbs cytokinesis. ENST00000577440 is connected to Cerebellar Hemisphere and shows a significant increase between weeks 7 to 9.

### Genes for cytokine

```{r,fig.height=3.2,fig.width=3.2}
# ENST00000337225
data.ENST00000337225 = as.data.frame(embryo_df_tissues %>%
                             filter(ensembl.transcript == str_extract(ENST, "ENST00000337225")))
data.ENST00000337225 = data.ENST00000337225[,1:18]
week_4 =data.ENST00000337225[1:3]
week_4 = apply(week_4,1,median)
week_5 =data.ENST00000337225[4:6]
week_5 = apply(week_5,1,median)
week_6 =data.ENST00000337225[7:9]
week_6 = apply(week_6,1,median)
week_7 =data.ENST00000337225[10:12]
week_7 = apply(week_7,1,median)
week_8 =data.ENST00000337225[13:15]
week_8 = apply(week_8,1,median)
week_9 =data.ENST00000337225[16:18]
week_9 = apply(week_9,1,median)
data.ENST00000337225 = c(week_4,week_5,week_6,week_7,week_8,week_9) 
data.ENST00000337225 =cbind(data.ENST00000337225,week)

# ENST00000579298
data.ENST00000579298 = as.data.frame(embryo_df_tissues %>%
                             filter(ensembl.transcript == str_extract(ENST, "ENST00000579298")))
data.ENST00000579298 = data.ENST00000579298[,1:18]
week_4 =data.ENST00000579298[1:3]
week_4 = apply(week_4,1,median)
week_5 =data.ENST00000579298[4:6]
week_5 = apply(week_5,1,median)
week_6 =data.ENST00000579298[7:9]
week_6 = apply(week_6,1,median)
week_7 =data.ENST00000579298[10:12]
week_7 = apply(week_7,1,median)
week_8 =data.ENST00000579298[13:15]
week_8 = apply(week_8,1,median)
week_9 =data.ENST00000579298[16:18]
week_9 = apply(week_9,1,median)
data.ENST00000579298 = c(week_4,week_5,week_6,week_7,week_8,week_9) 
data.ENST00000579298 =cbind(data.ENST00000579298,week)

# ENST00000555247
data.ENST00000555247 = as.data.frame(embryo_df_tissues %>%
                             filter(ensembl.transcript == str_extract(ENST, "ENST00000555247")))
data.ENST00000555247 = data.ENST00000555247[,1:18]
week_4 =data.ENST00000555247[1:3]
week_4 = apply(week_4,1,median)
week_5 =data.ENST00000555247[4:6]
week_5 = apply(week_5,1,median)
week_6 =data.ENST00000555247[7:9]
week_6 = apply(week_6,1,median)
week_7 =data.ENST00000555247[10:12]
week_7 = apply(week_7,1,median)
week_8 =data.ENST00000555247[13:15]
week_8 = apply(week_8,1,median)
week_9 =data.ENST00000555247[16:18]
week_9 = apply(week_9,1,median)
data.ENST00000555247 = c(week_4,week_5,week_6,week_7,week_8,week_9) 
data.ENST00000555247 =cbind(data.ENST00000555247,week)

# ENST00000590261
data.ENST00000590261 = as.data.frame(embryo_df_tissues %>%
                             filter(ensembl.transcript == str_extract(ENST, "ENST00000590261")))
data.ENST00000590261 = data.ENST00000590261[,1:18]
week_4 =data.ENST00000590261[1:3]
week_4 = apply(week_4,1,median)
week_5 =data.ENST00000590261[4:6]
week_5 = apply(week_5,1,median)
week_6 =data.ENST00000590261[7:9]
week_6 = apply(week_6,1,median)
week_7 =data.ENST00000590261[10:12]
week_7 = apply(week_7,1,median)
week_8 =data.ENST00000590261[13:15]
week_8 = apply(week_8,1,median)
week_9 =data.ENST00000590261[16:18]
week_9 = apply(week_9,1,median)
data.ENST00000590261 = c(week_4,week_5,week_6,week_7,week_8,week_9) 
data.ENST00000590261 =cbind(data.ENST00000590261,week)

# Summary
plot(data.ENST00000337225[,2],data.ENST00000337225[,1],type = "b",lwd=1,pch=19, main = "Geneexpression level during week 4 to 9", xlab = "weeks",ylab = "Expressionlevel",col = "red",ylim = c(7.6,9.7))
par(new=TRUE)
plot(data.ENST00000579298[,2],data.ENST00000579298[,1],type = "b",lwd=1,pch=19, main = "Geneexpression level during week 4 to 9", xlab = "weeks",ylab = "Expressionlevel",col = "blue",ylim = c(7.6,9.7)) 
par(new=TRUE)
plot(data.ENST00000555247[,2],data.ENST00000555247[,1],type = "b",lwd=1,pch=19, main = "Geneexpression level during week 4 to 9", xlab = "weeks",ylab = "Expressionlevel",col = "orange",ylim = c(7.6,9.7))
par(new=TRUE)
plot(data.ENST00000590261[,2],data.ENST00000590261[,1],type = "b",lwd=1,pch=19, main = "Geneexpression level during week 4 to 9", xlab = "weeks",ylab = "Expressionlevel",col = "green", ylim = c(7.6,9.7))
legend(3.8, 9.8, legend=c("ENST00000337225", "ENST00000579298", "ENST00000555247", "ENST00000590261"),col=c("red", "blue", "orange", "green"), lty=1, cex=0.3,title="Expressed genes")
```

Cytokines are important signaling molecules. Especially chemokines are ment to take part in signaling process during neuronal development (Tiveron 2008).

#### Interleukin related genes

Two interleukin related genes were discovered. ENST00000555247 encodes for IL11RA (interleukin 11 receptor subunit alpha) which is a receptor for the cytokine, Interleukin 11. The IL-11 receptor is a member of the hematopoietic cytokine receptor family. ENST00000555247 is connected to the spinal cord and shows a significant increase in expression during weeks 4 to 8. ENST00000590261 encodes for LF3 (interleukin enhancer binding factor 3) and forms a heterodimer with a 45 kDa transcription factor. This complex is necessary for T-cell expression of interleukin 2. ENST00000590261 is related to Substantia nigra and shows a decline over the weeks.

#### Chemokine related genes

There is a lot of research targeting the role of chemokines during neuronal development. We discovered ENST00000337225 which encodes for CXCL14 (alpha class chemokin ligand). ENST00000337225 shows a significant increase in expression between weeks 4 to 6 and is related to Anterior cingulate cortex. ENST00000579298 encodes for NUP85 (nucleoporin 85) a protein component of the Nup107-160 subunit of the nuclear pore complex. NUP85 can bind to CCR2 (a receptor for beta class chemokines) and promote chemotaxis of monocytes. ENST00000579298 is related to the Frontal Cortex and shows a decline between weeks 5 to 9.



## TRAs can infer a basic timeline of organ development

### Differentially expressed transcripts can be linked to all analyzed tissues

The TRA Data covers 53 distinct tissues. For all of those, we found at least 40 differentially expressed transcripts within our dataset. The minimum was found with 46 stomach-linked transcripts, the maximum were 837 TRAs for the testes.

```{r, fig.width=8, fig.height=6}
# Create a barplot for the TRA count of each tissue based on our dataset of differentially expressed genes

all_tissues = c("Adipose - Subcutaneous", "Adipose - Visceral", "Cells - Transformed fibroblasts", "Skin - Not Sun Exposed", "Skin - Sun Exposed",  "Muscle - Skeletal", "Bladder", "Kidney - Cortex", "Colon - Sigmoid", "Colon - Transverse", "Esophagus - Gastroesophageal Junction", "Esophagus - Mucosa", "Esophagus - Muscularis", "Liver", "Minor Salivary Gland", "Pancreas", "Small Intestine - Terminal Ileum", "Spleen", "Stomach", "Breast - Mammary Tissue", "Cervix - Ectocervix", "Cervix - Endocervix",  "Fallopian Tube", "Ovary", "Uterus", "Vagina", "Prostate", "Testis", "Adrenal Gland", "Pituitary", "Thyroid", "Artery - Aorta", "Artery - Coronary", "Artery - Tibial", "Heart - Atrial Appendage", "Heart - Left Ventricle", "Lung", "Whole Blood", "Cells - EBV-transformed lymphocytes", "Brain - Amygdala", "Brain - Anterior cingulate cortex", "Brain - Caudate", "Brain - Cerebellar Hemisphere", "Brain - Cerebellum", "Brain - Cortex", "Brain - Frontal Cortex", "Brain - Hippocampus", "Brain - Hypothalamus", "Brain - Nucleus accumbens", "Brain - Putamen", "Brain - Spinal cord", "Brain - Substantia nigra",  "Nerve - Tibial")
tissue_count = data.frame(tissue = all_tissues, count = sapply(all_tissues, function(y) {sum(sapply(diff_genes_ann_0.01$tissues, function(x) {grepl(y, toString(x))}))}))

tissue_count_barplot =  ggplot(tissue_count, aes(x = tissue, y = count, fill = tissue))  + geom_bar(stat = "identity") + labs(x = "Tissue", y = "Number of Transcripts")  + theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1)) + guides(fill = FALSE) 
```

```{r, fig.width=7, fig.height=7}
library(ggplot2)
library(treemapify)

select_tissues = c("Adipose", "Cells - Transformed fibroblasts", "Skin",  "Muscle - Skeletal", "Bladder", "Kidney - Cortex",  "Colon", "Esophagus", "Liver", "Minor Salivary Gland", "Pancreas", "Small Intestine - Terminal Ileum", "Spleen", "Stomach", "Breast - Mammary Tissue", "Cervix", "Fallopian Tube", "Ovary", "Uterus", "Vagina", "Prostate", "Testis", "Adrenal Gland", "Pituitary", "Thyroid",  "Artery", "Heart", "Lung", "Whole Blood", "Cells - EBV-transformed lymphocytes", "Brain - Spinal cord", "Brain", "Nerve - Tibial")



count_data = data.frame(tissue = select_tissues, count = NA)
for (tissue in select_tissues){
  
  count_data[count_data$tissue == tissue, "count"] = sum(bool_contained = sapply(diff_genes_ann_0.01$tissues, function(x) {grepl(tissue, toString(x))}))

}
cols = rainbow(33, s=.6, v=.9) #[sample(1:33,33)] #61
count_data$color = cols
count_data
treemap = ggplot(count_data, aes(area = count, label = tissue, fill = tissue)) + geom_treemap() + geom_treemap_text(colour = "white", place = "top", reflow = T) + scale_fill_manual(values=cols) + guides(fill = FALSE)
```

```{r, fig.width=9, fig.height=6, fig.subcap= "A. The number of transcripts associated with each tissue, including subtissues, is displayed.  B. The share of TRAs associated with each tissue. Subtissues are subsumed under their main tissue."}
library(grid)
library(gridExtra)
library(cowplot)
lay = rbind(c(1,1,2), c(1,1, NA))
plot = arrangeGrob(grobs = list(tissue_count_barplot, treemap),layout_matrix = lay)
plot_output = as_ggplot(plot) + draw_plot_label(label = c("A", "B"), size = 15, x = c(0.05, 0.7), y = c(0.1,0.1)) 
annotate_figure(plot_output, top = text_grob("Differentially expressed transcripts of TRAs by tissue", color = "black", face = "bold", size = 16))
```

```{r}
# Mean number of tissues associated with each transcript
mean(sapply(diff_genes_ann_0.01$tissues, function(x) {length(x)}))
```

These numbers are sufficient for further analysis of the gene expression within individual tissues. Therefore, all further analysis will be based on our dataset with differentially expressed genes from limma analysis. Nonetheless, it should be noted that there is a significant overlap between the TRAs associated with different tissues, especially as each transcript is on average linked to 7.4 different sub-tissues or tissues. This overlap is further illustrated by fig. ????.

```{r,fig.width=9, fig.height=7, fig.subcap= "Each tissue is displayed as a node with its size representing the number of transcripts associated with it. The edges show the shared TRAs between the linked tissues, with only links corresponding to more than 100 common TRAs visible."}
library(igraph)
library(tidyverse)
library(ggnet)
library(viridis)

nodes = c("Adipose", "Cells - Transformed fibroblasts", "Skin",  "Muscle - Skeletal", "Bladder", "Kidney - Cortex",  "Colon", "Esophagus", "Liver", "Minor Salivary Gland", "Pancreas", "Small Intestine - Terminal Ileum", "Spleen", "Stomach", "Breast - Mammary Tissue", "Cervix", "Fallopian Tube", "Ovary", "Uterus", "Vagina", "Prostate", "Testis", "Adrenal Gland", "Pituitary", "Thyroid",  "Artery", "Heart", "Lung", "Whole Blood", "Cells - EBV-transformed lymphocytes", "Brain - Spinal cord", "Brain", "Nerve - Tibial")

edges = data.frame (rep(nodes, 32), NA)
for (i in 1:32) {
  a = c(edges[(i+1):33,1], edges[1:i,1])
  edges[(33*(i-1)+1):(33*i),2] = a
}
colnames(edges) = c("Tissue1", "Tissue2")
edges
graph1 = graph_from_data_frame(edges, directed = FALSE)
plot(graph1)


node_weight = data.frame(tissue = nodes, weight = sapply(nodes, function(y) {sum(sapply(diff_genes_ann_0.01$tissues, function(x) {grepl(y, toString(x))}))}))

edge_combined = edges %>% unite(tissues, c("Tissue1", "Tissue2"), sep = "|")
edge_combined
edge_weight = data.frame(edges, weight = apply(edges,1, function(y) {sum(sapply(diff_genes_ann_0.01$tissues, function(x) {grepl(y[1], toString(x)) && grepl(y[2], toString(x))}))}))

show(edge_weight)
min(edge_weight$weight)
edge_weight_adj = edge_weight
edge_weight_adj$weight = sapply(edge_weight_adj$weight, function(x) { if(x < 100) {x = 0} else {x = x/100}})
edge_weight_cleared = edge_weight_adj[!(edge_weight_adj$weight == 0),]

edge_weight_cleared

graph2 = ggnet2(edge_weight_cleared[,1:2], label = TRUE, edge.size = edge_weight_cleared$weight)
node_size = sapply(graph2$data$label, function(x) {node_weight[x,2]})

graph2 = ggnet2(edge_weight_cleared[,1:2], color = graph2$data$label, label = TRUE, color.palette = c() ,edge.size = edge_weight_cleared$weight, size = (ceiling(node_size/30))) + guides(size = FALSE, color = FALSE) + scale_color_viridis(option = "D", discrete = TRUE) + labs(title = "Overlap between the TRAs associated with different tissues")
graph2
```

#### Das würde ich in den Anhang packen.

```{r, fig.width=14, fig.height=14}
library(pheatmap)

tissues =  c("Adipose", "Cells - Transformed fibroblasts", "Skin",  "Muscle - Skeletal", "Bladder", "Kidney - Cortex",  "Colon", "Esophagus", "Liver", "Minor Salivary Gland", "Pancreas", "Small Intestine - Terminal Ileum", "Spleen", "Stomach", "Breast - Mammary Tissue", "Cervix", "Fallopian Tube", "Ovary", "Uterus", "Vagina", "Prostate", "Testis", "Adrenal Gland", "Pituitary", "Thyroid",  "Artery", "Heart", "Lung", "Whole Blood", "Cells - EBV-transformed lymphocytes", "Brain - Spinal cord", "Brain", "Nerve - Tibial")

node_weight = data.frame(tissue = tissue, weight = sapply(nodes, function(y) {sum(sapply(diff_genes_ann_0.01$tissues, function(x) {grepl(y, toString(x))}))}))

weights = matrix(nrow =length(tissues), ncol = length(tissues))
rownames(weights) = tissues
colnames(weights) = tissues
for (i in 1:33) {
  for (j in 1:33) {
    weights[i,j] = (sum(sapply(diff_genes_ann_0.01$tissues, function(x) {grepl(tissues[i], toString(x)) && grepl(tissues[j], toString(x))}))/node_weight[i,2])
  }
} 

heatmap_links <- pheatmap(weights, display_numbers = TRUE, fontsize_number = 8, legend = TRUE, breaks = c(seq(0,0.4,0.005), seq(0.41, 0.8, 0.025), 0.85, 0.9, 0.95,1))
```

### The expression of all TRAs associated with a tissue cannot be used to infer organ development

In this research, we attempt to draw conclusions about the developmental state of a tissue based on the expression of genes associated with it alone. Therefore, we analyzed the share of differentially expressed transcripts above a certain expression level over time, as shown in Fig. ???A. Furthermore, we observed trends within the median expression of all differentially expressed transcripts associated with a tissue (Fig. ???B). Since both metrics only showed in miniscule changes, we hypothesized that distinct, counteracting trends in expression exsisted within one tissue. Thus, k-means clustering was used to determine groups of TRAs with similar expression patterns. For each of these clusters, the median expression was plotted as shown in Fig. ???C.

```{r}
# Create dataframe with median expression levels for each transcript and week (combining the replicates)
embryo_df_tissues_median  = data.frame(Week4 = apply(embryo_df_tissues[1:3], 1, median), Week5 =  apply(embryo_df_tissues[4:6], 1, median), Week6 =  apply(embryo_df_tissues[7:9], 1, median), Week7=apply(embryo_df_tissues[10:12], 1, median), Week8 =  apply(embryo_df_tissues[13:15], 1, median), Week9 =  apply(embryo_df_tissues[16:18], 1, median))
embryo_df_tissues_median = cbind(embryo_df_tissues_median, embryo_df_tissues[,c(19,29)])


tissue = "Spleen"
threshold = c(6.6,6.8, 7.0, 7.2, 7.4, 7.6, 8.0)

# Select the data from a tissues
bool_contained = sapply(embryo_df_tissues_median$tissues, function(x) {grepl(tissue, toString(x))})
data_temp = embryo_df_tissues_median[bool_contained,]

#Determine the percentage of expressed genes
count = 0
a = c()
data_plot = data.frame(Week = c(sapply(c(4:9), function(x) {rep(x, length(threshold))})), Threshold = factor(c(rep(threshold, 6))), Percentage = NA)
  for(s in 4:9) {
    for(t in 1:length(threshold)) {
      count = count +1
      a  = c(a,(sum(data_temp[,str_glue("Week", s)] > threshold[t]))/(dim(data_temp)[1]))
    }
  }
data_plot$Percentage= a

  
#Plot the data
spleen_percentage_plot =  ggplot(data_plot, aes(x = Week, y= Percentage, group = Threshold))   + geom_line(aes(colour = Threshold)) + geom_point(aes(colour = Threshold, shape = Threshold)) + labs(title = "Share over threshold") + theme_classic() + ylim(0,1)

```

```{r}
#Plot of median expression level of all spleen-related genes over time

#Select genes only of target tissue
bool_contained = sapply(diff_genes_ann_0.01$tissues, function(x) {"Spleen" %in% x})

#Graph of mean gene expression over time
spleen_genes = diff_genes_ann_0.01[bool_contained,]
spleen_genes = spleen_genes[, 1:19]
spleen_genes_mean = data.frame(week4=rowMeans(spleen_genes[,1:3]),week5=rowMeans(spleen_genes[,4:6]), week6=rowMeans(spleen_genes[,7:9]),week7=rowMeans(spleen_genes[,10:12]), week8=rowMeans(spleen_genes[,13:15]), week9=rowMeans(spleen_genes[,16:18]))
spleen_genes_graph = apply(spleen_genes_mean, 2, median)
spleen_genes_graph= data.frame (Expression = spleen_genes_graph, Week = c(4:9))

spleen_genes_plot = ggplot(spleen_genes_graph, aes(x = Week, y= Expression))  + geom_line(color = "#0000ff") + geom_point(color = "#0000ff") + labs(title = paste("Median expression"))  + theme_classic() +  ylim(6.5,8.5)

```

```{r}
#Clustered plot of expression levels over time for the spleen

library(factoextra)
library(cluster)
library(stringr)
library(ggplot2)
library(ggsci)
library(grid)
library(gridExtra)

tissue = "Spleen"

# Functions for later use
expression_difference = function(x,y) {
  return ((x-y)/((x+y)*0.5))
}

avg_sil = function(k, data) {
  km = kmeans(data, centers = k, nstart = 25)
  ss = silhouette(km$cluster, dist_data)
  mean(ss[, 3])
}

create_plot_data = function(dataset) {
  data_plot = data.frame(Week = c(sapply(c(4:9), function(x) {rep(x, n_cluster)})), Cluster = as.character(c(rep(c(1:n_cluster), 6))), Expression = NA)
  count = 0
  for(s in 4:9) {
    for(t in 1:n_cluster) {
      count = count + 1
      values = dataset[dataset$Cluster == t, str_glue("Week", s)]
      data_plot[count, "Expression"] = median(values)
      data_plot[count, "Cluster"] = str_glue(data_plot[count, "Cluster"], " [", km$size[t], " genes]")
   }
  }
  return(data_plot)
}



# Select an modify the data used for kmeans
bool_contained = sapply(diff_genes_ann_0.01$tissues, function(x) {grepl(tissue, toString(x))})
data_temp = diff_genes_ann_0.01[bool_contained,]
data_abs = data.frame(Week4 = apply(data_temp[1:3], 1, median), Week5 =  apply(data_temp[4:6], 1, median), Week6 =  apply(data_temp[7:9], 1, median),
                        Week7=apply(data_temp[10:12], 1, median), Week8 =  apply(data_temp[13:15], 1, median), Week9 =  apply(data_temp[16:18], 1, median))
data = data.frame("5_4" = expression_difference(data_abs[,"Week5"], data_abs[,"Week4"]), 
                  "6_5" = expression_difference(data_abs[,"Week6"], data_abs[,"Week5"]),
                  "7_6" = expression_difference(data_abs[,"Week7"], data_abs[,"Week6"]),
                  "8_7" = expression_difference(data_abs[,"Week8"], data_abs[,"Week7"]),
                  "9_8" = expression_difference(data_abs[,"Week9"], data_abs[,"Week8"]))
row.names(data) = rownames(data_abs)
  
# Select optimum cluster number
data = scale(data)
dist_data = get_dist(data, method = "pearson")
n_cluster = which.max(sapply(c(2:10), function(k) {avg_sil(k, data)})) +1
  
# Create the data for our plot
km = kmeans(data, centers = n_cluster, nstart = 25)
data_plot_raw = cbind(data_abs, Cluster = km$cluster)
plot_data = create_plot_data(data_plot_raw)
  
# Create the Plot
spleen_clustered_plot =  ggplot(plot_data, aes(x = Week, y= Expression, group = Cluster))  + geom_line(aes(colour = Cluster)) +
          geom_point(aes(colour = Cluster, shape = Cluster)) + labs(title = "Clustered expression") +
          scale_color_npg() + theme_classic()  + ylim(6.5,8.5)
   
```

```{r, fig.width= 10, fig.height= 3, fig.subcap= "A. For different expression thresholds the share of differentially expressed transcripts with higher expressions than the threshold is depicted. B. The median expression for all spleen-associated TRAs is shown for each point in time. C. For k-means clustring, the silhouette score determined an optimum of two clusters. The median expression analog to B is plottet for each of these clusters. "}
library(grid)
library(gridExtra)
library(cowplot)
plot = arrangeGrob(grobs = list(spleen_percentage_plot, spleen_genes_plot, spleen_clustered_plot), nrow=1, widths = c(3.3,2.7, 4), heights = c(0.7))
plot_output = as_ggplot(plot) + draw_plot_label(label = c("A", "B", "C"), size = 15, x = c(0.03, 0.35, 0.63), y = c(0.13, 0.13, 0.13)) 
annotate_figure(plot_output, top = text_grob("Expression over time of Spleen-related differentially expressed transcripts", color = "black", face = "bold", size = 16))
```


For many tissues, as shown here exemplary with the spleen, the clustering showed two or more clusters that could each be characterized as either an upregulation or a downregulation. In order to analyze the indications for organ development, we analyzed the functions of the transcripts belonging to the two clusters. 


### The clusters of up- and downregulated transcripts can be linked to distinct gene functions. 

For all differentially expressed spleen-associated transcripts, we used the NCBI gene database to get a functional annotation. 

```{r}
#Get the spleen dataset with the clusters
expression_difference = function(x,y) {
  return ((x-y)/((x+y)*0.5))
}

avg_sil = function(k, data) {
  km = kmeans(data, centers = k, nstart = 25)
  ss = silhouette(km$cluster, dist_data)
  mean(ss[, 3])
}
bool_contained = sapply(diff_genes_ann_0.01$tissues, function(x) {grepl("Spleen", toString(x))})
  data_temp = diff_genes_ann_0.01[bool_contained,]
  data_abs = data.frame(Week4 = apply(data_temp[1:3], 1, median), Week5 =  apply(data_temp[4:6], 1, median), Week6 =  apply(data_temp[7:9], 1, median),
                        Week7=apply(data_temp[10:12], 1, median), Week8 =  apply(data_temp[13:15], 1, median), Week9 =  apply(data_temp[16:18], 1, median))
  data = data.frame("5_4" = expression_difference(data_abs[,"Week5"], data_abs[,"Week4"]), 
                  "6_5" = expression_difference(data_abs[,"Week6"], data_abs[,"Week5"]),
                  "7_6" = expression_difference(data_abs[,"Week7"], data_abs[,"Week6"]),
                  "8_7" = expression_difference(data_abs[,"Week8"], data_abs[,"Week7"]),
                  "9_8" = expression_difference(data_abs[,"Week9"], data_abs[,"Week8"]))
  row.names(data_abs) = data_temp$ensembl.transcript
  row.names(data) = rownames(data_abs)
  
  data = scale(data)
  dist_data = get_dist(data, method = "pearson")
  n_cluster = which.max(sapply(c(2:10), function(k) {avg_sil(k, data)})) +1
  
  km = kmeans(data, centers = n_cluster, nstart = 25)
  data_plot_raw = cbind(data_abs, Cluster = km$cluster)

  spleen_genes_clustered = data_plot_raw

```

```{r, fig.width= 5, fig.height= 3}
library(ggplot2)
#Compare the initial expression values of both clusters
spleen_genes_clustered_1 = spleen_genes_clustered[spleen_genes_clustered$Cluster==1, ]
spleen_genes_clustered_2 = spleen_genes_clustered[spleen_genes_clustered$Cluster==2, ]
spleen_boxplot_data = data.frame(Expression = NA, Week = NA, Cluster = NA)
i = 0
for (genelist in list(spleen_genes_clustered_1$Week4, spleen_genes_clustered_1$Week9, spleen_genes_clustered_2$Week4,  spleen_genes_clustered_2$Week9)) {
  i = i+1
  spleen_df = data.frame(Expression = genelist, Week = factor(rep((if(i%%2 == 0) {9} else {4}),length(genelist))), Cluster = rep((if(i<3) {"Up"} else
    {"Down"}),length(genelist)))
  spleen_boxplot_data = rbind(spleen_boxplot_data, spleen_df)
}
spleen_boxplot_data = spleen_boxplot_data[-1,]
spleen_boxplot = ggplot(spleen_boxplot_data, aes(x = Cluster, y = Expression, fill = Week)) + geom_boxplot(position=position_dodge(0.8)) + theme_classic() + labs(title = "Expression of spleen-associated transcripts")
spleen_boxplot
#boxplot(spleen_genes_clustered_1$Week4, spleen_genes_clustered_1$Week9, spleen_genes_clustered_2$Week4,  spleen_genes_clustered_2$Week9)
```

```{r}
get_gene_info = function(transcript_ids){
  library(rentrez)
  library(XML)
  table = data.frame(Transcript = NA, Gene = NA, Protein = NA, Summary = NA)
  for(x in transcript_ids) {
    table = rbind(table, get_single_transcript(x))
  }
  table
}

get_single_transcript = function(single_transcript) {
   res = entrez_search(db = "gene", term = single_transcript)
   if(!length(res$ids)==0) {
     results = entrez_summary(db = "gene", id = res$ids)
      c(single_transcript, as.character(results[2]),  as.character(results[3]),  as.character(results[17]))
   }
   else{
        c(single_transcript, NA, NA, NA)
   }
}

spleen1_gene_description = get_gene_info(rownames(spleen_genes_clustered_1))
spleen2_gene_description = get_gene_info(rownames(spleen_genes_clustered_2))
```

```{r}
#Table verschönern
library(gt)
library(gtExtras)
library(ggplot2)
library(tidyverse)
library(svglite)

Expression =  data.frame(c(1:length(spleen_genes_clustered_1[,1])))
Expression$Expression_over_time = sapply(c(1:length(spleen_genes_clustered_1[,1])), function(x) {list(as.numeric(c(as.numeric(spleen_genes_clustered_1[x,1]), as.numeric(spleen_genes_clustered_1[x,2]), as.numeric(spleen_genes_clustered_1[x,3]),
           as.numeric(spleen_genes_clustered_1[x,4]),as.numeric(spleen_genes_clustered_1[x,5]),as.numeric(spleen_genes_clustered_1[x,6]))))})

spleen1_gene_description_spl = cbind(spleen1_gene_description[-1,], Expression)
spleen1_gene_description_spl = spleen1_gene_description_spl[,-5]



table_spleen1 = gt(spleen1_gene_description_spl)|>  tab_header(title= "Spleen - Upregulated", subtitle = "List of differentially epxressed genes") |> fmt_markdown(columns = c(Transcript, Gene, Protein, Summary)) |> gt_theme_538() |> gt_plt_sparkline( column = Expression_over_time, type = "ref_median")
```


```{r}
# Jetzt noch für Spleen2
library(gt)
library(gtExtras)
library(ggplot2)
library(tidyverse)
library(svglite)
library(webshot)

Expression =  data.frame(c(1:length(spleen_genes_clustered_2[,1])))
Expression$Expression_over_time = sapply(c(1:length(spleen_genes_clustered_2[,1])), function(x) {list(as.numeric(c(as.numeric(spleen_genes_clustered_2[x,1]), as.numeric(spleen_genes_clustered_2[x,2]), as.numeric(spleen_genes_clustered_2[x,3]),
           as.numeric(spleen_genes_clustered_2[x,4]),as.numeric(spleen_genes_clustered_2[x,5]),as.numeric(spleen_genes_clustered_2[x,6]))))})

spleen2_gene_description_spl = cbind(spleen2_gene_description[-1,], Expression)
spleen2_gene_description_spl = spleen2_gene_description_spl[,-5]


table_spleen2 = gt(spleen2_gene_description_spl)|>  tab_header(title= "Spleen - Downregulated", subtitle = "List of differentially epxressed genes") |> fmt_markdown(columns = c(Transcript, Gene, Protein, Summary)) |> gt_theme_538() |> gt_plt_sparkline( column = Expression_over_time, type = "shaded")

```





# Discussion

## Hypothesis: Specific TRAs can be used as key biomarkers for the development of individual organs

## Hypothesis: Neural TRA expression patterns reflect morphological brain development


First point to discuss is the signifikant increase in the gene expression of Ca(2+)ATPases (SLN sarcolipin) by factor 2 in logarithmic scale. A reason therefor is the process of neuronal migration which starts at week 9. Ca(2+) is an important cofaktor for actin dependent cell migration.

Another point is the strong correlation between the expression of SNAP91 genes and one SYBU gene (ENST00000276646). Both proteins were associated with the Cerebellar Hemisphere and contribute to endocytosis, which is essential for functional neurons and contributes to neuronal survival (Overhoff, 2020).
One point to mention here is, that ENST00000521485 and ENST00000518312 were both associated with SNAP91, nevertheless to different isoforms. But both show identical correlation in expression, hence a failure in annotation might be possible.

*Quelle:https://onlinelibrary.wiley.com/doi/10.1111/jnc.15194*

In addition we identified a signifikant increase in expression of NXPH1 this refers to the NXPH1-promoted adhesion between axons and dendrites. An upregulation of this supplementary factor can prepare the process of synapse formation which starts at week 11.

We further identified that two transkripts associated to two different tissues show identical correlation, this can be caused by a false annotations. Nevertheless, they show a stong downregulation of BDNF. This factor normally promotes neuronal survival. A downregulation of BDNF can be a preparation for the phase of programmed cell death, which starts at week 20 in neuronal development.

The neuronal cell migration is stongly dependent on chemoattractors like chemokines (Tiveron 2008). We identified a signifikant increase in CXCL14 between week 5 to 6 and a maintaining at a high expression level for the following weeks. This could be an accumulation for neuronal migration, starting at week 9. 



## Hypothesis: TRAs can infer a timeline of organ development similar to the results by Yi et al. 2010


# Supplementary

```{r,fig.height=9,fig.width=12}

# Distribution control for Cerebellar Hemisphere 
Tissues = embryo_df_tissues$max.tissue

#Boxplot for all replicates over all weeks
Cerebellar_Hemisphere = as.data.frame(embryo_df_tissues %>%
                             filter(max.tissue == str_extract(Tissues, "Brain - Cerebellar Hemisphere")))
Cerebellar_Hemisphere_names = Cerebellar_Hemisphere$HGNC.symbol
Cerebellar_Hemisphere =Cerebellar_Hemisphere[,1:18]
Cerebellar_Hemisphere = t(Cerebellar_Hemisphere)
colnames(Cerebellar_Hemisphere) <- Cerebellar_Hemisphere_names
Cerebellar_Hemisphere = as.data.frame(Cerebellar_Hemisphere)
Cerebellar_Hemisphere_sorted = Cerebellar_Hemisphere %>% select(sort(colnames(Cerebellar_Hemisphere)))
Cerebellar_Hemisphere_sorted
boxplot(Cerebellar_Hemisphere_sorted, main="Expression of Brain - Cerebellar Hemisphere genes",col= rainbow(542),cex.axis=0.5)

# for week 4 only
Cerebellar_Hemisphere_week4= Cerebellar_Hemisphere_sorted[1:3,]
boxplot(Cerebellar_Hemisphere_week4, main="Expression of Brain - Cerebellar Hemisphere genes (week 4)",col= rainbow(542),cex.axis=0.5)

# for week 5 only
Cerebellar_Hemisphere_week5= Cerebellar_Hemisphere_sorted[4:6,]
boxplot(Cerebellar_Hemisphere_week5, main="Expression of Brain - Cerebellar Hemisphere genes (week 5)",col= rainbow(542),cex.axis=0.5)

# for week 6 only
Cerebellar_Hemisphere_week6= Cerebellar_Hemisphere_sorted[7:9,]
boxplot(Cerebellar_Hemisphere_week6, main="Expression of Brain - Cerebellar Hemisphere genes (week 6)",col= rainbow(542),cex.axis=0.5)

# for week 7 only
Cerebellar_Hemisphere_week7= Cerebellar_Hemisphere_sorted[10:12,]
boxplot(Cerebellar_Hemisphere_week7, main="Expression of Brain - Cerebellar Hemisphere genes (week 7)",col= rainbow(542),cex.axis=0.5)

# for week 8 only
Cerebellar_Hemisphere_week8= Cerebellar_Hemisphere_sorted[13:15,]
boxplot(Cerebellar_Hemisphere_week8, main="Expression of Brain - Cerebellar Hemisphere genes (week 8)",col= rainbow(542),cex.axis=0.5)

# for week 9 only
Cerebellar_Hemisphere_week9= Cerebellar_Hemisphere_sorted[16:18,]
boxplot(Cerebellar_Hemisphere_week9, main="Expression of Brain - Cerebellar Hemisphere genes (week 9)",col= rainbow(542),cex.axis=0.5)

# Distribution Kontrol for ENST00000276646, ENST00000518312 & ENST00000521485
ENST = embryo_df_tissues$ensembl.transcript

# ENST00000276646
data.ENST00000276646 = as.data.frame(embryo_df_tissues %>%
                             filter(ensembl.transcript == str_extract(ENST, "ENST00000276646")))
week_4 =t(data.ENST00000276646[1:3])
week_5 =t(data.ENST00000276646[4:6])
week_6 =t(data.ENST00000276646[7:9])
week_7 =t(data.ENST00000276646[10:12])
week_8 =t(data.ENST00000276646[13:15])
week_9 =t(data.ENST00000276646[16:18])
data.ENST00000276646 = cbind(week_4,week_5,week_6,week_7,week_8,week_9)
colnames(data.ENST00000276646) <- c("week_4","week_5","week_6","week_7","week_8","week_9")
boxplot(data.ENST00000276646, main="Expression of ENST00000276646",col= rainbow(6),cex.axis=0.5)

# ENST00000518312
data.ENST00000518312 = as.data.frame(embryo_df_tissues %>%
                             filter(ensembl.transcript == str_extract(ENST, "ENST00000518312")))
week_4 =t(data.ENST00000518312[1:3])
week_5 =t(data.ENST00000518312[4:6])
week_6 =t(data.ENST00000518312[7:9])
week_7 =t(data.ENST00000518312[10:12])
week_8 =t(data.ENST00000518312[13:15])
week_9 =t(data.ENST00000518312[16:18])
data.ENST00000518312 = cbind(week_4,week_5,week_6,week_7,week_8,week_9)
colnames(data.ENST00000518312) <- c("week_4","week_5","week_6","week_7","week_8","week_9")
boxplot(data.ENST00000518312, main="Expression of ENST00000518312",col= rainbow(6),cex.axis=0.5)

# ENST00000521485
data.ENST00000521485 = as.data.frame(embryo_df_tissues %>%
                             filter(ensembl.transcript == str_extract(ENST, "ENST00000521485")))
week_4 =t(data.ENST00000521485[1:3])
week_5 =t(data.ENST00000521485[4:6])
week_6 =t(data.ENST00000521485[7:9])
week_7 =t(data.ENST00000521485[10:12])
week_8 =t(data.ENST00000521485[13:15])
week_9 =t(data.ENST00000521485[16:18])
data.ENST00000521485 = cbind(week_4,week_5,week_6,week_7,week_8,week_9)
colnames(data.ENST00000521485) <- c("week_4","week_5","week_6","week_7","week_8","week_9")
boxplot(data.ENST00000521485, main="Expression of ENST00000521485",col= rainbow(6),cex.axis=0.5)

```


