---
title: ""

#author: "Paul Christmann, Joshua Eigenmann, David Jewanski, Verena Merke"
#date: "`r format(Sys.time(), '%B %d, %Y')`"
output:
  bookdown::pdf_document2:
    number_sections: TRUE
    toc: TRUE
    theme: united
geometry: "left=1.5cm,right=1.5cm,top=1cm,bottom=1.5cm"
header-includes:
  - \usepackage[font={small,it}, labelfont={bf}]{caption}
  - \usepackage{float}
  - \renewcommand{\topfraction}{.95}
  - \renewcommand{\bottomfraction}{.9}
  - \renewcommand{\textfraction}{.05}
  - \renewcommand{\floatpagefraction}{.77}
  - \setcounter{topnumber}{3}
  - \setcounter{bottomnumber}{3}
  - \setcounter{totalnumber}{4}
  #- \floatplacement{figure}{H}
  - \usepackage{caption}
  - \captionsetup[Fig.]{font=footnotesize}
---
\thispagestyle{empty}
\hrule
\vspace{0.3cm}
\begin{center}
	\vspace{1cm}
  \huge
	\begin{tabular}[c]{l}
	\\
	Final Report - Datascience MoBi 2022\\
	\\
	\\
  \textbf{Gaining insight on human early organogenesis}\\
  \textbf{through TRA expression analysis}\\
	\\
	Topic 04 - Team 01\\
	\\
  Paul Christmann, Joshua Eigenmann,\\ David Jewanski, Verena Merke \\
  \\
  \\
  \\
  \Large Date: June 17, 2022\\
  \\
  \\
  \\
	\large Supervisor:
	\large Dr. Maria Dinkelacker \\
	\\
	\large Tutor:
	\large Ian Fichtner \\
	\\
	\\
	\large Institute of Pharmacy and Molecular Biotechnology\\
	\large Heidelberg University 
	\vspace{0.3cm} \\
	\end{tabular}
	\end{center}

---


```{r knitr_init, echo=FALSE, cache=FALSE}
library(knitr)
#library(rmdformats)
## Global options
#options(max.print="120")
opts_chunk$set(echo=FALSE,
	             cache=TRUE,
               prompt=FALSE,
               tidy=TRUE,
               comment=NA,
               message=FALSE,
               warning=FALSE)
#opts_knit$set(width=120)
#opts_knit$set(root.dir  = "~/")
```

```{r, include=FALSE}
library(affy)
library(AnnotationDbi)
library(biomaRt)
library(cluster)
library(clusterProfiler)
library(cowplot)
library(dplyr)
library(enrichplot)
library(factoextra)
library(ggbiplot)
library(ggforce)
library(GGally)
library(ggplot2)
library(ggplotify)
library(ggpubr)
library(ggrepel)
library(ggsci)
library(ggupset)
library(grid)
library(gridExtra)
library(gt)
library(gtExtras)
library(hexbin)
library(hgu133plus2hsenstcdf)
library(hgu133plus2hsenstprobe)
library(igraph)
library(kableExtra)
library(limma)
library(magick)
library(magrittr)
library(org.Hs.eg.db)
library(pheatmap)
library(png)
library(Rcpp)
library(RCurl)
library(readxl)
library(rentrez)
library(Rfssa)
library(stringr)
library(svglite)
library(tidyverse)
library(treemapify)
library(VennDiagram)
library(viridis)
library(vsn)
library(webshot)
library(XML)
```

\newpage

# Abstract

A single cell turns into a complex multicellular organism during embryogenesis. While the morphological steps are mostly understood, the role of molecular pathways as well as transcriptional regulation in embryonic development is still a topic of current research. Tissue restricted antigens (TRAs) may be a key in understanding the connection between gene expression and morphological consequence. Since the expression of a TRA is specific to its relative tissue, we could draw conclusions about organ development from TRA expression patterns between week 4 and 9 of embryogenesis. Using microarray data from  Yi H *et al.* (2010), we created a dataset of differentially expressed transcripts, including genes not found by Yi H *et al.*, through limma analysis. Based on this data, we could successfully link developmental steps to TRA expression patterns for some of the analyzed organs. Furthermore, we found individual genes associated with processes in neurogenesis. Finally, we attempted without success to define specific TRAs as biomarkers for organogenesis.



\newpage

# Introduction

## TRAs as a tool to gain insight into embryonic development {#intro-tra}

Tissues restricted antigens (TRAs) are originally a concept from immunology. There, they are sets of genes for auto-antigens that enable a negative selection in the thymus, which prevents autoimmune reactions. TRAs are ordered in clusters and controlled by the autoimmune regulator in medullary epithelial cells. They represent the diversity of antigens in the different tissues of an organism (Dinkelacker, 2019; Murphy & Weaver, 2018). For our research, we use a the broader definition of TRAs as genes with more than 5 times the median gene expression in less than 5 different tissues (Dinkelacker, 2019). In contrast to housekeeping genes, this makes TRAs highly specific for individual tissues. Therefore, the existence and functionality of an organ should be correlated with the expression of its associated TRAs. We will use this connection to gain insight into organ development based on gene expression data between week 4 and 9 of embryogenesis. 

## Embryonic development during the observed timeframe {#intro-tissues}

In order to compare our results based on expression patterns to the morphological processes, it is essential to give an overview about organogenesis during this time. We will summarize the developments in following organs: the liver, testes, spleen, heart, stomach, skin, skeletal muscles and brain.\
First, the liver sprout begins to form in week 3 after gestation. From week 4, it develops hepatocyte precursors and is innervated by veins. Between week 5 and 9, the production of gallic acid starts. Glycogen granules develop by week 8 and glycogen synthesis starts in the following week (Deutsch, 2013).\
The testes initially develop as non sex-specific gonads. At week 5, the first germ cells appear in the gonades. The gender-specific development into ovaries and testes starts at week 7 (Benninghoff, 1993).\
The spleen appears at week 6 of embryonic development. Blood vessels in the organ develop from week 8 to 9 (James & Jones 1983). Within the human immune system, the spleen plays a major role. To that end, B-lymphocytes are present within this tissue from week 12, while T-cells can be found not earlier than week 14. Previously, these cells develop in the liver starting week 9 and in the thymus from week 7 respectively (Hayward, 1983).\
At week 3, the heart consists only of a preliminary tube. From then on, it undergoes extensive growth and development with chambers and ventricles forming. The fundamental layout is already present by week 5. Then, further remodeling takes place until around week 7, at which point the major steps are already completed (Ulfig, 2009; Hikspoors *et al.*, 2022).\
The stomach develops from the foregut. The primitive gut divides into foregut, midgut and hindgut by week 4. The stomach is first visible at the end of week 4 (Kluth *et al.*, 2013). Gastric pits form by week 8, while essential cell types like enteroendocrine cells and mucous cells appear between the 10th and 15th week. Stomach acid is only secreted from week 32 onward (Esrefoglu *et al.*, 2017).\
Skin development starts immediately after gastrulation at week 3. The ectoderm further develops into the nervous system and skin epithelium. There, the epidermal differentiation is illustrated through the expression of keratin genes. Adhered cells (periderm) create a protective layer for the ectoderm during weeks 4 to 8 (Hu *et al.*, 2018).\
The skeletal muscles develop from the mesoderm first in form of myoblasts. Later, between week 10 and 13, they fuse to form myotubes and then differentiated muscle fibers. The proteins necessary for muscle formation appear at the earliest after 7 weeks, with more being expressed from week 9 and 10 on. Muscle fibers only form from week 15 onwards (Romero *et al.*, 2013).\
While neurulation happens around week 4, major parts of the brain are already visible by week 9. Between theses stages, characteristic steps of neuronal development such as neuronal proliferation and differentiation starting at week 4 as well as neuronal migration and synapse formation beginning week 9 take place (National Research Council and Institute of Medicine, 2009; MÃ¼ller & Hassel, 2018). These processes are influenced through signals provided for instance by chemokines.

## Chemokines and brain development {#intro-brain}

Chemokines are a group of small proteins acting as chemoattractors on effector cells. They are classified in 4 groups labeled alpha to delta, depending on the position of their first cysteines (Cys). In the alpha group (CXC), they are separated by a single aminoacid. In the beta group (CC), the Cys are located next to each other. In the gamma group (C), only one Cys is present. In the delta group (CX3C), they are separated by three aminoacids (Yusuf *et al.*, 2005). Chemokines induce cell migration by binding to their respective receptors. Furthermore, the CXCL12/CXCR4 signalling pathway plays an important role in the neuronal cell migration during embryonic development (Tiveron & Cremer 2008).

# Methods

## Programming language and Libraries

We used the programming language R version 4.2.0 and its IDE RStudio to draw statistical conclusions. We installed several library packages (Table 1) from CRAN and bioconductor, an open software library for statistical genomics. Annotation packages for the microarrays were provided by brainarray.

```{=latex}
\begin{table}[htb]
\caption{All libraries used for the code of this report, libraries installed from CRAN, Bioconductor and brainarray}
\resizebox{\textwidth}{!}{
\begin{tabular}{@{}ll|ll|ll|ll@{}}
\toprule
\textbf{Library}       & \textbf{Version} & \textbf{Library} & \textbf{Version} & \textbf{Library} & \textbf{Version} & \textbf{Library}     & \textbf{Version} \\ \midrule
affy                   & 1.74             & AnnotationDbi    & 1.58             & biomaRt          & 2.52             & cluster              & 2.1.3            \\
clusterProfiler        & 4.4.4            & cowplot          & 1.1.1            & dplyr            & 1.0.9            & enrichplot           & 1.16.1           \\
factoextra             & 1.0.7            & ggbiplot         & 0.55             & ggforce          & 0.3.3            & GGally               & 2.1.2            \\
ggplot2                & 3.3.6            & ggplotify        & 0.1.0            & ggpubr           & 0.4.0            & ggrepel              & 0.9.1            \\
ggsci                  & 2.9              & ggupset          & 0.3.0            & grid             & 4.2.0            & gridExtra            & 2.3              \\
gt                     & 0.6.0            & gtExtras         & 0.4.1            & hexbin           & 1.28.2           & hgu133plus2hsenstcdf & 25.0             \\
hgu133plus2hsenstprobe & 25.0             & igraph           & 1.3.2            & kableExtra       & 1.3.4            & limma                & 3.52             \\
magick                 & 2.7.3            & magrittr         & 2.0.3            & org.Hs.eg.db     & 3.15             & pheatmap             & 1.0.12           \\
png                    & 0.7              & Rcpp             & 1.0.9            & RCurl            & 1.98             & readxl               & 1.4              \\
rentrez                & 1.2.3            & Rfssa            & 2.0.1            & stringr          & 1.4              & svglite              & 2.1              \\
tidyverse              & 1.3.1            & treemapify       & 2.5.5            & VennDiagram      & 1.7.3            & viridis              & 0.6.2            \\
vsn                    & 3.64             & webshot          & 0.5.3            & XML              & 3.99             &                      &                  \\ \bottomrule
\end{tabular}}
\end{table}
```

## Dataset

We obtained our data set from Yi H *et al.* (2010). It contains human embryonic data which cover every week between the 4th and 9th week with three replica at each point in time. Hence, acquired we the data from 18 embryos. The time frame covers the Carnegie stages 10-23, containing the processes of embryogenesis and organogenesis. This period of embryonic development is highly regulated resulting in considerable differential gene expression. Overall, the data set suits the requirements for our purpose.

### Affymetrix U133 plus 2.0 human GeneChip array and importing its data

The data were generated using Affymetrix U133 plus 2.0 human GeneChip arrays, coted slides with matrices for screening purposes. [The HG-U133 Plus 2.0](https://www.affymetrix.com/support/technical/datasheets/human_datasheet.pdf) allows the detection of about 50,000 transcripts and include 62 control transcripts.

We downloaded the raw data from the Gene Expression Omnibus with the accession number  [GSE15744](https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE15744). We imported it with the library package *affy*, which allows more manageable data analysis and manipulation of microarray intensity values.\
To access the data remotely, we uploaded it to the cloud-based repository hosting service github.

```{r importing-the-data-from-harddrive, eval = FALSE}
setwd("C:/Users/david/Documents/Data Analysis Lokal/Raw-Data/CEL_files")
data.GSE15744 <- ReadAffy()
data.GSE15744@cdfName <- "HGU133Plus2_Hs_ENST"
```



```{r loading-data-from-github, include = FALSE}
#make data accessible for everyone
githubURL <- "https://github.com/datascience-mobi-2022/2022-topic-04-team-01/blob/main/Report/data.GSE15744.RData"
load_github_data(githubURL)
```

### Quality control of the suface images and of RNA Degradation

As shown in Fig. \@ref(fig:QC-surface-images-and-RNA-Degradation-Plot)A, the surfaces of the microarrays show no spatial artefacts, fingerprints, irregular dye or stripes. Some differences in overall brightness are visible but marginal.

```{r QC-surface-images, fig.width= 4, fig.height= 4, fig.cap= "**Quality control: a selected surface image of the microarrays showing no damage or artefacts.** The microarray inspection shows no irregularities and every chip is accepted for further data analysis", include=FALSE}
png(filename = "Surface image.png")
surfaceimage.GSM394525 <- image(data.GSE15744[,13], col=rainbow(100, start = 0, end = 0.75)[100:1])
dev.off()
## To generate all images please use:
#for (i in 1:18) {
#  image(data.GSE15744[,i], col=rainbow(100, start = 0, end = 0.75)[100:1])
#} # GSM394501 looks a bit brighter, GSM394527 looks bright
```

Furthermore, we tested for low RNA quality chips. Coated matrices degrade under unfavorable conditions, which negatively affects raw intensities (Fasold & Binder 2013). By plotting the RNA degradation for the 3'--5' strand, we compared the different chips (Fig. \@ref(fig:QC-surface-images-and-RNA-Degradation-Plot)B) and verified the overall chip quality for data analysis.

```{r RNA-Degradation-Plot, fig.width= 4, fig.height= 4, fig.cap= "**Quality control: RNA degradation plot shows slight irregularities and verifies the data.** Some crossing lines can be seen, especally the microarray GSM394519. We decided that the inconsistencies are minor though, and kept all microarrays to avoid the loss of potentionally relevant data.", include=FALSE}
data.rnadegradation <- AffyRNAdeg(data.GSE15744)

png(filename = "RNA degradation plot.png")
plotAffyRNAdeg(data.rnadegradation, col=rainbow(18))
title(sub="RNA Degradation of 18 microarrays, GSE15744")
dev.off()

## Not scaled
#plotAffyRNAdeg(data.rnadegradation, col=rainbow(18), transform = "shift.only")
#title(sub="Human embryo (GSE15744, Raw data)")
```

```{r QC-surface-images-and-RNA-Degradation-Plot,  fig.width= 8, fig.height= 3, fig.cap= "**Quality control: Selected surface image of a microarray shows no damage or artefacts and RNA degradation plot shows slight irregularities and verifies the data.** **A**: The microarray inspection shows no irregularities and every chip is accepted for further data analysis. **B**: Some crossing lines can be seen, especally the microarray GSM394519. We decided that the inconsistencies are minor though, and kept all microarrays to avoid the loss of potentionally relevant data."}
surface_image_plot = ggdraw() + draw_image("Surface image.png")
RNA_degradation_plot = ggdraw() + draw_image("RNA degradation plot.png")

OC_chip_plot = arrangeGrob(grobs = list(surface_image_plot, RNA_degradation_plot),
                                    nrow=1, widths = c(3,3), heights = c(3))

OC_chip_plot_output = as_ggplot(OC_chip_plot) +
  draw_plot_label(label = c("A", "B"), size = 15, x = c(0.09, 0.58), y = c(0.17, 0.17)) 

annotate_figure(OC_chip_plot_output, top = text_grob("Quality control: verifying the surface\nimage and RNA degradation", color = "black", face = "bold", size = 12))
```

## Normalising the data set

Intensity values of different chips are affected by statistical variance and random fluctuation. To access the biological relevant variation, the raw data are normalised. We chose the vsn rma normalization with its library *vsn* according to Huber *et al.* (2002). This library is designed to process microarray intensity values. It calibrates data and applies *generalized log*-transformation, which is an adjusted natural logarithm and preserves statistical significance.

```{r vsn-normalization, include=FALSE}
data.GSE15744.vsnrma <- vsnrma(data.GSE15744)
```

```{r silent-Data-Cleaning, include=FALSE}
#adjusting the microarray names by cutting the .cel
rownames(pData(data.GSE15744.vsnrma)) <- substr(rownames(pData(data.GSE15744)), 1, nchar(rownames(pData(data.GSE15744)))-4)
colnames(exprs(data.GSE15744.vsnrma)) <- substr(colnames(exprs(data.GSE15744)), 1, nchar(colnames(exprs(data.GSE15744)))-4)

temp_embryo_df = as.data.frame(exprs(data.GSE15744.vsnrma))
colnames(temp_embryo_df) = rownames(pData(data.GSE15744.vsnrma))
embryo_df = temp_embryo_df[63:95721,]
```

### Quality control of the vsn normalization {#QC-normalization}

To verify the transformed data intensity values, severah tests can be performed (suppl. Fig. \@ref(fig:QC-normalization-plots)). After the normalization, the rank of the mean of the intensity values and their standard deviation should not correlate. To control this, we plotted the rank of the mean against the standard deviation, which resulted in a reasonable horizontal line indicated in red (suppl. Fig. \@ref(fig:QC-normalization-plots)A).\ To further control the normalization we visualized the intensity values. We used boxplots to compare each of the 18 microarrays separately by its mean, median and variance (suppl. \@ref(fig:QC-normalization-plots)B). From these, we deduced that all arrays fit. Another method gives us the ability to zoom in even further. The intensity levels of three replica should be the same, since they were taken at the same time. We can use scatterplots to compare single intensity levels. With one of the replicas applied on the x and y axis respectively, we should see a scatterplot following the linear function y = x since the same transcript should show the same intensity in both replicas (supplementary Fig. \@ref(fig:QC-normalization-plots)C).

```{r mean-sd-plot, include=FALSE}
meansd.GSE15744.vsnrma <- meanSdPlot(data.GSE15744.vsnrma, xlab="Rank (mean)", ylab="Standard Deviation")
meansdplot.GSE15744.vsnrma <- meansd.GSE15744.vsnrma$gg + ggtitle("Mean-Sd-Plot of vsn normalized data\nGSE15744") + theme(aspect.ratio = 1)
```

```{r boxplot-vsn-normalization, include=FALSE}
weeks <- c("Week 4.1", "Week 4.2", "Week 4.3", "Week 5.1", "Week 5.2", "Week 5.3", "Week 6.1", "Week 6.2", "Week 6.3", "Week 7.1", "Week 7.2", "Week 7.3", "Week 8.1", "Week 8.2", "Week 8.3", "Week 9.1", "Week 9.2", "Week 9.3")
png(filename = "boxplot_GSE15744_vsnrma.png", res = 115)
boxplot(exprs(data.GSE15744.vsnrma), col = rainbow(15),
        cex.axis=0.5,
        las=2,
        names=weeks,
        ylab="Relative expression",
        main="Boxplots of microarrays with\nvsn normalized intensities")

dev.off()

boxplot.GSE15744.vsnrma = ggdraw() + draw_image("boxplot_GSE15744_vsnrma.png") 


```

```{r Scatter-plot, include=FALSE}
exprs.GSE15744.vsnrma <- as.data.frame(exprs(data.GSE15744.vsnrma))
scatterplot.GSE15744.vsnrma <- ggplot(data = exprs.GSE15744.vsnrma) +
  theme_bw() +
  geom_point(mapping = aes(exprs.GSE15744.vsnrma[,4], exprs.GSE15744.vsnrma[,5]), size = 0.5) +
  geom_abline(intersept = 0, slope = 1, col = "red", aes(size=1)) +
  labs(title = "Scatterplot of microarrays,\ntwo replica of week 5",
       x = colnames(data.GSE15744.vsnrma)[4],
       y = colnames(data.GSE15744.vsnrma)[5])+ theme(aspect.ratio = 1) 
```



## Analytical methods
1. **Annotation**:
To make sense of the intensity values, they need to be associated to data with known properties. We applied the data frame *ensembl_103.txt* provided by Dr. Dinkelacker in order to annotate our data and yield the appropriate transcript ID for eacg the Probe ID in the microarray. To further annotate for TRAs, we used another data frame by Dr Dinkelacker called *tra.2017.human.gtex.5x.table.tsv*.

```{r Annotation, include=FALSE}
file_annotation = getURL("https://raw.githubusercontent.com/datascience-mobi-2022/2022-topic-04-team-01/main/R%20Scripts/ensembl_103_human.txt")
ensembl = as.data.frame(read.csv(text=file_annotation, sep = ","))   # Load the ensembl annotation data
TranscriptID_orig = rownames(embryo_df)
ensembl = ensembl %>% distinct(Transcript.stable.ID, .keep_all = TRUE) # Remove duplicate transcripts


embryo_df["Transcript.stable.ID"] = gsub("[//._].*$" ,"", rownames(embryo_df)) # Add column for mergign
embryo_df_ann = left_join(embryo_df, ensembl, by="Transcript.stable.ID")   # Combine the Datasets by their Transcript ID
rownames(embryo_df_ann) <- embryo_df_ann$Transcript.stable.ID
colnames(embryo_df_ann) = c("Week4_1", "Week4_2", "Week4_3", "Week5_1", "Week5_2", "Week5_3", "Week6_1", "Week6_2", "Week6_3", "Week7_1", "Week7_2", "Week7_3", "Week8_1", "Week8_2", "Week8_3", "Week9_1", "Week9_2", "Week9_3", "Transcript.stable.ID", "Gene.stable.ID", "Chromosome.scaffold.name", "HGNC.symbol", "AFFY.HG.U133.Plus.2.probe")


file_tra = getURL("https://raw.githubusercontent.com/datascience-mobi-2022/2022-topic-04-team-01/main/R%20Scripts/tra.2017.human.gtex.5x.table.tsv")
tra_gtex = as.data.frame(read.csv(text = file_tra, sep = "\t"))
embryo_df_tra = embryo_df_ann[embryo_df_ann$Transcript.stable.ID %in% tra_gtex$ensembl.transcript ,]
embryo_df_tra = embryo_df_tra %>%  dplyr::rename(ensembl.transcript = Transcript.stable.ID)
embryo_df_tra =  left_join(embryo_df_tra, tra_gtex, by="ensembl.transcript") 


#Make Tissue names more accessible by creating a vector for them
a = separate(embryo_df_tra, tissues, into = as.character(c(1:30)), sep="/")
create_tissue_vector = function(f) {
  f = unlist(f)
  f = f[!is.na(f)]
  return(f)
}
tissues = apply(a[,31:60],1, create_tissue_vector)
tissues = sapply(tissues, unlist)
a = a[,c(1:30, 61)] 
a %>% add_column(tissues = NA)
a$tissues = tissues
embryo_df_tissues = a
```

```{r further-silent-Data-Cleaning}
embryo_df_tissues <- embryo_df_tissues[,!names(embryo_df_tissues) %in% c("ensembl.gene", "ensembl.symbol", "ensembl.chrom")] 

#Chromosome Spalte aufhÃ¼bschen und unnÃ¶tiges lÃ¶schen
embryo_df_tissues$Chromosome.scaffold.name <- as.factor(embryo_df_tissues$Chromosome.scaffold.name) 
embryo_df_tissues <- embryo_df_tissues %>%
  filter(Chromosome.scaffold.name != "CHR_HSCHR10_1_CTG2" & Chromosome.scaffold.name != "CHR_HSCHR11_1_CTG7"& Chromosome.scaffold.name!= "CHR_HSCHR15_4_CTG8"& Chromosome.scaffold.name!= "CHR_HSCHR22_1_CTG3"& Chromosome.scaffold.name!= "CHR_HSCHR22_1_CTG7"& Chromosome.scaffold.name!=  "CHR_HSCHR4_6_CTG12"& Chromosome.scaffold.name!= "CHR_HSCHR7_2_CTG4_4"& Chromosome.scaffold.name!=  "CHR_HSCHR7_2_CTG6" )
embryo_df_tissues$Chromosome.scaffold.name <- factor(embryo_df_tissues$Chromosome.scaffold.name, levels = c("1", "2","3","4","5","6","7","8","9","10","11","12","13","14","15","16","17","18","19", "20", "21","22", "X","Y"))

# Nach NA values suchen
#sum(is.na(embryo_df_tissues$ensembl.entrez)) # diese Spalte enthÃ¤lt 501 NA - Values
```

2. **Limma**
The *limma* package determines among other things the changes of gene expression over time from intensity values of our microarrays. It facilitates advanced statistical algorithms to calculate the necessary coefficients of a linear model for every intensity value in the data set. It uses information borrowing, quantitative weighting, variance modeling and data preprocessing without subsetting the data (Ritchie. *et al.* 2015). Because the linear model was cast on every intensity value, a statistical test called Empirical Bayes can determine differential expressed transcripts via t-statistics and their associated p-values. 

3. **Over representation analysis**
The statistical method overrepresentation analysis determines which functional gene sets in a subset of genes are overrepresented compared to a mother dataset. Categories for functions can be accessed via gene ontology.

# Results

## Limma analysis {#limma}

To filter our data for biologically interesting information, we performed *limma* analysis to extract differentially expressed transcripts. Our threshold for significance is a Benjamini-Hochberg adjusted p-value of 0.01 or below. We found changes of gene expression in 1,814 transcripts. For all individual tissues within the TRA data, we detected at least 40 differentially expressed transcripts. (Fig. \@ref(fig:bar-tissues)). These numbers are sufficient for a further analysis within individual tissues.  \
A volcano plot for differentially expressed transcripts between week 4 and 9 was made to evaluate the limma analysis. There, transcripts are categorized as up-regulated, down-regulated or not differentially expressed (suppl. Fig. \@ref(fig:Volcano-Plot))). 
For all further results, we used the dataset of all transcripts that are differentially expressed between any weeks with an adjusted p-value smaller than 0.01. 


```{r, include = FALSE}

limma_data = as.data.frame(embryo_df_tissues[1:18], rownames = embryo_df_tissues$ensembl.transcript)

weeks = c("week4", "week5", "week6", "week7", "week8", "week9")
f = factor(c("week4", "week4", "week4", "week5", "week5", "week5", "week6", "week6", "week6", "week7", "week7", "week7", "week8", "week8", "week8", "week9", "week9", "week9"), levels =weeks)
design = model.matrix(~0+f)
colnames(design) =  weeks
fit_2 = lmFit(limma_data, design)
contrast_all = makeContrasts(week9-week8, week9-week7, week9-week6, week9-week5, week9-week4, week8-week7,week8-week6, week8-week5, week8-week4, week7-week6, week7-week5, week7-week4, week6-week5, week6-week4, week5-week4, levels = design)
fit_all = contrasts.fit(fit_2, contrast_all)
fit_all = eBayes(fit_all)
results_all = decideTests(fit_all)
#summary(results_all)


#Create list of all differentially expressed transcripts
diff_genes_list_0.01 = c()

for (i in 1:15) {
 top.table = topTable(fit_all,i, n = Inf)
 n = sapply(rownames(top.table), function(x){
  embryo_df_tissues$ensembl.transcript[as.numeric(x)]
  })
  row.names(top.table) = n
  diff_genes_list_0.01 = c(diff_genes_list_0.01, rownames(top.table[which(top.table$adj.P.Val < 0.01), ]))
}
length(diff_genes_list_0.01) #4522
diff_genes_list_0.01 = diff_genes_list_0.01[!duplicated(diff_genes_list_0.01)]
length(diff_genes_list_0.01) #1814

#Select differential genes from annotated dataframe
diff_genes_ann_0.01 = embryo_df_tissues[embryo_df_tissues$ensembl.transcript %in% diff_genes_list_0.01, ]

```

```{r, eval = FALSE}
#List with 5%

diff_genes_list_0.05 = c()

for (i in 1:15) {
 top.table = topTable(fit_all,i, n = Inf)
 n = sapply(rownames(top.table), function(x){
  embryo_df_tissues$ensembl.transcript[as.numeric(x)]
  })
  row.names(top.table) = n
  diff_genes_list_0.05 = c(diff_genes_list_0.05, rownames(top.table[which(top.table$adj.P.Val < 0.05), ]))
}
diff_genes_list_0.05 = diff_genes_list_0.05[!duplicated(diff_genes_list_0.05)]
```







## Comparison of differentially expressed transcripts between the original paper and our method {#paper-comp}

We acquired the data from the original paper's supplemental materials. These data were imported it into a data frame and then annotated it using our dat set because the both have the same probe set IDs. \
To determine the number of differentially expressed transcripts which overlap between the list from the paper and our data, we plotted a Venn diagram (suppl. Fig. \@ref(fig:Venn-Diagram)). The intersection showed a number of transcripts worthy of further analysis.

```{r importing-data-paper}
#importing the data from xls file
file_paper = getURL("https://raw.githubusercontent.com/datascience-mobi-2022/2022-topic-04-team-01/main/Report/differentially_expressed_transcripts_paper.csv")
paper_df_diff <- as.data.frame(read.csv(text = file_paper, sep = ";")) # 5385 Probe IDs
for (i in 1:length(paper_df_diff)) {
  vec1 <- sapply(paper_df_diff[,i], function(x){x==""})
  paper_df_diff[vec1,i] = NA
}
for (i in seq.int(1,length(paper_df_diff$Probe_Set_ID))) { # die abgeschnittenen Zellen hinzugefÃ¼gt
  if(is.na(paper_df_diff$Probe_Set_ID[i])){
    j=1
    while (is.na(paper_df_diff$Probe_Set_ID[i-j])) {
      j = j+1
    }
    paper_df_diff$Gene_Title[i-j] <- paste(paper_df_diff$Gene_Title[i-j], paper_df_diff$Gene_Title[i])
  }
}
paper_df_diff <- paper_df_diff %>% drop_na(Probe_Set_ID) # Ã¼berflÃ¼ssige Reihen mit restlichem String entfernt
paper_df_diff$Gene_Title[paper_df_diff$Gene_Title == "---"] <- NA # set missing values to NA
paper_df_diff$Gene_Symbol[paper_df_diff$Gene_Symbol == "---"] <- NA
#Datacleaning complete

#Annotation:
paper_df_ann <- left_join(paper_df_diff, embryo_df_tissues, by = c("Probe_Set_ID" = "AFFY.HG.U133.Plus.2.probe"))
paper_df_intersec <- paper_df_ann %>% drop_na(ensembl.transcript) # we are able to compare 3203 Transcripts with TRA data

our_df_diff <- as.data.frame(diff_genes_list_0.01)
our_df_diff$select <- "diff"
paper_df_ann_diff <- left_join(paper_df_ann, our_df_diff, by = c("ensembl.transcript" = "diff_genes_list_0.01"))
paper_df_intersec_diff <- paper_df_ann_diff %>% drop_na(select)

number_vec = sapply(c(1:sum(is.na(paper_df_ann$ensembl.transcript))), function(x){str_glue("no_value", x)} )
paper_df_ann$ensembl.transcript[is.na(paper_df_ann$ensembl.transcript)] <- number_vec

string_ENST <- toString(paper_df_ann$ensembl.transcript)
contain_bool_diff = sapply(diff_genes_list_0.01, function(x) {grepl(x, string_ENST)})
df_unique_diff <- diff_genes_ann_0.01[!contain_bool_diff,]
```

```{r Venn-Diagram-code, include=FALSE}
# Make Venn diagram from list of groups
colors <- c("#ff4059", "#2cff21", "yellow")
venn.diagram(x = list(paper_df_ann$ensembl.transcript, embryo_df_tissues$ensembl.transcript, diff_genes_ann_0.01$ensembl.transcript) ,
            category.names = c("data from paper", "our TRA data", "our diff. data"),
            filename = 'venn_diagram_comparison_paper.png',
            output=TRUE,
            imagetype="png",
            #na = "none",
            #scaled = FALSE,
            height = 1000 , 
            width = 1000 ,
            resolution = 300,
            compression = "lzw",
            col = "black",
            fill = colors,
            cat.col = colors,
            cat.cex = 1.4,
            cat.pos = c(350,10,0),
            #margin = 0
            main = "Venn diagram of \n differentially expressed transcripts",
            main.fontface = "bold"
            
)
```

In contrast to our method using limma, the authors of the original paper used *one-way analysis of variance* with a p-value of 0.05 to determine the differentially expressed transcripts (Yi H *et. al*, 2010). In the paper, they provided an annotation of transcripts that were regulated *up*, *down* or showed an *arch*. To verify this dataset, we looked at the trends of the pattern-annotated data. We used k-means clustering to confirm whether we see these trends in our data (Fig. \@ref(fig:paper-trends-cluster)). \
Fig. \@ref(fig:paper-trends-cluster)A and B show high conformity to the annotated pattern. We repeated both plots with our differentially expressed transcripts and the results were very similar. The figure can be found on our github. \
For Fig. \@ref(fig:paper-trends-cluster)C, clusters 1, 6, 7, 8 clearly show an *arch* regulated pattern, but for clusters 4 and 5 the transcripts appear to be *down* regulated or in the case of cluster 10 *up* regulated. The final clusters 2, 3 and 9 with a total of 156 transcripts are not differentially expressed at all. When we cluster our differentially expressed transcripts annotated with the *arch* pattern (Fig. \@ref(fig:paper-trends-cluster)D), we receive only 10% of the amount of transcripts which are clearly *arch* regulated. \
Additionally, Fig. \@ref(fig:paper-trends-cluster)E reveals, that there a clearly *up* and *down* regulated genes of 646 transcripts or about 10% that are missed by the method used by the paper's authors.




```{r paper-trends-cluster, fig.width=17, fig.height=8, fig.cap = "**Clustering of the common transcripts reveals similar trends ,but our additional data show missing differentially expressed transcripts.** We chose data with transcript IDs both in the dataset of the original paper and in our TRA data. Additionally, they had to be annotated as *up*, *down*, or *arch* by the authors of the paper. The *up* and *down* regulated transcript-data (**A** and **B**) follow the same pattern as postulated in the paper. For the *arch* regulated transcript-data (**C**) we see some clusters matching the *arch* pattern while others are rather undetermined. Of 396 transcripts, only 39 of our differentially expressed transcripts through *limma* analyis share this *arch* property (**D**). Here, we clearly see differential gene expression. In **E** we plotted data of our differentially expressed transcripts not included in the paper's differential expression gene list. There are *up* and *down* regulated genes determined by our *limma* analysis that are not included in the dataset of the authors of the original paper"}

#Clustered plot of expression levels over time for the spleen
paper_df_up <- filter(paper_df_intersec, Expression_pattern == "UP") # 759
paper_df_down <- filter(paper_df_intersec, Expression_pattern == "DOWN") # 1096
paper_df_arch <- filter(paper_df_intersec, Expression_pattern == "ARCH") # 396
paper_df_arch_diff <- filter(paper_df_intersec_diff, Expression_pattern == "ARCH")

# Functions for later use
expression_difference = function(x,y) {
  return ((x-y)/((x+y)*0.5))
}

avg_sil = function(k, data) {
  km = kmeans(data, centers = k, nstart = 25)
  ss = silhouette(km$cluster, dist_data)
  mean(ss[, 3])
}

create_plot_data = function(dataset) {
  data_plot = data.frame(Week = c(sapply(c(4:9), function(x) {rep(x, n_cluster)})), Cluster = as.character(c(rep(c(1:n_cluster), 6))), Expression = NA)
  count = 0
  for(s in 4:9) {
    for(t in 1:n_cluster) {
      count = count + 1
      values = dataset[dataset$Cluster == t, str_glue("Week", s)]
      data_plot[count, "Expression"] = median(values)
      data_plot[count, "Cluster"] = str_glue(data_plot[count, "Cluster"], " [", km$size[t], "]")
   }
  }
  return(data_plot)
}

paper_df_diff_list = list(as.data.frame(paper_df_up[8:25]),as.data.frame(paper_df_down[8:25]),as.data.frame(paper_df_arch[8:25]),as.data.frame(paper_df_arch_diff[8:25]),  as.data.frame(df_unique_diff[,1:18]))
# Select an modify the data used for kmeans
i = 0
names <- c("Up-regulated", "Down-regulated","Arch-regulated","Arch-regulated which intersect our differentially expressed transcripts","Differentially expressed transcripts not included" )
for (data_temp in paper_df_diff_list){
  i= i+1
data_abs = data.frame(Week4 = apply(data_temp[1:3], 1, median), Week5 =  apply(data_temp[4:6], 1, median), Week6 =  apply(data_temp[7:9], 1, median),
                        Week7=apply(data_temp[10:12], 1, median), Week8 =  apply(data_temp[13:15], 1, median), Week9 =  apply(data_temp[16:18], 1, median))
data = data.frame("5_4" = expression_difference(data_abs[,"Week5"], data_abs[,"Week4"]), 
                  "6_5" = expression_difference(data_abs[,"Week6"], data_abs[,"Week5"]),
                  "7_6" = expression_difference(data_abs[,"Week7"], data_abs[,"Week6"]),
                  "8_7" = expression_difference(data_abs[,"Week8"], data_abs[,"Week7"]),
                  "9_8" = expression_difference(data_abs[,"Week9"], data_abs[,"Week8"]))
row.names(data) = rownames(data_abs)
  
# Select optimum cluster number
data = scale(data)
dist_data = get_dist(data, method = "pearson")
n_cluster = which.max(sapply(c(2:10), function(k) {avg_sil(k, data)})) +1
  
# Create the data for our plot
km = kmeans(data, centers = n_cluster, nstart = 25)
data_plot_raw = cbind(data_abs, Cluster = km$cluster)
plot_data = create_plot_data(data_plot_raw)

assign(str_glue( "plot_", i), ggplot(plot_data, aes(x = Week, y= Expression, group = Cluster))  + geom_line(aes(colour = Cluster)) +geom_point(aes(colour = Cluster, shape = Cluster)) + labs(title = names[i]) + scale_color_npg() + theme_classic()  + ylim(6.5,8.5) + theme(aspect.ratio = 1, plot.margin = unit(c(0.5,0,0.5,0),"cm")))
}

#Paper_clustering_plot = arrangeGrob(grobs = list(plot_1, plot_2, plot_3),
#                                    nrow=1, widths = c(5,5,5), heights = c(5))

lay = rbind(c(1,2,3), c(4,5,NA))

Paper_clustering_plot = arrangeGrob(grobs = list(plot_1, plot_2, plot_3, plot_4, plot_5),
                                    layout_matrix = lay)

Paper_clustering_plot_output = as_ggplot(Paper_clustering_plot) +
  draw_plot_label(label = c("A", "B", "C", "D", "E"), size = 15, x = c(0.01, 0.34, 0.68, 0.01, 0.34), y = c(0.55, 0.55, 0.55, 0.05, 0.05)) 

annotate_figure(Paper_clustering_plot_output, top = text_grob("Intensity values of transcripts determined to be differentially expressed by authors of the paper and supplemented by our data", color = "black", face = "bold", size = 16))


```



## TRAs can infer a basic timeline of organ development {#organ}

### Differentially expressed transcripts can be linked to all analyzed tissues {#organ-overview}

```{r, fig.width=8, fig.height=6, include=FALSE}
# Create a barplot for the TRA count of each tissue based on our dataset of differentially expressed transcripts

all_tissues = c("Adipose - Subcutaneous", "Adipose - Visceral", "Cells - Transformed fibroblasts", "Skin - Not Sun Exposed", "Skin - Sun Exposed",  "Muscle - Skeletal", "Bladder", "Kidney - Cortex", "Colon - Sigmoid", "Colon - Transverse", "Esophagus - Gastroesophageal Junction", "Esophagus - Mucosa", "Esophagus - Muscularis", "Liver", "Minor Salivary Gland", "Pancreas", "Small Intestine - Terminal Ileum", "Spleen", "Stomach", "Breast - Mammary Tissue", "Cervix - Ectocervix", "Cervix - Endocervix",  "Fallopian Tube", "Ovary", "Uterus", "Vagina", "Prostate", "Testis", "Adrenal Gland", "Pituitary", "Thyroid", "Artery - Aorta", "Artery - Coronary", "Artery - Tibial", "Heart - Atrial Appendage", "Heart - Left Ventricle", "Lung", "Whole Blood", "Cells - EBV-transformed lymphocytes", "Brain - Amygdala", "Brain - Anterior cingulate cortex", "Brain - Caudate", "Brain - Cerebellar Hemisphere", "Brain - Cerebellum", "Brain - Cortex", "Brain - Frontal Cortex", "Brain - Hippocampus", "Brain - Hypothalamus", "Brain - Nucleus accumbens", "Brain - Putamen", "Brain - Spinal cord", "Brain - Substantia nigra",  "Nerve - Tibial")
tissue_count = data.frame(tissue = all_tissues, count = sapply(all_tissues, function(y) {sum(sapply(diff_genes_ann_0.01$tissues, function(x) {grepl(y, toString(x))}))}))

tissue_count_barplot =  ggplot(tissue_count, aes(x = tissue, y = count, fill = tissue))  + geom_bar(stat = "identity") + labs(x = "Tissue", y = "Number of Transcripts")  + theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1)) + guides(fill = FALSE) 
```

```{r, fig.width=5, fig.height=5, include=FALSE}

select_tissues = c("Adipose", "Cells - Transformed fibroblasts", "Skin",  "Muscle - Skeletal", "Bladder", "Kidney - Cortex",  "Colon", "Esophagus", "Liver", "Minor Salivary Gland", "Pancreas", "Small Intestine - Terminal Ileum", "Spleen", "Stomach", "Breast - Mammary Tissue", "Cervix", "Fallopian Tube", "Ovary", "Uterus", "Vagina", "Prostate", "Testis", "Adrenal Gland", "Pituitary", "Thyroid",  "Artery", "Heart", "Lung", "Whole Blood", "Cells - EBV-transformed lymphocytes", "Brain - Spinal cord", "Brain", "Nerve - Tibial")



count_data = data.frame(tissue = select_tissues, count = NA)
for (tissue in select_tissues){
  
  count_data[count_data$tissue == tissue, "count"] = sum(bool_contained = sapply(diff_genes_ann_0.01$tissues, function(x) {grepl(tissue, toString(x))}))

}
cols = rainbow(33, s=.6, v=.9) #[sample(1:33,33)] #61
count_data$color = cols
treemap = ggplot(count_data, aes(area = count, label = tissue, fill = tissue)) + geom_treemap() + geom_treemap_text(colour = "white", place = "top", reflow = T) + scale_fill_manual(values=cols) + guides(fill = FALSE)
```


```{r bar-tissues, fig.width=13, fig.height=6, fig.cap= "**A**. The number of transcripts associated with each tissue, including subtissues, is displayed.  **B**. The plot shows the share of TRAs associated with each tissue. Subtissues are subsumed under their main tissue."}
lay = rbind(c(1,1,1,2), c(1,1,1, NA))
plot = arrangeGrob(grobs = list(tissue_count_barplot, treemap),layout_matrix = lay)
plot_output = as_ggplot(plot) + draw_plot_label(label = c("A", "B"), size = 15, x = c(0.05, 0.77), y = c(0.1,0.1)) + theme (plot.margin = unit(c(0.2,1.2,0,1.2),"cm"))
annotate_figure(plot_output, top = text_grob("Differentially expressed transcripts of TRAs by tissue", color = "black", face = "bold", size = 16))
```

```{r, eval = FALSE}
# Mean number of tissues associated with each transcript
mean(sapply(diff_genes_ann_0.01$tissues, function(x) {length(x)}))
```


The TRA dataset covers 53 distinct tissues (Fig. \@ref(fig:bar-tissues)). The minimum of differentially expressed transcripts inone  tissue were 46 stomach-linked transcripts, the highest reached 837 TRAs for the testes. \
Nonetheless, there is a significant overlap between the TRAs associated with different tissues,  as each transcript is on average linked to 7.4 distinct sub-tissues or tissues. This overlap is illustrated by suppl. Fig. \@ref(fig:tissue-links). Furthermore, a detailed heatmap for the shared TRAs between tissues is shown in the suppl. Fig. \@ref(fig:heatmap-tissues).


 
### The expression of all TRAs associated with a tissue cannot be used to infer organ development {#organ-clustering}

In this research, we attempted to draw conclusions about the developmental state of a tissue based on the expression of genes associated with it. For all differencially expressed transcripts associated with one tissue, we analyzed the share of transcripts above a certain expression level over time, as shown in Fig. \@ref(fig:expression-plot)A. Furthermore, we observed trends within the median expression of these trasnscripts (Fig. \@ref(fig:expression-plot)B). Since both metrics only depicted miniscule changes, we hypothesized that distinct, counteracting trends in expression exist within one tissue. Thus, k-means clustering was used to determine groups of TRAs with similar expression patterns. For each of these clusters, the median expression was plotted as shown in Fig. \@ref(fig:expression-plot)C. The clustering graphs for all tissues can be found on our github repository.

```{r expression-threshold, include=FALSE}
# Create dataframe with median expression levels for each transcript and week (combining the replicates)
embryo_df_tissues_median  = data.frame(Week4 = apply(embryo_df_tissues[1:3], 1, median), Week5 =  apply(embryo_df_tissues[4:6], 1, median), Week6 =  apply(embryo_df_tissues[7:9], 1, median), Week7=apply(embryo_df_tissues[10:12], 1, median), Week8 =  apply(embryo_df_tissues[13:15], 1, median), Week9 =  apply(embryo_df_tissues[16:18], 1, median))
embryo_df_tissues_median = cbind(embryo_df_tissues_median, embryo_df_tissues[,c(19,29)])


tissue = "Spleen"
threshold = c(6.6,6.8, 7.0, 7.2, 7.4, 7.6, 8.0)

# Select the data from a tissues
bool_contained = sapply(embryo_df_tissues_median$tissues, function(x) {grepl(tissue, toString(x))})
data_temp = embryo_df_tissues_median[bool_contained,]

#Determine the percentage of expressed transcripts
count = 0
a = c()
data_plot = data.frame(Week = c(sapply(c(4:9), function(x) {rep(x, length(threshold))})), Threshold = factor(c(rep(threshold, 6))), Percentage = NA)
  for(s in 4:9) {
    for(t in 1:length(threshold)) {
      count = count +1
      a  = c(a,(sum(data_temp[,str_glue("Week", s)] > threshold[t]))/(dim(data_temp)[1]))
    }
  }
data_plot$Percentage= a

  
#Plot the data
spleen_percentage_plot =  ggplot(data_plot, aes(x = Week, y= Percentage, group = Threshold))   + geom_line(aes(colour = Threshold)) + geom_point(aes(colour = Threshold, shape = Threshold)) + labs(title = "Share over threshold") + theme_classic() + ylim(0,1)

```

```{r expression-median, include=FALSE}
#Plot of median expression level of all spleen-related genes over time

#Select genes only of target tissue
bool_contained = sapply(diff_genes_ann_0.01$tissues, function(x) {"Spleen" %in% x})

#Graph of mean gene expression over time
spleen_genes = diff_genes_ann_0.01[bool_contained,]
spleen_genes = spleen_genes[, 1:19]
spleen_genes_mean = data.frame(week4=rowMeans(spleen_genes[,1:3]),week5=rowMeans(spleen_genes[,4:6]), week6=rowMeans(spleen_genes[,7:9]),week7=rowMeans(spleen_genes[,10:12]), week8=rowMeans(spleen_genes[,13:15]), week9=rowMeans(spleen_genes[,16:18]))
spleen_genes_graph = apply(spleen_genes_mean, 2, median)
spleen_genes_graph= data.frame (Expression = spleen_genes_graph, Week = c(4:9))

spleen_genes_plot = ggplot(spleen_genes_graph, aes(x = Week, y= Expression))  + geom_line(color = "#0000ff") + geom_point(color = "#0000ff") + labs(title = paste("Median expression"))  + theme_classic() +  ylim(6.5,8.5)

```

```{r expression-clustered, include=FALSE}
#Clustered plot of expression levels over time for the spleen

tissue = "Spleen"

# Functions for later use
expression_difference = function(x,y) {
  return ((x-y)/((x+y)*0.5))
}

avg_sil = function(k, data) {
  km = kmeans(data, centers = k, nstart = 25)
  ss = silhouette(km$cluster, dist_data)
  mean(ss[, 3])
}

create_plot_data = function(dataset) {
  data_plot = data.frame(Week = c(sapply(c(4:9), function(x) {rep(x, n_cluster)})), Cluster = as.character(c(rep(c(1:n_cluster), 6))), Expression = NA)
  count = 0
  for(s in 4:9) {
    for(t in 1:n_cluster) {
      count = count + 1
      values = dataset[dataset$Cluster == t, str_glue("Week", s)]
      data_plot[count, "Expression"] = median(values)
      data_plot[count, "Cluster"] = str_glue(data_plot[count, "Cluster"], " [", km$size[t], " genes]")
   }
  }
  return(data_plot)
}



# Select an modify the data used for kmeans
bool_contained = sapply(diff_genes_ann_0.01$tissues, function(x) {grepl(tissue, toString(x))})
data_temp = diff_genes_ann_0.01[bool_contained,]
data_abs = data.frame(Week4 = apply(data_temp[1:3], 1, median), Week5 =  apply(data_temp[4:6], 1, median), Week6 =  apply(data_temp[7:9], 1, median),
                        Week7=apply(data_temp[10:12], 1, median), Week8 =  apply(data_temp[13:15], 1, median), Week9 =  apply(data_temp[16:18], 1, median))
data = data.frame("5_4" = expression_difference(data_abs[,"Week5"], data_abs[,"Week4"]), 
                  "6_5" = expression_difference(data_abs[,"Week6"], data_abs[,"Week5"]),
                  "7_6" = expression_difference(data_abs[,"Week7"], data_abs[,"Week6"]),
                  "8_7" = expression_difference(data_abs[,"Week8"], data_abs[,"Week7"]),
                  "9_8" = expression_difference(data_abs[,"Week9"], data_abs[,"Week8"]))
row.names(data) = rownames(data_abs)
  
# Select optimum cluster number
data = scale(data)
dist_data = get_dist(data, method = "pearson")
n_cluster = which.max(sapply(c(2:10), function(k) {avg_sil(k, data)})) +1
  
# Create the data for our plot
km = kmeans(data, centers = n_cluster, nstart = 25)
data_plot_raw = cbind(data_abs, Cluster = km$cluster)
plot_data = create_plot_data(data_plot_raw)
  
# Create the Plot
spleen_clustered_plot =  ggplot(plot_data, aes(x = Week, y= Expression, group = Cluster))  + geom_line(aes(colour = Cluster)) +
          geom_point(aes(colour = Cluster, shape = Cluster)) + labs(title = "Clustered expression") +
          scale_color_npg() + theme_classic()  + ylim(6.5,8.5)
   
```

```{r expression-plot, fig.width= 10, fig.height= 3, fig.cap= "**A**. For different expression thresholds the share of differentially expressed transcripts with a higher expression than the threshold is depicted. **B**. The median expression for all spleen-associated TRAs is shown for each point in time. **C**. For k-means clustring, the silhouette score determined an optimum of two clusters. The median expression analog to B is plottet for each of these clusters. "}

plot = arrangeGrob(grobs = list(spleen_percentage_plot, spleen_genes_plot, spleen_clustered_plot), nrow=1, widths = c(3.3,2.7, 4), heights = c(0.7))
plot_output = as_ggplot(plot) + draw_plot_label(label = c("A", "B", "C"), size = 15, x = c(0.03, 0.35, 0.63), y = c(0.13, 0.13, 0.13)) 
annotate_figure(plot_output, top = text_grob("Expression over time of Spleen-related differentially expressed transcripts", color = "black", face = "bold", size = 14))
```

For many tissues, as shown here exemplary with the spleen, the clustering revealed two or more clusters that could be characterized as either an upregulation or a downregulation. In order to determine the indications for organ development, we analyzed the functions of the transcripts belonging to the two clusters.

### The clusters of up- and downregulated transcripts can be linked to distinct gene functions. {#organ-tables}

```{r spleen-clusters}
#Get the spleen dataset with the clusters
expression_difference = function(x,y) {
  return ((x-y)/((x+y)*0.5))
}

avg_sil = function(k, data) {
  km = kmeans(data, centers = k, nstart = 25)
  ss = silhouette(km$cluster, dist_data)
  mean(ss[, 3])
}
bool_contained = sapply(diff_genes_ann_0.01$tissues, function(x) {grepl("Spleen", toString(x))})
  data_temp = diff_genes_ann_0.01[bool_contained,]
  data_abs = data.frame(Week4 = apply(data_temp[1:3], 1, median), Week5 =  apply(data_temp[4:6], 1, median), Week6 =  apply(data_temp[7:9], 1, median),
                        Week7=apply(data_temp[10:12], 1, median), Week8 =  apply(data_temp[13:15], 1, median), Week9 =  apply(data_temp[16:18], 1, median))
  data = data.frame("5_4" = expression_difference(data_abs[,"Week5"], data_abs[,"Week4"]), 
                  "6_5" = expression_difference(data_abs[,"Week6"], data_abs[,"Week5"]),
                  "7_6" = expression_difference(data_abs[,"Week7"], data_abs[,"Week6"]),
                  "8_7" = expression_difference(data_abs[,"Week8"], data_abs[,"Week7"]),
                  "9_8" = expression_difference(data_abs[,"Week9"], data_abs[,"Week8"]))
  row.names(data_abs) = data_temp$ensembl.transcript
  row.names(data) = rownames(data_abs)
  
  data = scale(data)
  dist_data = get_dist(data, method = "pearson")
  n_cluster = which.max(sapply(c(2:10), function(k) {avg_sil(k, data)})) +1
  
  km = kmeans(data, centers = n_cluster, nstart = 25)
  data_plot_raw = cbind(data_abs, Cluster = km$cluster)

  spleen_genes_clustered = data_plot_raw

```

The spleen is a clear example of two dinstict clusters (suppl. Fig. \@ref(fig:spleen-boxplot)), with one consisting of upregulated previously inactive genes and one with downregulated highly active genes. For all these differentially expressed transcripts, we used the NCBI gene database to get a functional annotation. Of the 98 upregulated genes, 48 had a functional annotation. 17 of those were clearly associated with immune system or blood functions and thus relevant for the functional spleen. We further found 157 downregulated genes. There, 70 were annotated and 45 of those displayed a relation to the cell cycle or cell division. The tables of the transcripts with a relevant function are visible in the supplementary material (suppl. Fig. \@ref(fig:spleen-up-table) and Fig. \@ref(fig:spleen-down-table)).

### Overrepresentation Analysis can create plots that signify organ development {#organ-ora}

For this analysis, the eight tissues with the most meaningful results were chosen. In Fig. \@ref(fig:ORA-plot), the most important functions for these tissues were determined through overrepresentation analysis. In addition, the expression of the associated transcripts was plotted.

```{r ORA-code, include = FALSE}
#Rearrange the selected plots so that we can combine the table with a plot 


# Spleen 
tissue = "Spleen"
selected = c(1,2,3,4,5)
file_tissue = "https://raw.githubusercontent.com/datascience-mobi-2022/2022-topic-04-team-01/main/Report/Spleen_table.png"
#Overrepresentation Analysis
bool_contained = sapply(embryo_df_tissues$tissues, function(x) {grepl(tissue, toString(x))})
embryo_df_tissue = embryo_df_tissues[bool_contained,]
deGenes = as.character(embryo_df_tissue$ensembl.entrez[!duplicated(embryo_df_tissue$Gene.stable.ID)])
geneUniverse = as.character(embryo_df_tissues$ensembl.entrez[!duplicated(embryo_df_tissues$Gene.stable.ID)])
enr_GO_tissue1 = enrichGO(gene = deGenes, ont = "BP",  OrgDb ="org.Hs.eg.db", universe = geneUniverse, readable=TRUE, pvalueCutoff = 0.05)
enr_GO_tissue = as.data.frame(enr_GO_tissue1)
#Create the table of most significant functions
if (dim(enr_GO_tissue)[1] <10) {npath = dim(enr_GO_tissue)[1]} else {npath = 10}
tissue_table = enr_GO_tissue[1:npath,c(2,6)]
colnames(tissue_table) = c("Function", "Adj_p_value")
#assign(str_glue(tissue,"_table"), gt(tissue_table)|>  tab_header(title= tissue) |> fmt_markdown(columns = c(Function)) |> gt_theme_538() |> fmt_scientific(columns = c(Adj_p_value), decimals = 0) |>  tab_style(style = list(cell_text(weight = "bold")), locations = cells_body(columns = Function, rows = selected)))
#gtsave(get(str_glue(tissue, "_table")),str_glue(tissue, "_table.png"))
download.file(file_tissue, str_glue(tissue, "_table.png"), mode = "wb")
assign(str_glue(tissue, "_t"), (ggdraw() + draw_image(str_glue(tissue, "_table.png"))))
#Plot the chosen functions
data_plot = data.frame(Week = c(sapply(c(4:9), function(x) {rep(x, length(selected))})), Function =
                             as.character(c(rep(enr_GO_tissue$Description[selected], 6)), Expression=NA))
for (i in selected) {
  xx = strsplit(enr_GO_tissue$geneID[i], "/")
  bool_contained1 = sapply(diff_genes_ann_0.01$HGNC.symbol, function(x) {grepl(x, toString(xx))})
  bool_contained1[is.na(bool_contained1)] = FALSE
  data_temp = diff_genes_ann_0.01[bool_contained1,]
  data_abs = data.frame(Week4 = apply(data_temp[1:3], 1, median), Week5 = apply(data_temp[4:6], 1, median), Week6 =  
                              apply(data_temp[7:9], 1, median), Week7=apply(data_temp[10:12], 1, median), Week8 = 
                              apply(data_temp[13:15], 1, median), Week9 =  apply(data_temp[16:18], 1, median))
  expr_path = colMeans(data_abs[1:6,])
  data_plot[data_plot$Function == enr_GO_tissue$Description[i],"Expression"] = expr_path
}
assign(str_glue(tissue, "_plot"), ggplot(data_plot, aes(x = Week, y= Expression, group = Function))  + geom_line(aes(colour =Function), position = position_dodge(0)) + geom_point(aes(colour = Function, shape = Function), position = position_dodge(0)) + labs(title = tissue) +  theme_classic() + scale_color_npg()  + theme(legend.title = element_text(size = 9), legend.text = element_text(size = 7), aspect.ratio = 1, legend.position = "bottom", legend.spacing.y = unit(0.1, "cm"),legend.key.size = unit(0.4, "cm"), plot.margin = unit(c(0.5,0.8,0.2,0),"cm")) + guides(colour = guide_legend(nrow = length(selected), title.position = "top",byrow = TRUE), shape = guide_legend(nrow = length(selected), title.position = "top",byrow = TRUE)))




# Brain 
tissue = "Brain"
selected = c(1,5,7)
file_tissue = "https://raw.githubusercontent.com/datascience-mobi-2022/2022-topic-04-team-01/main/Report/Brain_table.png"
#Overrepresentation Analysis
bool_contained = sapply(embryo_df_tissues$tissues, function(x) {grepl(tissue, toString(x))})
embryo_df_tissue = embryo_df_tissues[bool_contained,]
deGenes = as.character(embryo_df_tissue$ensembl.entrez[!duplicated(embryo_df_tissue$Gene.stable.ID)])
geneUniverse = as.character(embryo_df_tissues$ensembl.entrez[!duplicated(embryo_df_tissues$Gene.stable.ID)])
enr_GO_tissue1 = enrichGO(gene = deGenes, ont = "BP",  OrgDb ="org.Hs.eg.db", universe = geneUniverse, readable=TRUE, pvalueCutoff = 0.05)
enr_GO_tissue = as.data.frame(enr_GO_tissue1)
#Create the table of most significant functions
if (dim(enr_GO_tissue)[1] <10) {npath = dim(enr_GO_tissue)[1]} else {npath = 10}
tissue_table = enr_GO_tissue[1:npath,c(2,6)]
colnames(tissue_table) = c("Function", "Adj_p_value")
#assign(str_glue(tissue,"_table"), gt(tissue_table)|>  tab_header(title= tissue) |> fmt_markdown(columns = c(Function)) |> gt_theme_538() |> fmt_scientific(columns = c(Adj_p_value), decimals = 0) |>  tab_style(style = list(cell_text(weight = "bold")), locations = cells_body(columns = Function, rows = selected)))
#gtsave(get(str_glue(tissue, "_table")),str_glue(tissue, "_table.png"))
download.file(file_tissue, str_glue(tissue, "_table.png"), mode = "wb")
assign(str_glue(tissue, "_t"), (ggdraw() + draw_image(str_glue(tissue, "_table.png"))))
#Plot the chosen functions
data_plot = data.frame(Week = c(sapply(c(4:9), function(x) {rep(x, length(selected))})), Function =
                             as.character(c(rep(enr_GO_tissue$Description[selected], 6)), Expression=NA))
for (i in selected) {
  xx = strsplit(enr_GO_tissue$geneID[i], "/")
  bool_contained1 = sapply(diff_genes_ann_0.01$HGNC.symbol, function(x) {grepl(x, toString(xx))})
  bool_contained1[is.na(bool_contained1)] = FALSE
  data_temp = diff_genes_ann_0.01[bool_contained1,]
  data_abs = data.frame(Week4 = apply(data_temp[1:3], 1, median), Week5 = apply(data_temp[4:6], 1, median), Week6 =  
                              apply(data_temp[7:9], 1, median), Week7=apply(data_temp[10:12], 1, median), Week8 = 
                              apply(data_temp[13:15], 1, median), Week9 =  apply(data_temp[16:18], 1, median))
  expr_path = colMeans(data_abs[1:6,])
  data_plot[data_plot$Function == enr_GO_tissue$Description[i],"Expression"] = expr_path
}
assign(str_glue(tissue, "_plot"), ggplot(data_plot, aes(x = Week, y= Expression, group = Function))  + geom_line(aes(colour =Function), position = position_dodge(0)) + geom_point(aes(colour = Function, shape = Function), position = position_dodge(0)) + labs(title = tissue) +  theme_classic() + scale_color_npg()  + theme(legend.title = element_text(size = 9), legend.text = element_text(size = 7), aspect.ratio = 1, legend.position = "bottom", legend.spacing.y = unit(0.1, "cm"),legend.key.size = unit(0.4, "cm"), plot.margin = unit(c(0.5,0.8,0.2,0),"cm")) + guides(colour = guide_legend(nrow = length(selected), title.position = "top",byrow = TRUE), shape = guide_legend(nrow = length(selected), title.position = "top",byrow = TRUE)))


# Liver 
tissue = "Liver"
selected = c(1,3,5,7,9)
file_tissue = "https://raw.githubusercontent.com/datascience-mobi-2022/2022-topic-04-team-01/main/Report/Liver_table.png"
#Overrepresentation Analysis
bool_contained = sapply(embryo_df_tissues$tissues, function(x) {grepl(tissue, toString(x))})
embryo_df_tissue = embryo_df_tissues[bool_contained,]
deGenes = as.character(embryo_df_tissue$ensembl.entrez[!duplicated(embryo_df_tissue$Gene.stable.ID)])
geneUniverse = as.character(embryo_df_tissues$ensembl.entrez[!duplicated(embryo_df_tissues$Gene.stable.ID)])
enr_GO_tissue1 = enrichGO(gene = deGenes, ont = "BP",  OrgDb ="org.Hs.eg.db", universe = geneUniverse, readable=TRUE, pvalueCutoff = 0.05)
enr_GO_tissue = as.data.frame(enr_GO_tissue1)
#Create the table of most significant functions
if (dim(enr_GO_tissue)[1] <10) {npath = dim(enr_GO_tissue)[1]} else {npath = 10}
tissue_table = enr_GO_tissue[1:npath,c(2,6)]
colnames(tissue_table) = c("Function", "Adj_p_value")
#assign(str_glue(tissue,"_table"), gt(tissue_table)|>  tab_header(title= tissue) |> fmt_markdown(columns = c(Function)) |> gt_theme_538() |> fmt_scientific(columns = c(Adj_p_value), decimals = 0) |>  tab_style(style = list(cell_text(weight = "bold")), locations = cells_body(columns = Function, rows = selected)))
#gtsave(get(str_glue(tissue, "_table")),str_glue(tissue, "_table.png"))
download.file(file_tissue, str_glue(tissue, "_table.png"), mode = "wb")
assign(str_glue(tissue, "_t"), (ggdraw() + draw_image(str_glue(tissue, "_table.png"))))
#Plot the chosen functions
data_plot = data.frame(Week = c(sapply(c(4:9), function(x) {rep(x, length(selected))})), Function =
                             as.character(c(rep(enr_GO_tissue$Description[selected], 6)), Expression=NA))
for (i in selected) {
  xx = strsplit(enr_GO_tissue$geneID[i], "/")
  bool_contained1 = sapply(diff_genes_ann_0.01$HGNC.symbol, function(x) {grepl(x, toString(xx))})
  bool_contained1[is.na(bool_contained1)] = FALSE
  data_temp = diff_genes_ann_0.01[bool_contained1,]
  data_abs = data.frame(Week4 = apply(data_temp[1:3], 1, median), Week5 = apply(data_temp[4:6], 1, median), Week6 =  
                              apply(data_temp[7:9], 1, median), Week7=apply(data_temp[10:12], 1, median), Week8 = 
                              apply(data_temp[13:15], 1, median), Week9 =  apply(data_temp[16:18], 1, median))
  expr_path = colMeans(data_abs[1:6,])
  data_plot[data_plot$Function == enr_GO_tissue$Description[i],"Expression"] = expr_path
}
assign(str_glue(tissue, "_plot"), ggplot(data_plot, aes(x = Week, y= Expression, group = Function))  + geom_line(aes(colour =Function), position = position_dodge(0)) + geom_point(aes(colour = Function, shape = Function), position = position_dodge(0)) + labs(title = tissue) +  theme_classic() + scale_color_npg()  + theme(legend.title = element_text(size = 9), legend.text = element_text(size = 7), aspect.ratio = 1, legend.position = "bottom", legend.spacing.y = unit(0.1, "cm"),legend.key.size = unit(0.4, "cm"), plot.margin = unit(c(0.5,0.8,0.2,0),"cm")) + guides(colour = guide_legend(nrow = length(selected), title.position = "top",byrow = TRUE), shape = guide_legend(nrow = length(selected), title.position = "top",byrow = TRUE)))


# Heart 
tissue = "Heart"
selected = c(1,2,3,6,10)
file_tissue = "https://raw.githubusercontent.com/datascience-mobi-2022/2022-topic-04-team-01/main/Report/Heart_table.png"
#Overrepresentation Analysis
bool_contained = sapply(embryo_df_tissues$tissues, function(x) {grepl(tissue, toString(x))})
embryo_df_tissue = embryo_df_tissues[bool_contained,]
deGenes = as.character(embryo_df_tissue$ensembl.entrez[!duplicated(embryo_df_tissue$Gene.stable.ID)])
geneUniverse = as.character(embryo_df_tissues$ensembl.entrez[!duplicated(embryo_df_tissues$Gene.stable.ID)])
enr_GO_tissue1 = enrichGO(gene = deGenes, ont = "BP",  OrgDb ="org.Hs.eg.db", universe = geneUniverse, readable=TRUE, pvalueCutoff = 0.05)
enr_GO_tissue = as.data.frame(enr_GO_tissue1)
#Create the table of most significant functions
if (dim(enr_GO_tissue)[1] <10) {npath = dim(enr_GO_tissue)[1]} else {npath = 10}
tissue_table = enr_GO_tissue[1:npath,c(2,6)]
colnames(tissue_table) = c("Function", "Adj_p_value")
#assign(str_glue(tissue,"_table"), gt(tissue_table)|>  tab_header(title= tissue) |> fmt_markdown(columns = c(Function)) |> gt_theme_538() |> fmt_scientific(columns = c(Adj_p_value), decimals = 0) |>  tab_style(style = list(cell_text(weight = "bold")), locations = cells_body(columns = Function, rows = selected)))
#gtsave(get(str_glue(tissue, "_table")),str_glue(tissue, "_table.png"))
download.file(file_tissue, str_glue(tissue, "_table.png"), mode = "wb")
assign(str_glue(tissue, "_t"), (ggdraw() + draw_image(str_glue(tissue, "_table.png"))))
#Plot the chosen functions
data_plot = data.frame(Week = c(sapply(c(4:9), function(x) {rep(x, length(selected))})), Function =
                             as.character(c(rep(enr_GO_tissue$Description[selected], 6)), Expression=NA))
for (i in selected) {
  xx = strsplit(enr_GO_tissue$geneID[i], "/")
  bool_contained1 = sapply(diff_genes_ann_0.01$HGNC.symbol, function(x) {grepl(x, toString(xx))})
  bool_contained1[is.na(bool_contained1)] = FALSE
  data_temp = diff_genes_ann_0.01[bool_contained1,]
  data_abs = data.frame(Week4 = apply(data_temp[1:3], 1, median), Week5 = apply(data_temp[4:6], 1, median), Week6 =  
                              apply(data_temp[7:9], 1, median), Week7=apply(data_temp[10:12], 1, median), Week8 = 
                              apply(data_temp[13:15], 1, median), Week9 =  apply(data_temp[16:18], 1, median))
  expr_path = colMeans(data_abs[1:6,])
  data_plot[data_plot$Function == enr_GO_tissue$Description[i],"Expression"] = expr_path
}
assign(str_glue(tissue, "_plot"), ggplot(data_plot, aes(x = Week, y= Expression, group = Function))  + geom_line(aes(colour =Function), position = position_dodge(0)) + geom_point(aes(colour = Function, shape = Function), position = position_dodge(0)) + labs(title = tissue) +  theme_classic() + scale_color_npg()  + theme(legend.title = element_text(size = 9), legend.text = element_text(size = 7), aspect.ratio = 1, legend.position = "bottom", legend.spacing.y = unit(0.1, "cm"),legend.key.size = unit(0.4, "cm"), plot.margin = unit(c(0.5,0.8,0.2,0),"cm")) + guides(colour = guide_legend(nrow = length(selected), title.position = "top",byrow = TRUE), shape = guide_legend(nrow = length(selected), title.position = "top",byrow = TRUE)))


# Muscle - Skeletal
tissue = "Muscle - Skeletal"
selected = c(1,2,4,5,7)
file_tissue = "https://raw.githubusercontent.com/datascience-mobi-2022/2022-topic-04-team-01/main/Report/Muscle%20-%20Skeletal_table.png"
#Overrepresentation Analysis
bool_contained = sapply(embryo_df_tissues$tissues, function(x) {grepl(tissue, toString(x))})
embryo_df_tissue = embryo_df_tissues[bool_contained,]
deGenes = as.character(embryo_df_tissue$ensembl.entrez[!duplicated(embryo_df_tissue$Gene.stable.ID)])
geneUniverse = as.character(embryo_df_tissues$ensembl.entrez[!duplicated(embryo_df_tissues$Gene.stable.ID)])
enr_GO_tissue1 = enrichGO(gene = deGenes, ont = "BP",  OrgDb ="org.Hs.eg.db", universe = geneUniverse, readable=TRUE, pvalueCutoff = 0.05)
enr_GO_tissue = as.data.frame(enr_GO_tissue1)
#Create the table of most significant functions
if (dim(enr_GO_tissue)[1] <10) {npath = dim(enr_GO_tissue)[1]} else {npath = 10}
tissue_table = enr_GO_tissue[1:npath,c(2,6)]
colnames(tissue_table) = c("Function", "Adj_p_value")
#assign(str_glue(tissue,"_table"), gt(tissue_table)|>  tab_header(title= tissue) |> fmt_markdown(columns = c(Function)) |> gt_theme_538() |> fmt_scientific(columns = c(Adj_p_value), decimals = 0) |>  tab_style(style = list(cell_text(weight = "bold")), locations = cells_body(columns = Function, rows = selected)))
#gtsave(get(str_glue(tissue, "_table")),str_glue(tissue, "_table.png"))
download.file(file_tissue, str_glue(tissue, "_table.png"), mode = "wb")
assign(str_glue(tissue, "_t"), (ggdraw() + draw_image(str_glue(tissue, "_table.png"))))
#Plot the chosen functions
data_plot = data.frame(Week = c(sapply(c(4:9), function(x) {rep(x, length(selected))})), Function =
                             as.character(c(rep(enr_GO_tissue$Description[selected], 6)), Expression=NA))
for (i in selected) {
  xx = strsplit(enr_GO_tissue$geneID[i], "/")
  bool_contained1 = sapply(diff_genes_ann_0.01$HGNC.symbol, function(x) {grepl(x, toString(xx))})
  bool_contained1[is.na(bool_contained1)] = FALSE
  data_temp = diff_genes_ann_0.01[bool_contained1,]
  data_abs = data.frame(Week4 = apply(data_temp[1:3], 1, median), Week5 = apply(data_temp[4:6], 1, median), Week6 =  
                              apply(data_temp[7:9], 1, median), Week7=apply(data_temp[10:12], 1, median), Week8 = 
                              apply(data_temp[13:15], 1, median), Week9 =  apply(data_temp[16:18], 1, median))
  expr_path = colMeans(data_abs[1:6,])
  data_plot[data_plot$Function == enr_GO_tissue$Description[i],"Expression"] = expr_path
}
assign(str_glue(tissue, "_plot"), ggplot(data_plot, aes(x = Week, y= Expression, group = Function))  + geom_line(aes(colour =Function), position = position_dodge(0)) + geom_point(aes(colour = Function, shape = Function), position = position_dodge(0)) + labs(title = tissue) +  theme_classic() + scale_color_npg()  + theme(legend.title = element_text(size = 9), legend.text = element_text(size = 7), aspect.ratio = 1, legend.position = "bottom", legend.spacing.y = unit(0.1, "cm"),legend.key.size = unit(0.4, "cm"), plot.margin = unit(c(0.5,0.8,0.2,0),"cm")) + guides(colour = guide_legend(nrow = length(selected), title.position = "top",byrow = TRUE), shape = guide_legend(nrow = length(selected), title.position = "top",byrow = TRUE)))


# Testis
tissue = "Testis"
selected = c(1,4,7)
file_tissue = "https://raw.githubusercontent.com/datascience-mobi-2022/2022-topic-04-team-01/main/Report/Testis_table.png"
#Overrepresentation Analysis
bool_contained = sapply(embryo_df_tissues$tissues, function(x) {grepl(tissue, toString(x))})
embryo_df_tissue = embryo_df_tissues[bool_contained,]
deGenes = as.character(embryo_df_tissue$ensembl.entrez[!duplicated(embryo_df_tissue$Gene.stable.ID)])
geneUniverse = as.character(embryo_df_tissues$ensembl.entrez[!duplicated(embryo_df_tissues$Gene.stable.ID)])
enr_GO_tissue1 = enrichGO(gene = deGenes, ont = "BP",  OrgDb ="org.Hs.eg.db", universe = geneUniverse, readable=TRUE, pvalueCutoff = 0.05)
enr_GO_tissue = as.data.frame(enr_GO_tissue1)
#Create the table of most significant functions
if (dim(enr_GO_tissue)[1] <10) {npath = dim(enr_GO_tissue)[1]} else {npath = 10}
tissue_table = enr_GO_tissue[1:npath,c(2,6)]
colnames(tissue_table) = c("Function", "Adj_p_value")
#assign(str_glue(tissue,"_table"), gt(tissue_table)|>  tab_header(title= tissue) |> fmt_markdown(columns = c(Function)) |> gt_theme_538() |> fmt_scientific(columns = c(Adj_p_value), decimals = 0) |>  tab_style(style = list(cell_text(weight = "bold")), locations = cells_body(columns = Function, rows = selected)))
#gtsave(get(str_glue(tissue, "_table")),str_glue(tissue, "_table.png"))
download.file(file_tissue, str_glue(tissue, "_table.png"), mode = "wb")
assign(str_glue(tissue, "_t"), (ggdraw() + draw_image(str_glue(tissue, "_table.png"))))
#Plot the chosen functions
data_plot = data.frame(Week = c(sapply(c(4:9), function(x) {rep(x, length(selected))})), Function =
                             as.character(c(rep(enr_GO_tissue$Description[selected], 6)), Expression=NA))
for (i in selected) {
  xx = strsplit(enr_GO_tissue$geneID[i], "/")
  bool_contained1 = sapply(diff_genes_ann_0.01$HGNC.symbol, function(x) {grepl(x, toString(xx))})
  bool_contained1[is.na(bool_contained1)] = FALSE
  data_temp = diff_genes_ann_0.01[bool_contained1,]
  data_abs = data.frame(Week4 = apply(data_temp[1:3], 1, median), Week5 = apply(data_temp[4:6], 1, median), Week6 =  
                              apply(data_temp[7:9], 1, median), Week7=apply(data_temp[10:12], 1, median), Week8 = 
                              apply(data_temp[13:15], 1, median), Week9 =  apply(data_temp[16:18], 1, median))
  expr_path = colMeans(data_abs[1:6,])
  data_plot[data_plot$Function == enr_GO_tissue$Description[i],"Expression"] = expr_path
}
assign(str_glue(tissue, "_plot"), ggplot(data_plot, aes(x = Week, y= Expression, group = Function))  + geom_line(aes(colour =Function), position = position_dodge(0)) + geom_point(aes(colour = Function, shape = Function), position = position_dodge(0)) + labs(title = tissue) +  theme_classic() + scale_color_npg()  + theme(legend.title = element_text(size = 9), legend.text = element_text(size = 7), aspect.ratio = 1, legend.position = "bottom", legend.spacing.y = unit(0.1, "cm"),legend.key.size = unit(0.4, "cm"), plot.margin = unit(c(0.5,0.8,0.2,0),"cm")) + guides(colour = guide_legend(nrow = length(selected), title.position = "top",byrow = TRUE), shape = guide_legend(nrow = length(selected), title.position = "top",byrow = TRUE)))

#Stomach
tissue = "Stomach"
selected = c(1,2,4)
file_tissue = "https://raw.githubusercontent.com/datascience-mobi-2022/2022-topic-04-team-01/main/Report/Stomach_table.png"
#Overrepresentation Analysis
bool_contained = sapply(embryo_df_tissues$tissues, function(x) {grepl(tissue, toString(x))})
embryo_df_tissue = embryo_df_tissues[bool_contained,]
deGenes = as.character(embryo_df_tissue$ensembl.entrez[!duplicated(embryo_df_tissue$Gene.stable.ID)])
geneUniverse = as.character(embryo_df_tissues$ensembl.entrez[!duplicated(embryo_df_tissues$Gene.stable.ID)])
enr_GO_tissue1 = enrichGO(gene = deGenes, ont = "BP",  OrgDb ="org.Hs.eg.db", universe = geneUniverse, readable=TRUE, pvalueCutoff = 0.05)
enr_GO_tissue = as.data.frame(enr_GO_tissue1)
#Create the table of most significant functions
if (dim(enr_GO_tissue)[1] <10) {npath = dim(enr_GO_tissue)[1]} else {npath = 10}
tissue_table = enr_GO_tissue[1:npath,c(2,6)]
colnames(tissue_table) = c("Function", "Adj_p_value")
#assign(str_glue(tissue,"_table"), gt(tissue_table)|>  tab_header(title= tissue) |> fmt_markdown(columns = c(Function)) |> gt_theme_538() |> fmt_scientific(columns = c(Adj_p_value), decimals = 0) |>  tab_style(style = list(cell_text(weight = "bold")), locations = cells_body(columns = Function, rows = selected)))
#gtsave(get(str_glue(tissue, "_table")),str_glue(tissue, "_table.png"))
download.file(file_tissue, str_glue(tissue, "_table.png"), mode = "wb")
assign(str_glue(tissue, "_t"), (ggdraw() + draw_image(str_glue(tissue, "_table.png"))))
#Plot the chosen functions
data_plot = data.frame(Week = c(sapply(c(4:9), function(x) {rep(x, length(selected))})), Function =
                             as.character(c(rep(enr_GO_tissue$Description[selected], 6)), Expression=NA))
for (i in selected) {
  xx = strsplit(enr_GO_tissue$geneID[i], "/")
  bool_contained1 = sapply(diff_genes_ann_0.01$HGNC.symbol, function(x) {grepl(x, toString(xx))})
  bool_contained1[is.na(bool_contained1)] = FALSE
  data_temp = diff_genes_ann_0.01[bool_contained1,]
  data_abs = data.frame(Week4 = apply(data_temp[1:3], 1, median), Week5 = apply(data_temp[4:6], 1, median), Week6 =  
                              apply(data_temp[7:9], 1, median), Week7=apply(data_temp[10:12], 1, median), Week8 = 
                              apply(data_temp[13:15], 1, median), Week9 =  apply(data_temp[16:18], 1, median))
  expr_path = colMeans(data_abs[1:6,])
  data_plot[data_plot$Function == enr_GO_tissue$Description[i],"Expression"] = expr_path
}
assign(str_glue(tissue, "_plot"), ggplot(data_plot, aes(x = Week, y= Expression, group = Function))  + geom_line(aes(colour =Function), position = position_dodge(0)) + geom_point(aes(colour = Function, shape = Function), position = position_dodge(0)) + labs(title = tissue) +  theme_classic() + scale_color_npg()  + theme(legend.title = element_text(size = 9), legend.text = element_text(size = 7), aspect.ratio = 1, legend.position = "bottom", legend.spacing.y = unit(0.1, "cm"),legend.key.size = unit(0.4, "cm"), plot.margin = unit(c(0.5,0.8,0.8,0),"cm")) + guides(colour = guide_legend(nrow = length(selected), title.position = "top",byrow = TRUE), shape = guide_legend(nrow = length(selected), title.position = "top",byrow = TRUE)))


#Skin
tissue = "Skin"
selected = c(1,2,3,4)
file_tissue = "https://raw.githubusercontent.com/datascience-mobi-2022/2022-topic-04-team-01/main/Report/Skin_table.png"
#Overrepresentation Analysis
bool_contained = sapply(embryo_df_tissues$tissues, function(x) {grepl(tissue, toString(x))})
embryo_df_tissue = embryo_df_tissues[bool_contained,]
deGenes = as.character(embryo_df_tissue$ensembl.entrez[!duplicated(embryo_df_tissue$Gene.stable.ID)])
geneUniverse = as.character(embryo_df_tissues$ensembl.entrez[!duplicated(embryo_df_tissues$Gene.stable.ID)])
enr_GO_tissue1 = enrichGO(gene = deGenes, ont = "BP",  OrgDb ="org.Hs.eg.db", universe = geneUniverse, readable=TRUE, pvalueCutoff = 0.05)
enr_GO_tissue = as.data.frame(enr_GO_tissue1)
#Create the table of most significant functions
if (dim(enr_GO_tissue)[1] <10) {npath = dim(enr_GO_tissue)[1]} else {npath = 10}
tissue_table = enr_GO_tissue[1:npath,c(2,6)]
colnames(tissue_table) = c("Function", "Adj_p_value")
#assign(str_glue(tissue,"_table"), gt(tissue_table)|>  tab_header(title= tissue) |> fmt_markdown(columns = c(Function)) |> gt_theme_538() |> fmt_scientific(columns = c(Adj_p_value), decimals = 0) |>  tab_style(style = list(cell_text(weight = "bold")), locations = cells_body(columns = Function, rows = selected)))
#gtsave(get(str_glue(tissue, "_table")),str_glue(tissue, "_table.png"))
download.file(file_tissue, str_glue(tissue, "_table.png"), mode = "wb")
assign(str_glue(tissue, "_t"), (ggdraw() + draw_image(str_glue(tissue, "_table.png"))))
#Plot the chosen functions
data_plot = data.frame(Week = c(sapply(c(4:9), function(x) {rep(x, length(selected))})), Function =
                             as.character(c(rep(enr_GO_tissue$Description[selected], 6)), Expression=NA))
for (i in selected) {
  xx = strsplit(enr_GO_tissue$geneID[i], "/")
  bool_contained1 = sapply(diff_genes_ann_0.01$HGNC.symbol, function(x) {grepl(x, toString(xx))})
  bool_contained1[is.na(bool_contained1)] = FALSE
  data_temp = diff_genes_ann_0.01[bool_contained1,]
  data_abs = data.frame(Week4 = apply(data_temp[1:3], 1, median), Week5 = apply(data_temp[4:6], 1, median), Week6 =  
                              apply(data_temp[7:9], 1, median), Week7=apply(data_temp[10:12], 1, median), Week8 = 
                              apply(data_temp[13:15], 1, median), Week9 =  apply(data_temp[16:18], 1, median))
  expr_path = colMeans(data_abs[1:6,])
  data_plot[data_plot$Function == enr_GO_tissue$Description[i],"Expression"] = expr_path
}
assign(str_glue(tissue, "_plot"), ggplot(data_plot, aes(x = Week, y= Expression, group = Function))  + geom_line(aes(colour =Function), position = position_dodge(0)) + geom_point(aes(colour = Function, shape = Function), position = position_dodge(0)) + labs(title = tissue) +  theme_classic() + scale_color_npg()  + theme(legend.title = element_text(size = 9), legend.text = element_text(size = 7), aspect.ratio = 1, legend.position = "bottom", legend.spacing.y = unit(0.1, "cm"),legend.key.size = unit(0.4, "cm"), plot.margin = unit(c(0.5,0.8,0.8,0),"cm")) + guides(colour = guide_legend(nrow = length(selected), title.position = "top",byrow = TRUE), shape = guide_legend(nrow = length(selected), title.position = "top",byrow = TRUE)))
```


```{r ORA-plot, fig.width = 12, fig.height=15, fig.cap = "For eight tissues A to H, the tables show the 10 most significant results from overrepresentation analysis. There, the functional GO annotation of all transcripts in our dataset (not only the differentially expressed ones) was compared to the functions associated with all TRAs of one tissue. The significance was determined by an adjusted p-value. Of those 10 gene sets, up to 5 were chosen to represent different groups of functions and avoid the overlap of highly related processes. For each of those, the median expression for each week was calculated and plotted, as seen in the expression graphs."}
#Arrange the plots for all eight tissues
lay = rbind(c(1,2,2,3,4,4),c(5,6,6,7,8,8), c(9,10,10,11,12,12),c(13,14,14,15,16,16))
plot = arrangeGrob(grobs = list(Spleen_t,Spleen_plot, Brain_t, Brain_plot, Heart_t, Heart_plot, Liver_t, Liver_plot,get("Muscle - Skeletal_t"), get("Muscle - Skeletal_plot"), Testis_t, Testis_plot, Stomach_t, Stomach_plot, Skin_t, Skin_plot), norw =4)
plot_output = as_ggplot(plot) + draw_plot_label(label = c("A", "B", "C", "D", "E", "F", "G", "H"), size = 15, x = c(0.01, 0.51,0.01, 0.51,0.01, 0.51,0.01, 0.51), y = c(0.79,0.79,0.52,0.52,0.28,0.28,0.02,0.02)) 
annotate_figure(plot_output, top = text_grob("Main functions from overrepresentation analysis plotted by tissue", color = "black", face = "bold", size = 16))
```

For the spleen (Fig. \@ref(fig:ORA-plot)A), we determined a largely constant expression of immune-related genes throughout the time frame, with a slight increase in some functions from week 7-9.
The brain (Fig. \@ref(fig:ORA-plot)B) showed an increase in neuron projection morphogenesis from a previously inactive state (expression \< 6.8) in week 4 to a significant expression (\>7.5) by week 8. Synaptic signaling stayed at relatively constant expression levels.
Heart-associated functions (Fig. \@ref(fig:ORA-plot)C) can be grouped into two categories. Cardiac functions (cardiac muscle tissue development, heart contraction) are highly expressed in week 4 and fall continously until week 8. In contrast, general muscle gene sets are rising from originally lower expression levels during the observed time.
The liver (Fig. \@ref(fig:ORA-plot)D) shows no clear expression patterns, with some metabolic functions increasing through time (organic hydroxy compound metabolic process), while others stay mostly constant (cellular amino acid metabolic process) or fall (organic acid catabolic process). In contrast, skeletal muscle gene sets (Fig. \@ref(fig:ORA-plot)E) show a very clear trend. After a mostly slight increase between week 4 and 8, a sharp rise in expression levels is visible from week 8 to 9.
The testis-associated sets (Fig. \@ref(fig:ORA-plot)F) continuously decrease in expression from week 5 onward.
For the stomach (Fig. \@ref(fig:ORA-plot)G), we found a initially high expression in week 4 that falls until week 6 and then increases again towards week 9.
Finally, the skin-associated functions (Fig. \@ref(fig:ORA-plot)H) all displayed a constant rise in expression levels from week 5 to 9.

## Brain subtissue-specific gene expression reflects neuronal development processes {#brain}

Our dataset covers 13 different brain subtissues, whose TRAs show a signficant overlap. Hence, we filtered the differentially expressed transcripts for those that are only related to one of all 13 subtissues. The discovered genes were examined using the NCBI database. The resulting genes of interest can be categorized as follows: \
1. Genes of Ion channels  2. Genes for neuronal development  3. Genes of cytokines

```{r gene-expression-brain,fig.height=4,fig.width=12,fig.cap=" The gene expression of different transcripts coding for an ion channel (left), neuronal development and function proteins (middle) and chemokine-related genes (right) was plotted for week 4 to 9."}
# A vektor of ENST numbers and weeks was created
ENST = embryo_df_tissues$ensembl.transcript
week = c(4,5,6,7,8,9)

# Data for Ion channel geneexpression was generated
# ENST00000531293
data.ENST00000531293 = as.data.frame(embryo_df_tissues %>%
                             filter(ensembl.transcript == str_extract(ENST, "ENST00000531293")))
data.ENST00000531293 = data.ENST00000531293[,1:18]
week_4 =data.ENST00000531293[1:3]
week_4 = apply(week_4,1,median)
week_5 =data.ENST00000531293[4:6]
week_5 = apply(week_5,1,median)
week_6 =data.ENST00000531293[7:9]
week_6 = apply(week_6,1,median)
week_7 =data.ENST00000531293[10:12]
week_7 = apply(week_7,1,median)
week_8 =data.ENST00000531293[13:15]
week_8 = apply(week_8,1,median)
week_9 =data.ENST00000531293[16:18]
week_9 = apply(week_9,1,median)
data.ENST00000531293 = c(week_4,week_5,week_6,week_7,week_8,week_9)
data.ENST00000531293 =cbind(data.ENST00000531293,week)

# Data for neuronal development and function geneexpression was generated
# ENST00000602349
data.ENST00000602349 = as.data.frame(embryo_df_tissues %>%
                             filter(ensembl.transcript == str_extract(ENST, "ENST00000602349")))
data.ENST00000602349 = data.ENST00000602349[,1:18]
week_4 =data.ENST00000602349[1:3]
week_4 = apply(week_4,1,median)
week_5 =data.ENST00000602349[4:6]
week_5 = apply(week_5,1,median)
week_6 =data.ENST00000602349[7:9]
week_6 = apply(week_6,1,median)
week_7 =data.ENST00000602349[10:12]
week_7 = apply(week_7,1,median)
week_8 =data.ENST00000602349[13:15]
week_8 = apply(week_8,1,median)
week_9 =data.ENST00000602349[16:18]
week_9 = apply(week_9,1,median)
data.ENST00000602349 = c(week_4,week_5,week_6,week_7,week_8,week_9) 
data.ENST00000602349 =cbind(data.ENST00000602349,week)

# ENST00000276646 & ENST00000529690
data.ENST00000276646 = as.data.frame(embryo_df_tissues %>%
                             filter(ensembl.transcript == str_extract(ENST, "ENST00000276646")))
data.ENST00000276646 = data.ENST00000276646[,1:18]
week_4 =data.ENST00000276646[1:3]
week_4 = apply(week_4,1,median)
week_5 =data.ENST00000276646[4:6]
week_5 = apply(week_5,1,median)
week_6 =data.ENST00000276646[7:9]
week_6 = apply(week_6,1,median)
week_7 =data.ENST00000276646[10:12]
week_7 = apply(week_7,1,median)
week_8 =data.ENST00000276646[13:15]
week_8 = apply(week_8,1,median)
week_9 =data.ENST00000276646[16:18]
week_9 = apply(week_9,1,median)
data.ENST00000276646 = c(week_4,week_5,week_6,week_7,week_8,week_9) 
data.ENST00000276646 =cbind(data.ENST00000276646,week)

data.ENST00000529690 = as.data.frame(embryo_df_tissues %>%
                             filter(ensembl.transcript == str_extract(ENST, "ENST00000529690")))
data.ENST00000529690 = data.ENST00000529690[,1:18]
week_4 =data.ENST00000529690[1:3]
week_4 = apply(week_4,1,median)
week_5 =data.ENST00000529690[4:6]
week_5 = apply(week_5,1,median)
week_6 =data.ENST00000529690[7:9]
week_6 = apply(week_6,1,median)
week_7 =data.ENST00000529690[10:12]
week_7 = apply(week_7,1,median)
week_8 =data.ENST00000529690[13:15]
week_8 = apply(week_8,1,median)
week_9 =data.ENST00000529690[16:18]
week_9 = apply(week_9,1,median)
data.ENST00000529690 = c(week_4,week_5,week_6,week_7,week_8,week_9) 
data.ENST00000529690 =cbind(data.ENST00000529690,week)

# ENST00000356660
data.ENST00000356660 = as.data.frame(embryo_df_tissues %>%
                             filter(ensembl.transcript == str_extract(ENST, "ENST00000356660")))
data.ENST00000356660 = data.ENST00000356660[,1:18]
week_4 =data.ENST00000356660[1:3]
week_4 = apply(week_4,1,median)
week_5 =data.ENST00000356660[4:6]
week_5 = apply(week_5,1,median)
week_6 =data.ENST00000356660[7:9]
week_6 = apply(week_6,1,median)
week_7 =data.ENST00000356660[10:12]
week_7 = apply(week_7,1,median)
week_8 =data.ENST00000356660[13:15]
week_8 = apply(week_8,1,median)
week_9 =data.ENST00000356660[16:18]
week_9 = apply(week_9,1,median)
data.ENST00000356660 = c(week_4,week_5,week_6,week_7,week_8,week_9) 
data.ENST00000356660 =cbind(data.ENST00000356660,week)

# ENST00000518312 & ENST00000521485
data.ENST00000518312 = as.data.frame(embryo_df_tissues %>%
                             filter(ensembl.transcript == str_extract(ENST, "ENST00000518312")))
data.ENST00000518312 = data.ENST00000518312[,1:18]
week_4 =data.ENST00000518312[1:3]
week_4 = apply(week_4,1,median)
week_5 =data.ENST00000518312[4:6]
week_5 = apply(week_5,1,median)
week_6 =data.ENST00000518312[7:9]
week_6 = apply(week_6,1,median)
week_7 =data.ENST00000518312[10:12]
week_7 = apply(week_7,1,median)
week_8 =data.ENST00000518312[13:15]
week_8 = apply(week_8,1,median)
week_9 =data.ENST00000518312[16:18]
week_9 = apply(week_9,1,median)
data.ENST00000518312 = c(week_4,week_5,week_6,week_7,week_8,week_9) 
data.ENST00000518312 =cbind(data.ENST00000518312,week)

data.ENST00000521485 = as.data.frame(embryo_df_tissues %>%
                             filter(ensembl.transcript == str_extract(ENST, "ENST00000521485")))
data.ENST00000521485 = data.ENST00000521485[,1:18]
week_4 =data.ENST00000521485[1:3]
week_4 = apply(week_4,1,median)
week_5 =data.ENST00000521485[4:6]
week_5 = apply(week_5,1,median)
week_6 =data.ENST00000521485[7:9]
week_6 = apply(week_6,1,median)
week_7 =data.ENST00000521485[10:12]
week_7 = apply(week_7,1,median)
week_8 =data.ENST00000521485[13:15]
week_8 = apply(week_8,1,median)
week_9 =data.ENST00000521485[16:18]
week_9 = apply(week_9,1,median)
data.ENST00000521485 = c(week_4,week_5,week_6,week_7,week_8,week_9) 
data.ENST00000521485 =cbind(data.ENST00000521485,week)

# ENST00000439476
data.ENST00000439476 = as.data.frame(embryo_df_tissues %>%
                             filter(ensembl.transcript == str_extract(ENST, "ENST00000439476")))
data.ENST00000439476 = data.ENST00000439476[,1:18]
week_4 =data.ENST00000439476[1:3]
week_4 = apply(week_4,1,median)
week_5 =data.ENST00000439476[4:6]
week_5 = apply(week_5,1,median)
week_6 =data.ENST00000439476[7:9]
week_6 = apply(week_6,1,median)
week_7 =data.ENST00000439476[10:12]
week_7 = apply(week_7,1,median)
week_8 =data.ENST00000439476[13:15]
week_8 = apply(week_8,1,median)
week_9 =data.ENST00000439476[16:18]
week_9 = apply(week_9,1,median)
data.ENST00000439476 = c(week_4,week_5,week_6,week_7,week_8,week_9) 
data.ENST00000439476 =cbind(data.ENST00000439476,week)


# ENST00000539563
data.ENST00000539563 = as.data.frame(embryo_df_tissues %>%
                             filter(ensembl.transcript == str_extract(ENST, "ENST00000539563")))
data.ENST00000539563 = data.ENST00000539563[,1:18]
week_4 =data.ENST00000539563[1:3]
week_4 = apply(week_4,1,median)
week_5 =data.ENST00000539563[4:6]
week_5 = apply(week_5,1,median)
week_6 =data.ENST00000539563[7:9]
week_6 = apply(week_6,1,median)
week_7 =data.ENST00000539563[10:12]
week_7 = apply(week_7,1,median)
week_8 =data.ENST00000539563[13:15]
week_8 = apply(week_8,1,median)
week_9 =data.ENST00000539563[16:18]
week_9 = apply(week_9,1,median)
data.ENST00000539563 = c(week_4,week_5,week_6,week_7,week_8,week_9) 
data.ENST00000539563 =cbind(data.ENST00000539563,week)

# Data for chemokine geneexpression was generated 
# ENST00000337225
data.ENST00000337225 = as.data.frame(embryo_df_tissues %>%
                             filter(ensembl.transcript == str_extract(ENST, "ENST00000337225")))
data.ENST00000337225 = data.ENST00000337225[,1:18]
week_4 =data.ENST00000337225[1:3]
week_4 = apply(week_4,1,median)
week_5 =data.ENST00000337225[4:6]
week_5 = apply(week_5,1,median)
week_6 =data.ENST00000337225[7:9]
week_6 = apply(week_6,1,median)
week_7 =data.ENST00000337225[10:12]
week_7 = apply(week_7,1,median)
week_8 =data.ENST00000337225[13:15]
week_8 = apply(week_8,1,median)
week_9 =data.ENST00000337225[16:18]
week_9 = apply(week_9,1,median)
data.ENST00000337225 = c(week_4,week_5,week_6,week_7,week_8,week_9) 
data.ENST00000337225 =cbind(data.ENST00000337225,week)

# ENST00000579298
data.ENST00000579298 = as.data.frame(embryo_df_tissues %>%
                             filter(ensembl.transcript == str_extract(ENST, "ENST00000579298")))
data.ENST00000579298 = data.ENST00000579298[,1:18]
week_4 =data.ENST00000579298[1:3]
week_4 = apply(week_4,1,median)
week_5 =data.ENST00000579298[4:6]
week_5 = apply(week_5,1,median)
week_6 =data.ENST00000579298[7:9]
week_6 = apply(week_6,1,median)
week_7 =data.ENST00000579298[10:12]
week_7 = apply(week_7,1,median)
week_8 =data.ENST00000579298[13:15]
week_8 = apply(week_8,1,median)
week_9 =data.ENST00000579298[16:18]
week_9 = apply(week_9,1,median)
data.ENST00000579298 = c(week_4,week_5,week_6,week_7,week_8,week_9) 
data.ENST00000579298 =cbind(data.ENST00000579298,week)

# Plotting of the gained geneexpression data
par(mfrow = c(1, 3))

#Plot for ion channel gene
plot(data.ENST00000531293[,2],data.ENST00000531293[,1],type = "b",lwd=1,pch=19, main = "Ion channel gene", xlab = "weeks",ylab = "Expression level",col = "red")
legend(3.8, 9.25, legend=c("ENST00000531293"),col=c("red"), lty=1, cex=0.8,title="expressed transcripts")


#Plot for neuronal development and function genes 
plot(data.ENST00000602349[,2],data.ENST00000602349[,1],type = "b",lwd=1,pch=19, main = "Neuronal development and function genes", xlab = "weeks",ylab = "Expression level",col = "red", ylim = c(6.3,8.4))
lines(data.ENST00000276646[,2],data.ENST00000276646[,1],type = "b",lwd=1,pch=19, col = "blue")
lines(data.ENST00000529690[,2],data.ENST00000529690[,1],type = "b",lwd=1,pch=19, col = "orange")
lines(data.ENST00000356660[,2],data.ENST00000356660[,1],type = "b",lwd=1,pch=19,col = "green")
lines(data.ENST00000518312[,2],data.ENST00000518312[,1],type = "b",lwd=1,pch=19,col = "black")
lines(data.ENST00000521485[,2],data.ENST00000521485[,1],type = "b",lwd=1,pch=19,col = "pink")
lines(data.ENST00000439476[,2],data.ENST00000439476[,1],type = "b",lwd=1,pch=19,col = "purple")
lines(data.ENST00000539563[,2],data.ENST00000539563[,1],type = "b",lwd=1,pch=19,col = "yellowgreen")
legend(3.8, 8.5, legend=c("ENST00000602349", "ENST00000276646", "ENST00000529690", "ENST00000356660", "ENST00000518312", "ENST00000521485",  "ENST00000439476", "ENST00000539563"),col=c("red", "blue", "orange", "green","black", "pink", "purple", "yellowgreen"), lty=1, cex=0.8,title="expressed transcripts")


# Plot for chemokine related genes
plot(data.ENST00000337225[,2],data.ENST00000337225[,1],type = "b",lwd=1,pch=19, main = "Chemokine-related genes", xlab = "weeks",ylab = "Expression level",col = "red",ylim = c(7.6,9.7))
lines(data.ENST00000579298[,2],data.ENST00000579298[,1],type = "b",lwd=1,pch=19,col = "blue")
legend(3.8, 9.8, legend=c("ENST00000337225", "ENST00000579298"),col=c("red", "blue"), lty=1, cex=0.8,title="expressed transcripts")

```

### Ion channel

Ion channels play an important role in the neuronal function. We discovered that ENST00000531293 is highly expressed in the nucleus accumbens. It shows a significant increase between weeks 7 to 9 and codes for SLN sarcolipin, which is a sarcoplasmic reticulum Ca-ATPase (Fig. \@ref(fig:gene-expression-brain) left).

### Genes for neuronal development and function

The second group are genes with a specific role in neuronal development and function. There, we discovered that ENST00000276646 and ENST00000529690 expression increases significantly over time. Both transcripts are associated with the cerebellar hemisphere and code for SYBU (syntabulin), a protein that contributes to activity-dependent presynaptic assembly in neuronal development.\
Furthermore, we found four transcripts for axon goudance: ENST00000602349 codes for NXPH1 (neurexophilin 1) which forms a tight complex with neurexins. These proteins promote adhesion between axons and dendrites.  The transcript shows a strong rise in expression, especially from weeks 7 to 8, and is connected to  the anterior cingulate cortex (Fig. \@ref(fig:gene-expression-brain) middle).\
ENST00000518312 and ENST00000521485 encode for SNAP91 (synaptosome associated protein 91), which plays a role in the regulation of clathrin-dependent endocytosis. Therefore, SNAP91 is important for essential axonal functions of neurons like postsynaptic density (Overhoff et al. 2020). The gene is associated with the cerebellar hemisphere and also shows a significant increase between weeks 7 to 8 (Fig. \@ref(fig:gene-expression-brain) middle). The two transcripts have a highly similar distribution of expression values (suppl. Fig. \@ref(fig:Cerebellar-Hemisphere-specific) middle and right).\
In addition, ENST00000539563  encodes for LSAMP (limbic system associated membrane protein), which plays a role in axon guidance. The encoded preprotein is processed into a neuronal surface glycoprotein which functions as an adhesion molecule during axon guidance and neuronal growth in the developing limbic system. ENST00000539563 is associated with the  Putamen, a part of the basal ganglia that are associated with the limbic system. \
ENST00000356660 and ENST00000439476 code for BDNF (brain derived neurotrophic factor). A binding of BDNF to its receptor promotes neuronal survival. Both transcripts show an identical decline in expression over the weeks. Nonetheless, ENST00000356660 is connected to the cerebellar hemisphere while  ENST00000439476 to related to the hippocampus.

### Chemokine related genes

The last group are chemokines, proteins with an important role in the signaling process during neuronal development (Tiveron 2008). We discovered ENST00000337225, which encodes for CXCL14 (alpha class chemokin ligand). It shows a strong increase in expression between week 4 to 6 and is associated with the anterior cingulate cortex. ENST00000579298 encodes for NUP85 (nucleoporin 85), a protein component of the Nup107-160 subunit of the nuclear pore complex. NUP85 can bind to CCR2 (a receptor for beta class chemokines). ENST00000579298 is related to the frontal cortex and shows a decline from week 5 to 9 (Fig. \@ref(fig:gene-expression-brain) right).


## No TRAs could be identified as key biomarkers for developmental progression

Principal component analysis (PCA) was performed on a matrix containing all differentially expressed genes throughout the six weeks. This was performed to reduce dimensions while keeping most of the data's variance. To get a better grasp of how much variance is explained by each principal component (PC), the percentage variance of each PC together as well as the cumulative variance were plotted. This visualization helps to determine how many PCs are needed to explain a significant amount of the data's variation. The chosen PCs can then be used for further analysis. The first three PCs already explain over 80% of the data's variance, with the first PC alone accounting for over 60% (suppl. Fig. \@ref(fig:PCA-analysis) A).

```{r, include=FALSE}
#pca on rows (genes)
pca_genes <- prcomp(
  t(diff_genes_ann_0.01[,1:18]), # transposed matrix
  scale. = TRUE # data scaled to have unit variance
  )

#taking a first look
summary(pca_genes)
```


```{r pca-variance, fig.width = 5, fig.height=5}
#visualizing the variance of the PCAs
pc_eigenvalues <- pca_genes$sdev^2

pc_eigenvalues <- tibble(PC = factor(1:length(pc_eigenvalues)), 
                         variance = pc_eigenvalues) %>% 
  #column with percent variance
  mutate(pct = variance/sum(variance)*100) %>% 
  #column with cumulative variance explained
  mutate(pct_cum = cumsum(pct))


#actual plot
PC_contribution <- pc_eigenvalues %>% 
  ggplot(aes(x = PC)) +
  geom_col(aes(y = pct)) +
  geom_line(aes(y = pct_cum, group = 1)) + 
  geom_point(aes(y = pct_cum)) +
  ggtitle('PC variance') +
  theme(plot.title = element_text(hjust = 0.5,face = "bold")) +
  labs(x = "Principal component", y = "Fraction variance explained") +
  theme(aspect.ratio = 1)
```

```{r, include=FALSE}
 loadings = pca_genes$rotation[,"PC1"]
pca_df = data.frame(diff_genes_ann_0.01[,c("ensembl.transcript", "HGNC.symbol", "tissues")], Loadings = loadings)
pca_df[order(abs(pca_df$Loadings), decreasing = TRUE),]
```

```{r, include=FALSE}
#PCA for Heart transripts
#tissue = "Heart"
#bool_contained = sapply(diff_genes_ann_0.01$tissues, function(x) {grepl(tissue, toString(x))})
test_genes = diff_genes_ann_0.01  #[bool_contained,]

#pca on rows (genes)
pca_genes <- prcomp(
  t(test_genes[,1:18]), # transposed matrix
  scale. = TRUE # data scaled to have unit variance
  )

#taking a first look
summary(pca_genes)

loadings = pca_genes$rotation[,"PC1"]
pca_df = data.frame(test_genes[,c("ensembl.transcript", "HGNC.symbol", "tissues")], Loadings = loadings)
pca_df[order(abs(pca_df$Loadings), decreasing = TRUE),]
test_genes
ranks_pca = data.frame(test_genes[,c("ensembl.transcript", "HGNC.symbol")], PC1 = NA, PC2 = NA, PC3 = NA, PC4 = NA, PC5 = NA)
PC_vector = c("PC1", "PC2", "PC3", "PC4", "PC5")
for (i in PC_vector) {
  loadings = pca_genes$rotation[,i]
  pca_df = data.frame(test_genes$ensembl.transcript, Loadings = loadings, Order = c(1:length(test_genes$ensembl.transcript)))
  pca_df = pca_df[order(abs(pca_df$Loadings), decreasing = TRUE),]
  pca_df$rank = c(1:length(test_genes$ensembl.transcript))
  pca_df = pca_df[order(pca_df$Order),]
  ranks_pca[,i] =  pca_df$rank
}

ranks_pca$sum = ranks_pca$PC1*4 - ranks_pca$PC2 - ranks_pca$PC3 - ranks_pca$PC4 - ranks_pca$PC5
ranks_pca[order(ranks_pca$sum),]

```

For further analysis, each transcript was awarded a rank depending on how much it contributes to each PC. Then, a score was calculated for maximum contribution to PC1 and minimal loadings for all other relevant PCs (PC2-5). We took a closer look at the five best socring transcripts for PC1, as this PC alone already explains the brunt of the variance of our data. These transcripts encode for three different proteins: Low-Density Lipoprotein Receptor Class A Domain-Containing Protein 4 (LDLRAD4) encoded by ENST00000399848, Calcium-activated potassium channel subunit beta-2 (KCNMB2) encoded by ENST00000432997 and ENST00000452583 as well as the SLIT and NTRK-Like Protein 3 (SLITRK3) encoded by ENST00000241274 and ENST00000475390.

```{r, fig.width=5,fig.height=5, include = FALSE, eval = FALSE}

PC1_first_five <- diff_genes_ann_0.01[c('9703','11797','13044','616','15025'),0:19]


#median of the transcripts for each week
PC1_4 <- PC1_first_five[,c(1,2,3)]
week4 <- t(data.frame(apply(PC1_4,1,median)))

PC1_5 <- PC1_first_five[,c(4,5,6)]
week5 <- t(data.frame(apply(PC1_5,1,median)))

PC1_6 <- PC1_first_five[,c(7,8,9)]
week6 <- t(data.frame(apply(PC1_6,1,median)))

PC1_7 <- PC1_first_five[,c(10,11,12)]
week7 <- t(data.frame(apply(PC1_7,1,median)))

PC1_8 <- PC1_first_five[,c(13,14,15)]
week8 <- t(data.frame(apply(PC1_8,1,median)))

PC1_9 <- PC1_first_five[,c(16,17,18)]
week9 <- t(data.frame(apply(PC1_9,1,median)))


#combining in a dataframe
first_five <- rbind(week4,week5,week6,week7,week8,week9)

#plotting the expression over weeks

png(filename = "PC1_five.png", res = 100)
par(mar=c(4, 4, 4, 7), xpd=TRUE)
matplot(first_five, 
        type = 'b',
        pch=16,
        col = c('black','red','green','blue','orange'), 
        xlab = 'Weeks', 
        ylab = 'Expression Level', 
        main = 'Most contribution to PC1' , 
        xaxt='n'
)
axis(side = 1,
     at = 1:6, 
     labels = c(4,5,6,7,8,9)
)
legend('topright', 
       inset=c(-0.45,0), 
       legend = c('ENST00000399848 \n (LDLRAD4)\n ','ENST00000432997 \n (KCNMB2) \n','ENST00000452583 \n (KCNMB2) \n','ENST00000241274 \n (SLITRK3)\n','ENST00000475390 \n (SLITRK3)\n'),
       col=c('black','red','green','blue','orange'), 
       pch=16, 
       cex=0.6,
       title="expressed transcripts"
)

dev.off()

```


The transcripts of ENST00000432997 and ENST00000452583 encode for the same Protein (KCNMB2). Correspondingly, both consistently display the same expression level over all weeks. The same applies to the transcripts of ENST00000241274 and ENST00000475390, as they also encode for the same protein (SLITRK3) and therefore also display the same expression levels. The transcripts for LDLRAD4 show the highest overall expression. Their expression pattern is also highly dissimilar to those of KCNMB2 and SLITRK3. In contrast, KCNMB2 and SLITRK3 show a rather similar curve over all weeks (suppl. Fig. \@ref(fig:PCA-analysis) B).

# Discussion

## Comparison with original paper

In contrast to our *limma* analysis to generate data with biological relevance, the authors of the original paper (Yi H *et al.*, 2010) of our dataset used *one-way analysis of variance* to acquire differentially expressed transcripts. We compared both sets by k-means clustering followed by plotting the overlapping intensity values (Fig. \@ref(fig:paper-trends-cluster)). By looking at the trends of the clusters for *up* and *down* regulated gene expression, we can validate the method used by the authors. In contrast, the *arch* regulated genes-clusters do not present a clear result and show a weakness of the method used. It falsely detects some genes as differentially expressed, assuming our *vsn* normalization is correct, which is indicated by the quality control (section \@ref(QC-normalization)).
Furthermore, we showed in our Venn diagram (Fig. \@ref(fig:Venn-Diagram)), that our method detected differentially expressed transcripts not included in their data. This questions the significance of either the papers results or again our *vsn* normalization and subsequent *limma* analysis, which is rather unlikely given our quality control and plots.
The discrepancy could be explained by the publication date, which is year 2010. Now, there are more advanced algorithms that determine differentially expressed transcripts more precisely. The *vsn* and the *limma* package are up to date with frequent advances (Ritchie. et al. 2015).\
Overall, this ensures the quality of our data and differentially expressed transcripts while also pointing out some flaws of the dataset of the paper.

## TRAs can infer a basic timeline of organ development {#dis-organ}

In our analysis, we have shown that a number of TRAs are differentially expressed (section \@ref(organ-overview)) between week 4 and 9 of human embryonic development in each of the analyzed tissues. Nonetheless, the expression levels of TRAs associated with one tissue do not constitute a useful metric for the organ's development (section \@ref(organ-clustering)). This can be explained by the fact that within one tissue's TRAs, there are multiple groups of genes both distinct in expression patterns (clustering in section \@ref(organ-clustering)) and function (analysis of spleen gene functions in section \@ref(organ-tables)). Thus, we determined that the expression over time of functional gene sets linked to specific tissues through overrepresentation analysis is a more meaningful metric for organ development.

This approach was used in section \@ref(organ-ora) for eight different tissues. 
For the spleen, the results of our analysis (Fig. \@ref(fig:ORA-plot)A) largely do not reflect the embryonic development (section \@ref(intro-tissues)). While some of the immune-related gene sets are already expressed in week 4, the spleen only develops in week 6 and contains immune cells by week 12. This shows that while the spleen plays a role in the immune system and such gene sets are therefore rightly linked to the spleen, the expression of these transcripts alone does not necessarily relate to the development of the organ. It is still noteworthy that functions related to the adaptive immune system increase in expression from week 7 onward, which correlates with the beginning of T-cell development in the thymus. 
The observed timeframe is an important part of brain development (section \@ref(intro-tissues)). This is also visible in the expression data (Fig. \@ref(fig:ORA-plot)B), with an already high but still continously increasing expression of synaptic gene sets. Furthermore, as the brain starts to form, the expression of neuron projection morphogenesis transcripts increases continuously from week 5 to 8. 
At week 4, the clearly heart-associated gene sets (Fig. \@ref(fig:ORA-plot)C) are at their highest expression level and decrease until week 8. The cardiac muscle tissue development transcripts still remain highly expressed (>7.5). This corresponds to the early development of the heart as noted in the introduction (section \@ref(intro-tissues)). It is noteworthy that the heart contraction gene set rises in expression again from week 8 to 9, but here an explanation is not possible without further analyzing the individual genes. 
The liver-associated TRAs showed no clear expression pattern (Fig. \@ref(fig:ORA-plot)D). Thus, even though the liver forms mostly during the analyzed timeframe (section \@ref(intro-tissues)), we cannot link the gene expression to the organ's development. The detected functions are mostly metabolic pathways, whose activity could also be related to processes outside the liver. As a result, it is plausible that their expression is independent of liver development. 
The skeletal muscle functions are expressed only late within the observed time, as shown by the large increase in expression from  week 8 to 9 (Fig. \@ref(fig:ORA-plot)E). As muscle fibers begin to develop later than week 9 and the first related proteins appear from week 7 on (section \@ref(intro-tissues)), these expression data correspond well to the embryonic development. 
The testis gene sets decrease in expression from week 5 onward (Fig. \@ref(fig:ORA-plot)F). This is in contrast to the embryonic development, where the gonads start to form at around the same time (section \@ref(intro-tissues)). 
For the stomach, the expression pattern depicts a decrease until week 6 followed by rising expression levels until week 9 (Fig. \@ref(fig:ORA-plot)G). However, the literature indicates that these results are unrelated to the stomach development. Functions like digestion or peptide hormone secretion are impossible to occure at this time, since the specific cells needed for this only appear later in embryogenesis (section \@ref(intro-tissues)). Therfore, the cause of the changing expression would have to be determined through a more in-depth analysis of the involved genes. 
Finally, the skin shows an increased expression of related genes sets from week 5 to 9 (Fig. \@ref(fig:ORA-plot)H). This broadly reflects the embryonic development, with the epidermis starting to form in week 4 (section \@ref(intro-tissues)). We also found this expression pattern in the keratinization gene set that is suggested by literature as a good indicator for skin formation.  

## Brain subtissue-specific gene expression reflects neuronal development processes {#dis-brain}

First, we found a significant increase in the expression of SLN sarcolipin (a Ca-ATPase) (Fig. \@ref(fig:gene-expression-brain) left). Since Ca(2+) is an essential cofactor for actin dependent cell migration, this might be related to neuronal migration starting in week 9 (section \@ref(intro-tissues)). 
Additionally, the strong correlation between the expression of SNAP91 genes and one SYBU gene (ENST00000276646) (Fig. \@ref(fig:gene-expression-brain) middle) was noteworthy. Both proteins are associated with the cerebellar hemisphere and contribute to endocytosis (Overhoff et al. 2020). 
In addition, we identified a significant increase in expression of NXPH1 (Fig. \@ref(fig:gene-expression-brain) middle). An upregulation of this factor can prepare the process of synapse formation which starts at week 11 (section \@ref(intro-tissues)). 
We further identified a strong LSAMP expression associated with the putamen as well as  NXPH1 linked with the anterior cingulate cortex. (Fig. \@ref(fig:gene-expression-brain) middle). As adhesion molecules for axon guidance, their increases in expression (from week 6/7 on) could be in preparation for synapse formation at week 9 (section \@ref(intro-tissues)). 
The neuronal cell migration is strongly dependent on chemoattractors like chemokines (Tiveron 2008). We identified a significant increase in CXCL14 between weeks 5 to 6 that maintains a high expression level afterwards. Fig. \@ref(fig:gene-expression-brain) right). This could be an accumulation for neuronal migration, starting by week 9 (section \@ref(intro-tissues)). A decline in  NUP85 expression is also notable (Fig. \@ref(fig:gene-expression-brain) right). NUP85 can bind to CCR2, hence a decline reducse the chances for this binding. A consequence might be that more CCR2  receptors are free for beta-type chemokin mediated signals. 
Overall, there were two cases were two closely related transcripts (linked to the same gene) showed identical expression values. This phenomenon could be caused by failures in annotation, e.g. associating two transcripts with one probe. 

## No TRAs could be identified as key biomarkers for developmental progression {#dis-pca}

Principal component analysis helped to identify transcripts that have a high contribution to our data's variance. The high variance explained by PC1 alone made it possible to focus our further analysis only on this PC. Ranking the transcripts according to their contribution led us to identify the three proteins LDLRAD4, KCNMB2 and SLITRK3 as main contributors to only principal component 1. Nonetheless, the loadings for these genes are not significantly larger than those of other genes. Therefore, the information gained trough PCA was not sufficient to reliably classify these or any TRAs as biomarkers for embryonic development. More in-depth analysis is necessary to determine whether PCA is an insufficient method to identify these biomarkers, or if using TRAs as markers is an unsuccessful approach in general.

## Conclusion

The overarching goal of this research was to gain insight on human embryonic development between week 4 and 9 through the expression patterns of tissue-restricted antigens (TRAs). Under this main objective, we divided our research into distinct parts. First, we managed to find differentially expressed TRAs associated with all tissues (section \@ref(limma)) and determined that our method of analysis even has advantages compared to the original paper (section \@ref(paper-comp)). Then, we used these results to try to determine the developmental steps occurring during the observed time frame (section \@ref(organ)). While not all TRA-specific expression patterns could be linked to morphological events, we were able to show that the expression patterns for some organ-specific gene sets are highly correlated with milestones in development (section \@ref(dis-organ)). An in-depth analysis of brain subtissues proved diffcult due to the high degree of TRA overlap between them. Still, we found genes related to neuronal function that were TRAs for only one subtissue (section \@ref(brain)). Based on their expression, we could draw some conclusions on neuronal development (section \@ref(dis-brain)). Finally, we tried to define individual genes as biomarkers for embryonic development through PCA (section \@ref(dis-pca)). There, we could not define such markers effectively based on the selection of genes through principal component loadings. All in all, we were still able to show that the morphological events during embryogenesis are reflected in gene expression and prove that differential expression analysis of TRAs can be a valid method for embryonic research.

\newpage

# References

Benninghoff, A. (1993), Makroskopische Anatomie, Embryologie und Histologie des Menschen, 15. Auflage, MÃ¼nchen, Wien, Baltimore, Urban und Schwarzenberg.

Deutsch J. (2013), Embryologie und Physiologie der Leber, PÃ¤diatrische Gastroenterologie, Hepatologie und ErnÃ¤hrung, 375--87, Berlin Deutschland.

Dinkelacker M., (2019), Chromosomal clustering of tissue restricted antigens.

Esrefoglu M., Taslindere E. & Cetin A. (2017), Development of the Esophagus and Stomach, Bezmialem sci. *5*, 175-82.

Fasold M. & Binder H., (2013), AffyRNADegradation: control and correction of RNA quality effects in GeneChip expression data, Bioinformatics, *29*, 129-31. 

Hayward AR., (1983), The human fetus and newborn: development of the immune response, Birth Defects Orig. Artic. Ser. *19*, 289-94. 

Hikspoors J.P.J.M., Kruepunga N., Mommen G.M.C. et al., (2022), A pictorial account of the human embryonic heart between 3.5 and 8 weeks of development, Commun. Biol. *5*, 226.

Hu MS., Borrelli MR., Hong WX., Malhotra S., Cheung ATM., Ransom RC., Rennert RC., Morrison SD., Lorenz HP. & Longaker MT., (2018) Embryonic skin development and repair, Organogenesis *14*, 46-63.

Huber W., von Heydebreck A., SÃ¼ltmann H., Poustka A., Vingron M., (2002), Variance stabilization applied to microarray data calibration and to the quantification of differential expression, Bioinformatics. *18* ,Suppl 1, 96-104.

James F. & Jones M.D., (1983), Development of the Spleen, Lymphology *16*, 83-89, Georg Thieme Verlag Stuttgart, New York.

Kluth D., Jaeschke-Melli S. & Fiegel H., (2003),The embryology of gut rotation, Semin. Pediatr. Surg. *12*, 275-279.

MÃ¼ller W. A. & Hassel M., (2018), Entwicklungsbiologie und Reproduktionsbiologie des Menschen und bedeutender Modellorganismen, 6. Auflage, Springer-Verlag GmbH Deutschland, Berlin, Deutschland.

Murphy K. & Weaver C., (2018), Janeway Immunologie, 9. Auflage, Springer-Verlag GmbH Deutschland, Berlin, Deutschland.

National Research Council and Institute of Medicine, (2009), Preventing Mental, Emotional, and Behavioral Disorders Among Young People: Progress and Possibilities, Washington, DC: The National Academies Press.

Overhoff M., De Bruyckere E. & Kononenko N- L., (2020), Mechanisms of neuronal survival safeguarded by endocytosis and autophagy, J. Neurochem. *157*, 263-296.

Ritchie ME., Phipson B., Wu D., Hu Y., Law CW., Shi W. & Smyth GK., (2015), limma powers differential expression analyses for RNA-sequencing and microarray studies, Nucleic Acids Res. *43*, e47.

Romero N. B., Mezmezian M .& FidziaÅska A. (2013), Pediatric Neurology Part III, Chapter 137 - Main steps of skeletal muscle development in the human: Morphological analysis and ultrastructural characteristics of developing human muscle, Handb. Clin. Neurol. *113*, 1299-1310.

Tiveron M.C. & Cremer H., (2008), CXCL12/CXCR4 signalling in neuronal cell migration, Curr. Opin. Neurobiol. *18*, 237-244.

Ulfig N., (2009), Kurzlehrbuch Embryologie, Georg Thieme Verlag Stuttgart,New York.

Yi H., Xue L., Guo MX., Ma J., Zeng Y., Wang W., Cai JY., Hu HM., Shu HB., Shi YB. et al., (2010), Gene expression atlas for human embryogenesis, FASEB. J. *24*, 3341-50.. 

Yusuf F., Rehimi R., Dai F. & Brand-Saberi B., (2005), Expression of chemokine receptor CXCR4 during chick embryo development, Anat. Embryol. *210*, 35-41.

\newpage

# Supplementary

```{r QC-normalization-plots, fig.width=12, fig.height=5, fig.cap= "The plots support the use of the vsn normalization on our data set. **A**: The red line is close to horizontal, although it shows some correlation at high intensity leves. **B**: The boxplots show good alignment of the mean intensity values. Some outliers are present but can neglected given the 0.25 and 0.75 quantile. **C**: A selected Scatterplot is shown. The very slightly banana shaped structure can be seen, but is only marginal. Overall the quality control confirmes successful vsn normalization"}


OC_normalization_plot = arrangeGrob(grobs = list(meansdplot.GSE15744.vsnrma, boxplot.GSE15744.vsnrma, scatterplot.GSE15744.vsnrma),
                                    nrow=1, widths = c(3,3,3), heights = c(5))

OC_normalization_plot_output = as_ggplot(OC_normalization_plot) +
  draw_plot_label(label = c("A", "B", "C"), size = 15, x = c(0.01, 0.36, 0.71), y = c(0.13, 0.13, 0.13)) 

annotate_figure(OC_normalization_plot_output, top = text_grob("Quality control: verifying the normalization at different levels of detail", color = "black", face = "bold", size = 16))
```



```{r, fig.width = 18, fig.height = 10, include = FALSE}
# Create a dataset for the volcano plot from the limma analysis
# Showing the toptable from week 9 to 4 comparison
top.table = topTable(fit_all,5, n = Inf)


# Create a function for volcano plot
volcanizator <- function(data, logFC_thresh_upper, logFC_thresh_lower, max_pValue){
  
  # Point colours
  mycolours <- c("blue", "red", "black")
  names(mycolours) <- c("DOWN", "UP", "NO")
  
  # Add a column to the data frame to specify if they are UP- or DOWN- regulated (log2FoldChange respectively positive or negative)
  data$diffexpressed <- "NO"
  data$diffexpressed[data$logFC > logFC_thresh_upper & data$adj.P.Val < max_pValue] <- "UP"
  data$diffexpressed[data$logFC < logFC_thresh_lower & data$adj.P.Val < max_pValue] <- "DOWN"
  
  # Create a new column with names of differential expressed transcripts
  data$delabel <- NA
  #data$delabel[data$diffexpressed != "NO"] <- data$HGNC.symbol[data$diffexpressed != "NO"]
  
  # Plot
  ggplot(data, aes(x=logFC, y=-log10(adj.P.Val))) +
    geom_point(aes(colour = diffexpressed), size=0.9) +
    theme_minimal() +
    geom_text_repel(aes(label = delabel)) + #, force = 5, point.padding	= unit(5, "cm")
    geom_vline(xintercept=c(logFC_thresh_lower, logFC_thresh_upper), col="red") + # log2FoldChange thresholds
    geom_hline(yintercept=-log10(max_pValue), col="red") + # p-value threshold
    scale_colour_manual(values = mycolours) +
    geom_text(aes(label = delabel)) +
    labs(title = paste("Differential gene expression: week 4 to 9"),
         subtitle = "GEO dataset: GSE15744",
         caption = "p Value adjusted: Benjamini & Hochberg (1995)"
         ) +
    xlab("log2FC") +
    ylab("-log10(adj.P.Val)")
}
volcano <- volcanizator(top.table, 1, -1, 0.01)
png(filename = "volcano.png", width = 1000, height = 600)
volcano
dev.off()
```

```{r Volcano-Plot,fig.width = 12, fig.height = 6, fig.cap="Volcano plot for differential gene expression between week 4 and 9. The adjusted P-value boundary was set at 0.01. The logFC boundaries were -1 and 1, which reflects a doubling in expression. Genes are up-regulated if the logFC value is larger than 1 and the adjusted P-value is smaller than 0.01 and down-regulated if the logFC value is smaller than 1 and the adjusted P-value is smaller than 0.01. Genes which didnât belong in those groups were declared as not differentially expressed."}
ggdraw() + draw_image("volcano.png")
```

```{r heatmap-code, fig.width=14, fig.height=14, include = FALSE, eval = FALSE}

tissues =  c("Adipose", "Cells - Transformed fibroblasts", "Skin",  "Muscle - Skeletal", "Bladder", "Kidney - Cortex",  "Colon", "Esophagus", "Liver", "Minor Salivary Gland", "Pancreas", "Small Intestine - Terminal Ileum", "Spleen", "Stomach", "Breast - Mammary Tissue", "Cervix", "Fallopian Tube", "Ovary", "Uterus", "Vagina", "Prostate", "Testis", "Adrenal Gland", "Pituitary", "Thyroid",  "Artery", "Heart", "Lung", "Whole Blood", "Cells - EBV-transformed lymphocytes", "Brain - Spinal cord", "Brain", "Nerve - Tibial")

node_weight = data.frame(tissue = tissue, weight = sapply(nodes, function(y) {sum(sapply(diff_genes_ann_0.01$tissues, function(x) {grepl(y, toString(x))}))}))

weights = matrix(nrow =length(tissues), ncol = length(tissues))
rownames(weights) = tissues
colnames(weights) = tissues
for (i in 1:33) {
  for (j in 1:33) {
    weights[i,j] = (sum(sapply(diff_genes_ann_0.01$tissues, function(x) {grepl(tissues[i], toString(x)) && grepl(tissues[j], toString(x))}))/node_weight[i,2])
  }
} 

heatmap_links <- pheatmap(weights, display_numbers = TRUE, fontsize_number = 8, legend = TRUE, breaks = c(seq(0,0.4,0.005), seq(0.41, 0.8, 0.025), 0.85, 0.9, 0.95,1))
```

```{r, include = FALSE}

file_heatmap = "https://raw.githubusercontent.com/datascience-mobi-2022/2022-topic-04-team-01/main/Plots/Organ%20development/General%20Explanation/Heatmap%20TRA%20Overlap_edited.png"
download.file(file_heatmap, "TRA_overlap_edited.png", mode = "wb")
 plot = ggdraw() + draw_image("TRA_overlap_edited.png")
 
```

```{r heatmap-tissues, fig.width=14, fig.height=12, fig.cap = "For each of the tissues on the right, the share of its TRAs that are also associated with the tissue in the respective column is displayed."}
plot_2 = grid.arrange(plot) + theme(plot.margin = unit(c(0,0,5,0,0.5),"cm"))
```



```{r spleen-boxplot, fig.width= 12, fig.height= 3, fig.cap = "A further look on the expression of transcripts in the up- and downregulated clusters shows that the upregulated transcripts are close to the minimum expression level between 6 and 7 in week 4 and showing expressions between 7 and 8.5 by week 9. In contrast, the downregulated genes have very high expression levels (8-9) by week 4 and decrease to a more moderate expression between 7 and 8.5 analogous to the upregulated transcripts."}
library(ggplot2)
#Compare the initial expression values of both clusters
spleen_genes_clustered_1 = spleen_genes_clustered[spleen_genes_clustered$Cluster==1, ]
spleen_genes_clustered_2 = spleen_genes_clustered[spleen_genes_clustered$Cluster==2, ]
spleen_boxplot_data = data.frame(Expression = NA, Week = NA, Cluster = NA)
i = 0
for (genelist in list(spleen_genes_clustered_1$Week4, spleen_genes_clustered_1$Week9, spleen_genes_clustered_2$Week4,  spleen_genes_clustered_2$Week9)) {
  i = i+1
  spleen_df = data.frame(Expression = genelist, Week = factor(rep((if(i%%2 == 0) {9} else {4}),length(genelist))), Cluster = rep((if(i<3) {"Up"} else
    {"Down"}),length(genelist)))
  spleen_boxplot_data = rbind(spleen_boxplot_data, spleen_df)
}
spleen_boxplot_data = spleen_boxplot_data[-1,]
spleen_boxplot = ggplot(spleen_boxplot_data, aes(x = Cluster, y = Expression, fill = Week)) + geom_boxplot(position=position_dodge(0.8)) + theme_classic() + labs(title = "Expression of spleen-associated transcripts") + theme(aspect.ratio = 1, plot.title = element_text(hjust = 0.5))
spleen_boxplot

```

```{r tissue-links,fig.width= 10, fig.height=5, fig.cap= "Each tissue is displayed as a node with its size representing the number of transcripts associated with it. The edges show the shared TRAs between the linked tissues, with only links corresponding to more than 100 common TRAs visible."}

nodes = c("Adipose", "Cells - Transformed fibroblasts", "Skin",  "Muscle - Skeletal", "Bladder", "Kidney - Cortex",  "Colon", "Esophagus", "Liver", "Minor Salivary Gland", "Pancreas", "Small Intestine - Terminal Ileum", "Spleen", "Stomach", "Breast - Mammary Tissue", "Cervix", "Fallopian Tube", "Ovary", "Uterus", "Vagina", "Prostate", "Testis", "Adrenal Gland", "Pituitary", "Thyroid",  "Artery", "Heart", "Lung", "Whole Blood", "Cells - EBV-transformed lymphocytes", "Brain - Spinal cord", "Brain", "Nerve - Tibial")

edges = data.frame (rep(nodes, 32), NA)
for (i in 1:32) {
  a = c(edges[(i+1):33,1], edges[1:i,1])
  edges[(33*(i-1)+1):(33*i),2] = a
}
colnames(edges) = c("Tissue1", "Tissue2")
graph1 = graph_from_data_frame(edges, directed = FALSE)


node_weight = data.frame(tissue = nodes, weight = sapply(nodes, function(y) {sum(sapply(diff_genes_ann_0.01$tissues, function(x) {grepl(y, toString(x))}))}))

edge_combined = edges %>% unite(tissues, c("Tissue1", "Tissue2"), sep = "|")
edge_weight = data.frame(edges, weight = apply(edges,1, function(y) {sum(sapply(diff_genes_ann_0.01$tissues, function(x) {grepl(y[1], toString(x)) && grepl(y[2], toString(x))}))}))

edge_weight_adj = edge_weight
edge_weight_adj$weight = sapply(edge_weight_adj$weight, function(x) { if(x < 100) {x = 0} else {x = x/100}})
edge_weight_cleared = edge_weight_adj[!(edge_weight_adj$weight == 0),]


graph2 = ggnet2(edge_weight_cleared[,1:2], label = TRUE, edge.size = edge_weight_cleared$weight)
node_size = sapply(graph2$data$label, function(x) {node_weight[x,2]})

graph2 = ggnet2(edge_weight_cleared[,1:2], color = graph2$data$label, label = TRUE, color.palette = c() ,edge.size = edge_weight_cleared$weight, size = (ceiling(node_size/30))) + guides(size = FALSE, color = FALSE) + scale_color_viridis(option = "D", discrete = TRUE) + labs(title = "Overlap between the TRAs associated with different tissues") +  theme(plot.title = element_text(hjust = 0.5, face = "bold"), plot.margin = unit(c(0.2,2,0,2),"cm"))
graph2
```


```{r spleen-tables-basics, eval=FALSE}
get_gene_info = function(transcript_ids){
  table = data.frame(Transcript = NA, Gene = NA, Protein = NA, Summary = NA)
  for(x in transcript_ids) {
    table = rbind(table, get_single_transcript(x))
  }
  table
}

get_single_transcript = function(single_transcript) {
   res = entrez_search(db = "gene", term = single_transcript)
   if(!length(res$ids)==0) {
     results = entrez_summary(db = "gene", id = res$ids)
      c(single_transcript, as.character(results[2]),  as.character(results[3]),  as.character(results[17]))
   }
   else{
        c(single_transcript, NA, NA, NA)
   }
}

spleen1_gene_description = get_gene_info(rownames(spleen_genes_clustered_1))
spleen2_gene_description = get_gene_info(rownames(spleen_genes_clustered_2))
```


```{r spleen-up-code, include = FALSE, eval = FALSE}
# Create a useful table for all upregulated genes

Expression =  data.frame(c(1:length(spleen_genes_clustered_1[,1])))
Expression$Expression_over_time = sapply(c(1:length(spleen_genes_clustered_1[,1])), function(x) {list(as.numeric(c(as.numeric(spleen_genes_clustered_1[x,1]), as.numeric(spleen_genes_clustered_1[x,2]), as.numeric(spleen_genes_clustered_1[x,3]),
           as.numeric(spleen_genes_clustered_1[x,4]),as.numeric(spleen_genes_clustered_1[x,5]),as.numeric(spleen_genes_clustered_1[x,6]))))})

spleen1_gene_description_spl = cbind(spleen1_gene_description[-1,], Expression)
spleen1_gene_description_spl = spleen1_gene_description_spl[,-5]

spleen1_gene_description_sum = spleen1_gene_description_spl[c(2,5,6,8,11,17,20,26,29, 33,37,38,41,42,49,71,96),]
functions_1 = c("Member of the immunoglobulin superfamily", "Mediates polarization of T cells and neutrophils", "Secreted by cytptpxic lymphocytes", "Can upregulate blood pressure", "B lymphocyte chemoattractant", "IL-11 receptor", "", "Contains von Willebrand factor domains", "May function as a cytokine", "Plays a role in immune surveillance", "Regulating the complement cascade", "HLA class II", "Endocytosis of pathogen", "Receptors for interferons alpha and beta", "Fibroblast growth factor", "Regulating the complement cascade", "Protect the antibody from degradation")
spleen1_gene_description_sum$Summary = functions_1

table_spleen1 = gt(spleen1_gene_description_sum)|>  tab_header(title= "Spleen - Upregulated", subtitle = "List of differentially epxressed genes") |> fmt_markdown(columns = c(Transcript, Gene, Protein, Summary)) |> gt_theme_538() |> gt_plt_sparkline( column = Expression_over_time, type = "ref_median") |> tab_options(container.width = 1300, container.height = 800)
gtsave(table_spleen1, "spleen_up.png")
```

```{r spleen-up-table, fig.width = 12, fig.height = 7, fig.cap = "Table of all spleen-associated genes from the upregulated cluster with a function related to the spleens overall purpose (immune and blood-related genes)."}
# Load the table from Github
file_spleen_up = "https://raw.githubusercontent.com/datascience-mobi-2022/2022-topic-04-team-01/main/Report/spleen_up.png"
download.file(file_spleen_up, "spleen_up.png", mode = "wb")
ggdraw() + draw_image("spleen_up.png")
```


 \newpage

```{r spleen-down-code, fig.width=12, include = FALSE, eval = FALSE}
# Table for downregulated genes

Expression =  data.frame(c(1:length(spleen_genes_clustered_2[,1])))
Expression$Expression_over_time = sapply(c(1:length(spleen_genes_clustered_2[,1])), function(x) {list(as.numeric(c(as.numeric(spleen_genes_clustered_2[x,1]), as.numeric(spleen_genes_clustered_2[x,2]), as.numeric(spleen_genes_clustered_2[x,3]),
           as.numeric(spleen_genes_clustered_2[x,4]),as.numeric(spleen_genes_clustered_2[x,5]),as.numeric(spleen_genes_clustered_2[x,6]))))})

spleen2_gene_description_spl = cbind(spleen2_gene_description[-1,], Expression)
spleen2_gene_description_spl = spleen2_gene_description_spl[,-5]


spleen2_gene_description_sum = spleen2_gene_description_spl[c(2,3,7,8,9,15,20,22,23,25,27,29,31,34,35,37,38,39,40,41,43,45,46,50,54,56,57,58,59,64,65,67,69,72,73,75,77,78,83,113,114,116,121,122,143),]
functions_2 = c("Component of  the centromere", "Interphase chromosome condensation", "Chromosome condensation and stabilization", "Spindle microtubule organization", "Spindle dynamics", "Spindle checkpoint function", "Highly expressed during mitosis", "Central role in mitosis", "", "Nuclear movement in the cell cycle", "Stabilization of the mitotic spindle", "Regulator of mitosis", "Stem cell pluripotency", "Involved with cellular proliferation", "Mismatch repair", "Regulates G1/S transition", "Moves chromosomes", "Anaphase-promoting complex substrate", "Unwinds Holliday junction", "Induces apoptotic chromatin condensation", "Promotes cell growth", "Transcriptional activator in proliferation", "Associates with centromer-kinetochore complex", "Initiation of DNA replication", "Regulatory in the cell cycle", "Maintaining chromosome integrity in mitosis", "Stable spindle microtubule capture", "CpG methylation", "Initiation of DNA replication", "Subunit of M-phase promoting factor", "Involved in kinetochore function", "Cell cycle progression", "Cell cycle checkpoint regulator", "Spindle checkpoint function", "Spindle microtubule organisation", "Inducing cell cycle arrest", "Subunit of condensin complex","Cell division cycle protein", "DNA polymerase accessory", "Centromere-microtubule complex", "Mismatch repair","Anaphase-promoting complex substrate","Segregation of chromosomes", "Central role in mitosis", "Contributor to developmental programs")
spleen2_gene_description_sum$Summary = functions_2
gtsave((gt(spleen2_gene_description_sum)|>  tab_header(title= "Spleen - Downregulated", subtitle = "List of differentially epxressed genes") |> fmt_markdown(columns = c(Transcript, Gene, Protein, Summary)) |> gt_theme_538() |> gt_plt_sparkline(column = Expression_over_time, type = "shaded") |> tab_options(container.width = 1300, container.height = 1800)), "spleen_down.png")
```

```{r spleen-down-table, fig.width = 12, fig.height = 16, fig.cap = "Table of all spleen-associated genes from the downregulated cluster with a function related to the cell cycle or cellular division."}
file_spleen_down = "https://raw.githubusercontent.com/datascience-mobi-2022/2022-topic-04-team-01/main/Report/spleen_down.png"
download.file(file_spleen_down, "spleen_down.png", mode = "wb")
ggdraw() + draw_image("spleen_down.png")

```

\newpage

```{r Venn-Diagram, fig.width=12, fig.height=3, fig.cap = "**The Venn diagram shows overlapping sections between our data and the data from the original paper.** According to the paper,  6,833 of its total transcripts are differentially expressed. This overlapps with our general data with all TRAs (24,749 transcripts) and with our differentially expressed (diff.) data according to our limma analysis with a p-value of 0.01 (1,837 transcripts)"}
ggdraw() + draw_image("venn_diagram_comparison_paper.png")
```


```{r boxplot-Cerebral-Hemisphere, fig.height=9,fig.width=12,fig.cap="A boxplot of all Cerebellar Hemisphere genes was made to evaluate their scattering. For the samples of all weeks at the top and only for the three samples of week 4 on the bottom.", eval = FALSE}

# Distribution control for Cerebellar Hemisphere 
Tissues = embryo_df_tissues$max.tissue


#Boxplot for all replicates over all weeks
Cerebellar_Hemisphere = as.data.frame(embryo_df_tissues %>%
                             filter(max.tissue == str_extract(Tissues, "Brain - Cerebellar Hemisphere")))
Cerebellar_Hemisphere_names = Cerebellar_Hemisphere$HGNC.symbol
Cerebellar_Hemisphere =Cerebellar_Hemisphere[,1:18]
Cerebellar_Hemisphere = t(Cerebellar_Hemisphere)
colnames(Cerebellar_Hemisphere) <- Cerebellar_Hemisphere_names
Cerebellar_Hemisphere = as.data.frame(Cerebellar_Hemisphere)
Cerebellar_Hemisphere_sorted = Cerebellar_Hemisphere %>% select(sort(colnames(Cerebellar_Hemisphere)))

# As contrast, only for week 4 only
Cerebellar_Hemisphere_week4= Cerebellar_Hemisphere_sorted[1:3,]

# Summary plot 
par(mfrow=c(2,1))
boxplot(Cerebellar_Hemisphere_sorted, main="Expression of Brain - Cerebellar Hemisphere genes",col= rainbow(542),cex.axis=0.5)
boxplot(Cerebellar_Hemisphere_week4, main="Expression of Brain - Cerebellar Hemisphere genes (week 4)",col= rainbow(542),cex.axis=0.5)
```

```{r Cerebellar-Hemisphere-specific,fig.height=4,fig.width=12,fig.cap="A boxplot of ENST00000276646, ENST00000518312 and ENST00000521485 was made to evaluate the scattering of the three replicas per week for the 6 weeks from our dataset."}
# Distribution Kontrol for ENST00000276646, ENST00000518312 & ENST00000521485
ENST = embryo_df_tissues$ensembl.transcript

# ENST00000276646
data.ENST00000276646 = as.data.frame(embryo_df_tissues %>%
                             filter(ensembl.transcript == str_extract(ENST, "ENST00000276646")))
week.4 =t(data.ENST00000276646[1:3])
week.5 =t(data.ENST00000276646[4:6])
week.6 =t(data.ENST00000276646[7:9])
week.7 =t(data.ENST00000276646[10:12])
week.8 =t(data.ENST00000276646[13:15])
week.9 =t(data.ENST00000276646[16:18])
data.ENST00000276646 = cbind(week.4,week.5,week.6,week.7,week.8,week.9)
colnames(data.ENST00000276646) <- c("week 4","week 5","week 6","week 7","week 8","week 9")

# ENST00000518312
data.ENST00000518312 = as.data.frame(embryo_df_tissues %>%
                             filter(ensembl.transcript == str_extract(ENST, "ENST00000518312")))
week.4 =t(data.ENST00000518312[1:3])
week.5 =t(data.ENST00000518312[4:6])
week.6 =t(data.ENST00000518312[7:9])
week.7 =t(data.ENST00000518312[10:12])
week.8 =t(data.ENST00000518312[13:15])
week.9 =t(data.ENST00000518312[16:18])
data.ENST00000518312 = cbind(week.4,week.5,week.6,week.7,week.8,week.9)
colnames(data.ENST00000518312) <- c("week 4","week 5","week 6","week 7","week 8","week 9")

# ENST00000521485
data.ENST00000521485 = as.data.frame(embryo_df_tissues %>%
                             filter(ensembl.transcript == str_extract(ENST, "ENST00000521485")))
week.4 =t(data.ENST00000521485[1:3])
week.5 =t(data.ENST00000521485[4:6])
week.6 =t(data.ENST00000521485[7:9])
week.7 =t(data.ENST00000521485[10:12])
week.8 =t(data.ENST00000521485[13:15])
week.9 =t(data.ENST00000521485[16:18])
data.ENST00000521485 = cbind(week.4,week.5,week.6,week.7,week.8,week.9)
colnames(data.ENST00000521485) <- c("week 4","week 5","week 6","week 7","week 8","week 9")

# Summmary plot
par(mfrow= c(1,3))
boxplot(data.ENST00000276646, main="Expression of ENST00000276646",col= rainbow(6),cex.axis=0.5)
boxplot(data.ENST00000518312, main="Expression of ENST00000518312",col= rainbow(6),cex.axis=0.5)
boxplot(data.ENST00000521485, main="Expression of ENST00000521485",col= rainbow(6),cex.axis=0.5)
```

```{r PCA-analysis, fig.width = 10, fig.height = 4, fig.cap = "**A**. Variance explained by each individual PC as well as cumulative variance explained by the PCs together. **B**. Expression levels of the five transcripts that contribute the most to PC1."}
file_fiveplot = "https://raw.githubusercontent.com/datascience-mobi-2022/2022-topic-04-team-01/main/Report/PC1_five.png"
download.file(file_fiveplot, "PC1_five.png", mode = "wb")
first_five_plot <- ggdraw() + draw_image("PC1_five.png") 

lay = rbind(c(1,2))
PCA_plots = arrangeGrob(grobs = list(PC_contribution, first_five_plot),layout_matrix = lay)

PCA_plots_output = as_ggplot(PCA_plots) +
  draw_plot_label(label = c("A", "B"), 
                  size = 15, 
                  x = c(0.01, 0.5), 
                  y = c(0.17, 0.17)
                  ) 

annotate_figure(PCA_plots_output, 
                top = text_grob("Principal component analysis", 
                                color = "black", 
                                face = "bold", 
                                size = 16
                                )
                )
```
